---
title: "FigureGen_10222021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



Install/Load Libraries
```{r}
#install.packages("reticulate")
#install.packages("robustbase")
#install.packages("facetscales")
#install.packages("ggpointdensity")
#install_github("https://github.com/vdemichev/diann-rpackage")
source("plexDIA_Functions.R")
library(facetscales)
library(robustbase)
library(plyr)
library(reticulate)
library(stringi)
library(stringr)
library(dplyr)
library(tidyr)
library(patchwork)
library(ggplot2)
library(reshape2)
library(Cairo)
library(gridGraphics)
library(data.table)
library(devtools)
library(diann)
library(gridExtra)
library(ggpubr)
library(ggpointdensity)
library(viridis)
library(scales)
library(data.table)
library(matrixStats)


'%!in%' <- function(x,y)!('%in%'(x,y))

fpath <- "/v1.8b7_10102021/plexDIA_MS1/Report.tsv" #plexDIA file path (MS1)
LF_fpath <- "/v1.8b7_10102021/LF_MS1/Report.tsv"  #LF-DIA file path (MS1)
DDA_fpath <- "/wJD750_752/evidence.txt"  #DDA file path (from MQ)
CC_data <- "/CC_MS1_MS2_with_Contam_FASTA/Report.tsv" #plexDIA cell cycle data path
INPUT <- "/v1.8b7_10102021/plexDIA_MS1/Report.pr_matrix_channels_ms1_translated.tsv" #used just to get column names
YE_fpath <- "/v1.8b7_10102021/YE_missingE/Report.tsv" 
DDA_prot_fpath <- "wJD750_752/proteinGroups.txt"
MS2_fpath <- "/v1.8b7_10102021/plexDIA_MS2/Report.tsv" #file path
LF_MS2_fpath <- "/v1.8b7_10102021/LF_MS2/Report.tsv"  #LF-DIA file path
Bench_fpath <- "/v1.8b7_10102021/Navaro_LFQBench/Report (1).tsv" #LFQ bench file path

GO_DB_Path <- "/GOA_db.txt" #GO database


plexDIA_run <- "wJD804"
LF_runs <- c("wJD757_re", "wJD760_re", "wJD763_re") # LF-DIA runs you are interested in plotting

#######
tqval <- 1.1 #Translated.Q.Value filter (set to 1.1 to not filter anything)


#########
clust <- function(x) {
  umelt <- tidyr::spread(x, prot_info, Prot_norm_int)
  rownames(umelt) <- umelt$C_Q
  umelt1 <- umelt
  rownames(umelt1) <- umelt1$C_Q
#2) calculate euclidian dist
  t_umelt <- t(as.matrix(umelt1))[-c(1:3),]
  ord <- hclust( dist(t_umelt, method = "euclidean"), method = "ward.D" )$order
  umelt <- melt(t_umelt)
  umelt$Var1 <- factor(umelt$Var1, levels= rownames(t_umelt)[ord])
  return(umelt)
}

```


Figure 1: Schematic
```{r}

#Figure 1a: created in Abode Illustrator

###################################################################
###################################################################
###################################################################
###################################################################

#Figure 2a:
d0 <- 778.64  #mTRAQ delta 0 cost
d4 <- 2348.04 #mTRAQ delta 4 cost
d8 <- 2343.62 #mTRAQ delta 8 cost

total_prot <- 50*3*0.1 #50 aliquots * 3 labels * 0.1 mg protein/aliquot

total_cost <- d0+d4+d8

cost_per_ug <- total_prot*1000/total_cost

plexDIA <- (100*1.5)/3
LFDIA <- (100*1.5)
cost <- c(LFDIA, plexDIA, (cost_per_ug/3)*1.5)
Application_Reagent <- c("LC-MS/MS cost","LC-MS/MS cost","Label cost")
type <- c("LF-DIA", "plexDIA","plexDIA")

df <- data.frame(cost = cost, Application_Reagent = Application_Reagent,type = type)
#df$error <- c(25,75)

df$Application_Reagent <- factor(df$Application_Reagent, levels=c("Label cost", "LC-MS/MS cost"))

ggplot(df, aes(x=type, y=cost, fill=type)) + 
  theme_classic() + labs(x="", y="Estimated cost (USD) per sample") +
  geom_bar(data = df %>% filter(type == "plexDIA"),#colour="black", 
           stat="identity", aes(x=type, y =cost, fill=Application_Reagent), width = 0.45, alpha =1) + 
  geom_bar(data = df %>% filter(type == "LF-DIA"),#colour="black", 
           stat="identity", aes(x=type, y =cost, fill=Application_Reagent), width = 0.45, alpha =1) + 
  scale_y_continuous(labels=scales::dollar_format()) + 
  scale_fill_manual(values = c("red","grey25")) +
  theme(legend.position = c(0.7,0.8),
        legend.title = element_blank(),
        legend.text = element_text(size=11),
        axis.text.x = element_text(size=13, face = "bold", color = "black"),
        axis.text.y = element_text(size=14, face = "bold", color = "black"),
        axis.title.x = element_text(size=12, color = "black")) + coord_flip()

ggsave("Figure1b.pdf", width=4.1, height=1.9, dpi=500)

###################################################################
###################################################################
###################################################################
###################################################################

#Figure 3a: created in Adobe Illustrator

```

Figure 2: Coverage
```{r}

# Figure 2, Proteomic coverage and overlap between samples and runs of plexDIA and LF-DIA:
mTRAQ <- read.delim(fpath)
mTRAQ<- mTRAQ %>% filter(Translated.Q.Value<=tqval) 
mTRAQ$Label <- "mTRAQ"
mTRAQ$seqcharge <- paste0(mTRAQ$Modified.Sequence, mTRAQ$Precursor.Charge)
mTRAQ$seqcharge_run <- paste0(mTRAQ$seqcharge, "_", mTRAQ$Run)
mTRAQ$prot_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ0\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ4\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ8\\)")
mTRAQ_uni_prot <- unique(mTRAQ$Protein.Group)

ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$Precursor.Id),] %>% distinct(seqcharge_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated","Ms1.Area", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_4) <- c("Run","Protein.Group_d4","seqcharge_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4", "seqcharge_run_d4", "prot_run_d4","Lib.PG.Q.Value_d4")
ev2_8 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated","Ms1.Area", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_8) <- c("Run","Protein.Group_d8","seqcharge_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8", "seqcharge_run_d8", "prot_run_d8","Lib.PG.Q.Value_d8")

mTRAQ_int <- ev2_0 %>% inner_join(ev2_4, by =c("seqcharge" = "seqcharge_d4"))
mTRAQ_int <- mTRAQ_int %>% inner_join(ev2_8, by =c("seqcharge" = "seqcharge_d8"))

ev2_0_prot <- ev2_0[which(ev2_0$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
ev2_0_prot$Sample <- paste0("\u0394","0-A")
ev2_4_prot <- ev2_4[which(ev2_4$Lib.PG.Q.Value_d4 < 0.01),] %>% distinct(prot_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4","prot_run_d4")
colnames(ev2_4_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_4_prot$Sample <- paste0("\u0394","4-B")
ev2_8_prot <- ev2_8[which(ev2_8$Lib.PG.Q.Value_d8 < 0.01),] %>% distinct(prot_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8","prot_run_d8")
colnames(ev2_8_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_8_prot$Sample <- paste0("\u0394","8-C")

ev2_0_pep <- ev2_0 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_0_pep$Sample <- paste0("\u0394","0-A")
ev2_4_pep <- ev2_4 %>% distinct(seqcharge_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4", "seqcharge_d4", "seqcharge_run_d4")
colnames(ev2_4_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_4_pep$Sample <- paste0("\u0394","4-B")
ev2_8_pep <- ev2_8 %>% distinct(seqcharge_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8", "seqcharge_d8", "seqcharge_run_d8")
colnames(ev2_8_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_8_pep$Sample <- paste0("\u0394","8-C")

mTRAQ_prots <- rbind(ev2_0_prot,ev2_4_prot,ev2_8_prot)
mTRAQ_prots$Sample_new <- mTRAQ_prots$Sample

mTRAQ_peps <- rbind(ev2_0_pep,ev2_4_pep,ev2_8_pep)
mTRAQ_peps$Sample_new <- mTRAQ_peps$Sample

p_mTRAQ <- mTRAQ_peps %>% dplyr::count(Run, Sample_new)
p_mTRAQ$Condition <- "plexDIA"

prot_mTRAQ <- mTRAQ_prots %>% dplyr::count(Run, Sample_new)
prot_mTRAQ$Condition <- "plexDIA"



##############
##############
#######################
########################
##########################  Label-free

unlab <- read.delim(LF_fpath)
unlab<- unlab %>% filter(Translated.Q.Value<=tqval) 
unlab_1 <- unlab[grepl("wJD756_re|wJD757_re|wJD758_re", unlab$Run),]
unlab_1$seqcharge <- paste0(unlab_1$Modified.Sequence, unlab_1$Precursor.Charge)
unlab_1$seqcharge_run <- paste0(unlab_1$seqcharge, "_", unlab_1$Run)
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_", unlab_1$Run)

unlab_1_prot <- unlab_1[which(unlab_1$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_1_prot$Sample <- paste0("\u0394","0-A")
unlab_1_prot$Sample_new <- paste0("A")

unlab_1_pep <- unlab_1 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_1_pep$Sample <- paste0("\u0394","0-A")
unlab_1_pep$Sample_new <- paste0("A")

##############
unlab_2 <- unlab[grepl("wJD759_re|wJD760_re|wJD761_re", unlab$Run),]
unlab_2$seqcharge <- paste0(unlab_2$Modified.Sequence, unlab_2$Precursor.Charge)
unlab_2$seqcharge_run <- paste0(unlab_2$seqcharge, "_", unlab_2$Run)
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_", unlab_2$Run)

unlab_2_prot <- unlab_2[which(unlab_2$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_2_prot$Sample <- paste0("\u0394","4-B")
unlab_2_prot$Sample_new <- paste0("B")

unlab_2_pep <- unlab_2 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_2_pep$Sample <- paste0("\u0394","4-B")
unlab_2_pep$Sample_new <- paste0("B")

##############
unlab_3 <- unlab[grepl("wJD762_re|wJD763_re|wJD764_re", unlab$Run),]
unlab_3$seqcharge <- paste0(unlab_3$Modified.Sequence, unlab_3$Precursor.Charge)
unlab_3$seqcharge_run <- paste0(unlab_3$seqcharge, "_", unlab_3$Run)
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_", unlab_3$Run)

unlab_3_prot <- unlab_3[which(unlab_3$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_3_prot$Sample <- paste0("\u0394","8-C")
unlab_3_prot$Sample_new <- paste0("C")


unlab_3_pep <- unlab_3 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_3_pep$Sample <- paste0("\u0394","8-C")
unlab_3_pep$Sample_new <- paste0("C")

###################

all_unlab_pep <- rbind(unlab_1_pep, unlab_2_pep, unlab_3_pep)
all_unlab_prot <- rbind(unlab_1_prot, unlab_2_prot, unlab_3_prot)

p_unlab <- all_unlab_pep %>% dplyr::count(Run, Sample_new)
p_unlab$Condition <- "LF-DIA"

prot_unlab <- all_unlab_prot %>% dplyr::count(Run, Sample_new)
prot_unlab$Condition <- "LF-DIA"

##############
##############
#######################
########################
##########################  mTRAQ DDA

#get prots which are 1% FDR
prot_DDAFDR <- read.delim(DDA_prot_fpath)
prot_DDAFDR <- prot_DDAFDR[which(prot_DDAFDR$Q.value<0.01),] %>% filter(!grepl("REV|CON", Protein.IDs))
prot_DDAFDR$Protein.Group <- gsub(";.*","",prot_DDAFDR$Majority.protein.IDs)

DDA <- read.delim(DDA_fpath)
DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
## find FDR 1 % for each run:
DDA1 <- DDA %>% group_by(Raw.file) %>% arrange(PEP) %>% 
  mutate(rec = 1) %>% 
  mutate(rollavg = cumsum(PEP)/cumsum(rec)) %>% 
  select(-rec) %>% ungroup()
DDA1 <- DDA1 %>% dplyr::select("Modified.sequence","PEP","Raw.file","rollavg")
DDAfil <- DDA1 %>% filter(rollavg < 0.01) %>% group_by(Raw.file) %>% 
  summarise(PEP = max(PEP))

#filter for 1% FDR
DDA_list <- list()
for( i in 1:3){
  a <- as.numeric(paste0(DDAfil[i,2]))
  temp <- DDA[grepl(paste0(DDAfil[i,1]), DDA$Raw.file),] %>% filter(PEP < a)
  DDA_list[[i]] <- temp
}
DDA <- do.call("rbind",DDA_list) #combine all vectors into a matrix

DDA_d0 <- DDA[which(DDA$Intensity.L > 0),]
DDA_d0$seqcharge <- paste0(DDA_d0$Modified.sequence, DDA_d0$Charge)
DDA_d0$seqcharge_run <- paste0(DDA_d0$seqcharge, "_", DDA_d0$Raw.file)
DDA_d0$Protein.Group <- gsub(";.*","",DDA_d0$Leading.razor.protein)
DDA_d0$prot_run <- paste0(DDA_d0$Leading.razor.protein, "_", DDA_d0$Raw.file)

DDA_d0_prot <- DDA_d0 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d0_prot$Sample <- paste0("A")
DDA_d0_prot$Sample_new <- paste0("\u0394","0 DDA-A")

DDA_d0_pep <- DDA_d0 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d0_pep$Sample <- paste0("A")
DDA_d0_pep$Sample_new <- paste0("\u0394","0 DDA-A")

####
DDA_d4 <- DDA[which(DDA$Intensity.M > 0),]
DDA_d4$seqcharge <- paste0(DDA_d4$Modified.sequence, DDA_d4$Charge)
DDA_d4$seqcharge_run <- paste0(DDA_d4$seqcharge, "_", DDA_d4$Raw.file)
DDA_d4$Protein.Group <- gsub(";.*","",DDA_d4$Leading.razor.protein)
DDA_d4$prot_run <- paste0(DDA_d4$Leading.razor.protein, "_", DDA_d4$Raw.file)

DDA_d4_prot <- DDA_d4 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d4_prot$Sample <- paste0("B")
DDA_d4_prot$Sample_new <- paste0("\u0394","4 DDA-B")

DDA_d4_pep <- DDA_d4 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d4_pep$Sample <- paste0("B")
DDA_d4_pep$Sample_new <- paste0("\u0394","4 DDA-B")

####
DDA_d8 <- DDA[which(DDA$Intensity.H > 0),]
DDA_d8$seqcharge <- paste0(DDA_d8$Modified.sequence, DDA_d8$Charge)
DDA_d8$seqcharge_run <- paste0(DDA_d8$seqcharge, "_", DDA_d8$Raw.file)
DDA_d8$Protein.Group <- gsub(";.*","",DDA_d8$Leading.razor.protein)
DDA_d8$prot_run <- paste0(DDA_d8$Leading.razor.protein, "_", DDA_d8$Raw.file)

DDA_d8_prot <- DDA_d8 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d8_prot$Sample <- paste0("C")
DDA_d8_prot$Sample_new <- paste0("\u0394","8 DDA-C")

DDA_d8_pep <- DDA_d8 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d8_pep$Sample <- paste0("C")
DDA_d8_pep$Sample_new <- paste0("\u0394","8 DDA-C")


##########################
##########################  data-wrangling for plotting
all_DDA_pep <- rbind(DDA_d0_pep, DDA_d4_pep, DDA_d8_pep)
all_DDA_prot <- rbind(DDA_d0_prot, DDA_d4_prot, DDA_d8_prot)
all_DDA_prot <- all_DDA_prot[which(all_DDA_prot$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

p_DDA <- all_DDA_pep %>% dplyr::count(Raw.file, Sample_new)
colnames(p_DDA) <- c("Run", "Sample_new", "n")
p_DDA$Condition <- "mTRAQ\nDDA"

prot_DDA <- all_DDA_prot %>% dplyr::count(Raw.file, Sample_new)
colnames(prot_DDA) <- c("Run", "Sample_new", "n")
prot_DDA$Condition <- "mTRAQ\nDDA"

###################
all_pep <- rbind(p_mTRAQ, p_unlab, p_DDA)
all_pep$Sample <- gsub(".*-","",all_pep$Sample_new)

p_all_tab <- all_pep %>% group_by(Sample_new) %>% mutate(mean = mean(n)) %>%
  mutate(se = sd(n)/sqrt(3)) %>% distinct(Sample_new, .keep_all=T)

a <- p_all_tab[3,6]
b <- p_all_tab[2,6] + p_all_tab[3,6]
c <- p_all_tab[2,6] + p_all_tab[1,6] + p_all_tab[3,6]

p_all_tab$er_max <- p_all_tab$mean + p_all_tab$se
p_all_tab[1,8] <- c+as.numeric(p_all_tab[1,7])
p_all_tab[2,8] <- b+as.numeric(p_all_tab[2,7])
p_all_tab[3,8] <- a+p_all_tab[3,7]

p_all_tab$er_min <- p_all_tab$mean - p_all_tab$se
p_all_tab[1,9] <- c-as.numeric(p_all_tab[1,7])
p_all_tab[2,9] <- b-as.numeric(p_all_tab[2,7])
p_all_tab[3,9] <- a-p_all_tab[3,7]


## for DDA
a <- p_all_tab[9,6]
b <- p_all_tab[8,6] + p_all_tab[9,6]
c <- p_all_tab[8,6] + p_all_tab[7,6] + p_all_tab[9,6]

p_all_tab[7,8] <- c+as.numeric(p_all_tab[7,7])
p_all_tab[8,8] <- b+as.numeric(p_all_tab[8,7])
p_all_tab[9,8] <- a+p_all_tab[9,7]

p_all_tab[7,9] <- c-as.numeric(p_all_tab[7,7])
p_all_tab[8,9] <- b-as.numeric(p_all_tab[8,7])
p_all_tab[9,9] <- a-p_all_tab[9,7]


#### text 
df_n <- p_all_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+7000
my_fin_colors <- c("turquoise3","palevioletred3","lightslategrey")

p1_cov <- ggplot(p_all_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = p_all_tab %>% filter(Condition == "plexDIA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "mTRAQ\nDDA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "LF-DIA"),# colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Precursors / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12),
        legend.position = c(0.7,0.8),
        legend.direction = "horizontal",
        legend.spacing.x = unit(1.0, 'line')) +
  guides(fill = guide_legend(override.aes= list(alpha = 0.7),
                             title = "Samples",
                             label.position = "bottom",
                             title.position = "top", title.vjust = 1, title.hjust=0.5)) +
  geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3
  ) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3, stat="identity", position=position_dodge(1.17))

p1_cov
ggsave("Figure2a.pdf",p1_cov,width=5.1,height=4.6, dpi=500)


############ prots

###################
all_prot <- rbind(prot_mTRAQ, prot_unlab, prot_DDA)
all_prot$Sample <- gsub(".*-","",all_prot$Sample_new)

all_prot_tab <- all_prot %>% group_by(Sample_new) %>% mutate(mean = mean(n)) %>%
  mutate(se = sd(n)/sqrt(3)) %>% distinct(Sample_new, .keep_all=T)

a <- all_prot_tab[3,6]
b <- all_prot_tab[2,6] + all_prot_tab[3,6]
c <- all_prot_tab[2,6] + all_prot_tab[1,6] + all_prot_tab[3,6]

all_prot_tab$er_max <- all_prot_tab$mean + all_prot_tab$se
all_prot_tab[1,8] <- c+as.numeric(all_prot_tab[1,7])
all_prot_tab[2,8] <- b+as.numeric(all_prot_tab[2,7])
all_prot_tab[3,8] <- a+all_prot_tab[3,7]

all_prot_tab$er_min <- all_prot_tab$mean - all_prot_tab$se
all_prot_tab[1,9] <- c-as.numeric(all_prot_tab[1,7])
all_prot_tab[2,9] <- b-as.numeric(all_prot_tab[2,7])
all_prot_tab[3,9] <- a-all_prot_tab[3,7]


## for DDA
a <- all_prot_tab[9,6]
b <- all_prot_tab[8,6] + all_prot_tab[9,6]
c <- all_prot_tab[8,6] + all_prot_tab[7,6] + all_prot_tab[9,6]

all_prot_tab[7,8] <- c+as.numeric(all_prot_tab[7,7])
all_prot_tab[8,8] <- b+as.numeric(all_prot_tab[8,7])
all_prot_tab[9,8] <- a+all_prot_tab[9,7]

all_prot_tab[7,9] <- c-as.numeric(all_prot_tab[7,7])
all_prot_tab[8,9] <- b-as.numeric(all_prot_tab[8,7])
all_prot_tab[9,9] <- a-all_prot_tab[9,7]


#### text 
df_n <- all_prot_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+700

my_fin_colors <- c("turquoise3","palevioletred3","lightslategrey")

p2_cov <- ggplot(all_prot_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = all_prot_tab %>% filter(Condition == "plexDIA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "mTRAQ\nDDA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "LF-DIA"),# colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Protein Data Points / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.position = "none") +
   geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2, stat="identity", position=position_dodge(1.17))

p2_cov
ggsave("Figure2b.pdf",p2_cov,width=5.1,height=4.6, dpi=500)


################## JACCARD 

mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]
mTRAQ <- mTRAQ[which(mTRAQ$Lib.Q.Value < 0.01),]

mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0),]  #only keep precursors with quant info

ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$Precursor.Id),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("mTRAQ8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8","Protein.Group")

d0_1 <- ev2_0[grepl("wJD803", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("wJD803", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_1 <- ev2_8[grepl("wJD803", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("wJD804", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("wJD804", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_2 <- ev2_8[grepl("wJD804", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("wJD815", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("wJD815", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_3 <- ev2_8[grepl("wJD815", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value

################################### unlabeled
unlab <- read.delim(LF_fpath)
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]
unlab <- unlab[which(unlab$Lib.Q.Value < 0.01),]

df_t <- data.frame(c("wJD756_re", "wJD757_re", "wJD758_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c("wJD759_re", "wJD760_re", "wJD761_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0),]  #only keep precursors with quant info

##

df_t <- data.frame(c("wJD762_re", "wJD763_re", "wJD764_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_3 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_",unlab_3$rep)
unlab_3 <- unlab_3[which(unlab_3$Ms1.Area>0),]  #only keep precursors with quant info

#####
s1_1 <- unlab_1[grepl("wJD756_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("wJD757_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("wJD758_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)

s2_1 <- unlab_2[grepl("wJD759_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("wJD760_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("wJD761_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)

s3_1 <- unlab_3[grepl("wJD762_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_2 <- unlab_3[grepl("wJD763_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_3 <- unlab_3[grepl("wJD764_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3,s3_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3,s3_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s3_1","s1_2","s2_2","s3_2","s1_3","s2_3","s3_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################



############### Jaccard index  DDA

DDA <- read.delim(DDA_fpath)

#filter for 1% FDR
DDA_list <- list()
for( i in 1:3){
  a <- as.numeric(paste0(DDAfil[i,2]))
  temp <- DDA[grepl(paste0(DDAfil[i,1]), DDA$Raw.file),] %>% filter(PEP < a)
  DDA_list[[i]] <- temp
}
DDA <- do.call("rbind",DDA_list) #combine all vectors into a matrix
DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_1 <- ev2_8[grepl("wJD750", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_2 <- ev2_8[grepl("wJD751", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_3 <- ev2_8[grepl("wJD752", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")


jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                                           ifelse(grepl("_3", Var1), "Rep 3", "remove"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                                           ifelse(grepl("_3", Var2), "Rep 3", "remove"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "remove"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "remove"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="MultiDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  


############## LF
m_jac_LF <- m_jac_LF %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="Label-Free DIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  


## combine
library(gridExtra)
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                            p1 + theme(legend.position="none"), NULL,
                            pDDA +theme(legend.position="none"),
                            nrow=1, widths=c(1,0.1,1,0.1,1)))

ggsave("Figure2c.pdf", p3, width=12.5, height=5.2, dpi=500)


############################ other plots:
jac_dist_mTRAQ_m$Type <- "plexDIA"
m_jac_LF$Type <- "LF-DIA"

jac_dist_mTRAQ_m$run_a <- stri_sub(jac_dist_mTRAQ_m$Var1,-1,-1)
jac_dist_mTRAQ_m$run_b <- stri_sub(jac_dist_mTRAQ_m$Var2,-1,-1)
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("set" = ifelse(run_a == run_b, "Within a run", "Across runs"))

m_jac_LF$run_a <- stri_sub(m_jac_LF$Var1,-1,-1)
m_jac_LF$run_b <- stri_sub(m_jac_LF$Var2,-1,-1)
m_jac_LF <- m_jac_LF %>% mutate("set" = "Across runs")

both <- bind_rows(jac_dist_mTRAQ_m, m_jac_LF)
both <- both[which(both$value<1),]  #remove overlaps to self

both$Var1a <- substr(both$Var1,1,nchar(as.character(both$Var1))-1)
both$Var2a <- substr(both$Var2,1,nchar(as.character(both$Var2))-1)

both <- both %>% mutate("Groups" = ifelse(Var1a == Var2a, "Same samples", "Different samples"))
both$set <- factor(both$set, levels=c("Within a run", "Across runs"))
both$Type <- factor(both$Type, levels=c("plexDIA", "LF-DIA"))
both$missingness <- (1-both$value)*100
both$Groups <- factor(both$Groups, levels=c("Same samples", "Different samples"))

ggplot(both, aes(x=as.factor(Type),y=(missingness), color=set)) + 
  geom_point(size=2.3, alpha =0.7, position=position_jitter(width=0.25, height=0)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA")) + 
  scale_color_manual(values = c("steelblue", "tomato1")) + #"tan2"
  theme_bw() +
  facet_grid(~Groups, scales = "free") +
  theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=16),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12),
        legend.text = element_text(size=14),
        legend.title = element_blank())+
  labs(x = "", y = "Missing data between pairs, %", fill = "") + ylim(0, 35)

ggsave("Figure2d.pdf", width=3.9, height=4, dpi=500)

```

Figure 3: Quantitative accuracy
```{r}
#1% FDR:
dat <- mTRAQ
plexDIA_run <- "wJD804"
dat <- dat[grepl("wJD804", dat$Run),]
df <- dat
nrow(df[which(df$Ms1.Area>0),])
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                      
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)
df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

pDIAa <- paste0(plexDIA_run,"mTRAQ0")
pDIAb <- paste0(plexDIA_run,"mTRAQ4")
pDIAc <- paste0(plexDIA_run,"mTRAQ8")

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAa]])/as.numeric(protein.groups_n[[pDIAb]])


protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio

######################## LF ########################################

dat_LF <- fread(LF_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                                     "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value","PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated",
                                     "Ms1.Area", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                                     "Mass.Evidence", "CScore", "Fragment.Correlations"))

df <- dat_LF
dat_LF<- dat_LF %>% filter(Translated.Q.Value<=tqval) 

df <- df[grepl(paste0(LF_runs[1],"|",LF_runs[2]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_runs[1]), Run), "mTRAQ0",
                                            ifelse(grepl(paste0(LF_runs[2]), Run), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                    
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

LFDIAa <- paste0(LF_runs[1],"mTRAQ0")
LFDIAb <- paste0(LF_runs[2],"mTRAQ4")
LFDIAc <- paste0(LF_runs[3],"mTRAQ8")


protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAa]])/as.numeric(protein.groups_n[[LFDIAb]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio

################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAb), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAb), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF_a) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF_a$sample2_int <- as.numeric(protein.groups_n_omit_LF_a$sample2_int)


#intersect LF and multiDIA
protein.groups_n_omit_noint <- protein.groups_n_omit
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF

protein.groups_n_omit_a <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF_a$PGs,]
protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF_a[protein.groups_n_omit_LF_a$PGs%in%protein.groups_n_omit$PGs,]


LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_a, protein.groups_n_omit_LF_a)

################################# plotting #########################

x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
a2 <- ggplot(protein.groups_n_omit_LF_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y="Log2, delta0:delta4",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())
a2


a4 <- ggplot(protein.groups_n_omit_LF_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a4
####################################
#d0d4
a1 <- ggplot(protein.groups_n_omit_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y=expression(paste(Log["2"],", A/B")),title = "plexDIA")+
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))

a1 

a3 <- ggplot(protein.groups_n_omit_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a3

################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

a_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+300, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = "Protein Ratios (A/B)", fill = "")

a_cover



################################################################
################################################################
########################## d0:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl(paste0(plexDIA_run), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAa]])/as.numeric(protein.groups_n[[pDIAc]])
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl(paste0(LF_runs[1],"|",LF_runs[3]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_runs[1]), Run), "mTRAQ0",
                                            ifelse(grepl(paste0(LF_runs[3]), Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAa]])/as.numeric(protein.groups_n[[LFDIAc]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA


protein.groups_n_omit_b <- protein.groups_n_omit
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF

protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[!grepl("H.", protein.groups_n_omit_LF_b$species),]
protein.groups_n_omit_b <- protein.groups_n_omit_b[!grepl("H.", protein.groups_n_omit_b$species),]

protein.groups_n_omit_noint <- protein.groups_n_omit_b
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_b

protein.groups_n_omit_b <- protein.groups_n_omit_b[protein.groups_n_omit_b$PGs%in%protein.groups_n_omit_LF_b$PGs,]
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[protein.groups_n_omit_LF_b$PGs%in%protein.groups_n_omit_b$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_b, protein.groups_n_omit_LF_b)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
b2 <- ggplot(protein.groups_n_omit_LF_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta0:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


b4 <- ggplot(protein.groups_n_omit_LF_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
b1 <- ggplot(protein.groups_n_omit_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_bolor_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", A/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


b3 <- ggplot(protein.groups_n_omit_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])



################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("MultiDIA", "Label-Free", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("MultiDIA", "Label-Free", "Intersected"))
library(Cairo)

b_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = "Protein Ratios (A/C)", fill = "", title = expression(paste(italic(H.~sapiens),' excluded')))
b_cover



################################################################
################################################################
########################## d4:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl(paste0(plexDIA_run), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAb]])/as.numeric(protein.groups_n[[pDIAc]])
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl(paste0(LF_runs[2],"|",LF_runs[3]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_runs[2]), Run), "mTRAQ4",
                                            ifelse(grepl(paste0(LF_runs[3]), Run), "mTRAQ8", "mTRAQ0")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAb]])/as.numeric(protein.groups_n[[LFDIAc]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)


#intersect LF and multiDIA
protein.groups_n_omit_c <- protein.groups_n_omit
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF

protein.groups_n_omit_noint <- protein.groups_n_omit_c
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_c

protein.groups_n_omit_c <- protein.groups_n_omit_c[protein.groups_n_omit_c$PGs%in%protein.groups_n_omit_LF_c$PGs,]
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF_c[protein.groups_n_omit_LF_c$PGs%in%protein.groups_n_omit_c$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_c, protein.groups_n_omit_LF_c)

################################# plotting #########################

x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
c2 <- ggplot(protein.groups_n_omit_LF_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta4:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


c4 <- ggplot(protein.groups_n_omit_LF_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
c1 <- ggplot(protein.groups_n_omit_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", B/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


c3 <- ggplot(protein.groups_n_omit_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])


################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

c_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  theme_classic() +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "top",
        legend.direction = "vertical",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = "Protein Ratios (B/C)", fill = "", title = expression(paste(italic(H.~sapiens)," excluded")))
c_cover

############## combine
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(a3)
mylegend1<-g_legend(c_cover)


a_fin <- grid.arrange(mylegend, nrow=2,heights=c(1, 10),
                      arrangeGrob(a_cover,
                                  a1 + theme(legend.position="none"),
                                  a2 + theme(legend.position="none"), 
                                  a3 + theme(legend.position="none"),
                                  a4 + theme(legend.position="none"), nrow = 1, widths=c(1.68,2.25,2,0.7,0.7)))

c_fin <- grid.arrange(nrow=1,
                      arrangeGrob(c_cover + theme(legend.position="none"),
                                  c1 + theme(legend.position="none"),
                                  c2 + theme(legend.position="none"), 
                                  c3 + theme(legend.position="none"),
                                  c4 + theme(legend.position="none"), nrow = 1, widths=c(1.68,2.25,2,0.7,0.7)))

b_fin <- grid.arrange(nrow=1,
                      arrangeGrob(b_cover + theme(legend.position="none"),
                                  b1 + theme(legend.position="none"),
                                  b2 + theme(legend.position="none"), 
                                  b3 + theme(legend.position="none"),
                                  b4 + theme(legend.position="none"), nrow = 1, widths=c(1.68,2.25,2,0.7,0.7)))


figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 5,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1))

ggsave("Figure3abc.pdf", figure_fin, width=11,height=9, dpi=400)




################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########       Figure D: In-set vs Out-of-Set      ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 


dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area","Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))

df_names <- dat %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

df <- dat
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)


df1 <- df[grepl("wJD803", df$Run),]
protein.groups1 <- diann_maxlfq(df1[df1$Lib.Q.Value <= 0.01 & df1$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")

df2 <- df[grepl("wJD804", df$Run),]
protein.groups2 <- diann_maxlfq(df2[df2$Lib.Q.Value <= 0.01 & df2$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")


protein.groups1_new <- protein.groups1
protein.groups1_new[,1] <- protein.groups1[,2]
protein.groups1_new[,2] <- protein.groups1[,1]
protein.groups1_new[,3] <- protein.groups1[,3]
protein.groups1 <-protein.groups1_new


colnames(protein.groups1) <- c("d0", "d4", "d8")
colnames(protein.groups2) <- c("d0", "d4", "d8")

rep1_d0d4_pt <- protein.groups2
PGs <- rownames(rep1_d0d4_pt)
rep1_d0d4_pt <- cbind(rep1_d0d4_pt, PGs)
rep1_d0d4_pt <- data.frame(rep1_d0d4_pt)
rep1_d0d4_pt$Prot_rat <- as.numeric(rep1_d0d4_pt$d0)/as.numeric(rep1_d0d4_pt$d4)
rep1_d0d4_pt <- rep1_d0d4_pt[!is.na(rep1_d0d4_pt$Prot_rat), ]
rep1_d0d4_pt <- rep1_d0d4_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_pt$HY<-F
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("ECOLI",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names)] <-T
table(rep1_d0d4_pt$HY)
rep1_d0d4_pt <- rep1_d0d4_pt[grepl("FALSE", rep1_d0d4_pt$HY),]
rep1_d0d4_pt <- rep1_d0d4_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d4_pt <- rep1_d0d4_pt[!grepl("remove", rep1_d0d4_pt$species),]
med_human <- rep1_d0d4_pt[grepl("H. sapiens", rep1_d0d4_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_pt$Prot_rat <- rep1_d0d4_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_pt$Type <- "Within a run"

######## out of set:
rep1_d0d4_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- cbind(rep1_d0d4_out_pt, PGs)
rep1_d0d4_out_pt <- data.frame(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% dplyr::select("d0", "PGs")
d4_set2 <- rep1_d0d4_pt %>% dplyr::select("d4", "PGs")

rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% inner_join(d4_set2, by =c("PGs" = "PGs"))
rep1_d0d4_out_pt$Prot_rat <- as.numeric(rep1_d0d4_out_pt$d0)/as.numeric(rep1_d0d4_out_pt$d4)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!is.na(rep1_d0d4_out_pt$Prot_rat), ]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_out_pt$HY<-F
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names)] <-T
table(rep1_d0d4_out_pt$HY)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[grepl("FALSE", rep1_d0d4_out_pt$HY),]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!grepl("remove", rep1_d0d4_out_pt$species),]
med_human <- rep1_d0d4_out_pt[grepl("H. sapiens", rep1_d0d4_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_out_pt$Prot_rat <- rep1_d0d4_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_out_pt$Type <- "Across runs"


######################### d0:d8 ##################################

rep1_d0d8_pt <- protein.groups2
PGs <- rownames(rep1_d0d8_pt)
rep1_d0d8_pt <- cbind(rep1_d0d8_pt, PGs)
rep1_d0d8_pt <- data.frame(rep1_d0d8_pt)
rep1_d0d8_pt$Prot_rat <- as.numeric(rep1_d0d8_pt$d0)/as.numeric(rep1_d0d8_pt$d8)
rep1_d0d8_pt <- rep1_d0d8_pt[!is.na(rep1_d0d8_pt$Prot_rat), ]
rep1_d0d8_pt <- rep1_d0d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_pt$HY<-F
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("ECOLI",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names)] <-T
table(rep1_d0d8_pt$HY)
rep1_d0d8_pt <- rep1_d0d8_pt[grepl("FALSE", rep1_d0d8_pt$HY),]
rep1_d0d8_pt <- rep1_d0d8_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("remove", rep1_d0d8_pt$species),]
med_human <- rep1_d0d8_pt[grepl("H. sapiens", rep1_d0d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_pt$Prot_rat <- rep1_d0d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_pt_h <- rep1_d0d8_pt
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("H.", rep1_d0d8_pt$species),]
rep1_d0d8_pt$Type <- "Within a run"

######## out of set:
rep1_d0d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- cbind(rep1_d0d8_out_pt, PGs)
rep1_d0d8_out_pt <- data.frame(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% dplyr::select("d0", "PGs")
d8_set2 <- rep1_d0d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d0d8_out_pt$Prot_rat <- as.numeric(rep1_d0d8_out_pt$d0)/as.numeric(rep1_d0d8_out_pt$d8)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!is.na(rep1_d0d8_out_pt$Prot_rat), ]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_out_pt$HY<-F
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names)] <-T
table(rep1_d0d8_out_pt$HY)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[grepl("FALSE", rep1_d0d8_out_pt$HY),]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("remove", rep1_d0d8_out_pt$species),]
med_human <- rep1_d0d8_out_pt[grepl("H. sapiens", rep1_d0d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_out_pt$Prot_rat <- rep1_d0d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("H.", rep1_d0d8_out_pt$species),]
rep1_d0d8_out_pt$Type <- "Across runs"


######################################## d4:d8 ###########################

rep1_d4d8_pt <- protein.groups2
PGs <- rownames(rep1_d4d8_pt)
rep1_d4d8_pt <- cbind(rep1_d4d8_pt, PGs)
rep1_d4d8_pt <- data.frame(rep1_d4d8_pt)
rep1_d4d8_pt$Prot_rat <- as.numeric(rep1_d4d8_pt$d4)/as.numeric(rep1_d4d8_pt$d8)
rep1_d4d8_pt <- rep1_d4d8_pt[!is.na(rep1_d4d8_pt$Prot_rat), ]
rep1_d4d8_pt <- rep1_d4d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_pt$HY<-F
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("ECOLI",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names)] <-T
table(rep1_d4d8_pt$HY)
rep1_d4d8_pt <- rep1_d4d8_pt[grepl("FALSE", rep1_d4d8_pt$HY),]
rep1_d4d8_pt <- rep1_d4d8_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("remove", rep1_d4d8_pt$species),]
med_human <- rep1_d4d8_pt[grepl("H. sapiens", rep1_d4d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_pt$Prot_rat <- rep1_d4d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_pt_h <- rep1_d4d8_pt
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("H.", rep1_d4d8_pt$species),]
rep1_d4d8_pt$Type <- "Within a run"

######## out of set:
rep1_d4d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- cbind(rep1_d4d8_out_pt, PGs)
rep1_d4d8_out_pt <- data.frame(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% dplyr::select("d4", "PGs")
d8_set2 <- rep1_d4d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d4d8_out_pt$Prot_rat <- as.numeric(rep1_d4d8_out_pt$d4)/as.numeric(rep1_d4d8_out_pt$d8)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!is.na(rep1_d4d8_out_pt$Prot_rat), ]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_out_pt$HY<-F
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names)] <-T
table(rep1_d4d8_out_pt$HY)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[grepl("FALSE", rep1_d4d8_out_pt$HY),]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("remove", rep1_d4d8_out_pt$species),]
med_human <- rep1_d4d8_out_pt[grepl("H. sapiens", rep1_d4d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_out_pt$Prot_rat <- rep1_d4d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("H.", rep1_d4d8_out_pt$species),]
rep1_d4d8_out_pt$Type <- "Across runs"



###############
rep1_d0d4_pt <- rep1_d0d4_pt[rep1_d0d4_pt$PGs%in%rep1_d0d4_out_pt$PGs,]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[rep1_d0d4_out_pt$PGs%in%rep1_d0d4_pt$PGs,]

rep1_d0d8_pt <- rep1_d0d8_pt[rep1_d0d8_pt$PGs%in%rep1_d0d8_out_pt$PGs,]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[rep1_d0d8_out_pt$PGs%in%rep1_d0d8_pt$PGs,]

rep1_d4d8_pt <- rep1_d4d8_pt[rep1_d4d8_pt$PGs%in%rep1_d4d8_out_pt$PGs,]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[rep1_d4d8_out_pt$PGs%in%rep1_d4d8_pt$PGs,]

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))
df2<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(1/6), log2(6)))


d0d4 <- bind_rows(rep1_d0d4_pt, rep1_d0d4_out_pt)
d0d8 <- bind_rows(rep1_d0d8_pt, rep1_d0d8_out_pt)
d4d8 <- bind_rows(rep1_d4d8_pt, rep1_d4d8_out_pt)

d0d4 <- d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8 <- d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8 <- d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4$error <- abs(log2(d0d4$Prot_rat)-d0d4$Z)
d0d8$error <- abs(log2(d0d8$Prot_rat)-d0d8$Z)
d4d8$error <- abs(log2(d4d8$Prot_rat)-d4d8$Z)

all <- rbind(d0d4, d4d8, d0d8)
####

####


y_d0d4 <- all$error %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,0.99)))

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))

all$Type <- factor(all$Type, levels=c("Within a run", "Across runs"))
library(Cairo)


D_fig <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "") + ylim(y_d0d4_val[1],y_d0d4_val[2])

ggsave("Figure3d.pdf", width=3.7,height=3.7)

################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########        Figure E: R vs K quant MS2         ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 

dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area",
                               "Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))

ev <- dat[grepl("wJD804", dat$Run),]
ev$seqcharge <- paste0(ev$Modified.Sequence, ev$Precursor.Charge)
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ0\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ4\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ8\\)")
ev$seqcharge_file <- paste0(ev$seqcharge,ev$Run)
ev2_lim <- ev %>% mutate("label" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                          ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

########################### FIGURE E Cont. K vs R quant (MS2)
ev2_0 <- ev2_lim[grepl("mTRAQ0",ev2_lim$Precursor.Id),]
ev2_4 <- ev2_lim[grepl("mTRAQ4",ev2_lim$Precursor.Id),] %>% dplyr::select("seqcharge","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_4) <- c("seqcharge_d4", "seqcharge_file","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4")
ev2_8 <- ev2_lim[grepl("mTRAQ8",ev2_lim$Precursor.Id),] %>% dplyr::select("seqcharge","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_8) <- c("seqcharge_d8", "seqcharge_file","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8")
#
rep1_d0d4 <- MS2_t_mTRAQ_d0d4(ev2_0,ev2_4)
rep1_d0d8 <- MS2_t_mTRAQ_d0d8(ev2_0,ev2_8)
rep1_d0d4_MS1 <- MS1_mTRAQ_d0d4(ev2_0,ev2_4)
rep1_d0d8_MS1 <- MS1_mTRAQ_d0d8(ev2_0,ev2_8)


ev2_4 <- ev2_lim[grepl("mTRAQ4",ev2_lim$Precursor.Id),] %>% dplyr::select("Run","seqcharge","Protein.Names","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_4) <- c("Run","seqcharge", "Protein.Names","seqcharge_file","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4")
rep1_d4d8 <- MS2_t_mTRAQ_d4d8(ev2_4,ev2_8)
rep1_d4d8_MS1 <- MS1_mTRAQ_d4d8(ev2_4,ev2_8)



df1<- data.frame(species = c("ECOLI", "HUMAN", "YEAST"), Z = c(2, 0, -1))
df2<- data.frame(species = c("ECOLI", "YEAST"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("ECOLI", "YEAST"), Z = c(log2(1/6), log2(6)))

d0d4_MS2 <- rep1_d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8_MS2 <- rep1_d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8_Ms2 <- rep1_d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4_MS1 <- rep1_d0d4_MS1 %>% left_join(df1, by =c("species" = "species"))
d0d8_MS1 <- rep1_d0d8_MS1 %>% left_join(df2, by =c("species" = "species"))
d4d8_Ms1 <- rep1_d4d8_MS1 %>% left_join(df3, by =c("species" = "species"))


#### intersect
d0d4_MS2 <- d0d4_MS2[d0d4_MS2$seqcharge%in%d0d4_MS1$seqcharge,]
d0d4_MS1 <- d0d4_MS1[d0d4_MS1$seqcharge%in%d0d4_MS2$seqcharge,]

d0d8_MS2 <- d0d8_MS2[d0d8_MS2$seqcharge%in%d0d8_MS1$seqcharge,]
d0d8_MS1 <- d0d8_MS1[d0d8_MS1$seqcharge%in%d0d8_MS2$seqcharge,]

d4d8_Ms2 <- d4d8_Ms2[d4d8_Ms2$seqcharge%in%d4d8_Ms1$seqcharge,]
d4d8_Ms1 <- d4d8_Ms1[d4d8_Ms1$seqcharge%in%d4d8_Ms2$seqcharge,]

all <- list(d0d4_MS2, d0d8_MS2, d4d8_Ms2, d0d4_MS1, d0d8_MS1, d4d8_Ms1)

#normalize the data and remove PGs which map to multiple species
for(i in 1:6){
  temp <- all[[i]]
  temp$HY<-F
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names) & grepl("ECOLI",temp$Protein.Names)] <-T
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("ECOLI",temp$Protein.Names)] <-T
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names)] <-T
  temp$HY[grepl("ECOLI",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names)] <-T
  table(temp$HY)
  temp <- temp[grepl("FALSE", temp$HY),]
  med_human <- temp[grepl("HUMAN", temp$species),]
  med_human <- median(med_human$d0_d4_rat)
  scalar <- 1/med_human
  temp$d0_d4_rat <- temp$d0_d4_rat*scalar   #normalize human median on 1:1 ratio
  all[[i]] <- temp
}


d0d4_MS2 <- all[[1]]
d0d8_MS2 <- all[[2]]
d4d8_Ms2 <- all[[3]]

d0d4_MS1 <- all[[4]]
d0d8_MS1 <- all[[5]]
d4d8_Ms1 <- all[[6]]


all_MS2 <- bind_rows(d0d4_MS2, d0d8_MS2, d4d8_Ms2)

all_MS1 <- bind_rows(d0d4_MS1, d0d8_MS1, d4d8_Ms1)




#####
all_MS2$aa <- str_sub(all_MS2$Stripped.Sequence, -1)  #get C-terminal amino acid
all_MS2_K <- all_MS2[grepl("K", all_MS2$aa),]
all_MS2_R <- all_MS2[grepl("R", all_MS2$aa),]

all_MS2_K$Type <- "Lys precursors"
all_MS2_K <- all_MS2_K %>% add_count(species)
all_MS2_K$cond <- paste0("n=", all_MS2_K$n)

all_MS2_R$Type <- "Arg precursors"
all_MS2_R <- all_MS2_R %>% add_count(species)
all_MS2_R$cond <- paste0("n=", all_MS2_R$n)


####
all_MS1$Type <- "All precursors"
all_MS1 <- all_MS1 %>% add_count(species)
all_MS1$cond <- paste0("n=", all_MS1$n)











K_vs_R_MS1_mDIA_d0d4 <- bind_rows(all_MS2_K, all_MS2_R, all_MS1)
K_vs_R_MS1_mDIA_d0d4$error <- abs(log2(K_vs_R_MS1_mDIA_d0d4$d0_d4_rat)-K_vs_R_MS1_mDIA_d0d4$Z)

y_d0d4 <- log2(K_vs_R_MS1_mDIA_d0d4$error) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,.999)))


############ add in the MS1 quant for multiDIA precursors:
K_vs_R_MS1_mDIA_d0d4 <- K_vs_R_MS1_mDIA_d0d4 %>% mutate("species" = ifelse(grepl("ECOLI", species), "E. coli", 
                                                                           ifelse(grepl("HUMAN", species), "H. sapiens",
                                                                                  ifelse(grepl("YEAST", species), "S. cerevisiae", "unknown"))))

K_vs_R_MS1_mDIA_d0d4 <- K_vs_R_MS1_mDIA_d0d4[!grepl("unknown", K_vs_R_MS1_mDIA_d0d4$species),]
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% distinct(cond,.keep_all=T)
df$species<- factor(df$species)

df_n <- K_vs_R_MS1_mDIA_d0d4 %>% distinct(cond,.keep_all=T)
# df_n[1,1] <- 1.3
# df_n[2,1] <- 0.7
# df_n[3,1] <- 4.5
# df_n[4,1] <- 1.3
# df_n[5,1] <- 0.7
# df_n[6,1] <- 4.5

library(viridis)
K_vs_R_MS1_mDIA_d0d4$Type <- factor(K_vs_R_MS1_mDIA_d0d4$Type, levels=c("Lys precursors", "Arg precursors", "All precursors"))
library(Cairo)

E_fig <- ggplot(K_vs_R_MS1_mDIA_d0d4, aes(x=as.factor(Type),y=error, fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  scale_x_discrete(limits=c("Lys precursors","Arg precursors", "All precursors")) + 
  scale_fill_manual(values = c("dodgerblue3", "deepskyblue", "tan2")) +
  theme_classic() +
  facet_grid(~species, scales = "free") +
  ylim(0,y_d0d4_val[2]) + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 16, hjust = 0.5, face="bold"),
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y=expression(paste("|",Log["2"],", Precursor Ratio Error|")), fill = "",subtitle="",title="")
E_fig
ggsave("Figure3e.pdf", width=5.7,height=3.7)


library(gridExtra)
library(ggpubr)
library(gridGraphics)


others <- ggarrange(NULL, D_fig, NULL, E_fig, NULL, nrow = 1, widths=c(0.6,3,0.3,4.5,0.2))

others

figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin, NULL, others,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 7,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1,0.07,1.25))

ggsave("Figure3abcde.pdf", figure_fin, width=11,height=12.2, dpi=500)


```


Figure 4: Quantitative reproducibility (CVs and correlation to LF)
```{r}
dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated","Ms1.Area", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))
dat <- dat[grepl("wJD803|wJD804|wJD815", dat$Run),]

df <- dat
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- robustbase::colMedians(protein.groups, na.rm=T)
protein.groups_n_plex <- sweep(protein.groups, 2, cmeds, FUN = '/')

PGs <- rownames(protein.groups)
protein.groups_n_plex <- cbind(protein.groups_n_plex,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n_plex <- data.frame(protein.groups_n_plex)
protein.groups_n_plex <- protein.groups_n_plex %>% inner_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n_plex$HY<-F
protein.groups_n_plex$HY[grepl("HUMAN",protein.groups_n_plex$Protein.Names) & grepl("YEAST",protein.groups_n_plex$Protein.Names) & grepl("ECOLI",protein.groups_n_plex$Protein.Names)] <-T
protein.groups_n_plex$HY[grepl("HUMAN",protein.groups_n_plex$Protein.Names) & grepl("ECOLI",protein.groups_n_plex$Protein.Names)] <-T
protein.groups_n_plex$HY[grepl("HUMAN",protein.groups_n_plex$Protein.Names) & grepl("YEAST",protein.groups_n_plex$Protein.Names)] <-T
protein.groups_n_plex$HY[grepl("ECOLI",protein.groups_n_plex$Protein.Names) & grepl("YEAST",protein.groups_n_plex$Protein.Names)] <-T
table(protein.groups_n_plex$HY)
protein.groups_n_plex <- protein.groups_n_plex[grepl("FALSE", protein.groups_n_plex$HY),]

protein.groups_n_plex <- protein.groups_n_plex%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n_plex <- protein.groups_n_plex[!grepl("remove", protein.groups_n_plex$species),]



######################## LF ########################################

dat_LF <- fread(LF_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                                     "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value","PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                                     "Mass.Evidence", "CScore", "Fragment.Correlations"))

df <- dat_LF
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD757_re|wJD756_re|wJD758_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD759_re|wJD760_re|wJD761_re", Run), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- colMedians(protein.groups, na.rm=T)
protein.groups_n <- sweep(protein.groups, 2, cmeds, FUN = '/')

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

## intersect plexDIA and LF-DIA ###
protein.groups_n <- protein.groups_n[protein.groups_n$PGs%in%protein.groups_n_plex$PGs,]
protein.groups_n_plex <- protein.groups_n_plex[protein.groups_n_plex$PGs%in%protein.groups_n$PGs,]

rownames(protein.groups_n) <- protein.groups_n$PGs
rownames(protein.groups_n_plex) <- protein.groups_n_plex$PGs



####### plex cont.

protein.groups_n_plex <- protein.groups_n_plex %>% dplyr::select(-matches(c("PGs", "species", "HY", "Protein.Names")))
m_PGs <- melt(as.matrix(protein.groups_n_plex))
m_PGs <- m_PGs %>% mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                                                       ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs <- m_PGs[!is.na(m_PGs$value),]
m_PGs <- m_PGs %>% group_by(label, Var1) %>% add_count() %>% filter(n==3) %>% ungroup()
  
m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("Prot_CV" = prot_sd/prot_mean) %>% ungroup
m_PGs$uni <- paste0(m_PGs$Var1,m_PGs$label)
m_PGs <- m_PGs %>% distinct(uni, .keep_all=T)
m_PGs$Cond <- "plexDIA"


###### LF cont.
protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),] %>% dplyr::select(-matches(c("PGs","species", "HY", "Protein.Names")))

m_PGs_LF <- reshape2::melt(as.matrix(protein.groups_n))
m_PGs_LF <- m_PGs_LF %>% mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                           ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs_LF <- m_PGs_LF[!is.na(m_PGs_LF$value),]
m_PGs_LF <- m_PGs_LF %>% group_by(label, Var1) %>% dplyr::add_count() %>% filter(n==3) %>% ungroup()

m_PGs_LF <- m_PGs_LF %>% group_by(label, Var1) %>% mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% group_by(label, Var1) %>% mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% group_by(label, Var1) %>% mutate("Prot_CV" = prot_sd/prot_mean) %>% ungroup
m_PGs_LF$uni <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs_LF <- m_PGs_LF %>% distinct(uni, .keep_all=T)
m_PGs_LF$Cond <- "LF-DIA"

m_PGs_LF$Var1_lab <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs$Var1_lab <- paste0(m_PGs$Var1,m_PGs$label)

## last intersection:
m_PGs_LF <- m_PGs_LF[m_PGs_LF$Var1_lab%in%m_PGs$Var1_lab,]
m_PGs <- m_PGs[m_PGs$Var1_lab%in%m_PGs_LF$Var1_lab,]

##
################ select and bind:

mDIA_LF <- bind_rows(m_PGs,m_PGs_LF)
mDIA_LF <- mDIA_LF %>% group_by(Cond) %>% add_count()

mDIA_LF$Cond <- factor(mDIA_LF$Cond, levels=c("plexDIA", "LF-DIA"))
df1 <- mDIA_LF %>% mutate("med_CV" = round(median(Prot_CV,na.rm=T),3)) %>% distinct(Cond,.keep_all=T)

ymax <- mDIA_LF$Prot_CV %>% sort()
ymax <- as.numeric(quantile(ymax,probs=c(.01,.99)))

cvs <- df1$med_CV

cvs <- c("0.100", "0.108") #was getting an error in dynamically adding medians to plot... so here's a hack

CV1_fig <- ggplot(mDIA_LF, aes(x=Cond,y=Prot_CV, fill=Cond)) + geom_boxplot( alpha =c(0.6,0.4), outlier.shape=NA) +
  scale_fill_manual(values = c("red", "gray8")) +
  theme_classic() +
  geom_text(data=df1, aes(x=Cond, y=c(0.128,0.135), label=paste0("med = ",cvs)), parse = FALSE, size=4) +
  # facet_grid(~species, scales = "free") + geom_hline(data = df1, aes(yintercept = Z), color = "red", size = 1.052, linetype = "dashed") +
  theme(axis.text.x = element_text(size=14, face = "bold", color = "black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=14),
        axis.title.x = element_text(size=14),
        
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Sample-specific protein CV", fill = "") + ylim(0,ymax[2])


CV1_fig
ggsave("Figure4a_Ms1Area.pdf", width=3.4,height=3.2)


########################################################################################
########################################################################################
############################ Protein differential abundance ############################
########################################################################################
########################################################################################
data_prec <- diann_load(INPUT)
data_prec <- data_prec[,!grepl("raw", colnames(data_prec))]
pr <- data_prec
cnames <- colnames(data_prec)
LFINPUT <- "/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/dat/multiDIA/v1.8b7_10102021/LF_MS1/Report.tsv"
######################### LF:

LF <- fread(LFINPUT, select = c(cnames,"Run","Lib.PG.Q.Value","Ms1.Area"))
keep_LF <- LF
plex <- keep_LF
plex <- plex[which(plex$Lib.PG.Q.Value<0.01),]

# remove any proteins groups with >1 species
plex$HY<-F
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
plex$HY[grepl("ECOLI",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
table(plex$HY)
plex <- plex[grepl("FALSE", plex$HY),]
#######

plex <- plex %>% mutate("channel_name" = ifelse(grepl("wJD759|wJD760|wJD761", Run), "mTRAQ0",
                                            ifelse(grepl("wJD762|wJD763|wJD764", Run), "mTRAQ4", "mTRAQ8")))
plex$Unlabelled.Precursor <- paste0(plex$Modified.Sequence, plex$Precursor.Charge)
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ0\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ4\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ8\\)")
plex$precRun <- paste0(plex$Unlabelled.Precursor, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] #%>% add_count(precRun) %>% filter(n==2)
plex <- plex %>% filter(grepl("HUMAN", Protein.Names))
h <- plex %>% filter(grepl("HUMAN", Protein.Names)) %>% group_by(chanRun) %>% mutate(med_MS1 = median(Ms1.Area)) %>% 
  distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area>0),]
plex <- plex %>% mutate(Ms1.Area_cnorm = Ms1.Area/med_MS1)
plex <- plex %>% group_by(Unlabelled.Precursor) %>% mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("Unlabelled.Precursor", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group)
plex1 <- plex1 %>% group_by(channel_name, Protein.Group) %>% mutate("med_pep_ab" = median(Ms1.Area))%>% ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),] 

prots_LF <- plex1 %>% distinct(Protein.Group, .keep_all=T)
prots_LF <- prots_LF[prots_LF$Protein.Group%in%int$Protein.Group,]
prots_LF$pval <- 10000
unique_prots <- as.vector(prots_LF$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots_LF[i,7] <- as.numeric(pval_temp)
  
}

prots_LF$FC_qval <-  p.adjust(prots_LF$pval, method = 'BH') # qvalue
prots_LF <- prots_LF%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

write.table(prots_LF, "U_J_DA_LF_10182021.txt", sep = "\t", row.names = FALSE)

################### use this list of DA proteins to compare quant of LF-DIA and plexDIA


df <- dat
df <- df[grepl(paste0("wJD803"), df$Run),]
df <- df[(df$Translated.Q.Value <= tqval),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD803mTRAQ4)/as.numeric(protein.groups_n$wJD803mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[grepl("H.", protein.groups_n_omit$species),]

p_1 <- protein.groups_n_omit
#######

df <- dat
df <- df[grepl(paste0("wJD804"), df$Run),]
df <- df[(df$Translated.Q.Value <= tqval),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD804mTRAQ4)/as.numeric(protein.groups_n$wJD804mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[grepl("H.", protein.groups_n_omit$species),]

p_2 <- protein.groups_n_omit

###
df <- dat
df <- df[grepl(paste0("wJD815"), df$Run),]
df <- df[(df$Translated.Q.Value <= tqval),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD815mTRAQ4)/as.numeric(protein.groups_n$wJD815mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[grepl("H.", protein.groups_n_omit$species),]

p_3 <- protein.groups_n_omit
  
######################## LF ########################################

dat_LF <- fread(LF_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                                     "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value","PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                                     "Mass.Evidence", "CScore", "Fragment.Correlations"))
#evt <- read.delim(fpath)

#dat_1hr <- dat
df <- dat_LF
df <- df[grepl("wJD759_re|wJD762_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD759_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD762_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD759_remTRAQ0)/as.numeric(protein.groups_n$wJD762_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[grepl("H.", protein.groups_n_omit_LF$species),]

LF_1 <- protein.groups_n_omit_LF
#####

#dat_1hr <- dat
df <- dat_LF
df <- df[grepl("wJD760_re|wJD763_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD760_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD763_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD760_remTRAQ0)/as.numeric(protein.groups_n$wJD763_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[grepl("H.", protein.groups_n_omit_LF$species),]

LF_2 <- protein.groups_n_omit_LF

#######

#dat_1hr <- dat
df <- dat_LF
df <- df[grepl("wJD761_re|wJD764_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD761_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD764_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
df <- df[!grepl("YEAST|ECOLI", df$Protein.Names),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD761_remTRAQ0)/as.numeric(protein.groups_n$wJD764_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[grepl("H.", protein.groups_n_omit_LF$species),]

LF_3 <- protein.groups_n_omit_LF

LF_all <- bind_rows(LF_1, LF_2, LF_3) %>% group_by(PGs) %>%
  mutate("Prot_rat" = median(d0_d4)) %>% ungroup() %>% distinct(PGs, .keep_all=T) %>%
   dplyr::select( "Prot_rat", "PGs")

p_all <- bind_rows(p_1, p_2, p_3) %>% group_by(PGs) %>%
  mutate("Prot_rat" = median(d0_d4)) %>% ungroup() %>% distinct(PGs, .keep_all=T) %>%
   dplyr::select( "Prot_rat", "PGs")

################ intersect LF and multiDIA

protein.groups_n_omit_c <- p_all[p_all$PGs%in%LF_all$PGs,]
protein.groups_n_omit_LF_c <- LF_all[LF_all$PGs%in%p_all$PGs,]

##############
difprots <- prots_LF  #DE is the differentially expressed table we generated ~150 lines above

difprots <- difprots %>% dplyr::select("Protein.Group", "FC_qval")

protein.groups_n_omit <- protein.groups_n_omit_c %>% select("PGs", "Prot_rat") %>% ungroup()
protein.groups_n_omit$Prot_rat <- as.numeric(protein.groups_n_omit$Prot_rat)
protein.groups_unplex <- protein.groups_n_omit_LF_c %>% select("PGs", "Prot_rat") %>% ungroup()
colnames(protein.groups_unplex) <- c("Protein.Group", "Prot_rat_LF")

j_multiDIA_LF_human <- protein.groups_n_omit %>% inner_join(protein.groups_unplex, by =c("PGs" = "Protein.Group"))

j_multiDIA_LF_human <- j_multiDIA_LF_human %>% inner_join(difprots, by =c("PGs" = "Protein.Group"))

j_multiDIA_LF_human <- j_multiDIA_LF_human[j_multiDIA_LF_human$PGs%in%difprots$Protein.Group,]

j_multiDIA_LF_human$l2_prot <- log2(j_multiDIA_LF_human$Prot_rat)
j_multiDIA_LF_human$l2_prot_LF <- log2(j_multiDIA_LF_human$Prot_rat_LF)

plexDIA_LFDIA_cor <- cor(j_multiDIA_LF_human$l2_prot, j_multiDIA_LF_human$l2_prot_LF, method="spearman")
DE_m_LF <- j_multiDIA_LF_human[which(j_multiDIA_LF_human$FC_qval < 0.01),]
plexDIA_LFDIA_cor_DE <- cor(DE_m_LF$l2_prot, DE_m_LF$l2_prot_LF, method="spearman")


ggplot(j_multiDIA_LF_human, aes(x=log2(Prot_rat), y=log2(Prot_rat_LF))) +
  geom_point(data = j_multiDIA_LF_human %>% filter(FC_qval>=0.01), alpha=0.2)+
#  geom_point(data = j_multiDIA_LF_human %>% filter(FC_qval<0.01), alpha=0.4, color="red")+
  geom_pointdensity(data = j_multiDIA_LF_human %>% filter(FC_qval<0.01), aes(x=log2(Prot_rat), y=log2(Prot_rat_LF)), alpha=1)+
scale_colour_gradientn(
  colors =c("purple2","steelblue2", "springgreen3", "yellow", "mediumvioletred"),
  values = NULL,
  space = "Lab",
  na.value = "grey50",
  guide = "colourbar",
  aesthetics = "colour",
  colors
) +

  theme_classic() + theme(legend.position = "none") + 
  theme(axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14))+
  labs(x = expression(paste(Log["2"],", U937/Jurkat")), y=expression(paste(Log["2"],", U937/Jurkat")),
       colors = "dfd")+ ylim(-4,5) + xlim(-4,5) +
  geom_abline(slope=1, linetype = "dashed")

ggsave("Figure4b_10182021.pdf", width=3.7,height=3.7, dpi=500)
nrow(DE_m_LF[which(DE_m_LF$FC_qval<0.01),])
nrow(j_multiDIA_LF_human)



################# Empirical FDR #################
library(PRROC)
LFINPUT <- "/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/dat/multiDIA/v1.8b7_10102021/LF_MS1/Report.tsv"
fpath <- "/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/dat/multiDIA/v1.8b7_10102021/plexDIA_MS1/Report.tsv" #file path

plex <- fread(fpath, select = c(cnames,"Run","Lib.PG.Q.Value","Ms1.Area"))
keep <- plex
plex <- keep
plex <- plex[!grepl("wJD803|wJD805|wJD807|wJD815", plex$Run),]
plex <- plex[which(plex$Lib.PG.Q.Value<0.01),]

# remove any proteins groups with >1 species
plex$HY<-F
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
plex$HY[grepl("ECOLI",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
table(plex$HY)
plex <- plex[grepl("FALSE", plex$HY),]
#######
plex1 <- plex[grepl("803|804", plex$Run),]
plex2 <- plex[grepl("805|806", plex$Run),]
plex3 <- plex[grepl("807|808", plex$Run),]

plex1 <- plex1 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

plex2 <- plex2 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ8",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ0", "mTRAQ4")))

plex3 <- plex3 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ4",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ8", "mTRAQ0")))

plex <- rbind(plex1,plex2,plex3)

plex$Unlabelled.Precursor <- paste0(plex$Modified.Sequence, plex$Precursor.Charge)
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ0\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ4\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ8\\)")
plex$precRun <- paste0(plex$Unlabelled.Precursor, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] %>% add_count(precRun) %>% filter(n==2)
h <- plex %>% filter(grepl("HUMAN", Protein.Names)) %>% group_by(chanRun) %>% mutate(med_MS1 = median(Ms1.Area)) %>% 
  distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area>0),]
plex <- plex %>% mutate(Ms1.Area_cnorm = Ms1.Area/med_MS1)
plex <- plex %>% group_by(Unlabelled.Precursor) %>% mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("Unlabelled.Precursor", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")

d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group)
plex1 <- plex1 %>% group_by(channel_name, Protein.Group) %>% mutate("med_pep_ab" = median(Ms1.Area)) %>% ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),]

prots <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots <- prots[prots$Protein.Group%in%int$Protein.Group,]
prots$pval <- 10000
unique_prots <- as.vector(prots$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots[i,7] <- as.numeric(pval_temp)
  
}

prots$FC_qval <-  prots$pval #p.adjust(prots$pval, method = 'BH') # qvalue
prots <- prots%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))


######################### LF:

LF <- fread(LFINPUT, select = c(cnames,"Run","Lib.PG.Q.Value","Ms1.Area"))
keep_LF <- LF
plex <- keep_LF
#plex <- plex[grepl("wJD804", plex$Run),]
plex <- plex[which(plex$Lib.PG.Q.Value<0.01),]

# remove any proteins groups with >1 species
plex$HY<-F
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("ECOLI",plex$Protein.Names)] <-T
plex$HY[grepl("HUMAN",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
plex$HY[grepl("ECOLI",plex$Protein.Names) & grepl("YEAST",plex$Protein.Names)] <-T
table(plex$HY)
plex <- plex[grepl("FALSE", plex$HY),]
#######

plex <- plex %>% mutate("channel_name" = ifelse(grepl("wJD756|wJD757|wJD758", Run), "mTRAQ0",
                                            ifelse(grepl("wJD759|wJD760|wJD761", Run), "mTRAQ4", "mTRAQ8")))
plex$Unlabelled.Precursor <- paste0(plex$Modified.Sequence, plex$Precursor.Charge)
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ0\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ4\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ8\\)")
plex$precRun <- paste0(plex$Unlabelled.Precursor, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] #%>% add_count(precRun) %>% filter(n==2)
h <- plex %>% filter(grepl("HUMAN", Protein.Names)) %>% group_by(chanRun) %>% mutate(med_MS1 = median(Ms1.Area)) %>% 
  distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area>0),]
plex <- plex %>% mutate(Ms1.Area_cnorm = Ms1.Area/med_MS1)
plex <- plex %>% group_by(Unlabelled.Precursor) %>% mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("Unlabelled.Precursor", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group)
plex1 <- plex1 %>% group_by(channel_name, Protein.Group) %>% mutate("med_pep_ab" = median(Ms1.Area))%>% ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),] 

prots_LF <- plex1 %>% distinct(Protein.Group, .keep_all=T)
prots_LF <- prots_LF[prots_LF$Protein.Group%in%int$Protein.Group,]
prots_LF$pval <- 10000
unique_prots <- as.vector(prots_LF$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots_LF[i,7] <- as.numeric(pval_temp)
  
}

prots_LF$FC_qval <-  prots_LF$pval #p.adjust(prots_LF$pval, method = 'BH') # qvalue
prots_LF <- prots_LF%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))


##### plot PR curves:
DE_LF_lim <- na.omit(prots_LF)#[which(DE_LF$n>2),]
DE_lim <- na.omit(prots)#[which(DE$n>2),]


### LF DIA
neg <- DE_LF_lim[grepl("HUMAN", DE_LF_lim$Protein.Names),]
neg <- 1/(as.vector(neg$FC_qval))

pos <- DE_LF_lim[!grepl("HUMAN", DE_LF_lim$Protein.Names),]
pos <- 1/(as.vector(pos$FC_qval))

pr<-pr.curve(scores.class0 = pos, scores.class1 = neg,  curve=T)
pr
plot(pr)


##### plexDIA
neg_plex <- DE_lim[grepl("HUMAN", DE_lim$Protein.Names),]
neg_plex <- 1/as.vector(neg_plex$FC_qval)

pos_plex <- DE_lim[!grepl("HUMAN", DE_lim$Protein.Names),]
pos_plex <- 1/as.vector(pos_plex$FC_qval)

pr_plex<-pr.curve(scores.class0 = pos_plex, scores.class1 = neg_plex,  curve=T)
pr_plex

plot(pr_plex)

####
plot(pr_plex, max.plot = TRUE, min.plot = TRUE, rand.plot = T, fill.area = F, color=3, auc.main = T)

plot(pr, add = TRUE, color = 1)


####### remove human
prots_LF_YE <- prots_LF[!grepl("H. sapiens",prots_LF$species),]
prots_YE <- prots[!grepl("H. sapiens",prots$species),]

#####
plexdia <- data.frame(pr_plex$curve)
plexdia$numprots <- nrow(prots_YE)*(plexdia$X1)# - (1-plexdia$X2)*nrow(prots)*(plexdia$X1) #total proteins - human proteins = YE proteins

lfdia <- data.frame(pr$curve)
lfdia$numprots <- nrow(prots_LF_YE)*(lfdia$X1)# - (1-lfdia$X2)*nrow(prots_LF)*(lfdia$X1) #total proteins - human proteins = YE proteins


plexdia$type <- "plexDIA"
lfdia$type <- "LF-DIA"

both <- rbind(plexdia, lfdia)

ggplot(both, aes(x=100*(1-X2), y=as.numeric(numprots), colour=type)) + geom_point() + 
  labs(x = "Empirical FDR(%)", y = "Protein groups (true positives)", color="") +
  theme_bw()+
    theme(legend.position = "bottom",
        legend.text = element_text(size=16),
        axis.title.x = element_text(size=16),
        axis.title.y = element_text(size=16),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14)) + 
      scale_color_manual(values = c("black", "red"))
#ggsave("Prots_FDR_10112021.pdf",width=5,height=5)


#make it so the lines only increases (no jagged-ness)
b <- seq(0, 1, 0.001)
d <- data.frame(b)
d$max_prot <- 0

plexdia1 <- plexdia
plexdia1$FDR <- 1-plexdia1$X2
for(i in 1:nrow(d)){
  a <- d[i,1]
  temp <- plexdia1[which(plexdia1$FDR<=a),]
  m_temp <- max(temp$numprots)
  d[i,2] <- as.numeric(m_temp)
  
}

d$type <- "plexDIA"

b <- seq(0, 1, 0.001)
lf <- data.frame(b)
lf$max_prot <- 0

lfdia1 <- lfdia
lfdia1$FDR <- 1-lfdia1$X2
for(i in 1:nrow(lf)){
  a <- lf[i,1]
  temp <- lfdia1[which(lfdia1$FDR<=a),]
  m_temp <- max(temp$numprots)
  lf[i,2] <- as.numeric(m_temp)
  
}


lf$type <- "LF-DIA"

both1 <- rbind(d,lf)




ggplot(both1, aes(x=100*b, y=as.numeric(max_prot), colour=type)) + geom_line(size=3.9, alpha=0.6)+#alpha=0.025) + 
  labs(x = "Empirical FDR (%)", y = ("Differentially abundant \nprotein groups (true positives)"), color="") +
  theme_bw() + xlim(0,5) +
      theme(legend.position = c(0.2,0.88),
        legend.text = element_text(size=20),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        axis.text.x = element_text(size=16),
        axis.text.y = element_text(size=16)) + ylim(0,1400)+
  scale_color_manual(values = c("gray8", "red")) #+ geom_smooth(method = "loess", method.args = list(family = "symmetric"), size=1.25)
ggsave("Figure4c.pdf",width=5,height=4.5)

```


Figure5 version 2: Ms1 and MS2 cell cycle
```{r}
df_CC <- read.delim(CC_data)
fil <- df_CC[which(df_CC$Lib.PG.Q.Value<0.01),]
length(unique(fil$Protein.Group))
con <- c("P04745", "P13645", "P35527","P04264","P35908","Q15323","Q14532","O76011","Q92764","O76013","O76014","O76015","O76009","Q14525","Q14533","Q9NSB4","P78385","Q9NSB2","P78386","O43790")

df <- df_CC[which(df_CC$Protein.Group%!in%con),]

df <- df_CC[grepl("eJD903_re", df_CC$Run),]
df <- df[which(df$Translated.Q.Value <= tqval),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups <- protein.groups[complete.cases(protein.groups),]
protein.groups[,1] <- protein.groups[,1]/median(protein.groups[,1], na.rm=T)
protein.groups[,2] <- protein.groups[,2]/median(protein.groups[,2], na.rm=T)
protein.groups[,3] <- protein.groups[,3]/median(protein.groups[,3], na.rm=T)

rmeans <- rowMeans(protein.groups, na.rm=T)
protein.groupsx <- protein.groups/rmeans
protein.groups_n <- protein.groupsx                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groupsx)
protein.groups_n <- as.data.frame(cbind(protein.groups_n,PGs))

PGs_G1 <- protein.groups_n %>% dplyr::select("eJD903_remTRAQ0", "PGs")
PGs_G1$Condition <- "G1"
colnames(PGs_G1) <- c("Intensity", "Protein", "Condition")
PGs_S <- protein.groups_n %>% dplyr::select("eJD903_remTRAQ4", "PGs")
PGs_S$Condition <- "S"
colnames(PGs_S) <- c("Intensity", "Protein", "Condition")
PGs_G2 <- protein.groups_n %>% dplyr::select("eJD903_remTRAQ8", "PGs")
PGs_G2$Condition <- "G2"
colnames(PGs_G2) <- c("Intensity", "Protein", "Condition")

genes <- df %>% dplyr::select("Protein.Group", "Genes") %>% distinct(Protein.Group, .keep_all=T)
#------------
# Data import
#------------
# Your data should be formatted in terms of a melted list with the following characteristics:
# 1. Condition column
# 2. Relative Protein Abundance Column
# 3. Column of Uniprot IDs
# 4. (optional) Column of Genes
#
# Make sure to title the relevant columns: [Protein, Intensity, Condition, Gene]

DataPath <- rbind(PGs_G1, PGs_S, PGs_G2)
DataPath$Protein <- str_extract(DataPath$Protein,"[^-]+") 
DataPath <- DataPath %>% left_join(genes, by = c("Protein" = "Protein.Group"))
colnames(DataPath) <- c("Intensity", "Protein", "Condition", "Gene")
DataPath$Intensity <- as.numeric(DataPath$Intensity)
data <- DataPath

#Import the GO db
GO_db <- read.delim(GO_DB_Path, sep = " ")

data$Protein <- str_extract(data$Protein,"[^-]+") 


####### code for reactome analysis
#----------------
GO_db <- GO_db %>% dplyr::select(Gene,GO_term_name)
GO_db <- unique(GO_db)
GO_db$Gene <- toupper(GO_db$Gene)

#----------
# Kruskall Wallis GSEA
#----------

GO_db <- GO_db %>% group_by(GO_term_name) %>% add_count() %>% filter(n>1)

unique_GO_terms <- unique(GO_db$GO_term_name)
KW_output <- data.frame(GO_term = character(), pVal = numeric(),numberOfMatches= numeric(),fractionOfDB_Observed = numeric(),stringsAsFactors = FALSE)
for (i in 1:length(unique_GO_terms)){
  GO_term <- unique_GO_terms[i]
  GO_db_lim <- GO_db %>% dplyr::filter(GO_term_name == GO_term)
  
  data_matches <- data %>% dplyr::filter(Gene %in% GO_db_lim$Gene)
  matches_number <- length(unique(data_matches$Protein))
  total_Proteins_in_GOterm <- length(unique(GO_db_lim$Gene))
  fractionObserved <- matches_number/total_Proteins_in_GOterm
  dataProt_matches_medInt <- data_matches %>% dplyr::group_by(Condition)  %>% dplyr::summarize(medianInt = median(Intensity, na.rm=T))
  dataProt_matches_medInt$GO_term <- GO_term
  
  # Kruskall Wallis
  if(matches_number > 1){
    aav_test <- aov(Intensity ~ Condition, data = data_matches)
    sum_aav1 <- summary(aav_test)
    sum_aav1 <- data.frame(sum_aav1[[1]])
    pVal <- paste0(sum_aav1[1,5]) 
  }
  
  if(matches_number <2){
    pVal = NA
  }
    KW_output[i,1] <- GO_term
    KW_output[i,2] <- pVal 
    KW_output[i,3] <- matches_number
    KW_output[i,4] <- fractionObserved
    
    if(i == 1){
      KW_Int_output <- dataProt_matches_medInt[0,]
    }
    if(i > 1){
      KW_Int_output <- rbind(KW_Int_output, dataProt_matches_medInt)
    }
    
}  

# Multiple hypothesis testing correction
#KW_output_MS1_keep <- KW_output
KW_output <- KW_output_MS1_keep

KW_output <- KW_output[which(KW_output$numberOfMatches>3),]

#my changes
KW_output$qVal = p.adjust(KW_output$pVal, method = 'BH') # qvalue
KW_output1 <- KW_output %>% mutate("passes_5perc_FDR" = ifelse(qVal < 0.05, "TRUE", "FALSE"))
KW_output1 <- KW_output1[grepl("TRUE",KW_output1$passes_5perc_FDR),]
qval5_MS1 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.05),1]))
qval1_MS1 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.01),1]))

# Join Effect size
KW_output_comb <- KW_Int_output %>% left_join(KW_output, by = "GO_term")
KW_output_comb <- na.omit(KW_output_comb)
KW_output_comb <- ungroup(KW_output_comb)
KW_output_comb_effectSize <- KW_output_comb %>% group_by(GO_term) %>% dplyr::summarize(EffectSize = max(abs(medianInt)))
KW_output_comb <- KW_output_comb %>% left_join(KW_output_comb_effectSize, by = "GO_term")
# bounding effect size into quintiles
KW_Effect_quantile <- quantile(KW_output_comb_effectSize$EffectSize,c(.80,.60,.40,.20))

# Dot plot
KW_output_comb$Condition <- factor(KW_output_comb$Condition, levels=c("G1", "S", "G2"))
KW_output_comb_MS1 <- KW_output_comb
KW_output_comb_MS1$Quant <- "MS1"

########################################################################  
########################################################################  
########################      MS2 data         ######################### 
########################################################################  
########################################################################  
df <- df_CC[grepl("eJD904", df_CC$Run),]
df <- df[which(df$Translated.Q.Value <= tqval),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups <- protein.groups[complete.cases(protein.groups),]
protein.groups[,1] <- protein.groups[,1]/median(protein.groups[,1], na.rm=T)
protein.groups[,2] <- protein.groups[,2]/median(protein.groups[,2], na.rm=T)
protein.groups[,3] <- protein.groups[,3]/median(protein.groups[,3], na.rm=T)

rmeans <- rowMeans(protein.groups, na.rm=T)
protein.groupsx <- protein.groups/rmeans
protein.groups_n <- protein.groupsx                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groupsx)
protein.groups_n <- as.data.frame(cbind(protein.groups_n,PGs))

PGs_G1 <- protein.groups_n %>% dplyr::select("eJD904_remTRAQ0", "PGs")
PGs_G1$Condition <- "G1"
colnames(PGs_G1) <- c("Intensity", "Protein", "Condition")
PGs_S <- protein.groups_n %>% dplyr::select("eJD904_remTRAQ4", "PGs")
PGs_S$Condition <- "S"
colnames(PGs_S) <- c("Intensity", "Protein", "Condition")
PGs_G2 <- protein.groups_n %>% dplyr::select("eJD904_remTRAQ8", "PGs")
PGs_G2$Condition <- "G2"
colnames(PGs_G2) <- c("Intensity", "Protein", "Condition")

genes <- df %>% dplyr::select("Protein.Group", "Genes") %>% distinct(Protein.Group, .keep_all=T)
#------------
# Data import
#------------
# Your data should be formatted in terms of a melted list with the following characteristics:
# 1. Condition column
# 2. Relative Protein Abundance Column
# 3. Column of Uniprot IDs
# 4. (optional) Column of Genes
#
# Make sure to title the relevant columns: [Protein, Intensity, Condition, Gene]

DataPath <- rbind(PGs_G1, PGs_S, PGs_G2)
DataPath$Protein <- str_extract(DataPath$Protein,"[^-]+") 
DataPath <- DataPath %>% left_join(genes, by = c("Protein" = "Protein.Group"))
colnames(DataPath) <- c("Intensity", "Protein", "Condition", "Gene")
DataPath$Intensity <- as.numeric(DataPath$Intensity)
data <- DataPath

#Import the GO db
GO_db <- read.delim(GO_DB_Path, sep = " ")
data$Protein <- str_extract(data$Protein,"[^-]+") 

GO_db <- GO_db %>% dplyr::select(Gene,GO_term_name)
GO_db <- unique(GO_db)
GO_db$Gene <- toupper(GO_db$Gene)
GO_db <- GO_db %>% group_by(GO_term_name) %>% add_count() %>% filter(n>1)

unique_GO_terms <- unique(GO_db$GO_term_name)
KW_output <- data.frame(GO_term = character(), pVal = numeric(),numberOfMatches= numeric(),fractionOfDB_Observed = numeric(),stringsAsFactors = FALSE)
for (i in 1:length(unique_GO_terms)){
  GO_term <- unique_GO_terms[i]
  GO_db_lim <- GO_db %>% dplyr::filter(GO_term_name == GO_term)
  
  data_matches <- data %>% dplyr::filter(Gene %in% GO_db_lim$Gene)
  matches_number <- length(unique(data_matches$Protein))
  total_Proteins_in_GOterm <- length(unique(GO_db_lim$Gene))
  fractionObserved <- matches_number/total_Proteins_in_GOterm
  dataProt_matches_medInt <- data_matches %>% dplyr::group_by(Condition)  %>% dplyr::summarize(medianInt = median(Intensity, na.rm=T))
  dataProt_matches_medInt$GO_term <- GO_term
  
  # Kruskall Wallis
  if(matches_number > 1){
    aav_test <- aov(Intensity ~ Condition, data = data_matches)
    sum_aav1 <- summary(aav_test)
    sum_aav1 <- data.frame(sum_aav1[[1]])
    pVal <- paste0(sum_aav1[1,5]) 
  }
  
  if(matches_number <2){
    pVal = NA
  }
    KW_output[i,1] <- GO_term
    KW_output[i,2] <- pVal 
    KW_output[i,3] <- matches_number
    KW_output[i,4] <- fractionObserved
    
    if(i == 1){
      KW_Int_output <- dataProt_matches_medInt[0,]
    }
    if(i > 1){
      KW_Int_output <- rbind(KW_Int_output, dataProt_matches_medInt)
    }
    
}  

# Multiple hypothesis testing correction
KW_output_MS1_keep <- KW_output
KW_output <- KW_output_MS1_keep
KW_output <- KW_output[which(KW_output$numberOfMatches>3),]

#my changes
KW_output$qVal = p.adjust(KW_output$pVal, method = 'BH') # qvalue
KW_output1 <- KW_output %>% mutate("passes_5perc_FDR" = ifelse(qVal < 0.05, "TRUE", "FALSE"))
KW_output1 <- KW_output1[grepl("TRUE",KW_output1$passes_5perc_FDR),]
qval5_MS2 <- as.vector((KW_output1[which(KW_output1$qVal < 0.05),1]))
qval1_MS2 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.01),1]))

# Join Effect size
KW_output_comb <- KW_Int_output %>% left_join(KW_output, by = "GO_term")
KW_output_comb <- na.omit(KW_output_comb)
KW_output_comb <- ungroup(KW_output_comb)
KW_output_comb_effectSize <- KW_output_comb %>% group_by(GO_term) %>% dplyr::summarize(EffectSize = max(abs(medianInt)))
KW_output_comb <- KW_output_comb %>% left_join(KW_output_comb_effectSize, by = "GO_term")
# bounding effect size into quintiles
KW_Effect_quantile <- quantile(KW_output_comb_effectSize$EffectSize,c(.80,.60,.40,.20))

# Dot plot
KW_output_comb$Condition <- factor(KW_output_comb$Condition, levels=c("G1", "S", "G2"))
#KW_output_comb <- KW_output_comb[which(KW_output_comb$qVal < 0.05),]

Cellcycles <- KW_output_comb %>% #filter(EffectSize > KW_Effect_quantile[[4]]) %>%

ggplot(aes(x = Condition, y = reorder(GO_term, EffectSize), fill = log2(medianInt))) + geom_tile() + scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0)  + theme_classic()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.x = element_text(size = 12, face = "bold"),
            legend.position = "top")
Cellcycles

KW_output_comb_MS2 <- KW_output_comb
KW_output_comb_MS2$Quant <- "MS2"

uni_5qval <- unique(append(qval5_MS1,qval5_MS2))
uni_1qval <- unique(append(qval1_MS1,qval1_MS2))

KW_output_comb_both <- rbind(KW_output_comb_MS1, KW_output_comb_MS2)
KW_output_comb_both$C_Q <- paste0(KW_output_comb_both$Condition, "_",KW_output_comb_both$Quant)

KW_output_comb_both5 <- KW_output_comb_both[KW_output_comb_both$GO_term%in%uni_5qval,] #keep GO-terms which have atleast 1 condition at 5% FDR
KW_output_comb_both1 <- KW_output_comb_both[KW_output_comb_both$GO_term%in%uni_1qval,] #keep GO-terms which have atleast 1 condition at 1% FDR

#cluster:
#1) unmelt
umelt <- KW_output_comb_both5 %>% dplyr::select("Condition", "GO_term", "medianInt", "Quant", "C_Q")
umelt <- tidyr::spread(umelt, GO_term, medianInt)
rownames(umelt) <- umelt$C_Q

umelt1 <- umelt[,c(3,4,6,7,11,12,14,24,28,37,40,44,45,46,52,53,55:57,59,60,61,62,66,67,73,78,83,90,91,99,103,115,116,127)]

rownames(umelt1) <- umelt1$C_Q
#umelt1 <- umelt1[,-1]
#umelt <- umelt[,-c(1)]

#2) calculate euclidian dist
t_umelt <- t(as.matrix(umelt1))#[-c(1:3),]
t_umelt <- t_umelt[-1,]
ord <- hclust( dist(t_umelt, method = "euclidean"), method = "ward.D" )$order
rev_order <- rev(ord)
umelt <- melt(t_umelt)


umelt$Var1 <- factor(umelt$Var1, levels= rownames(t_umelt)[rev_order])

umelt$value[log2(as.numeric(umelt$value))>0.5] <- 2^0.4999
umelt$value[log2(as.numeric(umelt$value))< -0.5] <- 2^-0.4999



Cellcycles <- umelt %>%
  ggplot(aes(x = Var2, y = Var1, fill = log2(as.numeric(value)))) + geom_tile() +   
  #scale_fill_gradientn(colours = c("purple", "#077DEC", "white", "#EC7607", "red"))+ #limits=c(-0.5,0.5)) +
  scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0, limits=c(-0.5,0.5), breaks = c(-0.5,0,0.5)) +
  theme_classic()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.x = element_text(size = 12, face = "bold"),
            legend.position = "top") 
Cellcycles

ggsave("Figure5b_MS1_MS2_medium_10192021.pdf", Cellcycles, width = 6.65, height = 7.7)



#########################################################################
#########################################################################
############################ D.A. testing  ##############################
#########################################################################
#########################################################################
# 


############ functions for plotting

reorder_within <- function(x, by, within, fun = max, sep = "___", ...) {
  if (!is.list(within)) {
    within <- list(within)
  }
  new_x <- do.call(paste, c(list(x, sep = sep), within))
  stats::reorder(new_x, by, FUN = fun)
}


scale_x_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_x_discrete(labels = function(x) gsub(reg, "", x), ...)
}


scale_y_reordered <- function(..., sep = "___") {
  reg <- paste0(sep, ".+$")
  ggplot2::scale_y_discrete(labels = function(x) gsub(reg, "", x), ...)
}

#############
con <- c("P04745", "P13645", "P35527","P04264","P35908","Q15323","Q14532","O76011","Q92764","O76013","O76014","O76015","O76009","Q14525","Q14533","Q9NSB4","P78385","Q9NSB2","P78386","O43790") #contaminants

data_prec <- diann_load(INPUT)
data_prec <- data_prec[,!grepl("raw", colnames(data_prec))]
pr <- data_prec
cnames <- colnames(data_prec)

plex <- fread(CC_data, select = c(cnames, "Run","Lib.PG.Q.Value","Ms1.Area"))
plex <- plex[which(plex$Protein.Group%!in%con),]
plex_MS1 <- plex[grepl("903",plex$Run),]
plex_MS1 <- plex_MS1[which(plex_MS1$Lib.PG.Q.Value<0.01),]
plex_MS1 <- plex_MS1 %>% dplyr::rename("Intensity" = "Ms1.Area")

plex <- fread(CC_data, select = c(cnames,"Run","Lib.PG.Q.Value","Precursor.Translated"))
plex <- plex[which(plex$Protein.Group%!in%con),]
plex_MS2 <- plex[grepl("904",plex$Run),]
plex_MS2 <- plex_MS2[which(plex_MS2$Lib.PG.Q.Value<0.01),]
plex_MS2 <- plex_MS2 %>% dplyr::rename("Intensity" = "Precursor.Translated")

plex <- rbind(plex_MS1,plex_MS2)
plex <- plex %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

plex$Unlabelled.Precursor <- paste0(plex$Modified.Sequence, plex$Precursor.Charge)
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ0\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ4\\)")
plex$Unlabelled.Precursor <- str_remove_all(plex$Unlabelled.Precursor, "\\(mTRAQ8\\)")
plex$precRun <- paste0(plex$Unlabelled.Precursor, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)

h <- plex %>% group_by(chanRun) %>% mutate(med_int = median(Intensity)) %>% 
  distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_int")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Intensity>0),]
plex <- plex %>% mutate(Intensity_cnorm = Intensity/med_int)
plex <- plex %>% group_by(Unlabelled.Precursor, Run) %>% mutate(Intensity = Intensity_cnorm/mean(Intensity_cnorm)) %>% ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("Unlabelled.Precursor", "Intensity", "Protein.Group", "Genes", "Protein.Names","First.Protein.Description", "channel_name","Run")
d0_A <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% group_by(Protein.Group) %>% mutate(med_ab_d0 = median(Intensity)) %>% ungroup() %>%
  dplyr::select("Protein.Group", "med_ab_d0") %>% distinct(Protein.Group, .keep_all=T) 
d4_A <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% group_by(Protein.Group) %>% mutate(med_ab_d4 = median(Intensity)) %>% ungroup() %>%
  dplyr::select("Protein.Group", "med_ab_d4") %>% distinct(Protein.Group, .keep_all=T) 
d8_A <- plex1[grepl("mTRAQ8", plex1$channel_name),] %>% group_by(Protein.Group) %>% mutate(med_ab_d8 = median(Intensity)) %>% ungroup() %>%
  dplyr::select("Protein.Group", "med_ab_d8") %>% distinct(Protein.Group, .keep_all=T) 
d0_d4 <- d0_A %>% inner_join(d4_A, by =c("Protein.Group" = "Protein.Group"))
d0_d4_d8 <- d0_d4 %>% inner_join(d8_A, by =c("Protein.Group" = "Protein.Group"))

#require >1 precursor per protein group per sample
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group)
d8 <- plex1[grepl("mTRAQ8", plex1$channel_name),] %>% add_count(Protein.Group) %>% filter(n>1) %>% distinct(Protein.Group)

plex1 <- plex1 %>% group_by(channel_name, Protein.Group) %>% mutate("med_pep_ab" = median(Intensity)) %>% ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),]
int <- int[which(int$Protein.Group%in%d8$Protein.Group),]

prots <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots <- prots[prots$Protein.Group%in%int$Protein.Group,]
prots$pval <- 10000
unique_prots <- as.vector(prots$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Intensity~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots[i,10] <- as.numeric(pval_temp)
  
}

prots$FC_qval <-  p.adjust(prots$pval, method = 'BH') # qvalue
prots_sig <- prots[which(prots$FC_qval<0.01),] #%>% dplyr::select("Protein.Group", "FC_qval")
prots_sig <- prots%>% dplyr::select("Protein.Group", "FC_qval")
plex2 <- plex1 %>% inner_join(prots_sig, by =c("Protein.Group"= "Protein.Group"))
plex2 <- plex2 %>% group_by(Run, channel_name, Protein.Group) %>% mutate("median_prot" = median(Intensity, na.rm=T)) %>% ungroup()
plex_prot <- plex2 %>% mutate("prot_chanrun" = paste0(Run,channel_name,Protein.Group)) %>% distinct(prot_chanrun, .keep_all=T)
plex_prot <- plex_prot %>% group_by(Run, Protein.Group) %>% mutate("mean_prot_run" = mean(median_prot, na.rm=T)) %>% #take mean across channels
  mutate("Prot_norm_int" = median_prot/mean_prot_run) %>% ungroup()
plex_prot <- plex_prot %>% mutate("Condition" = ifelse(grepl("mTRAQ0", channel_name), "G1",
                                                       ifelse(grepl("mTRAQ4", channel_name), "S", "G2"))) %>%
  mutate("Quant" = ifelse(grepl("903", plex_prot$Run), "MS1", "MS2"))
plex_prot$C_Q <- paste0(plex_prot$Condition, plex_prot$Quant)
plex_prot_omit <- plex_prot[!grepl("Fez|C5orf", plex_prot$First.Protein.Description),]  #these proteins have missingness at MS1 and/or MS2
plex_prot_omit$prot_info <- paste0(plex_prot_omit$First.Protein.Description,", [",plex_prot_omit$Genes,"]")
#plex_prot_omit$prot_info <- paste0(plex_prot_omit$First.Protein.Description,", [",plex_prot_omit$Protein.Names,"]")
prots_sig <- prots[which(prots$FC_qval<0.01),]#%>% dplyr::select("Protein.Group", "FC_qval")

################# plot ##########################
umelt <- plex_prot_omit %>% dplyr::select("Condition", "prot_info","Prot_norm_int", "Quant", "C_Q")
umelt <- tidyr::spread(umelt, prot_info, Prot_norm_int)
rownames(umelt) <- umelt$C_Q
umelt1 <- umelt
rownames(umelt1) <- umelt1$C_Q

#2) calculate euclidian dist
t_umelt <- t(as.matrix(umelt1))#[-c(1:3),]
ord <- hclust( dist(t_umelt, method = "euclidean"), method = "ward.D" )$order
umelt <- melt(t_umelt)
umelt$Var1 <- factor(umelt$Var1, levels= rownames(t_umelt)[ord])
umelt$value[log2(as.numeric(umelt$value))>1] <- 2^0.999
umelt$value[log2(as.numeric(umelt$value))< -1] <- 2^-0.999
cond_order <- factor(umelt$Var2, level = c('G1MS1', 'G1MS2', 'SMS1', 'SMS2', 'G2MS1', 'G2MS2'))


DA_prot_HM <- umelt %>%
  ggplot(aes(x = cond_order, y = Var1, fill = log2(as.numeric(value)))) + geom_tile() +   
  #scale_fill_gradientn(colours = c("purple", "#077DEC", "white", "#EC7607", "red"))+ #limits=c(-0.5,0.5)) +
  scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0, limits=c(-1,1), breaks = c(-1,0,1)) +
  theme_classic()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.x = element_text(size = 12, face = "bold"),
            legend.position = "top") 
DA_prot_HM


ggsave("Figure5_DA_heatmap.pdf", DA_prot_HM, width = 7.5, height = 10)

length(unique(prots_sig$Protein.Group))
################ 
prots_sig <- prots_sig %>% dplyr::arrange(FC_qval)
denovo <- prots_sig[grepl("CMPK1|PFAS|ATIC", prots_sig$Genes),]
glyc <- prots_sig[grepl("PGAM1|TPI1|GLO1|ALDOA|ENO1|PGK1|TKT|GAPDH", prots_sig$Genes),]
dna_synth_catab <- prots_sig[grepl("HINT1|^RRM2$", prots_sig$Genes),]
cyto_org <- prots_sig[grepl("DSP|JUP|FLG2", prots_sig$Genes),]
noinfo <-  prots_sig[grepl("CDV3|JPT2", prots_sig$Genes),]
chemotaxis <- prots_sig[grepl("S100A7|^ANXA1$|PPIA|ACTN4|S100A9", prots_sig$Genes),]
#chrom_process <- prots_sig[c(19,47),]
spindle <- prots_sig[grepl("TPX2|KIF11|TACC3|TOP2A|MKI67|CLTC|CKAP5|NUSAP1", prots_sig$Genes),]
#cytokinesis <- prots_sig[c(10,38,63),]
DNA_repair <- prots_sig[grepl("^FH$|DHX9|PRKDC|PCLAF|BAZ1B", prots_sig$Genes),]
protect_ROS <- prots_sig[grepl("PARK7|PRDX1", prots_sig$Genes),]
misfold <- prots_sig[grepl("HSPA8|PDIA3|HSPA5|PFDN4|PFDN1|PFDN2|VBP1", prots_sig$Genes),]
APC <- prots_sig[grepl("PLK1|UBE2S|CCNA2", prots_sig$Genes),]
  
denovo$type <- "'de novo' nucleotide synthesis"
glyc$type <- "Glycolysis, gluconeogenesis, PPP"
protect_ROS$type <- "Protection against ROS"
dna_synth_catab$type <- "DNA synthesis & catabolism"
cyto_org$type <- "Intermediate filament organization"
noinfo$type <- "poorly characterized"
chemotaxis$type <- "Chemotaxis, migration, & motility"
spindle$type <- "Spindle interacting"
#cytokinesis$type <- "Cell division and cytokinesis"
DNA_repair$type <- "DNA repair"
misfold$type <- "Protein misfolding"
APC$type <- "Mitotic entry exit"

all_types <- rbind(denovo, glyc, protect_ROS, dna_synth_catab, cyto_org, noinfo, chemotaxis, spindle, DNA_repair,misfold, APC)
q <- plex_prot_omit %>% inner_join(all_types, by =c("Protein.Group" = "Protein.Group"))
q_info <- q %>% dplyr::select("prot_info", "type") %>% distinct(prot_info, .keep_all = T)


###### data wrangling:
#all_types <- q_info %>% dplyr::select("prot_info", "C_Q", "Prot_norm_int", "type")
#colnames(all_types) <- c("Var1", "Var2", "value","type")



####### alt:
q_all <- q %>% dplyr::select("prot_info", "C_Q", "Prot_norm_int", "type")
colnames(q_all) <- c("Var1", "Var2", "value","type")

######### combine:
# all_types <- rbind(chemotaxis1, glyc1, protect_ROS1, denovo1, dna_synth_catab1, cyto_org1, chrom_process1, spindle1, DNA_repair1, noinfo1)
# 
# q_all <- all_types %>% inner_join(q_info, by =c("Var1" = "prot_info"))

#########  plot:
cond_order <- factor(q_all$Var2, level = c('G1MS1', 'G1MS2', 'SMS1', 'SMS2', 'G2MS1', 'G2MS2'))

#q_all$value[log2(as.numeric(q_all$value))>2] <- 2^1.999
#q_all$value[log2(as.numeric(q_all$value))< -2] <- 2^-1.99
q_all$value[log2(as.numeric(q_all$value))>1] <- 2^0.9999
q_all$value[log2(as.numeric(q_all$value))< -1] <- 2^-0.9999

q_all_2 <- q_all
q_all_2$type<- factor(q_all_2$type, level = c("poorly characterized", "Chemotaxis, migration, & motility", "Glycolysis, gluconeogenesis, PPP", "Protection against ROS", "Protein misfolding","'de novo' nucleotide synthesis", "DNA synthesis & catabolism", "Intermediate filament organization", "DNA repair", "Spindle interacting", "Mitotic entry exit"))

DA_prot_HM <- q_all_2 %>%
  ggplot(aes(x = cond_order, y = reorder_within(Var1, value, type), fill = log2(as.numeric(value)))) + geom_tile() + 
  facet_grid(rows = vars(type), scales="free_y", space = "free_y", labeller = label_wrap_gen(width = 1, multi_line = TRUE))+
  #scale_fill_gradientn(colours = c("purple", "#077DEC", "white", "#EC7607", "red"))+ #limits=c(-0.5,0.5)) +
  scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0, limits=c(-1,1), breaks = c(-1,0,1)) +
  theme_bw()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.y = element_text(angle = 0, size = 8),
            legend.position = "top",
            strip.background = element_rect(fill="white"))+
  scale_y_reordered()
DA_prot_HM

ggsave("Figure5_DA_10212021.pdf", DA_prot_HM, width = 7.5, height = 9.8)

##########


##########
plex1$stripseq <- substr(plex1$Unlabelled.Precursor,1,nchar(plex1$Unlabelled.Precursor)-1)
vis <- plex1[grepl("CDV3|JPT2", plex1$Genes),] %>% distinct(stripseq, .keep_all=T) %>% dplyr::select(stripseq, Genes, First.Protein.Description)
write.table(vis, "/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/vis_CDV3_JPT2.txt", sep = "\t", row.names = FALSE)


############## XIC plots:
report <- read.delim("/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/dat/multiDIA/v1.8b7_10102021/CC_MS1_MS2_vis_command/Report.tsv")
report <- report[report$Stripped.Sequence%in%vis$stripseq,]
report <- report[grepl("903", report$File.Name),]
XIC <- read.delim("/Volumes/GoogleDrive/My Drive/MS/SCoPE/NUC/dat/multiDIA/v1.8b7_10102021/CC_MS1_MS2_vis_command/Report.XIC.tsv")

XIC_903 <- XIC[grepl("903", XIC$File.Name),]

#CVD3 <- XIC_903[grepl("\\(mTRAQ0\\)AVTK\\(mTRAQ0\\)DEDEWK\\(mTRAQ0\\)3", XIC_903$Precursor.Id),]
#CVD3 <- XIC_903[grepl("\\(mTRAQ4\\)VQAMQISSEK\\(mTRAQ4\\)EEDDNEK\\(mTRAQ4\\)3", XIC_903$Precursor.Id),]
CVD3 <- XIC_903[grepl("\\(mTRAQ4\\)SLDNFFAK\\(mTRAQ4\\)2", XIC_903$Precursor.Id),]
#CVD3 <- XIC_903[grepl("\\(mTRAQ4\\)LQLDNQYAVLENQK\\(mTRAQ4\\)3", XIC_903$Precursor.Id),]

CVD3$ion <- paste0(CVD3$FragmentType,CVD3$FragmentSeriesNumber,"_",CVD3$FragmentCharge)
CVD3_lim <- CVD3 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- CVD3_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_CVD3 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_CVD3$ion_type <- substr(df_CVD3$Ion,1,1)
### MS1
CVD3_MS1 <- CVD3_lim[1:2,]
CVD3_MS1 <- CVD3_MS1[,-1] #remove "ion"
CVD3_MS1_t <- t(as.matrix(CVD3_MS1))
colnames(CVD3_MS1_t) <- c("RT", "Intensity")
CVD3_MS1_t <- data.frame(CVD3_MS1_t)
CVD3_MS1_t$Ion <- "Precursor"
CVD3_MS1_t$Quant <- "MS1"
CVD3_MS1_t$ion_type <- "Precursor"

CVD3_all <- rbind(df_CVD3, CVD3_MS1_t)

#ggplot(CVD3_all, aes(x=RT, y=log10(Intensity), color=Ion)) + geom_line(aes(linetype = ion_type)) +
#  xlim(45.29,45.78) + labs(title = "AVTKDEDEWK3")


ggplot(CVD3_all, aes(x=RT, y=log2(Intensity), color=Ion)) + geom_line(aes(linetype = ion_type)) + #xlim(45.3,45.8)+
 # xlim(51,51.7)+
  # xlim(87.5,87.96)+
     xlim(84.85,85.3)+

  labs(title = "VQAMQISSEKEEDDNEK")


#############
JPT2_frag_0 <- report[grepl("\\(mTRAQ0\\)TSDIFGSPVTATSR2", report$Precursor.Id),]
JPT2_frag_0_cor <- unlist(strsplit(JPT2_frag_0$Fragment.Correlations, ";"))
JPT2_frag_0_info <- unlist(strsplit(JPT2_frag_0$Fragment.Info, ";"))
JPT2_frag_0_info <- sub("\\/.*","\\",JPT2_frag_0_info)
JPT2_frag_0_info <- str_replace(JPT2_frag_0_info, "\\^", "_")

JPT2_frag_4 <- report[grepl("\\(mTRAQ4\\)TSDIFGSPVTATSR2", report$Precursor.Id),]
JPT2_frag_4_cor <- unlist(strsplit(JPT2_frag_4$Fragment.Correlations, ";"))
JPT2_frag_4_info <- unlist(strsplit(JPT2_frag_4$Fragment.Info, ";"))
JPT2_frag_4_info <- sub("\\/.*","\\",JPT2_frag_4_info)
JPT2_frag_4_info <- str_replace(JPT2_frag_4_info, "\\^", "_")

JPT2_frag_8 <- report[grepl("\\(mTRAQ8\\)TSDIFGSPVTATSR2", report$Precursor.Id),]
JPT2_frag_8_cor <- unlist(strsplit(JPT2_frag_8$Fragment.Correlations, ";"))
JPT2_frag_8_info <- unlist(strsplit(JPT2_frag_8$Fragment.Info, ";"))
JPT2_frag_8_info <- sub("\\/.*","\\",JPT2_frag_8_info)
JPT2_frag_8_info <- str_replace(JPT2_frag_8_info, "\\^", "_")


JPT2_cor <- data.frame(c(JPT2_frag_0_cor, JPT2_frag_4_cor, JPT2_frag_8_cor))
colnames(JPT2_cor) <- "Cor"
JPT2_info <- data.frame(c(JPT2_frag_0_info, JPT2_frag_4_info, JPT2_frag_8_info))
colnames(JPT2_info) <- "ion"
JPT2_merge <- cbind(JPT2_info, JPT2_cor)
JPT2_merge <- JPT2_merge %>% group_by(ion) %>% mutate("mean_cor" = mean(as.numeric(Cor))) %>% ungroup()
JPT2_merge <- JPT2_merge[grepl("b", JPT2_merge$ion),]
JPT2_keep_ions <- JPT2_merge %>% distinct(ion, .keep_all=T) %>% arrange(desc(mean_cor)) %>% top_n(5) 
JPT2_keep_ions[6,] <- JPT2_merge[3,]

############ d0:
JPT2 <- XIC_903[grepl("\\(mTRAQ0\\)TSDIFGSPVTATSR2", XIC_903$Precursor.Id),]
JPT2$ion <- paste0(JPT2$FragmentType,JPT2$FragmentSeriesNumber,"_",JPT2$FragmentCharge)
JPT2_lim <- JPT2 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- JPT2_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_JPT2 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_JPT2$ion_type <- substr(df_JPT2$Ion,1,1)
df_JPT2 <- df_JPT2[df_JPT2$Ion%in%JPT2_keep_ions$ion,]

### MS1
JPT2_MS1 <- JPT2_lim[1:2,]
JPT2_MS1 <- JPT2_MS1[,-1] #remove "ion"
JPT2_MS1_t <- t(as.matrix(JPT2_MS1))
colnames(JPT2_MS1_t) <- c("RT", "Intensity")
JPT2_MS1_t <- data.frame(JPT2_MS1_t)
JPT2_MS1_t$Ion <- "Precursor"
JPT2_MS1_t$Quant <- "MS1"
JPT2_MS1_t$ion_type <- "Precursor"
JPT2_all_d0 <- rbind(df_JPT2, JPT2_MS1_t)

############# d4
JPT2 <- XIC_903[grepl("\\(mTRAQ4\\)TSDIFGSPVTATSR2", XIC_903$Precursor.Id),]
JPT2$ion <- paste0(JPT2$FragmentType,JPT2$FragmentSeriesNumber,"_",JPT2$FragmentCharge)
JPT2_lim <- JPT2 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- JPT2_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_JPT2 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_JPT2$ion_type <- substr(df_JPT2$Ion,1,1)
df_JPT2 <- df_JPT2[df_JPT2$Ion%in%JPT2_keep_ions$ion,]

### MS1
JPT2_MS1 <- JPT2_lim[1:2,]
JPT2_MS1 <- JPT2_MS1[,-1] #remove "ion"
JPT2_MS1_t <- t(as.matrix(JPT2_MS1))
colnames(JPT2_MS1_t) <- c("RT", "Intensity")
JPT2_MS1_t <- data.frame(JPT2_MS1_t)
JPT2_MS1_t$Ion <- "Precursor"
JPT2_MS1_t$Quant <- "MS1"
JPT2_MS1_t$ion_type <- "Precursor"

JPT2_all_d4 <- rbind(df_JPT2, JPT2_MS1_t)

############# d8
JPT2 <- XIC_903[grepl("\\(mTRAQ8\\)TSDIFGSPVTATSR2", XIC_903$Precursor.Id),]
JPT2$ion <- paste0(JPT2$FragmentType,JPT2$FragmentSeriesNumber,"_",JPT2$FragmentCharge)
JPT2_lim <- JPT2 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- JPT2_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_JPT2 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_JPT2$ion_type <- substr(df_JPT2$Ion,1,1)
df_JPT2 <- df_JPT2[df_JPT2$Ion%in%JPT2_keep_ions$ion,]

### MS1
JPT2_MS1 <- JPT2_lim[1:2,]
JPT2_MS1 <- JPT2_MS1[,-1] #remove "ion"
JPT2_MS1_t <- t(as.matrix(JPT2_MS1))
colnames(JPT2_MS1_t) <- c("RT", "Intensity")
JPT2_MS1_t <- data.frame(JPT2_MS1_t)
JPT2_MS1_t$Ion <- "Precursor"
JPT2_MS1_t$Quant <- "MS1"
JPT2_MS1_t$ion_type <- "Precursor"

JPT2_all_d8 <- rbind(df_JPT2, JPT2_MS1_t)
#JPT2_all$Intensity[as.numeric(JPT2_all$Intensity)==0] <- 1
ggplot(JPT2_all_d8, aes(x=RT, y=log2(Intensity), color=Ion)) + geom_line(aes(linetype = ion_type)) + xlim(73.8,74.22)+
   labs(title = "TSDIFGSPVTATSR2")



################################ CDV3 #########################


#############
CDV3_frag_0 <- report[grepl("\\(mTRAQ0\\)SLDNFFAK\\(mTRAQ0\\)2", report$Precursor.Id),]
CDV3_frag_0_cor <- unlist(strsplit(CDV3_frag_0$Fragment.Correlations, ";"))
CDV3_frag_0_info <- unlist(strsplit(CDV3_frag_0$Fragment.Info, ";"))
CDV3_frag_0_info <- sub("\\/.*","\\",CDV3_frag_0_info)
CDV3_frag_0_info <- str_replace(CDV3_frag_0_info, "\\^", "_")

CDV3_frag_4 <- report[grepl("\\(mTRAQ4\\)SLDNFFAK\\(mTRAQ4\\)2", report$Precursor.Id),]
CDV3_frag_4_cor <- unlist(strsplit(CDV3_frag_4$Fragment.Correlations, ";"))
CDV3_frag_4_info <- unlist(strsplit(CDV3_frag_4$Fragment.Info, ";"))
CDV3_frag_4_info <- sub("\\/.*","\\",CDV3_frag_4_info)
CDV3_frag_4_info <- str_replace(CDV3_frag_4_info, "\\^", "_")

CDV3_frag_8 <- report[grepl("\\(mTRAQ8\\)SLDNFFAK\\(mTRAQ8\\)2", report$Precursor.Id),]
CDV3_frag_8_cor <- unlist(strsplit(CDV3_frag_8$Fragment.Correlations, ";"))
CDV3_frag_8_info <- unlist(strsplit(CDV3_frag_8$Fragment.Info, ";"))
CDV3_frag_8_info <- sub("\\/.*","\\",CDV3_frag_8_info)
CDV3_frag_8_info <- str_replace(CDV3_frag_8_info, "\\^", "_")


CDV3_cor <- data.frame(c(CDV3_frag_0_cor, CDV3_frag_4_cor, CDV3_frag_8_cor))
colnames(CDV3_cor) <- "Cor"
CDV3_info <- data.frame(c(CDV3_frag_0_info, CDV3_frag_4_info, CDV3_frag_8_info))
colnames(CDV3_info) <- "ion"
CDV3_merge <- cbind(CDV3_info, CDV3_cor)
CDV3_merge <- CDV3_merge %>% group_by(ion) %>% mutate("mean_cor" = mean(as.numeric(Cor))) %>% ungroup()
CDV3_keep_ions <- CDV3_merge %>% distinct(ion, .keep_all=T) %>% arrange(desc(mean_cor)) %>% top_n(6) 
#CDV3_keep_ions[6,] <- CDV3_merge[3,]

############ d0:
CDV3 <- XIC_903[grepl("\\(mTRAQ0\\)SLDNFFAK\\(mTRAQ0\\)2", XIC_903$Precursor.Id),]
CDV3$ion <- paste0(CDV3$FragmentType,CDV3$FragmentSeriesNumber,"_",CDV3$FragmentCharge)
CDV3_lim <- CDV3 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- CDV3_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_CDV3 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_CDV3$ion_type <- substr(df_CDV3$Ion,1,1)
df_CDV3 <- df_CDV3[df_CDV3$Ion%in%CDV3_keep_ions$ion,]

### MS1
CDV3_MS1 <- CDV3_lim[1:2,]
CDV3_MS1 <- CDV3_MS1[,-1] #remove "ion"
CDV3_MS1_t <- t(as.matrix(CDV3_MS1))
colnames(CDV3_MS1_t) <- c("RT", "Intensity")
CDV3_MS1_t <- data.frame(CDV3_MS1_t)
CDV3_MS1_t$Ion <- "Precursor"
CDV3_MS1_t$Quant <- "MS1"
CDV3_MS1_t$ion_type <- "Precursor"
CDV3_all_d0 <- rbind(df_CDV3, CDV3_MS1_t)

############# d4
CDV3 <- XIC_903[grepl("\\(mTRAQ4\\)SLDNFFAK\\(mTRAQ4\\)2", XIC_903$Precursor.Id),]
CDV3$ion <- paste0(CDV3$FragmentType,CDV3$FragmentSeriesNumber,"_",CDV3$FragmentCharge)
CDV3_lim <- CDV3 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- CDV3_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_CDV3 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_CDV3$ion_type <- substr(df_CDV3$Ion,1,1)
df_CDV3 <- df_CDV3[df_CDV3$Ion%in%CDV3_keep_ions$ion,]

### MS1
CDV3_MS1 <- CDV3_lim[1:2,]
CDV3_MS1 <- CDV3_MS1[,-1] #remove "ion"
CDV3_MS1_t <- t(as.matrix(CDV3_MS1))
colnames(CDV3_MS1_t) <- c("RT", "Intensity")
CDV3_MS1_t <- data.frame(CDV3_MS1_t)
CDV3_MS1_t$Ion <- "Precursor"
CDV3_MS1_t$Quant <- "MS1"
CDV3_MS1_t$ion_type <- "Precursor"

CDV3_all_d4 <- rbind(df_CDV3, CDV3_MS1_t)

############# d8
CDV3 <- XIC_903[grepl("\\(mTRAQ8\\)SLDNFFAK\\(mTRAQ8\\)2", XIC_903$Precursor.Id),]
CDV3$ion <- paste0(CDV3$FragmentType,CDV3$FragmentSeriesNumber,"_",CDV3$FragmentCharge)
CDV3_lim <- CDV3 %>% dplyr::select(ion,X0,X1,X2,X3,X4,X5,X6,X7,X8,X9,X10,X11,X12,X13,X14,X15,X16,X17,X18,X19,X20,X21, X22,X23,X24)

mylist <- list() #create an empty list

for(i in 1:12){  
  temp <- CDV3_lim[(2*i+1):(2*i+2),]
  ion_temp <- paste0(temp[2,1])
  temp <- temp[,-1] #remove "ion"
  temp_t <- t(as.matrix(temp))
  colnames(temp_t) <- c("RT", "Intensity")
  temp_t <- data.frame(temp_t)
  temp_t$Ion <- as.character(paste0(ion_temp))
  temp_t$Quant <- "MS2"
  mylist[[i]] <- temp_t
}

df_CDV3 <- do.call("rbind",mylist) #combine all vectors into a matrix
df_CDV3$ion_type <- substr(df_CDV3$Ion,1,1)
df_CDV3 <- df_CDV3[df_CDV3$Ion%in%CDV3_keep_ions$ion,]

### MS1
CDV3_MS1 <- CDV3_lim[1:2,]
CDV3_MS1 <- CDV3_MS1[,-1] #remove "ion"
CDV3_MS1_t <- t(as.matrix(CDV3_MS1))
colnames(CDV3_MS1_t) <- c("RT", "Intensity")
CDV3_MS1_t <- data.frame(CDV3_MS1_t)
CDV3_MS1_t$Ion <- "Precursor"
CDV3_MS1_t$Quant <- "MS1"
CDV3_MS1_t$ion_type <- "Precursor"

CDV3_all_d8 <- rbind(df_CDV3, CDV3_MS1_t)
#CDV3_all$Intensity[as.numeric(CDV3_all$Intensity)==0] <- 1
ggplot(CDV3_all_d0, aes(x=RT, y=log2(Intensity), color=Ion)) + geom_line(aes(linetype = ion_type)) + xlim(84.85,85.3)+
   labs(title = "TSDIFGSPVTATSR2")


######### combine:
CDV3_all_d0$Channel <- paste0("G1-phase, ", "\u0394","0")
CDV3_all_d4$Channel <- paste0("S-phase, ", "\u0394","4")
CDV3_all_d8$Channel <- paste0("G2-phase, ", "\u0394","8")
CDV3_fin <- rbind(CDV3_all_d0, CDV3_all_d4, CDV3_all_d8)
CDV3_fin$Protein <- "CDV3"
CDV3_fin$Peptide <- "SLDNFFAK"
CDV3_fin$Precursor <- "SLDNFFAK2"
CDV3_fin <- CDV3_fin[which(CDV3_fin$RT>84.85 & CDV3_fin$RT<85.3),]


JPT2_all_d0$Channel <- paste0("G1-phase, ", "\u0394","0")
JPT2_all_d4$Channel <- paste0("S-phase, ", "\u0394","4")
JPT2_all_d8$Channel <- paste0("G2-phase, ", "\u0394","8")
JPT2_fin <- rbind(JPT2_all_d0, JPT2_all_d4, JPT2_all_d8)
JPT2_fin$Protein <- "JPT2"
JPT2_fin$Peptide <- "TSDIFGSPVTATSR"
JPT2_fin$Precursor <- "TSDIFGSPVTATSR2"
JPT2_fin <- JPT2_fin[which(JPT2_fin$RT>73.8 & JPT2_fin$RT<74.22),]
bp <- rbind(CDV3_fin, JPT2_fin)

################## plot:


scales_x <- list(
  `CDV3` = scale_x_continuous(limits = c(84.9, 85.3), breaks = seq(84.9, 85.3, 0.1)),
  `JPT2` = scale_x_continuous(limits = c(73.8, 74.22),breaks = seq(73.8, 74.22, 0.1))
)

bp$Channel<- factor(bp$Channel, level = c(paste0("G1-phase, ", "\u0394","0"), paste0("S-phase, ", "\u0394","4"), paste0("G2-phase, ", "\u0394","8")))

# ################

#############
bp$n <- rownames(bp)



bp <- bp %>% mutate(int_mirror = ifelse(grepl("S", Channel), -Intensity, Intensity))
bp <- bp[!grepl("G2", bp$Channel),]
bp$ion_chan <- paste0(bp$Ion,bp$Channel)
bp2 <- bp %>% left_join(bp1, by =c("ion_chan" = "ion_chan"))

scales_y <- list(
  `MS1` = scale_y_continuous(limits = c(-630000,630000), breaks = seq(-600000,600000, 200000), labels = scales::comma),
  `MS2` = scale_y_continuous(limits = c(-40000,40000), breaks = seq(-36000,36000, 12000), labels = scales::comma)
)

library(scales)
bp$ion_chan_quant <- paste0(bp$ion_chan, bp$Quant)
bp <- bp %>% group_by(Protein, Quant) %>% mutate("norm_int" = Intensity/max(Intensity)) %>% ungroup()

bp1 <- bp
bp$pred <- predict(loess(int_mirror~RT, data = bp1))





ggplot(bp, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  +  geom_point()+
  geom_line(stat = "smooth", method = "loess", aes(color=abs(..y..)/(log2(abs(..y..)))), size=1.5, alpha=0.7) + geom_hline(yintercept = 0)+
     facet_grid_sc(cols = vars(Protein), rows = vars(Quant), scales=list(x = scales_x, y=scales_y), labeller = label_wrap_gen(width = 1, multi_line = TRUE))+
  theme_bw()  +
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 14),
            axis.text.y = element_text(size = 14),
            strip.text.y = element_text(size = 18),
            strip.text.x = element_text(size = 18),
            axis.title.x = element_text(size = 24),
            axis.title.y = element_text(size = 24),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo") +
  scale_y_continuous(label = bp$int_mirror*100)


####################
ggplot(bp, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  + 
  geom_line(stat = "smooth", method = "loess", aes(color=abs(..y..)/(log2(abs(..y..)))), size=1.5, alpha=0.7) + geom_hline(yintercept = 0)+
     facet_grid_sc(cols = vars(Protein), rows = vars(Quant), scales=list(x = scales_x, y=scales_y), labeller = label_wrap_gen(width = 1, multi_line = TRUE))+
  theme_bw()  +
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 14),
            axis.text.y = element_text(size = 14),
            strip.text.y = element_text(size = 18),
            strip.text.x = element_text(size = 18),
            axis.title.x = element_text(size = 24),
            axis.title.y = element_text(size = 24),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo")

ggsave("Fig5_XIC_template.pdf")


bp_cdv3_MS1 <- bp[grepl("CDV3",bp$Protein),]
bp_cdv3_MS1 <- bp_cdv3_MS1[grepl("MS1", bp_cdv3_MS1$Quant),]

ggplot(bp_cdv3_MS1, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  + geom_point()+
  geom_line(stat = "smooth", method = "gam", aes(color=abs(..y..)), size=1.5, alpha=0.9, ) + geom_hline(yintercept = 0)+
  theme_bw() + #+ xlim(84.85,85.3)+ 
  scale_y_continuous(limits = c(-650000,650000), breaks = seq(-600000,600000, 300000), labels = scales::comma)+
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.y = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(size = 14),
            strip.text.y = element_text(size = 14),
            strip.text.x = element_text(size = 14),
            axis.title.x = element_text(size = 18),
            axis.title.y = element_text(size = 18),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo")
ggsave("Figure5_XICa_CDV3_MS1.pdf", width=3,height=4)

########
bp_cdv3_MS2 <- bp[grepl("CDV3",bp$Protein),]
bp_cdv3_MS2 <- bp_cdv3_MS2[grepl("MS2", bp_cdv3_MS2$Quant),]


ggplot(bp_cdv3_MS2, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  + geom_point()+
  geom_line(stat = "smooth", method = "loess", aes(color=abs(..y..)), size=1.5, alpha=0.9, ) + geom_hline(yintercept = 0)+
  theme_bw()  + #xlim(84.85,85.3)+ 
  scale_y_continuous(limits = c(-40000,40000), labels = scales::comma) +
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.y = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(size = 14),
            strip.text.y = element_text(size = 14),
            strip.text.x = element_text(size = 14),
            axis.title.x = element_text(size = 18),
            axis.title.y = element_text(size = 18),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo")
ggsave("Figure5_XICa_CDV3_MS2.pdf", width=3,height=4)

###########
bp_jpt2_MS1 <- bp[grepl("JPT2",bp$Protein),]
bp_jpt2_MS1 <- bp_jpt2_MS1[grepl("MS1", bp_jpt2_MS1$Quant),]

ggplot(bp_jpt2_MS1, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  + geom_point()+
  geom_line(stat = "smooth", method = "gam", aes(color=abs(..y..)), size=1.5, alpha=0.9, ) + geom_hline(yintercept = 0)+
  theme_bw()  + #xlim(84.85,85.3)+ 
  scale_y_continuous(limits = c(-180000,180000), breaks = seq(-180000,180000, 90000), labels = scales::comma)+
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.y = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(size = 14),
            strip.text.y = element_text(size = 14),
            strip.text.x = element_text(size = 14),
            axis.title.x = element_text(size = 18),
            axis.title.y = element_text(size = 18),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo")
ggsave("Figure5_XIC_JPT2_MS1.pdf", width=3,height=4)

########
bp_jpt2_MS2 <- bp[grepl("JPT2",bp$Protein),]
bp_jpt2_MS2 <- bp_jpt2_MS2[grepl("MS2", bp_jpt2_MS2$Quant),]


ggplot(bp_jpt2_MS2, aes(x=RT, y=int_mirror, group=ion_chan, color=norm_int))  + geom_point()+
  geom_line(stat = "smooth", method = "loess", aes(color=abs(..y..)), size=1.5, alpha=0.9, ) + geom_hline(yintercept = 0)+
  theme_bw()  + #xlim(84.85,85.3)+ 
  scale_y_continuous(limits = c(-22000,22000), labels = scales::comma) +
  labs(x = "Retention time (min)", y = "Intensity", fill = "Log2, Median Abundance") + 
      theme(axis.text.y = element_blank(),
            axis.ticks.y = element_blank(),
            axis.text.x = element_text(size = 14),
            strip.text.y = element_text(size = 14),
            strip.text.x = element_text(size = 14),
            axis.title.x = element_text(size = 18),
            axis.title.y = element_text(size = 18),
            panel.spacing.x = unit(1.3, "lines"),
            legend.position = "none") + scale_color_viridis(option="turbo")
ggsave("Figure5_XIC_JPT2_MS2.pdf", width=3,height=4)

```



#############
Supplementary
#############


Supp. Figure 1: Jaccard index (data completeness) of our data and Navarro et al.
```{r}



############### Jaccard index

mTRAQ <- read.delim(fpath)
mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0),]  #only keep precursors with quant info


ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$Precursor.Id),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Normalised","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Normalised_d4","Precursor.Translated_d4","Ms1.Area_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("mTRAQ8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Normalised","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Normalised_d8","Precursor.Translated_d8","Ms1.Area_d8","Protein.Group")


d0_1 <- ev2_0[grepl("wJD803", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("wJD803", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("wJD804", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("wJD804", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("wJD815", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("wJD815", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

all_peps <- rbind(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d0_2","d4_2","d0_3","d4_3")


jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value


################################### unlabeled
unlab <- read.delim(LF_fpath)
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]

df_t <- data.frame(c("wJD756_re", "wJD757_re", "wJD758_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c("wJD759_re", "wJD760_re", "wJD761_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0),]  #only keep precursors with quant info


#####
s1_1 <- unlab_1[grepl("wJD756_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("wJD757_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("wJD758_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)


s2_1 <- unlab_2[grepl("wJD759_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("wJD760_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("wJD761_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)

all_peps <- rbind(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s1_2","s2_2","s1_3","s2_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################


################################### unlabeled LFQBench
unlab <- read.delim(Bench_fpath)
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]
uni_run <- unique(unlab$Run)
uni_run
df <- unlab


df_t <- data.frame(c(paste0(uni_run[3]), paste0(uni_run[2]), paste0(uni_run[5])), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c(paste0(uni_run[4]), paste0(uni_run[1]), paste0(uni_run[6])), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0),]  #only keep precursors with quant info

#####
s1_1 <- unlab_1[grepl(paste0(uni_run[3]), unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl(paste0(uni_run[2]), unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl(paste0(uni_run[5]), unlab_1$Run),] %>% dplyr::select(Protein.Group)


s2_1 <- unlab_2[grepl(paste0(uni_run[4]), unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl(paste0(uni_run[1]), unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl(paste0(uni_run[6]), unlab_2$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s1_2","s2_2","s1_3","s2_3")

jac_dist_LF2_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF2 <- get_lower_tri(jac_dist_LF2_df)
m_jac_LF2 <- melt(m_jac_LF2, na.rm = TRUE)
m_jac_LF2$value <- 1-m_jac_LF2$value
#################


############### Jaccard index  DDA

#get prots which are 1% FDR
prot_DDAFDR <- read.delim(DDA_prot_fpath)
prot_DDAFDR <- prot_DDAFDR[which(prot_DDAFDR$Q.value<0.01),] %>% filter(!grepl("REV|CON", Protein.IDs))
prot_DDAFDR$Protein.Group <- gsub(";.*","",prot_DDAFDR$Majority.protein.IDs)

DDA <- read.delim(DDA_fpath)
DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
## find FDR 1 % for each run:
DDA1 <- DDA %>% group_by(Raw.file) %>% arrange(PEP) %>% 
  mutate(rec = 1) %>% 
  mutate(rollavg = cumsum(PEP)/cumsum(rec)) %>% 
  select(-rec) %>% ungroup()
DDA1 <- DDA1 %>% dplyr::select("Modified.sequence","PEP","Raw.file","rollavg")
DDAfil <- DDA1 %>% filter(rollavg < 0.01) %>% group_by(Raw.file) %>% 
  summarise(PEP = max(PEP))


DDA <- read.delim(DDA_fpath)

#filter for 1% FDR
DDA_list <- list()
for( i in 1:3){
  a <- as.numeric(paste0(DDAfil[i,2]))
  temp <- DDA[grepl(paste0(DDAfil[i,1]), DDA$Raw.file),] %>% filter(PEP < a)
  DDA_list[[i]] <- temp
}
DDA <- do.call("rbind",DDA_list) #combine all vectors into a df


DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

all_peps <- rbind(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d0_2","d4_2","d0_3","d4_3")


jac_dist_DDA <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
jac_dist_DDA <- get_lower_tri(jac_dist_DDA)
jac_dist_DDA_m <- melt(jac_dist_DDA, na.rm=T)
jac_dist_DDA_m$value <- 1-jac_dist_DDA_m$value



#################### call it A B C



######################### DDA
jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                ifelse(grepl("_2", Var1),"Rep 2",
                                                                       ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                ifelse(grepl("_2", Var2),"Rep 2",
                                                                       ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))

#labz <- c(paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"),paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"),paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"))
labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")

pDDA <- ggplot(jac_dist_DDA_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="mTRAQ DDA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  

pDDA

############### mTRAQ


jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))

#labz <- c(paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"),paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"),paste0(" \u0394","0A"), paste0(" \u0394","4B"), paste0(" \u0394","8C"))
labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="plexDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  



############## LF
m_jac_LF <- m_jac_LF %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="LF-DIA this study") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  




############## LFQ Bench
m_jac_LF2 <- m_jac_LF2 %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p4 <- ggplot(m_jac_LF2, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="LF-DIA from Navarro et al.") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  


## combine
library(gridExtra)
library(ggpubr)
library(gridGraphics)

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                          p1 + theme(legend.position="none"), NULL,
                          p4 + theme(legend.position="none"), NULL,
                          pDDA +theme(legend.position="none"),
                          nrow=1, widths=c(1,0.1,1,0.1,1, 0.1, 1)))

ggsave("SFigure1.pdf", p3, width=16.5, height=5.2, dpi=500)

```


Supp. Figure 2: Missing E. coli experiment to benchmark translated FDR
```{r}

ev <- read.delim(YE_fpath)
ev <- ev[which(ev$Translated.Q.Value<0.01),]  #filter for trans-q-value < 0.01 
ev$seqcharge <- paste0(ev$Modified.Sequence, ev$Precursor.Charge)
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ0\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ4\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ8\\)")
ev$seqcharge_run <- paste0(ev$seqcharge,ev$Run)

ev2_lim <- ev
ev2_lim <- ev2_lim[grepl("wJD813", ev$Run),]
ev2_lim <- ev2_lim[!grepl("mTRAQ8", ev2_lim$Precursor.Id),]
ev2_lim$HY<-F
ev2_lim$HY[grepl("HUMAN",ev2_lim$ProteinName) & grepl("YEAST",ev2_lim$ProteinName) & grepl("ECOLI",ev2_lim$ProteinName)] <-T
ev2_lim$HY[grepl("HUMAN",ev2_lim$ProteinName) & grepl("ECOLI",ev2_lim$ProteinName)] <-T
ev2_lim$HY[grepl("HUMAN",ev2_lim$ProteinName) & grepl("YEAST",ev2_lim$ProteinName)] <-T
ev2_lim$HY[grepl("ECOLI",ev2_lim$ProteinName) & grepl("YEAST",ev2_lim$ProteinName)] <-T
table(ev2_lim$HY)
ev2_lim <- ev2_lim[grepl("FALSE", ev2_lim$HY),]
ev2_lim <- ev2_lim %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli", "S. cerevisiae"))
ev2_lim <- ev2_lim %>% mutate("Label" = ifelse(grepl("mTRAQ4|mTRAQ4", Precursor.Id), paste0("\u0394","4"),  paste0("\u0394","0")))

d0 <- ev2_lim[grepl("0", ev2_lim$Label),] %>% dplyr::select("seqcharge", "Ms1.Area", "Precursor.Translated","Ms1.Profile.Corr", "Translated.Q.Value", "species", "Label")
d4 <- ev2_lim[grepl("4", ev2_lim$Label),] %>% dplyr::select("seqcharge", "Ms1.Area", "Precursor.Translated","Ms1.Profile.Corr", "Translated.Q.Value", "species", "Label")
#colnames(d4) <- c("seqcharge", "Ms1_d4", "MS2_d4", "QQ_d4","TQ")

#full_d0d4 <- d0 %>% full_join(d4, by =c("seqcharge" = "seqcharge"))
#################################################

#d4_again <- full_d0d4 %>% dplyr::select("seqcharge", "Ms1_d4", "MS2_d4", "QQ_d4","TQ", "species")
#colnames(d4_again) <- c("seqcharge", "Ms1.Area", "Precursor.Translated", "Ms1.Profile.Corr", "Translated.Q.Value","species")
#d4_again$Label <- paste0("\u0394","4, Yeast")
#d4_again$Label <- paste0("\u0394","4")

#d0_again <- full_d0d4 %>% dplyr::select("seqcharge", "Ms1.Area", "Precursor.Translated", "Ms1.Profile.Corr", "species", "Translated.Q.Value")
#d0_again$Label <- paste0("\u0394","0, Yeast + Bacteria")
#d0_again$Label <- paste0("\u0394","0")

#all <- rbind(d0_again, d4_again)
all <- rbind(d0, d4)

all$spec_lab <- paste0(all$Label," ",all$species)
all <- all %>% mutate(type = ifelse(grepl(paste0("\u0394","4 E. coli"), spec_lab), "3",
                                    ifelse(grepl(paste0("\u0394","0 E. coli"), spec_lab), "1",
                                    ifelse(grepl(paste0("\u0394","0 S. cerevisiae"), spec_lab), "2",
                                           ifelse(grepl(paste0("\u0394","4 S. cerevisiae"), spec_lab), "4","wrong")))))

# all$Ms1.Area[is.na(all$Ms1.Area)] <- 1
# all$Ms1.Area[which(all$Ms1.Area==0)] <- 1
# 
# all$Precursor.Translated[is.na(all$Precursor.Translated)] <- 1
# all$Precursor.Translated[which(all$Precursor.Translated==0)] <- 1
# 
# allnew <- all
# all <- 
################
all_lim <- all %>% dplyr::select("Ms1.Area", "type")
write.table(all_lim, "d0_d4_missingecoli.txt", sep = "\t", row.names = FALSE)

all_lim_MS2 <- all %>% dplyr::select("Precursor.Translated", "type")
write.table(all_lim_MS2, "d0_d4_missingecoli_MS2.txt", sep = "\t", row.names = FALSE)

# this data was then plotted in matlab

```


Supp. Figure 3: CV of plexDIA with same and different labels
```{r}

dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Lib.Q.Value","Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))


df <- dat
df <- df[grepl("wJD803|wJD804|wJD815", df$Run),]

df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- colMedians(protein.groups, na.rm=T)
protein.groups_n <- sweep(protein.groups, 2, cmeds, FUN = '/')

PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% inner_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n_same <- protein.groups_n[!grepl("remove", protein.groups_n$species),]


######################## label swaps ########################################

df <- dat
df1 <- df[grepl("wJD804", df$Run),]
df1 <- df1 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

df2 <- df[grepl("wJD806", df$Run),]
df2 <- df2 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ8",
                                              ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ0", "mTRAQ4")))

df3 <- df[grepl("wJD808", df$Run),]
df3 <- df3 %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ4",
                                              ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ8", "mTRAQ0")))

df <- rbind(df1,df2,df3)

df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- colMedians(protein.groups, na.rm=T)
protein.groups_n_dif <- sweep(protein.groups, 2, cmeds, FUN = '/')

PGs <- rownames(protein.groups)
protein.groups_n_dif <- cbind(protein.groups_n_dif,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n_dif <- data.frame(protein.groups_n_dif)
protein.groups_n_dif <- protein.groups_n_dif %>% inner_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n_dif$HY<-F
protein.groups_n_dif$HY[grepl("HUMAN",protein.groups_n_dif$Protein.Names) & grepl("YEAST",protein.groups_n_dif$Protein.Names) & grepl("ECOLI",protein.groups_n_dif$Protein.Names)] <-T
protein.groups_n_dif$HY[grepl("HUMAN",protein.groups_n_dif$Protein.Names) & grepl("ECOLI",protein.groups_n_dif$Protein.Names)] <-T
protein.groups_n_dif$HY[grepl("HUMAN",protein.groups_n_dif$Protein.Names) & grepl("YEAST",protein.groups_n_dif$Protein.Names)] <-T
protein.groups_n_dif$HY[grepl("ECOLI",protein.groups_n_dif$Protein.Names) & grepl("YEAST",protein.groups_n_dif$Protein.Names)] <-T
table(protein.groups_n_dif$HY)
protein.groups_n_dif <- protein.groups_n_dif[grepl("FALSE", protein.groups_n_dif$HY),]

protein.groups_n_dif <- protein.groups_n_dif%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n_dif <- protein.groups_n_dif[!grepl("remove", protein.groups_n_dif$species),]


## intersect plexDIA and LF-DIA ###
protein.groups_n_same <- protein.groups_n_same[protein.groups_n_same$PGs%in%protein.groups_n_dif$PGs,]
protein.groups_n_dif <- protein.groups_n_dif[protein.groups_n_dif$PGs%in%protein.groups_n_same$PGs,]

rownames(protein.groups_n_same) <- protein.groups_n_same$PGs
rownames(protein.groups_n_dif) <- protein.groups_n_dif$PGs

### plex cont.
protein.groups_n_same <- protein.groups_n_same %>% dplyr::select(-matches(c("PGs", "species", "HY", "Protein.Names")))
m_PGs <- melt(as.matrix(protein.groups_n_same))
m_PGs <- m_PGs %>% mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                           ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs <- m_PGs[!is.na(m_PGs$value),]
m_PGs <- m_PGs %>% group_by(label, Var1) %>% add_count() %>% filter(n==3) %>% ungroup()

m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs <- m_PGs %>% group_by(label, Var1) %>% mutate("Prot_CV" = prot_sd/prot_mean) %>% ungroup
m_PGs$uni <- paste0(m_PGs$Var1,m_PGs$label)
m_PGs <- m_PGs %>% distinct(uni, .keep_all=T)
m_PGs$Cond <- "Same Labels"

###### swap cont.
protein.groups_n <- protein.groups_n_dif %>% dplyr::select(-matches(c("PGs", "species", "HY", "Protein.Names")))
m_PGs_dif <- melt(as.matrix(protein.groups_n))
m_PGs_dif <- m_PGs_dif %>% mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                           ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs_dif <- m_PGs_dif[!is.na(m_PGs_dif$value),]
m_PGs_dif <- m_PGs_dif %>% group_by(label, Var1) %>% add_count() %>% filter(n==3) %>% ungroup()

m_PGs_dif <- m_PGs_dif %>% group_by(label, Var1) %>% mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs_dif <- m_PGs_dif %>% group_by(label, Var1) %>% mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs_dif <- m_PGs_dif %>% group_by(label, Var1) %>% mutate("Prot_CV" = prot_sd/prot_mean) %>% ungroup
m_PGs_dif$uni <- paste0(m_PGs_dif$Var1,m_PGs_dif$label)
m_PGs_dif <- m_PGs_dif %>% distinct(uni, .keep_all=T)
m_PGs_dif$Cond <- "Different Labels"

####
m_PGs_dif$Var1_lab <- paste0(m_PGs_dif$Var1,m_PGs_dif$label)
m_PGs$Var1_lab <- paste0(m_PGs$Var1,m_PGs$label)

## last intersection:
m_PGs_dif <- m_PGs_dif[m_PGs_dif$Var1_lab%in%m_PGs$Var1_lab,]
m_PGs <- m_PGs[m_PGs$Var1_lab%in%m_PGs_dif$Var1_lab,]

################ select and bind:

mDIA_LF <- bind_rows(m_PGs,m_PGs_dif)
mDIA_LF <- mDIA_LF %>% group_by(Cond) %>% add_count()

mDIA_LF$Cond <- factor(mDIA_LF$Cond, levels=c("Same Labels", "Different Labels"))
df1 <- mDIA_LF %>% mutate("med_CV" = round(median(Prot_CV,na.rm=T),3)) %>% distinct(Cond,.keep_all=T)

ymax <- mDIA_LF$Prot_CV %>% sort()
ymax <- as.numeric(quantile(ymax,probs=c(.01,.99)))

CV1_fig <- ggplot(mDIA_LF, aes(x=Cond,y=Prot_CV, fill=Cond)) + geom_boxplot( alpha =c(0.6,0.7), outlier.shape=NA) +
  scale_fill_manual(values = c("red", "mediumpurple2")) +
  theme_classic() +
  geom_text(data=df1, aes(x=Cond, y=c(0.14,0.18), label=paste0("med=",med_CV)), size=4) +
  # facet_grid(~species, scales = "free") + geom_hline(data = df1, aes(yintercept = Z), color = "red", size = 1.052, linetype = "dashed") +
  theme(axis.text.x = element_text(size=14, face = "bold", color = "black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=14),
        axis.title.x = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Sample-specific protein CV", fill = "") + ylim(0,ymax[2])


CV1_fig
ggsave("SFigure3.pdf", width=4.5,height=3.5)

```

Supp. Figure MS1 rep 1
```{r}

dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))


df <- dat
df <- df[grepl("wJD803", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD803mTRAQ0)/as.numeric(protein.groups_n$wJD803mTRAQ4)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio




######################## LF ########################################

dat_LF <- fread(LF_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                                     "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                                     "Mass.Evidence", "CScore", "Fragment.Correlations"))
#evt <- read.delim(fpath)

#dat_1hr <- dat
df <- dat_LF
df <- df[grepl("wJD756_re|wJD760_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD756_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD760_re", Run), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD756_remTRAQ0)/as.numeric(protein.groups_n$wJD760_remTRAQ4)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD803mTRAQ4", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF %>% dplyr::select("wJD760_remTRAQ4", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF_a) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF_a$sample2_int <- as.numeric(protein.groups_n_omit_LF_a$sample2_int)



#intersect LF and multiDIA
protein.groups_n_omit_noint <- protein.groups_n_omit
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF

protein.groups_n_omit_a <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF_a$PGs,]
protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF_a[protein.groups_n_omit_LF_a$PGs%in%protein.groups_n_omit$PGs,]


LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_a, protein.groups_n_omit_LF_a)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
a2 <- ggplot(protein.groups_n_omit_LF_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y="Log2, delta0:delta4",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())
a2


a4 <- ggplot(protein.groups_n_omit_LF_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a4
####################################
#d0d4
a1 <- ggplot(protein.groups_n_omit_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y=expression(paste(Log["2"],", A/B")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))

a1 

a3 <- ggplot(protein.groups_n_omit_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a3


################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

a_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+300, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (A/B)", fill = "")
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
a_cover
#ggsave("Quantifiable_prot_rats_07012021_fil_MaxLFQ.png", width=4.5,height=4)







################################################################
################################################################
########################## d0:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl("wJD803", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD803mTRAQ0)/as.numeric(protein.groups_n$wJD803mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl("wJD756_re|wJD762_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD756_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD762_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD756_remTRAQ0)/as.numeric(protein.groups_n$wJD762_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD803mTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select("wJD762_remTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA


protein.groups_n_omit_b <- protein.groups_n_omit
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF

protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[!grepl("H.", protein.groups_n_omit_LF_b$species),]
protein.groups_n_omit_b <- protein.groups_n_omit_b[!grepl("H.", protein.groups_n_omit_b$species),]

protein.groups_n_omit_noint <- protein.groups_n_omit_b
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_b

protein.groups_n_omit_b <- protein.groups_n_omit_b[protein.groups_n_omit_b$PGs%in%protein.groups_n_omit_LF_b$PGs,]
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[protein.groups_n_omit_LF_b$PGs%in%protein.groups_n_omit_b$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_b, protein.groups_n_omit_LF_b)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
b2 <- ggplot(protein.groups_n_omit_LF_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta0:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


b4 <- ggplot(protein.groups_n_omit_LF_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
b1 <- ggplot(protein.groups_n_omit_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_bolor_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", A/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


b3 <- ggplot(protein.groups_n_omit_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])





################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("MultiDIA", "Label-Free", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("MultiDIA", "Label-Free", "Intersected"))
library(Cairo)

b_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (A/C)", fill = "", title = expression(paste(italic(H.~sapiens),' excluded')))
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
b_cover
#ggsave("Quantifiable_prot_rats_07012021_fil_MaxLFQ.png", width=4.5,height=4)






########################## d4:d8 ###############################

df <- dat
df <- df[grepl("wJD803", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD803mTRAQ4)/as.numeric(protein.groups_n$wJD803mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl("wJD760_re|wJD762_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD760_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD762_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD760_remTRAQ0)/as.numeric(protein.groups_n$wJD762_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD803mTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select("wJD762_remTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA
protein.groups_n_omit_c <- protein.groups_n_omit
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF

protein.groups_n_omit_noint <- protein.groups_n_omit_c
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_c

protein.groups_n_omit_c <- protein.groups_n_omit_c[protein.groups_n_omit_c$PGs%in%protein.groups_n_omit_LF_c$PGs,]
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF_c[protein.groups_n_omit_LF_c$PGs%in%protein.groups_n_omit_c$PGs,]



LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_c, protein.groups_n_omit_LF_c)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
c2 <- ggplot(protein.groups_n_omit_LF_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta4:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


c4 <- ggplot(protein.groups_n_omit_LF_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
c1 <- ggplot(protein.groups_n_omit_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", B/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


c3 <- ggplot(protein.groups_n_omit_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])



################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

c_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "top",
        legend.direction = "vertical",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (B/C)", fill = "", title = expression(paste(italic(H.~sapiens)," excluded")))
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
c_cover
#ggsave("Quantifiable_prot_rats_07012021_fil_MaxLFQ.png", width=4.5,height=4)



############## combine


g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(a3)
mylegend1<-g_legend(c_cover)


a_fin <- grid.arrange(mylegend, nrow=2,heights=c(1, 10),
                      arrangeGrob(a_cover + theme(legend.position="none"),
                                  a1 + theme(legend.position="none"),
                                  a2 + theme(legend.position="none"), 
                                  a3 + theme(legend.position="none"),
                                  a4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

c_fin <- grid.arrange(nrow=1,
                      arrangeGrob(c_cover + theme(legend.position="none"),
                                  c1 + theme(legend.position="none"),
                                  c2 + theme(legend.position="none"), 
                                  c3 + theme(legend.position="none"),
                                  c4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

b_fin <- grid.arrange(nrow=1,
                      arrangeGrob(b_cover + theme(legend.position="none"),
                                  b1 + theme(legend.position="none"),
                                  b2 + theme(legend.position="none"), 
                                  b3 + theme(legend.position="none"),
                                  b4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

library(gridExtra)
library(ggpubr)

figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 5,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1))


ggsave("SFigure_MS1_run1.pdf", figure_fin, width=11,height=9, dpi=500)

```


Supp. Figure MS1 rep 3
```{r}

dat <- fread(fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))


df <- dat
df <- df[grepl("wJD815", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD815mTRAQ0)/as.numeric(protein.groups_n$wJD815mTRAQ4)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio




######################## LF ########################################

dat_LF <- fread(LF_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                                     "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                                     "Mass.Evidence", "CScore", "Fragment.Correlations"))
#evt <- read.delim(fpath)

#dat_1hr <- dat
df <- dat_LF
df <- df[grepl("wJD758_re|wJD761_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD758_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD761_re", Run), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD758_remTRAQ0)/as.numeric(protein.groups_n$wJD761_remTRAQ4)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD815mTRAQ4", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF %>% dplyr::select("wJD761_remTRAQ4", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF_a) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF_a$sample2_int <- as.numeric(protein.groups_n_omit_LF_a$sample2_int)



#intersect LF and multiDIA
protein.groups_n_omit_noint <- protein.groups_n_omit
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF

protein.groups_n_omit_a <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF_a$PGs,]
protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF_a[protein.groups_n_omit_LF_a$PGs%in%protein.groups_n_omit$PGs,]


LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_a, protein.groups_n_omit_LF_a)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
a2 <- ggplot(protein.groups_n_omit_LF_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y="Log2, delta0:delta4",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())
a2


a4 <- ggplot(protein.groups_n_omit_LF_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a4
####################################
#d0d4
a1 <- ggplot(protein.groups_n_omit_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y=expression(paste(Log["2"],", A/B")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))

a1 

a3 <- ggplot(protein.groups_n_omit_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a3


################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

a_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+300, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (A/B)", fill = "")
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
a_cover
#ggsave("Quantifiable_prot_rats_07012021_fil_MaxLFQ.png", width=4.5,height=4)







################################################################
################################################################
########################## d0:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl("wJD815", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD815mTRAQ0)/as.numeric(protein.groups_n$wJD815mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl("wJD758_re|wJD764_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD758_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD764_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD758_remTRAQ0)/as.numeric(protein.groups_n$wJD764_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD815mTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select("wJD764_remTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA


protein.groups_n_omit_b <- protein.groups_n_omit
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF

protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[!grepl("H.", protein.groups_n_omit_LF_b$species),]
protein.groups_n_omit_b <- protein.groups_n_omit_b[!grepl("H.", protein.groups_n_omit_b$species),]

protein.groups_n_omit_noint <- protein.groups_n_omit_b
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_b

protein.groups_n_omit_b <- protein.groups_n_omit_b[protein.groups_n_omit_b$PGs%in%protein.groups_n_omit_LF_b$PGs,]
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[protein.groups_n_omit_LF_b$PGs%in%protein.groups_n_omit_b$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_b, protein.groups_n_omit_LF_b)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
b2 <- ggplot(protein.groups_n_omit_LF_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta0:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


b4 <- ggplot(protein.groups_n_omit_LF_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
b1 <- ggplot(protein.groups_n_omit_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_bolor_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", A/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


b3 <- ggplot(protein.groups_n_omit_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])





################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("MultiDIA", "Label-Free", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("MultiDIA", "Label-Free", "Intersected"))
library(Cairo)

b_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (A/C)", fill = "", title = expression(paste(italic(H.~sapiens),' excluded')))
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
b_cover
#ggsave("Quantifiable_prot_rats_07012021_fil_MaxLFQ.png", width=4.5,height=4)






########################## d4:d8 ###############################

df <- dat
df <- df[grepl("wJD815", df$Run),]
#df <- df[(df$Ms1.Profile.Corr > 0.5),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD815mTRAQ4)/as.numeric(protein.groups_n$wJD815mTRAQ8)
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl("wJD761_re|wJD764_re", df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("wJD761_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD764_re", Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.Q.Value <= 0.01 & df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Ms1.Area")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n$wJD761_remTRAQ0)/as.numeric(protein.groups_n$wJD764_remTRAQ8)
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select("wJD815mTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select("wJD764_remTRAQ8", "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA
protein.groups_n_omit_c <- protein.groups_n_omit
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF

protein.groups_n_omit_noint <- protein.groups_n_omit_c
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_c

protein.groups_n_omit_c <- protein.groups_n_omit_c[protein.groups_n_omit_c$PGs%in%protein.groups_n_omit_LF_c$PGs,]
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF_c[protein.groups_n_omit_LF_c$PGs%in%protein.groups_n_omit_c$PGs,]



LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_c, protein.groups_n_omit_LF_c)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
c2 <- ggplot(protein.groups_n_omit_LF_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta4:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


c4 <- ggplot(protein.groups_n_omit_LF_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
c1 <- ggplot(protein.groups_n_omit_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", B/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


c3 <- ggplot(protein.groups_n_omit_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])



################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

c_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "top",
        legend.direction = "vertical",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (B/C)", fill = "", title = expression(paste(italic(H.~sapiens)," excluded")))
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
c_cover



############## combine


g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(a3)
mylegend1<-g_legend(c_cover)


a_fin <- grid.arrange(mylegend, nrow=2,heights=c(1, 10),
                      arrangeGrob(a_cover + theme(legend.position="none"),
                                  a1 + theme(legend.position="none"),
                                  a2 + theme(legend.position="none"), 
                                  a3 + theme(legend.position="none"),
                                  a4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

c_fin <- grid.arrange(nrow=1,
                      arrangeGrob(c_cover + theme(legend.position="none"),
                                  c1 + theme(legend.position="none"),
                                  c2 + theme(legend.position="none"), 
                                  c3 + theme(legend.position="none"),
                                  c4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

b_fin <- grid.arrange(nrow=1,
                      arrangeGrob(b_cover + theme(legend.position="none"),
                                  b1 + theme(legend.position="none"),
                                  b2 + theme(legend.position="none"), 
                                  b3 + theme(legend.position="none"),
                                  b4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

library(gridExtra)
library(ggpubr)

figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 5,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1))


ggsave("SFigure_MS1_Run3.pdf", figure_fin, width=11,height=9, dpi=500)


```


Supp Figure 5: MS2-optimized, Coverage and completeness removed s3 rep 3 for LFDIA
```{r}

plexDIA_run <- "eJD906" #plexDIA run name you are interested in plotting
LF_runs <- c("eJD909", "eJD912", "eJD915") # LF-DIA runs you are interested in plotting

# Figure 2, Proteomic coverage and overlap between samples and runs of plexDIA and LF-DIA:
mTRAQ <- fread(MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value","Precursor.Quantity", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated"))

mTRAQ<- mTRAQ %>% filter(Translated.Q.Value<=tqval) 
#mTRAQ <- mTRAQ[which(mTRAQ$Precursor.Translated>0),]  # must have non-zero Ms1 quant
mTRAQ$Label <- "mTRAQ"
mTRAQ$seqcharge <- paste0(mTRAQ$Modified.Sequence, mTRAQ$Precursor.Charge)
mTRAQ$seqcharge_run <- paste0(mTRAQ$seqcharge, "_", mTRAQ$Run)
mTRAQ$prot_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ0\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ4\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ8\\)")
mTRAQ_uni_prot <- unique(mTRAQ$Protein.Group)

ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$Precursor.Id),] %>% distinct(seqcharge_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_4) <- c("Run","Protein.Group_d4","seqcharge_d4","Precursor.Quantity_d4","Precursor.Translated_d4", "seqcharge_run_d4", "prot_run_d4","Lib.PG.Q.Value_d4")
ev2_8 <- mTRAQ[grepl("mTRAQ8",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_8) <- c("Run","Protein.Group_d8","seqcharge_d8","Precursor.Quantity_d8","Precursor.Translated_d8", "seqcharge_run_d8", "prot_run_d8","Lib.PG.Q.Value_d8")

mTRAQ_int <- ev2_0 %>% inner_join(ev2_4, by =c("seqcharge" = "seqcharge_d4"))
mTRAQ_int <- mTRAQ_int %>% inner_join(ev2_8, by =c("seqcharge" = "seqcharge_d8"))

ev2_0_prot <- ev2_0[which(ev2_0$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
ev2_0_prot$Sample <- paste0("\u0394","0-A")
ev2_4_prot <- ev2_4[which(ev2_4$Lib.PG.Q.Value_d4 < 0.01),] %>% distinct(prot_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4","prot_run_d4")
colnames(ev2_4_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_4_prot$Sample <- paste0("\u0394","4-B")
ev2_8_prot <- ev2_8[which(ev2_8$Lib.PG.Q.Value_d8 < 0.01),] %>% distinct(prot_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8","prot_run_d8")
colnames(ev2_8_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_8_prot$Sample <- paste0("\u0394","8-C")

ev2_0_pep <- ev2_0 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_0_pep$Sample <- paste0("\u0394","0-A")
ev2_4_pep <- ev2_4 %>% distinct(seqcharge_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4", "seqcharge_d4", "seqcharge_run_d4")
colnames(ev2_4_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_4_pep$Sample <- paste0("\u0394","4-B")
ev2_8_pep <- ev2_8 %>% distinct(seqcharge_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8", "seqcharge_d8", "seqcharge_run_d8")
colnames(ev2_8_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_8_pep$Sample <- paste0("\u0394","8-C")

mTRAQ_prots <- rbind(ev2_0_prot,ev2_4_prot,ev2_8_prot)
mTRAQ_prots$Sample_new <- mTRAQ_prots$Sample

mTRAQ_peps <- rbind(ev2_0_pep,ev2_4_pep,ev2_8_pep)
mTRAQ_peps$Sample_new <- mTRAQ_peps$Sample

p_mTRAQ <- mTRAQ_peps %>% dplyr::count(Run, Sample_new)
p_mTRAQ$Condition <- "plexDIA"

prot_mTRAQ <- mTRAQ_prots %>% dplyr::count(Run, Sample_new)
prot_mTRAQ$Condition <- "plexDIA"



##############
##############
#######################
########################
##########################  Label-free

unlab <- fread(LF_MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated"))

unlab<- unlab %>% filter(Translated.Q.Value<=tqval) 
#unlab <- unlab[which(unlab$Precursor.Translated>0),]  # must have non-zero Ms1 quant
unlab_1 <- unlab[grepl("eJD908|eJD909|eJD910", unlab$Run),]
unlab_1$seqcharge <- paste0(unlab_1$Modified.Sequence, unlab_1$Precursor.Charge)
unlab_1$seqcharge_run <- paste0(unlab_1$seqcharge, "_", unlab_1$Run)
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_", unlab_1$Run)

unlab_1_prot <- unlab_1[which(unlab_1$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_1_prot$Sample <- paste0("\u0394","0-A")
unlab_1_prot$Sample_new <- paste0("A")

unlab_1_pep <- unlab_1 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_1_pep$Sample <- paste0("\u0394","0-A")
unlab_1_pep$Sample_new <- paste0("A")

##############
unlab_2 <- unlab[grepl("eJD911|eJD912|eJD913", unlab$Run),]
unlab_2$seqcharge <- paste0(unlab_2$Modified.Sequence, unlab_2$Precursor.Charge)
unlab_2$seqcharge_run <- paste0(unlab_2$seqcharge, "_", unlab_2$Run)
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_", unlab_2$Run)

unlab_2_prot <- unlab_2[which(unlab_2$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_2_prot$Sample <- paste0("\u0394","4-B")
unlab_2_prot$Sample_new <- paste0("B")

unlab_2_pep <- unlab_2 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_2_pep$Sample <- paste0("\u0394","4-B")
unlab_2_pep$Sample_new <- paste0("B")

##############
unlab_3 <- unlab[grepl("eJD914|eJD915", unlab$Run),]
unlab_3$seqcharge <- paste0(unlab_3$Modified.Sequence, unlab_3$Precursor.Charge)
unlab_3$seqcharge_run <- paste0(unlab_3$seqcharge, "_", unlab_3$Run)
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_", unlab_3$Run)

unlab_3_prot <- unlab_3[which(unlab_3$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_3_prot$Sample <- paste0("\u0394","8-C")
unlab_3_prot$Sample_new <- paste0("C")


unlab_3_pep <- unlab_3 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_3_pep$Sample <- paste0("\u0394","8-C")
unlab_3_pep$Sample_new <- paste0("C")

###################

all_unlab_pep <- rbind(unlab_1_pep, unlab_2_pep, unlab_3_pep)
all_unlab_prot <- rbind(unlab_1_prot, unlab_2_prot, unlab_3_prot)

p_unlab <- all_unlab_pep %>% dplyr::count(Run, Sample_new)
p_unlab$Condition <- "LF-DIA"

prot_unlab <- all_unlab_prot %>% dplyr::count(Run, Sample_new)
prot_unlab$Condition <- "LF-DIA"

##############
##############
#######################
########################
##########################  mTRAQ DDA

#get prots which are 1% FDR
prot_DDAFDR <- read.delim(DDA_prot_fpath)
prot_DDAFDR <- prot_DDAFDR[which(prot_DDAFDR$Q.value<0.01),] %>% filter(!grepl("REV|CON", Protein.IDs))
prot_DDAFDR$Protein.Group <- gsub(";.*","",prot_DDAFDR$Majority.protein.IDs)

DDA <- read.delim(DDA_fpath)
DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
## find FDR 1 % for each run:
DDA1 <- DDA %>% group_by(Raw.file) %>% arrange(PEP) %>% 
  mutate(rec = 1) %>% 
  mutate(rollavg = cumsum(PEP)/cumsum(rec)) %>% 
  select(-rec) %>% ungroup()
DDA1 <- DDA1 %>% dplyr::select("Modified.sequence","PEP","Raw.file","rollavg")
DDAfil <- DDA1 %>% filter(rollavg < 0.01) %>% group_by(Raw.file) %>% 
  summarise(PEP = max(PEP))

#filter for 1% FDR
DDA_list <- list()
for( i in 1:3){
  a <- as.numeric(paste0(DDAfil[i,2]))
  temp <- DDA[grepl(paste0(DDAfil[i,1]), DDA$Raw.file),] %>% filter(PEP < a)
  DDA_list[[i]] <- temp
}
DDA <- do.call("rbind",DDA_list) #combine all vectors into a matrix

DDA_d0 <- DDA[which(DDA$Intensity.L > 0),]
DDA_d0$seqcharge <- paste0(DDA_d0$Modified.sequence, DDA_d0$Charge)
DDA_d0$seqcharge_run <- paste0(DDA_d0$seqcharge, "_", DDA_d0$Raw.file)
DDA_d0$Protein.Group <- gsub(";.*","",DDA_d0$Leading.razor.protein)
DDA_d0$prot_run <- paste0(DDA_d0$Leading.razor.protein, "_", DDA_d0$Raw.file)

DDA_d0_prot <- DDA_d0 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d0_prot$Sample <- paste0("A")
DDA_d0_prot$Sample_new <- paste0("\u0394","0 DDA-A")

DDA_d0_pep <- DDA_d0 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d0_pep$Sample <- paste0("A")
DDA_d0_pep$Sample_new <- paste0("\u0394","0 DDA-A")

####
DDA_d4 <- DDA[which(DDA$Intensity.M > 0),]
DDA_d4$seqcharge <- paste0(DDA_d4$Modified.sequence, DDA_d4$Charge)
DDA_d4$seqcharge_run <- paste0(DDA_d4$seqcharge, "_", DDA_d4$Raw.file)
DDA_d4$Protein.Group <- gsub(";.*","",DDA_d4$Leading.razor.protein)
DDA_d4$prot_run <- paste0(DDA_d4$Leading.razor.protein, "_", DDA_d4$Raw.file)

DDA_d4_prot <- DDA_d4 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d4_prot$Sample <- paste0("B")
DDA_d4_prot$Sample_new <- paste0("\u0394","4 DDA-B")

DDA_d4_pep <- DDA_d4 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d4_pep$Sample <- paste0("B")
DDA_d4_pep$Sample_new <- paste0("\u0394","4 DDA-B")

####
DDA_d8 <- DDA[which(DDA$Intensity.H > 0),]
DDA_d8$seqcharge <- paste0(DDA_d8$Modified.sequence, DDA_d8$Charge)
DDA_d8$seqcharge_run <- paste0(DDA_d8$seqcharge, "_", DDA_d8$Raw.file)
DDA_d8$Protein.Group <- gsub(";.*","",DDA_d8$Leading.razor.protein)
DDA_d8$prot_run <- paste0(DDA_d8$Leading.razor.protein, "_", DDA_d8$Raw.file)

DDA_d8_prot <- DDA_d8 %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d8_prot$Sample <- paste0("C")
DDA_d8_prot$Sample_new <- paste0("\u0394","8 DDA-C")

DDA_d8_pep <- DDA_d8 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d8_pep$Sample <- paste0("C")
DDA_d8_pep$Sample_new <- paste0("\u0394","8 DDA-C")


##########################
##########################  data-wrangling for plotting
all_DDA_pep <- rbind(DDA_d0_pep, DDA_d4_pep, DDA_d8_pep)
all_DDA_prot <- rbind(DDA_d0_prot, DDA_d4_prot, DDA_d8_prot)
all_DDA_prot <- all_DDA_prot[which(all_DDA_prot$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

p_DDA <- all_DDA_pep %>% dplyr::count(Raw.file, Sample_new)
colnames(p_DDA) <- c("Run", "Sample_new", "n")
p_DDA$Condition <- "mTRAQ\nDDA"

prot_DDA <- all_DDA_prot %>% dplyr::count(Raw.file, Sample_new)
colnames(prot_DDA) <- c("Run", "Sample_new", "n")
prot_DDA$Condition <- "mTRAQ\nDDA"

###################
all_pep <- rbind(p_mTRAQ, p_unlab, p_DDA)
all_pep$Sample <- gsub(".*-","",all_pep$Sample_new)

all_pep <- all_pep %>% group_by(Condition, Sample) %>% 
  add_count() %>% ungroup()
p_all_tab <- all_pep %>% group_by(Sample_new) %>% mutate(mean = mean(n)) %>%
  mutate(se = sd(n)/sqrt(nn)) %>% distinct(Sample_new, .keep_all=T)
p_all_tab <- p_all_tab[,-6]
a <- p_all_tab[3,6]
b <- p_all_tab[2,6] + p_all_tab[3,6]
c <- p_all_tab[2,6] + p_all_tab[1,6] + p_all_tab[3,6]

p_all_tab$er_max <- p_all_tab$mean + p_all_tab$se
p_all_tab[1,8] <- c+as.numeric(p_all_tab[1,7])
p_all_tab[2,8] <- b+as.numeric(p_all_tab[2,7])
p_all_tab[3,8] <- a+p_all_tab[3,7]

p_all_tab$er_min <- p_all_tab$mean - p_all_tab$se
p_all_tab[1,9] <- c-as.numeric(p_all_tab[1,7])
p_all_tab[2,9] <- b-as.numeric(p_all_tab[2,7])
p_all_tab[3,9] <- a-p_all_tab[3,7]


## for DDA
a <- p_all_tab[9,6]
b <- p_all_tab[8,6] + p_all_tab[9,6]
c <- p_all_tab[8,6] + p_all_tab[7,6] + p_all_tab[9,6]

p_all_tab[7,8] <- c+as.numeric(p_all_tab[7,7])
p_all_tab[8,8] <- b+as.numeric(p_all_tab[8,7])
p_all_tab[9,8] <- a+p_all_tab[9,7]

p_all_tab[7,9] <- c-as.numeric(p_all_tab[7,7])
p_all_tab[8,9] <- b-as.numeric(p_all_tab[8,7])
p_all_tab[9,9] <- a-p_all_tab[9,7]


#### text 
df_n <- p_all_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+10000
my_fin_colors <- c("turquoise3","palevioletred3","lightslategrey")

p1_cov <- ggplot(p_all_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = p_all_tab %>% filter(Condition == "plexDIA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "mTRAQ\nDDA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "LF-DIA"),# colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Precursors / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12),
        legend.position = c(0.7,0.8),
        legend.direction = "horizontal",
        legend.spacing.x = unit(1.0, 'line')) +
  guides(fill = guide_legend(override.aes= list(alpha = 0.7),
                             title = "Samples",
                             label.position = "bottom",
                             title.position = "top", title.vjust = 1, title.hjust=0.5)) +
  geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3
  ) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3, stat="identity", position=position_dodge(1.17))

p1_cov
ggsave("SFigure4a.pdf",p1_cov,width=5.6,height=5, dpi=500)


############ prots

###################
all_prot <- rbind(prot_mTRAQ, prot_unlab, prot_DDA)
all_prot$Sample <- gsub(".*-","",all_prot$Sample_new)
all_prot <- all_prot %>% group_by(Condition, Sample) %>% 
  add_count() %>% ungroup()
all_prot_tab <- all_prot %>% group_by(Sample_new) %>% mutate(mean = mean(n)) %>%
  mutate(se = sd(n)/sqrt(nn)) %>% distinct(Sample_new, .keep_all=T)
all_prot_tab <- all_prot_tab[,-6]

a <- all_prot_tab[3,6]
b <- all_prot_tab[2,6] + all_prot_tab[3,6]
c <- all_prot_tab[2,6] + all_prot_tab[1,6] + all_prot_tab[3,6]

all_prot_tab$er_max <- all_prot_tab$mean + all_prot_tab$se
all_prot_tab[1,8] <- c+as.numeric(all_prot_tab[1,7])
all_prot_tab[2,8] <- b+as.numeric(all_prot_tab[2,7])
all_prot_tab[3,8] <- a+all_prot_tab[3,7]

all_prot_tab$er_min <- all_prot_tab$mean - all_prot_tab$se
all_prot_tab[1,9] <- c-as.numeric(all_prot_tab[1,7])
all_prot_tab[2,9] <- b-as.numeric(all_prot_tab[2,7])
all_prot_tab[3,9] <- a-all_prot_tab[3,7]


## for DDA
a <- all_prot_tab[9,6]
b <- all_prot_tab[8,6] + all_prot_tab[9,6]
c <- all_prot_tab[8,6] + all_prot_tab[7,6] + all_prot_tab[9,6]

all_prot_tab[7,8] <- c+as.numeric(all_prot_tab[7,7])
all_prot_tab[8,8] <- b+as.numeric(all_prot_tab[8,7])
all_prot_tab[9,8] <- a+all_prot_tab[9,7]

all_prot_tab[7,9] <- c-as.numeric(all_prot_tab[7,7])
all_prot_tab[8,9] <- b-as.numeric(all_prot_tab[8,7])
all_prot_tab[9,9] <- a-all_prot_tab[9,7]


#### text 
df_n <- all_prot_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+700

my_fin_colors <- c("turquoise3","palevioletred3","lightslategrey")

p2_cov <- ggplot(all_prot_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = all_prot_tab %>% filter(Condition == "plexDIA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "mTRAQ\nDDA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "LF-DIA"),# colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Protein Data Points / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.position = "none") +
   geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2, stat="identity", position=position_dodge(1.17))

p2_cov
ggsave("SFigure4b.pdf",p2_cov,width=5.6,height=5, dpi=500)


################## JACCARD 

mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Precursor.Translated>0),]  #only keep precursors with quant info

ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$Precursor.Id),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("mTRAQ8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Protein.Group")

d0_1 <- ev2_0[grepl("eJD905", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("eJD905", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_1 <- ev2_8[grepl("eJD905", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("eJD906", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("eJD906", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_2 <- ev2_8[grepl("eJD906", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("eJD907", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("eJD907", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_3 <- ev2_8[grepl("eJD907", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value

################################### unlabeled
#unlab <- read.delim(LF_fpath)
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]

df_t <- data.frame(c("eJD908", "eJD909", "eJD910"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Precursor.Translated>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c("eJD911", "eJD912", "eJD913"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Precursor.Translated>0),]  #only keep precursors with quant info

##

df_t <- data.frame(c("eJD914", "eJD915"), 
                   c("1", "2"))
colnames(df_t) <- c("file", "rep")
unlab_3 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_",unlab_3$rep)
unlab_3 <- unlab_3[which(unlab_3$Precursor.Translated>0),]  #only keep precursors with quant info

#####
s1_1 <- unlab_1[grepl("eJD908", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("eJD909", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("eJD910", unlab_1$Run),] %>% dplyr::select(Protein.Group)

s2_1 <- unlab_2[grepl("eJD911", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("eJD912", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("eJD913", unlab_2$Run),] %>% dplyr::select(Protein.Group)

s3_1 <- unlab_3[grepl("eJD914", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_2 <- unlab_3[grepl("eJD915", unlab_3$Run),] %>% dplyr::select(Protein.Group)
#s3_3 <- unlab_3[grepl("eJD930", unlab_3$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3)
for(i in 1:8){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s3_1","s1_2","s2_2","s3_2","s1_3","s2_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################



############### Jaccard index  DDA

DDA <- read.delim(DDA_fpath)

#filter for 1% FDR
DDA_list <- list()
for( i in 1:3){
  a <- as.numeric(paste0(DDAfil[i,2]))
  temp <- DDA[grepl(paste0(DDAfil[i,1]), DDA$Raw.file),] %>% filter(PEP < a)
  DDA_list[[i]] <- temp
}
DDA <- do.call("rbind",DDA_list) #combine all vectors into a matrix
DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_1 <- ev2_8[grepl("wJD750", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_2 <- ev2_8[grepl("wJD751", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_3 <- ev2_8[grepl("wJD752", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")


jac_dist_DDA <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
jac_dist_DDA <- get_lower_tri(jac_dist_DDA)
jac_dist_DDA_m <- melt(jac_dist_DDA, na.rm=T)
jac_dist_DDA_m$value <- 1-jac_dist_DDA_m$value

################################# plot
######################### DDA
jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                ifelse(grepl("_2", Var1),"Rep 2",
                                                                       ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                ifelse(grepl("_2", Var2),"Rep 2",
                                                                       ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))


labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

pDDA <- ggplot(jac_dist_DDA_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="mTRAQ DDA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  #scale_fill_gradient2(high = "aquamarine", low = "maroon",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  

############### mTRAQ

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="plexDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  #scale_fill_gradient2(high = "aquamarine", low = "maroon",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) 
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))  


############## LF
m_jac_LF <- m_jac_LF %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="LF-DIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  #scale_fill_gradient2(high = "aquamarine", low = "maroon",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) 
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.79,limits=c(0.58, 1), breaks = c(0.6, 0.8, 1))  


## combine
library(gridExtra)
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                            p1 + theme(legend.position="none"), NULL,
                            pDDA +theme(legend.position="none"),
                            nrow=1, widths=c(1,0.1,1,0.1,1)))

#install.packages('svglite')

ggsave("SFigure4c.pdf", p3, width=12.5, height=5.2, dpi=500)





############################ other plots:
jac_dist_mTRAQ_m$Type <- "plexDIA"
m_jac_LF$Type <- "LF-DIA"

jac_dist_mTRAQ_m$run_a <- stri_sub(jac_dist_mTRAQ_m$Var1,-1,-1)
jac_dist_mTRAQ_m$run_b <- stri_sub(jac_dist_mTRAQ_m$Var2,-1,-1)
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("set" = ifelse(run_a == run_b, "Within a run", "Across runs"))

m_jac_LF$run_a <- stri_sub(m_jac_LF$Var1,-1,-1)
m_jac_LF$run_b <- stri_sub(m_jac_LF$Var2,-1,-1)
m_jac_LF <- m_jac_LF %>% mutate("set" = "Across runs")

both <- bind_rows(jac_dist_mTRAQ_m, m_jac_LF)
both <- both[which(both$value<1),]  #remove overlaps to self

both$Var1a <- substr(both$Var1,1,nchar(as.character(both$Var1))-1)
both$Var2a <- substr(both$Var2,1,nchar(as.character(both$Var2))-1)

both <- both %>% mutate("Groups" = ifelse(Var1a == Var2a, "Same samples", "Different samples"))
both$set <- factor(both$set, levels=c("Within a run", "Across runs"))
both$Type <- factor(both$Type, levels=c("plexDIA", "LF-DIA"))
both$missingness <- (1-both$value)*100
both$Groups <- factor(both$Groups, levels=c("Same samples", "Different samples"))

ggplot(both, aes(x=as.factor(Type),y=(missingness), color=set)) + 
  geom_point(size=2.3, alpha =0.7, position=position_jitter(width=0.25, height=0)) +
  #geom_text(data=df_n, aes(x=as.factor(Label), y=log2(d0_d4_rat), label=n), col='black', size=4) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA")) + 
  scale_color_manual(values = c("steelblue", "tomato1")) + #"tan2"
  theme_bw() +
  facet_grid(~Groups, scales = "free") +
  #ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=16),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12),
        legend.text = element_text(size=14),
        legend.title = element_blank())+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Missing data between pairs, %", fill = "") + ylim(0, 30)

ggsave("SFigure4d.pdf", width=3.9, height=4, dpi=500)

```


Supp. Figure 6: MS2 optimized method, quantitative accuracy
```{r}
#1% FDR:

MS2_plexDIA_run <- "eJD906"
LF_MS2_runs <- c("eJD909", "eJD912", "eJD915") # LF-DIA runs you are interested in plotting

dat <- mTRAQ
df <- dat
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
protein.groups_n <- protein.groups                                      
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)
df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]
plexDIA_run <- "eJD906"
pDIAa <- paste0(MS2_plexDIA_run,"mTRAQ0")
pDIAb <- paste0(MS2_plexDIA_run,"mTRAQ4")
pDIAc <- paste0(MS2_plexDIA_run,"mTRAQ8")

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAa]])/as.numeric(protein.groups_n[[pDIAb]])


protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio

######################## LF ########################################

dat_LF <- unlab

df <- dat_LF
df <- df[grepl(paste0(LF_MS2_runs[1],"|",LF_MS2_runs[2]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_MS2_runs[1]), Run), "mTRAQ0",
                                            ifelse(grepl(paste0(LF_MS2_runs[2]), Run), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
protein.groups_n <- protein.groups                                    
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

LFDIAa <- paste0(LF_MS2_runs[1],"mTRAQ0")
LFDIAb <- paste0(LF_MS2_runs[2],"mTRAQ4")
LFDIAc <- paste0(LF_MS2_runs[3],"mTRAQ8")


protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAa]])/as.numeric(protein.groups_n[[LFDIAb]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio

################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAb), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAb), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF_a) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF_a$sample2_int <- as.numeric(protein.groups_n_omit_LF_a$sample2_int)


#intersect LF and multiDIA
protein.groups_n_omit_noint <- protein.groups_n_omit
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF

protein.groups_n_omit_a <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF_a$PGs,]
protein.groups_n_omit_LF_a <- protein.groups_n_omit_LF_a[protein.groups_n_omit_LF_a$PGs%in%protein.groups_n_omit$PGs,]


LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_a, protein.groups_n_omit_LF_a)

################################# plotting #########################

x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
a2 <- ggplot(protein.groups_n_omit_LF_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y="Log2, delta0:delta4",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())
a2


a4 <- ggplot(protein.groups_n_omit_LF_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a4
####################################
#d0d4
a1 <- ggplot(protein.groups_n_omit_a, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", B")), y=expression(paste(Log["2"],", A/B")),title = "plexDIA")+
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))

a1 

a3 <- ggplot(protein.groups_n_omit_a, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF","#00BA38", "#F8766D")) +
  geom_hline(yintercept=2, linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=0, linetype="dashed", 
             color = "#00BA38", size=1)+
  geom_hline(yintercept=-1, linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])

a3

################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

a_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+300, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = "Protein Ratios (A/B)", fill = "")

a_cover



################################################################
################################################################
########################## d0:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl(paste0(MS2_plexDIA_run), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAa]])/as.numeric(protein.groups_n[[pDIAc]])
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl(paste0(LF_MS2_runs[1],"|",LF_MS2_runs[3]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_MS2_runs[1]), Run), "mTRAQ0",
                                            ifelse(grepl(paste0(LF_MS2_runs[3]), Run), "mTRAQ8", "mTRAQ4")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAa]])/as.numeric(protein.groups_n[[LFDIAc]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)



#intersect LF and multiDIA


protein.groups_n_omit_b <- protein.groups_n_omit
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF

protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[!grepl("H.", protein.groups_n_omit_LF_b$species),]
protein.groups_n_omit_b <- protein.groups_n_omit_b[!grepl("H.", protein.groups_n_omit_b$species),]

protein.groups_n_omit_noint <- protein.groups_n_omit_b
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_b

protein.groups_n_omit_b <- protein.groups_n_omit_b[protein.groups_n_omit_b$PGs%in%protein.groups_n_omit_LF_b$PGs,]
protein.groups_n_omit_LF_b <- protein.groups_n_omit_LF_b[protein.groups_n_omit_LF_b$PGs%in%protein.groups_n_omit_b$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_b, protein.groups_n_omit_LF_b)

################################# plotting #########################


x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
b2 <- ggplot(protein.groups_n_omit_LF_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta0:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


b4 <- ggplot(protein.groups_n_omit_LF_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
b1 <- ggplot(protein.groups_n_omit_b, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_bolor_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", A/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


b3 <- ggplot(protein.groups_n_omit_b, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(2/3), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(3), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])



################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("MultiDIA", "Label-Free", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("MultiDIA", "Label-Free", "Intersected"))
library(Cairo)

b_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (A/C)", fill = "", title = expression(paste(italic(H.~sapiens),' excluded')))
b_cover



################################################################
################################################################
########################## d4:d8 ###############################
################################################################
################################################################
################################################################

df <- dat
df <- df[grepl(paste0(MS2_plexDIA_run), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
protein.groups_n <- protein.groups                                      

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[pDIAb]])/as.numeric(protein.groups_n[[pDIAc]])
protein.groups_n_omit <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit[grepl("H. sapiens", protein.groups_n_omit$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit$d0_d4 <- protein.groups_n_omit$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit <- protein.groups_n_omit[!grepl("H.", protein.groups_n_omit$species),]



######################## LF ########################################

df <- dat_LF
df <- df[grepl(paste0(LF_MS2_runs[2],"|",LF_MS2_runs[3]), df$Run),]
df <- df %>% mutate("channel_name" = ifelse(grepl(paste0(LF_MS2_runs[2]), Run), "mTRAQ4",
                                            ifelse(grepl(paste0(LF_MS2_runs[3]), Run), "mTRAQ8", "mTRAQ0")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med

PGs <- rownames(protein.groups)

protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- df %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))

protein.groups_n$HY<-F
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("ECOLI",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("HUMAN",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
protein.groups_n$HY[grepl("ECOLI",protein.groups_n$Protein.Names) & grepl("YEAST",protein.groups_n$Protein.Names)] <-T
table(protein.groups_n$HY)
protein.groups_n <- protein.groups_n[grepl("FALSE", protein.groups_n$HY),]

protein.groups_n <- protein.groups_n %>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                   ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                          ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),]

protein.groups_n$d0_d4 <- as.numeric(protein.groups_n[[LFDIAb]])/as.numeric(protein.groups_n[[LFDIAc]])
protein.groups_n_omit_LF <- protein.groups_n[!is.na(protein.groups_n$d0_d4), ]
med_human <- protein.groups_n_omit_LF[grepl("H. sapiens", protein.groups_n_omit_LF$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
protein.groups_n_omit_LF$d0_d4 <- protein.groups_n_omit_LF$d0_d4*scalar   #normalize human median on 1:1 ratio
protein.groups_n_omit_LF <- protein.groups_n_omit_LF[!grepl("H.", protein.groups_n_omit_LF$species),]


################ select and bind:

protein.groups_n_omit <- protein.groups_n_omit %>% dplyr::select(all_of(pDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit$sample2_int <- as.numeric(protein.groups_n_omit$sample2_int)

protein.groups_n_omit_LF <- protein.groups_n_omit_LF %>% dplyr::select(all_of(LFDIAc), "species", "d0_d4", "PGs")
colnames(protein.groups_n_omit_LF) <- c("sample2_int", "species", "Prot_rat", "PGs")
protein.groups_n_omit_LF$sample2_int <- as.numeric(protein.groups_n_omit_LF$sample2_int)


#intersect LF and multiDIA
protein.groups_n_omit_c <- protein.groups_n_omit
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF

protein.groups_n_omit_noint <- protein.groups_n_omit_c
protein.groups_n_omit_LF_noint <- protein.groups_n_omit_LF_c

protein.groups_n_omit_c <- protein.groups_n_omit_c[protein.groups_n_omit_c$PGs%in%protein.groups_n_omit_LF_c$PGs,]
protein.groups_n_omit_LF_c <- protein.groups_n_omit_LF_c[protein.groups_n_omit_LF_c$PGs%in%protein.groups_n_omit_c$PGs,]

LF_mDIA_d0d4 <- rbind(protein.groups_n_omit_c, protein.groups_n_omit_LF_c)

################################# plotting #########################

x_d0d4 <- log2(LF_mDIA_d0d4$sample2_int) %>% sort()
x_d0d4_val <- as.numeric(quantile(x_d0d4,probs=c(.01,.99)))
y_d0d4 <- log2(LF_mDIA_d0d4$Prot_rat) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(.0025,.9975)))
####################################
#d0d4
c2 <- ggplot(protein.groups_n_omit_LF_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + 
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y="Log2, delta4:delta8",title = "LF-DIA")+
  #subtitle = paste0(nrow(LF_PGs_lim)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(hjust = 0.5, size=18),
        axis.ticks.y = element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.y=element_blank(),
        axis.text.x = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.text.y = element_blank())


c4 <- ggplot(protein.groups_n_omit_LF_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "LF-DIA")+ #subtitle="")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        plot.background=element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_text(size=32),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5)) +
  ylim(y_d0d4_val[1],y_d0d4_val[2])


####################################
#d0d4
c1 <- ggplot(protein.groups_n_omit_c, aes(x=log2(sample2_int), y=log2(Prot_rat), color = species, alpha = species)) +
  geom_point() + #scale_color_brewer(palette = "PuOr")+
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_alpha_manual(name = "category", values = c(0.3, 0.13, 0.22),guide = "none") +
  labs(x = expression(paste(Log["2"],", C")), y=expression(paste(Log["2"],", B/C")),title = "plexDIA")+
  #subtitle = paste0(nrow(d0_d4_pt)," Protein groups"))+ 
  geom_smooth(method = "loess", method.args = list(family = "symmetric")) + xlim(x_d0d4_val[1],x_d0d4_val[2])+
  theme_classic()+ ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(legend.position="none", 
        plot.title = element_text(size =18, hjust = 0.5),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14),
        legend.text = element_text(face = "italic"))


c3 <- ggplot(protein.groups_n_omit_c, aes(x=species, y=log2(Prot_rat), color = species))+
  geom_boxplot() +   
  scale_color_manual(values = c("#619CFF", "#F8766D")) +
  geom_hline(yintercept=log2(1/6), linetype="dashed", 
             color = "#619CFF", size=1)+
  geom_hline(yintercept=log2(6), linetype="dashed", 
             color = "#F8766D", size=1) +
  scale_x_discrete(labels=c("H. sapiens" = "", "E. coli" = "",
                            "S. cerevisiae" = "")) +
  labs(x="",title = "plexDIA", color = "Species:")+# subtitle = "")+
  theme(axis.ticks = element_blank(),
        axis.title.y=element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        panel.background=element_blank(), panel.border=element_blank(),panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        axis.title.x = element_text(size=32),
        plot.background=element_blank(),
        plot.subtitle = element_text(size =12, hjust = 0.5),
        plot.title = element_text(size =18, hjust = 0.5),
        legend.position = "top",
        legend.text = element_text(size=14, face= "italic"),
        legend.title = element_text(size=14),
        legend.spacing.x = unit(1.0, 'line')) +
  
  ylim(y_d0d4_val[1],y_d0d4_val[2])


################################## coverage of protein group ratios:
joined <- protein.groups_n_omit[protein.groups_n_omit$PGs%in%protein.groups_n_omit_LF$PGs,]

types <- c("plexDIA", "LF-DIA", "Intersected")
numbers <- c(nrow(protein.groups_n_omit_noint), nrow(protein.groups_n_omit_LF_noint), nrow(joined))
df_cover <- data.frame("Cond" = types, "IDs" = numbers)
df_cover

df_cover$Cond <- factor(df_cover$Cond, levels=c("plexDIA", "LF-DIA", "Intersected"))
library(Cairo)

c_cover <- ggplot(df_cover, aes(x=as.factor(Cond), y=IDs, fill=Cond)) + geom_bar(stat="identity", colour="black", alpha =0.7) +
  geom_text(data=df_cover, aes(x=as.factor(Cond), y=IDs+100, label=IDs), col='black', size=4.5) +
  scale_fill_manual(values = c("grey15", "grey50", "grey85")) +
  #scale_fill_manual(values = c("lightblue2", "lightpink2", "plum2")) +
  theme_classic() + #ylim(0,6000)+
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 13.5, hjust = 0.5, face="bold"),
        legend.position = "top",
        legend.direction = "vertical",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Protein Ratios (B/C)", fill = "", title = expression(paste(italic(H.~sapiens)," excluded")))
c_cover

############## combine
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(a3)
mylegend1<-g_legend(c_cover)


a_fin <- grid.arrange(mylegend, nrow=2, heights=c(1, 10),
                      arrangeGrob(a_cover,
                                  a1 + theme(legend.position="none"),
                                  a2 + theme(legend.position="none"), 
                                  a3 + theme(legend.position="none"),
                                  a4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

c_fin <- grid.arrange(nrow=1,
                      arrangeGrob(c_cover + theme(legend.position="none"),
                                  c1 + theme(legend.position="none"),
                                  c2 + theme(legend.position="none"), 
                                  c3 + theme(legend.position="none"),
                                  c4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))

b_fin <- grid.arrange(nrow=1,
                      arrangeGrob(b_cover + theme(legend.position="none"),
                                  b1 + theme(legend.position="none"),
                                  b2 + theme(legend.position="none"), 
                                  b3 + theme(legend.position="none"),
                                  b4 + theme(legend.position="none"), nrow = 1, widths=c(1.65,2.25,2,0.7,0.7)))


figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 5,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1))

ggsave("SFigure5abc.pdf", figure_fin, width=11,height=9, dpi=400)




################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########       Figure D: In-set vs Out-of-Set      ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 

dat <- fread(MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Lib.Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area","Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))

df_names <- dat %>% dplyr::select("Protein.Group", "Protein.Names") %>% distinct(Protein.Group, .keep_all=T)

df <- dat
df <- df %>% mutate("channel_name" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
df$Unlabelled.Precursor <- paste0(df$Modified.Sequence, df$Precursor.Charge)
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ0\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ4\\)")
df$Unlabelled.Precursor <- str_remove_all(df$Unlabelled.Precursor, "\\(mTRAQ8\\)")

df$File.Name <- paste0(df$Run, df$channel_name)


df1 <- df[grepl("eJD905", df$Run),]
protein.groups1 <- diann_maxlfq(df1[df1$Lib.Q.Value <= 0.01 & df1$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")

df2 <- df[grepl("eJD906", df$Run),]
protein.groups2 <- diann_maxlfq(df2[df2$Lib.Q.Value <= 0.01 & df2$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "Unlabelled.Precursor", quantity.header = "Precursor.Translated")


protein.groups1_new <- protein.groups1
protein.groups1_new[,1] <- protein.groups1[,3]
protein.groups1_new[,2] <- protein.groups1[,1]
protein.groups1_new[,3] <- protein.groups1[,2]
protein.groups1 <-protein.groups1_new


protein.groups2_new <- protein.groups2
protein.groups2_new[,1] <- protein.groups2[,2]
protein.groups2_new[,2] <- protein.groups2[,1]
protein.groups2_new[,3] <- protein.groups2[,3]
protein.groups2 <-protein.groups2_new


colnames(protein.groups1) <- c("d0", "d4", "d8")
colnames(protein.groups2) <- c("d0", "d4", "d8")

rep1_d0d4_pt <- protein.groups2
PGs <- rownames(rep1_d0d4_pt)
rep1_d0d4_pt <- cbind(rep1_d0d4_pt, PGs)
rep1_d0d4_pt <- data.frame(rep1_d0d4_pt)
rep1_d0d4_pt$Prot_rat <- as.numeric(rep1_d0d4_pt$d0)/as.numeric(rep1_d0d4_pt$d4)
rep1_d0d4_pt <- rep1_d0d4_pt[!is.na(rep1_d0d4_pt$Prot_rat), ]
rep1_d0d4_pt <- rep1_d0d4_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_pt$HY<-F
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("HUMAN",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names)] <-T
rep1_d0d4_pt$HY[grepl("ECOLI",rep1_d0d4_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_pt$Protein.Names)] <-T
table(rep1_d0d4_pt$HY)
rep1_d0d4_pt <- rep1_d0d4_pt[grepl("FALSE", rep1_d0d4_pt$HY),]
rep1_d0d4_pt <- rep1_d0d4_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d4_pt <- rep1_d0d4_pt[!grepl("remove", rep1_d0d4_pt$species),]
med_human <- rep1_d0d4_pt[grepl("H. sapiens", rep1_d0d4_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_pt$Prot_rat <- rep1_d0d4_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_pt$Type <- "Within a run"

######## out of set:
rep1_d0d4_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- cbind(rep1_d0d4_out_pt, PGs)
rep1_d0d4_out_pt <- data.frame(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% dplyr::select("d0", "PGs")
d4_set2 <- rep1_d0d4_pt %>% dplyr::select("d4", "PGs")

rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% inner_join(d4_set2, by =c("PGs" = "PGs"))
rep1_d0d4_out_pt$Prot_rat <- as.numeric(rep1_d0d4_out_pt$d0)/as.numeric(rep1_d0d4_out_pt$d4)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!is.na(rep1_d0d4_out_pt$Prot_rat), ]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_out_pt$HY<-F
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("HUMAN",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names)] <-T
rep1_d0d4_out_pt$HY[grepl("ECOLI",rep1_d0d4_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d4_out_pt$Protein.Names)] <-T
table(rep1_d0d4_out_pt$HY)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[grepl("FALSE", rep1_d0d4_out_pt$HY),]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!grepl("remove", rep1_d0d4_out_pt$species),]
med_human <- rep1_d0d4_out_pt[grepl("H. sapiens", rep1_d0d4_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_out_pt$Prot_rat <- rep1_d0d4_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_out_pt$Type <- "Across runs"


######################### d0:d8 ##################################

rep1_d0d8_pt <- protein.groups2
PGs <- rownames(rep1_d0d8_pt)
rep1_d0d8_pt <- cbind(rep1_d0d8_pt, PGs)
rep1_d0d8_pt <- data.frame(rep1_d0d8_pt)
rep1_d0d8_pt$Prot_rat <- as.numeric(rep1_d0d8_pt$d0)/as.numeric(rep1_d0d8_pt$d8)
rep1_d0d8_pt <- rep1_d0d8_pt[!is.na(rep1_d0d8_pt$Prot_rat), ]
rep1_d0d8_pt <- rep1_d0d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_pt$HY<-F
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("HUMAN",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names)] <-T
rep1_d0d8_pt$HY[grepl("ECOLI",rep1_d0d8_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_pt$Protein.Names)] <-T
table(rep1_d0d8_pt$HY)
rep1_d0d8_pt <- rep1_d0d8_pt[grepl("FALSE", rep1_d0d8_pt$HY),]
rep1_d0d8_pt <- rep1_d0d8_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("remove", rep1_d0d8_pt$species),]
med_human <- rep1_d0d8_pt[grepl("H. sapiens", rep1_d0d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_pt$Prot_rat <- rep1_d0d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_pt_h <- rep1_d0d8_pt
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("H.", rep1_d0d8_pt$species),]
rep1_d0d8_pt$Type <- "Within a run"

######## out of set:
rep1_d0d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- cbind(rep1_d0d8_out_pt, PGs)
rep1_d0d8_out_pt <- data.frame(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% dplyr::select("d0", "PGs")
d8_set2 <- rep1_d0d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d0d8_out_pt$Prot_rat <- as.numeric(rep1_d0d8_out_pt$d0)/as.numeric(rep1_d0d8_out_pt$d8)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!is.na(rep1_d0d8_out_pt$Prot_rat), ]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_out_pt$HY<-F
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("HUMAN",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names)] <-T
rep1_d0d8_out_pt$HY[grepl("ECOLI",rep1_d0d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d0d8_out_pt$Protein.Names)] <-T
table(rep1_d0d8_out_pt$HY)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[grepl("FALSE", rep1_d0d8_out_pt$HY),]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("remove", rep1_d0d8_out_pt$species),]
med_human <- rep1_d0d8_out_pt[grepl("H. sapiens", rep1_d0d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_out_pt$Prot_rat <- rep1_d0d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("H.", rep1_d0d8_out_pt$species),]
rep1_d0d8_out_pt$Type <- "Across runs"


######################################## d4:d8 ###########################

rep1_d4d8_pt <- protein.groups2
PGs <- rownames(rep1_d4d8_pt)
rep1_d4d8_pt <- cbind(rep1_d4d8_pt, PGs)
rep1_d4d8_pt <- data.frame(rep1_d4d8_pt)
rep1_d4d8_pt$Prot_rat <- as.numeric(rep1_d4d8_pt$d4)/as.numeric(rep1_d4d8_pt$d8)
rep1_d4d8_pt <- rep1_d4d8_pt[!is.na(rep1_d4d8_pt$Prot_rat), ]
rep1_d4d8_pt <- rep1_d4d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_pt$HY<-F
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("HUMAN",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names)] <-T
rep1_d4d8_pt$HY[grepl("ECOLI",rep1_d4d8_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_pt$Protein.Names)] <-T
table(rep1_d4d8_pt$HY)
rep1_d4d8_pt <- rep1_d4d8_pt[grepl("FALSE", rep1_d4d8_pt$HY),]
rep1_d4d8_pt <- rep1_d4d8_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                          ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                 ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("remove", rep1_d4d8_pt$species),]
med_human <- rep1_d4d8_pt[grepl("H. sapiens", rep1_d4d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_pt$Prot_rat <- rep1_d4d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_pt_h <- rep1_d4d8_pt
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("H.", rep1_d4d8_pt$species),]
rep1_d4d8_pt$Type <- "Within a run"

######## out of set:
rep1_d4d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- cbind(rep1_d4d8_out_pt, PGs)
rep1_d4d8_out_pt <- data.frame(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% dplyr::select("d4", "PGs")
d8_set2 <- rep1_d4d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d4d8_out_pt$Prot_rat <- as.numeric(rep1_d4d8_out_pt$d4)/as.numeric(rep1_d4d8_out_pt$d8)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!is.na(rep1_d4d8_out_pt$Prot_rat), ]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_out_pt$HY<-F
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("HUMAN",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names)] <-T
rep1_d4d8_out_pt$HY[grepl("ECOLI",rep1_d4d8_out_pt$Protein.Names) & grepl("YEAST",rep1_d4d8_out_pt$Protein.Names)] <-T
table(rep1_d4d8_out_pt$HY)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[grepl("FALSE", rep1_d4d8_out_pt$HY),]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt%>% mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("remove", rep1_d4d8_out_pt$species),]
med_human <- rep1_d4d8_out_pt[grepl("H. sapiens", rep1_d4d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_out_pt$Prot_rat <- rep1_d4d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("H.", rep1_d4d8_out_pt$species),]
rep1_d4d8_out_pt$Type <- "Across runs"



###############
rep1_d0d4_pt <- rep1_d0d4_pt[rep1_d0d4_pt$PGs%in%rep1_d0d4_out_pt$PGs,]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[rep1_d0d4_out_pt$PGs%in%rep1_d0d4_pt$PGs,]

rep1_d0d8_pt <- rep1_d0d8_pt[rep1_d0d8_pt$PGs%in%rep1_d0d8_out_pt$PGs,]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[rep1_d0d8_out_pt$PGs%in%rep1_d0d8_pt$PGs,]

rep1_d4d8_pt <- rep1_d4d8_pt[rep1_d4d8_pt$PGs%in%rep1_d4d8_out_pt$PGs,]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[rep1_d4d8_out_pt$PGs%in%rep1_d4d8_pt$PGs,]

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))
df2<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(1/6), log2(6)))


d0d4 <- bind_rows(rep1_d0d4_pt, rep1_d0d4_out_pt)
d0d8 <- bind_rows(rep1_d0d8_pt, rep1_d0d8_out_pt)
d4d8 <- bind_rows(rep1_d4d8_pt, rep1_d4d8_out_pt)

d0d4 <- d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8 <- d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8 <- d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4$error <- abs(log2(d0d4$Prot_rat)-d0d4$Z)
d0d8$error <- abs(log2(d0d8$Prot_rat)-d0d8$Z)
d4d8$error <- abs(log2(d4d8$Prot_rat)-d4d8$Z)

all <- rbind(d0d4, d4d8, d0d8)
####

####


y_d0d4 <- all$error %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,0.99)))

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))

all$Type <- factor(all$Type, levels=c("Within a run", "Across runs"))
library(Cairo)


D_fig <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  #geom_text(data=df_n, aes(x=as.factor(Label), y=log2(d0_d4_rat), label=n), col='black', size=4) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  #ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.title = element_text(size = 16, hjust = 0.5, face="bold"),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "", title = "plexDIA only") +
  ylim(y_d0d4_val[1],y_d0d4_val[2])

ggsave("SFigure5d_MS2.pdf", width=3.7,height=3.7)


################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########        Figure E: R vs K quant MS2         ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 

dat <- fread(MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area","Ms1.Translated", "Quantity.Quality", "Ms1.Profile.Corr", "Spectrum.Similarity", "Precursor.Quantity",
                               "Mass.Evidence", "CScore", "Fragment.Correlations"))

ev <- dat[grepl("eJD906", dat$Run),]
ev$seqcharge <- paste0(ev$Modified.Sequence, ev$Precursor.Charge)
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ0\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ4\\)")
ev$seqcharge <- str_remove_all(ev$seqcharge, "\\(mTRAQ8\\)")
ev$seqcharge_file <- paste0(ev$seqcharge,ev$Run)
ev2_lim <- ev %>% mutate("label" = ifelse(grepl("mTRAQ0", Modified.Sequence), "mTRAQ0",
                                          ifelse(grepl("mTRAQ4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))
#ev2_lim <- ev2_lim[!grepl("mTRAQ8", ev2_lim$label),]  #only keep d0 and d4

########################### FIGURE E Cont. K vs R quant (MS2)
ev2_0 <- ev2_lim[grepl("mTRAQ0",ev2_lim$Precursor.Id),]
ev2_4 <- ev2_lim[grepl("mTRAQ4",ev2_lim$Precursor.Id),] %>% dplyr::select("seqcharge","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_4) <- c("seqcharge_d4", "seqcharge_file","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4")
ev2_8 <- ev2_lim[grepl("mTRAQ8",ev2_lim$Precursor.Id),] %>% dplyr::select("seqcharge","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_8) <- c("seqcharge_d8", "seqcharge_file","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8")
#
rep1_d0d4 <- MS2_t_mTRAQ_d0d4(ev2_0,ev2_4)
rep1_d0d8 <- MS2_t_mTRAQ_d0d8(ev2_0,ev2_8)
rep1_d0d4_MS1 <- MS1_mTRAQ_d0d4(ev2_0,ev2_4)
rep1_d0d8_MS1 <- MS1_mTRAQ_d0d8(ev2_0,ev2_8)


ev2_4 <- ev2_lim[grepl("mTRAQ4",ev2_lim$Precursor.Id),] %>% dplyr::select("Run","seqcharge","Protein.Names","seqcharge_file","Precursor.Quantity","Precursor.Translated","Ms1.Area")
colnames(ev2_4) <- c("Run","seqcharge", "Protein.Names","seqcharge_file","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4")
rep1_d4d8 <- MS2_t_mTRAQ_d4d8(ev2_4,ev2_8)
rep1_d4d8_MS1 <- MS1_mTRAQ_d4d8(ev2_4,ev2_8)



df1<- data.frame(species = c("ECOLI", "HUMAN", "YEAST"), Z = c(2, 0, -1))
df2<- data.frame(species = c("ECOLI", "YEAST"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("ECOLI", "YEAST"), Z = c(log2(1/6), log2(6)))

d0d4_MS2 <- rep1_d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8_MS2 <- rep1_d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8_Ms2 <- rep1_d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4_MS1 <- rep1_d0d4_MS1 %>% left_join(df1, by =c("species" = "species"))
d0d8_MS1 <- rep1_d0d8_MS1 %>% left_join(df2, by =c("species" = "species"))
d4d8_Ms1 <- rep1_d4d8_MS1 %>% left_join(df3, by =c("species" = "species"))


#### intersect
d0d4_MS2 <- d0d4_MS2[d0d4_MS2$seqcharge%in%d0d4_MS1$seqcharge,]
d0d4_MS1 <- d0d4_MS1[d0d4_MS1$seqcharge%in%d0d4_MS2$seqcharge,]

d0d8_MS2 <- d0d8_MS2[d0d8_MS2$seqcharge%in%d0d8_MS1$seqcharge,]
d0d8_MS1 <- d0d8_MS1[d0d8_MS1$seqcharge%in%d0d8_MS2$seqcharge,]

d4d8_Ms2 <- d4d8_Ms2[d4d8_Ms2$seqcharge%in%d4d8_Ms1$seqcharge,]
d4d8_Ms1 <- d4d8_Ms1[d4d8_Ms1$seqcharge%in%d4d8_Ms2$seqcharge,]

all <- list(d0d4_MS2, d0d8_MS2, d4d8_Ms2, d0d4_MS1, d0d8_MS1, d4d8_Ms1)

#normalize the data and remove PGs which map to multiple species
for(i in 1:6){
  temp <- all[[i]]
  temp$HY<-F
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names) & grepl("ECOLI",temp$Protein.Names)] <-T
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("ECOLI",temp$Protein.Names)] <-T
  temp$HY[grepl("HUMAN",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names)] <-T
  temp$HY[grepl("ECOLI",temp$Protein.Names) & grepl("YEAST",temp$Protein.Names)] <-T
  table(temp$HY)
  temp <- temp[grepl("FALSE", temp$HY),]
  med_human <- temp[grepl("HUMAN", temp$species),]
  med_human <- median(med_human$d0_d4_rat)
  scalar <- 1/med_human
  temp$d0_d4_rat <- temp$d0_d4_rat*scalar   #normalize human median on 1:1 ratio
  all[[i]] <- temp
}


d0d4_MS2 <- all[[1]]
d0d8_MS2 <- all[[2]]
d4d8_Ms2 <- all[[3]]

d0d4_MS1 <- all[[4]]
d0d8_MS1 <- all[[5]]
d4d8_Ms1 <- all[[6]]


all_MS2 <- bind_rows(d0d4_MS2, d0d8_MS2, d4d8_Ms2)

all_MS1 <- bind_rows(d0d4_MS1, d0d8_MS1, d4d8_Ms1)




#####
all_MS2$aa <- str_sub(all_MS2$Stripped.Sequence, -1)  #get C-terminal amino acid
all_MS2_K <- all_MS2[grepl("K", all_MS2$aa),]
all_MS2_R <- all_MS2[grepl("R", all_MS2$aa),]

all_MS2_K$Type <- "Lys precursors"
all_MS2_K <- all_MS2_K %>% add_count(species)
all_MS2_K$cond <- paste0("n=", all_MS2_K$n)

all_MS2_R$Type <- "Arg precursors"
all_MS2_R <- all_MS2_R %>% add_count(species)
all_MS2_R$cond <- paste0("n=", all_MS2_R$n)


####
all_MS1$Type <- "All precursors"
all_MS1 <- all_MS1 %>% add_count(species)
all_MS1$cond <- paste0("n=", all_MS1$n)





K_vs_R_MS1_mDIA_d0d4 <- bind_rows(all_MS2_K, all_MS2_R, all_MS1)
K_vs_R_MS1_mDIA_d0d4$error <- abs(log2(K_vs_R_MS1_mDIA_d0d4$d0_d4_rat)-K_vs_R_MS1_mDIA_d0d4$Z)

y_d0d4 <- log2(K_vs_R_MS1_mDIA_d0d4$error) %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,.999)))


############ add in the MS1 quant for multiDIA precursors:
K_vs_R_MS1_mDIA_d0d4 <- K_vs_R_MS1_mDIA_d0d4 %>% mutate("species" = ifelse(grepl("ECOLI", species), "E. coli", 
                                                                           ifelse(grepl("HUMAN", species), "H. sapiens",
                                                                                  ifelse(grepl("YEAST", species), "S. cerevisiae", "unknown"))))

K_vs_R_MS1_mDIA_d0d4 <- K_vs_R_MS1_mDIA_d0d4[!grepl("unknown", K_vs_R_MS1_mDIA_d0d4$species),]
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% distinct(cond,.keep_all=T)
df$species<- factor(df$species)

df_n <- K_vs_R_MS1_mDIA_d0d4 %>% distinct(cond,.keep_all=T)
# df_n[1,1] <- 1.3
# df_n[2,1] <- 0.7
# df_n[3,1] <- 4.5
# df_n[4,1] <- 1.3
# df_n[5,1] <- 0.7
# df_n[6,1] <- 4.5

library(viridis)
K_vs_R_MS1_mDIA_d0d4$Type <- factor(K_vs_R_MS1_mDIA_d0d4$Type, levels=c("Lys precursors", "Arg precursors"))
library(Cairo)
K_vs_R_MS2_mDIA_d0d4 <- K_vs_R_MS1_mDIA_d0d4[grepl("MS2", K_vs_R_MS1_mDIA_d0d4$Quant),]

E_fig <- ggplot(K_vs_R_MS2_mDIA_d0d4, aes(x=as.factor(Type),y=error, fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  #geom_text(data=df_n, aes(x=as.factor(Label), y=log2(d0_d4_rat), label=n), col='black', size=4) +
  scale_x_discrete(limits=c("Lys precursors","Arg precursors")) + 
  scale_fill_manual(values = c("dodgerblue4", "deepskyblue")) +
  theme_classic() +
  facet_grid(~species, scales = "free") +
  ylim(0,y_d0d4_val[2]) + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 16, hjust = 0.5, face="bold"),
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y=expression(paste("|",Log["2"],", Precursor Ratio Error|")), fill = "",subtitle="",title="plexDIA only")
#scale_y_continuous(breaks=c(-2, -1, 0, 1, 2, 3), limits = c(y_d0d4_val[1],y_d0d4_val[2]))
E_fig
ggsave("SFigure5e.pdf", width=5.7,height=3.7)


library(gridExtra)
library(ggpubr)
library(gridGraphics)


others <- ggarrange(NULL, D_fig, NULL, E_fig, NULL, nrow = 1, widths=c(0.6,3,0.3,4.5,0.2))

others

figure_fin <- ggarrange(a_fin, NULL, b_fin, NULL, c_fin, NULL, others,
                        # labels = c("a", "b","","c", ""),
                        ncol = 1, nrow = 7,
                        # font.label = list(size = 28, color = "black", face = "bold", family = NULL),
                        heights=c(1.11, 0.1, 1, 0.1, 1,0.07,1.25))

ggsave("SFigure5abcde_MS2.pdf", figure_fin, width=11.5,height=13, dpi=500)


```



