---
title: "plexDIA_rev"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Install/Load Libraries
```{r}
source("plexDIA_Functions.R")
library(eulerr)
library(ggalt)
library(ggbeeswarm)
library(facetscales)
library(OrgMassSpecR)
library(robustbase)
library(plyr)
library(reticulate)
library(stringi)
library(stringr)
library(tidyr)
library(patchwork)
library(ggplot2)
library(reshape2)
library(Cairo)
library(gridGraphics)
library(data.table)
library(devtools)
library(diann)
library(gridExtra)
library(ggpubr)
library(ggpointdensity)
library(viridis)
library(scales)
library(data.table)
library(matrixStats)
library(dplyr)


########## Please download data search reports from MassIVE (MSV000089093) then update file-paths:

fpath <- "" # "plexDIA_HYE_V1_report"
LF_fpath <- ""  # "LFDIA_HYE_V1_report"
DDA_fpath <- ""  # "mTRAQ_DDA_evidence"
CC_data <- "" # "plexDIA_CC_V1_V2_report"
CC_data_XIC <-  "" # "plexDIA_CC_V1_V2_report_XIC"

YE_fpath <- "" # "plexDIA_YE_V1_report"
DDA_prot_fpath <- "" # "mTRAQ_DDA_PGs"
MS2_fpath <- "" # "plexDIA_HYE_V2_report"
LF_MS2_fpath <- ""  # "LFDIA_HYE_V2_report"
Bench_fpath <- "" # "Navarro_HYE_report"

###### single cell datasets:
b_ms1ext_fpath <- "" # "plexDIA_100cell_QExactive_report_Extracted"
b_fpath <- "" # "plexDIA_100cell_QExactive_report"
SC_ms1ext_fpath <- "" # "plexDIA_SC_nPOP_QExactive_Extracted"
SC_fpath <- "" # "plexDIA_SC_nPOP_QExactive_report"
SC_vis_XIC_fpath <-"" # "plexDIA_SC_nPOP_QExactive_report_XIC"
sn_fpath <- "" #plexDIA_signal_noise_SC_QExactive

tims_SC_fpath_ms1Ext <- "" # "plexDIA_SC_nPOP_timsTOFSCP_report_Extracted"
tims_SC_fpath <- "" # "plexDIA_SC_nPOP_timsTOFSCP_report"

#meta dataset info
meta_bench <- "Meta_Bulk_benchmarking.tsv"
meta_PCA_fpath <- "Meta_SingleCell.tsv"

#GOA_db
GO_DB_Path <- "GOA_db.txt" 

#########
```

Figure 1: Schematic
```{r}

#Figure 1a: created in Abode Illustrator

#Figure 1b: created in Abode Illustrator

#Figure 1c: created in Abode Illustrator

#Figure 1d:
d0 <- 778.64  #mTRAQ delta 0 cost
d4 <- 2348.04 #mTRAQ delta 4 cost
d8 <- 2343.62 #mTRAQ delta 8 cost

total_prot <- 50*3*0.1 #50 aliquots * 3 labels * 0.1 mg protein/aliquot

total_cost <- d0+d4+d8

cost_per_ug <- total_prot*1000/total_cost

plexDIA <- (100*1.5)/3
LFDIA <- (100*1.5)
cost <- c(LFDIA, plexDIA, (cost_per_ug/3)*1.5)
Application_Reagent <- c("LC-MS/MS cost","LC-MS/MS cost","Label cost")
type <- c("LF-DIA", "plexDIA","plexDIA")

df <- data.frame(cost = cost, Application_Reagent = Application_Reagent,type = type)
#df$error <- c(25,75)

df$Application_Reagent <- factor(df$Application_Reagent, levels=c("Label cost", "LC-MS/MS cost"))

ggplot(df, aes(x=type, y=cost, fill=type)) + 
  theme_classic() + labs(x="", y="Estimated cost (USD) per sample") +
  geom_bar(data = df %>% filter(type == "plexDIA"),#colour="black", 
           stat="identity", aes(x=type, y =cost, fill=Application_Reagent), width = 0.45, alpha =1) + 
  geom_bar(data = df %>% filter(type == "LF-DIA"),#colour="black", 
           stat="identity", aes(x=type, y =cost, fill=Application_Reagent), width = 0.45, alpha =1) + 
  scale_y_continuous(labels=scales::dollar_format()) + 
  scale_fill_manual(values = c("red","grey25")) +
  theme(legend.position = c(0.7,0.8),
        legend.title = element_blank(),
        legend.text = element_text(size=11),
        axis.text.x = element_text(size=13, face = "bold", color = "black"),
        axis.text.y = element_text(size=14, face = "bold", color = "black"),
        axis.title.x = element_text(size=12, color = "black")) + coord_flip()

ggsave("Figure1b.pdf", width=4.1, height=1.9, dpi=500)

#Figure 3a: created in Adobe Illustrator

```

Figure 2: Coverage
```{r}

# Figure 2, Proteomic coverage and overlap between samples and runs of plexDIA and LF-DIA:
mTRAQ <- data.frame(fread(fpath))
mTRAQ <- mTRAQ[grepl("wJD803|804|815", mTRAQ$Run),]
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0|mTRAQ$Precursor.Translated>0),]  # must have non-zero quant at MS1 or MS2
mTRAQ$Label <- "mTRAQ"
mTRAQ <- pD_seqcharge(mTRAQ)
mTRAQ <- pD_channel(mTRAQ)
mTRAQ$seqcharge_run <- paste0(mTRAQ$seqcharge, "_", mTRAQ$channel_name, "_", mTRAQ$Run)
mTRAQ$prot_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ_uni_prot <- unique(mTRAQ$Protein.Group)

ev2_0 <- mTRAQ[grepl("mTRAQ0",mTRAQ$channel_name),] %>% distinct(seqcharge_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("mTRAQ4",mTRAQ$channel_name),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated","Ms1.Area", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_4) <- c("Run","Protein.Group_d4","seqcharge_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4", "seqcharge_run_d4", "prot_run_d4","Lib.PG.Q.Value_d4")
ev2_8 <- mTRAQ[grepl("mTRAQ8",mTRAQ$channel_name),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated","Ms1.Area", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% distinct(seqcharge_run, .keep_all=T)
colnames(ev2_8) <- c("Run","Protein.Group_d8","seqcharge_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8", "seqcharge_run_d8", "prot_run_d8","Lib.PG.Q.Value_d8")


ev2_0_prot <- ev2_0[which(ev2_0$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
ev2_0_prot$Sample <- paste0("\u0394","0-A")
ev2_4_prot <- ev2_4[which(ev2_4$Lib.PG.Q.Value_d4 < 0.01),] %>% distinct(prot_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4","prot_run_d4")
colnames(ev2_4_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_4_prot$Sample <- paste0("\u0394","4-B")
ev2_8_prot <- ev2_8[which(ev2_8$Lib.PG.Q.Value_d8 < 0.01),] %>% distinct(prot_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8","prot_run_d8")
colnames(ev2_8_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_8_prot$Sample <- paste0("\u0394","8-C")

ev2_0_pep <- ev2_0 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_0_pep$Sample <- paste0("\u0394","0-A")
ev2_4_pep <- ev2_4 %>% distinct(seqcharge_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4", "seqcharge_d4", "seqcharge_run_d4")
colnames(ev2_4_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_4_pep$Sample <- paste0("\u0394","4-B")
ev2_8_pep <- ev2_8 %>% distinct(seqcharge_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8", "seqcharge_d8", "seqcharge_run_d8")
colnames(ev2_8_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_8_pep$Sample <- paste0("\u0394","8-C")

mTRAQ_prots <- rbind(ev2_0_prot,ev2_4_prot,ev2_8_prot)
mTRAQ_prots$Sample_new <- mTRAQ_prots$Sample

mTRAQ_peps <- rbind(ev2_0_pep,ev2_4_pep,ev2_8_pep)
mTRAQ_peps$Sample_new <- mTRAQ_peps$Sample

p_mTRAQ <- mTRAQ_peps %>% dplyr::count(Run, Sample_new)
p_mTRAQ$Condition <- "plexDIA"

prot_mTRAQ <- mTRAQ_prots %>% dplyr::count(Run, Sample_new)
prot_mTRAQ$Condition <- "plexDIA"

mTRAQ_int <-mTRAQ_prots %>% dplyr::group_by(Run, Protein.Group) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  dplyr::group_by(Run) %>% dplyr::add_count() %>% dplyr::distinct(Run, .keep_all=T)#require protein to be present across samples
mTRAQ_int$prots_per_run <- mTRAQ_int$nn/3
mTRAQ_int$type <- "plexDIA"

##############
##############
#######################
########################
##########################  Label-free

unlab <- data.frame(fread(LF_fpath))
unlab <- unlab[which(unlab$Ms1.Area>0|unlab$Precursor.Translated>0),]  # must have non-zero Ms1 or Ms2 quant
unlab_1 <- unlab[grepl("wJD756_re|wJD757_re|wJD758_re", unlab$Run),]
unlab_1$seqcharge <- paste0(unlab_1$Modified.Sequence, unlab_1$Precursor.Charge)
unlab_1$seqcharge_run <- paste0(unlab_1$seqcharge, "_", unlab_1$Run)
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_", unlab_1$Run)

unlab_1_prot <- unlab_1[which(unlab_1$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_1_prot$Sample <- paste0("\u0394","0-A")
unlab_1_prot$Sample_new <- paste0("A")

unlab_1_pep <- unlab_1 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_1_pep$Sample <- paste0("\u0394","0-A")
unlab_1_pep$Sample_new <- paste0("A")

##############
unlab_2 <- unlab[grepl("wJD759_re|wJD760_re|wJD761_re", unlab$Run),]
unlab_2$seqcharge <- paste0(unlab_2$Modified.Sequence, unlab_2$Precursor.Charge)
unlab_2$seqcharge_run <- paste0(unlab_2$seqcharge, "_", unlab_2$Run)
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_", unlab_2$Run)

unlab_2_prot <- unlab_2[which(unlab_2$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_2_prot$Sample <- paste0("\u0394","4-B")
unlab_2_prot$Sample_new <- paste0("B")

unlab_2_pep <- unlab_2 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_2_pep$Sample <- paste0("\u0394","4-B")
unlab_2_pep$Sample_new <- paste0("B")

##############
unlab_3 <- unlab[grepl("wJD762_re|wJD763_re|wJD764_re", unlab$Run),]
unlab_3$seqcharge <- paste0(unlab_3$Modified.Sequence, unlab_3$Precursor.Charge)
unlab_3$seqcharge_run <- paste0(unlab_3$seqcharge, "_", unlab_3$Run)
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_", unlab_3$Run)

unlab_3_prot <- unlab_3[which(unlab_3$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_3_prot$Sample <- paste0("\u0394","8-C")
unlab_3_prot$Sample_new <- paste0("C")


unlab_3_pep <- unlab_3 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_3_pep$Sample <- paste0("\u0394","8-C")
unlab_3_pep$Sample_new <- paste0("C")

###################

all_unlab_pep <- rbind(unlab_1_pep, unlab_2_pep, unlab_3_pep)
all_unlab_prot <- rbind(unlab_1_prot, unlab_2_prot, unlab_3_prot)

p_unlab <- all_unlab_pep %>% dplyr::count(Run, Sample_new)
p_unlab$Condition <- "LF-DIA"

prot_unlab <- all_unlab_prot %>% dplyr::count(Run, Sample_new)
prot_unlab$Condition <- "LF-DIA"

all_unlab_prot <- all_unlab_prot %>% dplyr::mutate("rep" = ifelse(grepl("756|759|762", Run),"rep1",
                                                                        ifelse(grepl("757|760|763", Run),"rep2",
                                                                                     ifelse(grepl("758|761|764", Run),"rep3", "remove"))))

LF_int <-all_unlab_prot %>% dplyr::group_by(rep, Protein.Group) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  dplyr::group_by(rep) %>% dplyr::add_count() %>% dplyr::distinct(rep, .keep_all=T)#require protein to be present across samples
LF_int$prots_per_run <- LF_int$nn/3
LF_int$type <- "LF-DIA"


##############
##############
#######################
########################
##########################  mTRAQ DDA

#get prots which are 1% FDR
prot_DDAFDR <- read.delim(DDA_prot_fpath)
prot_DDAFDR <- prot_DDAFDR[which(prot_DDAFDR$Q.value<0.01),] %>% dplyr::filter(!grepl("REV|CON", Protein.IDs))
prot_DDAFDR$Protein.Group <- gsub(";.*","",prot_DDAFDR$Majority.protein.IDs)

DDA <- read.delim(DDA_fpath)
DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA_d0 <- DDA[which(DDA$Intensity.L > 0),]
DDA_d0$seqcharge <- paste0(DDA_d0$Modified.sequence, DDA_d0$Charge)
DDA_d0$seqcharge_run <- paste0(DDA_d0$seqcharge, "_", DDA_d0$Raw.file)
DDA_d0$Protein.Group <- gsub(";.*","",DDA_d0$Leading.razor.protein)
DDA_d0$prot_run <- paste0(DDA_d0$Leading.razor.protein, "_", DDA_d0$Raw.file)

DDA_d0_prot <- DDA_d0 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d0_prot$Sample <- paste0("A")
DDA_d0_prot$Sample_new <- paste0("\u0394","0 DDA-A")

DDA_d0_pep <- DDA_d0 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d0_pep$Sample <- paste0("A")
DDA_d0_pep$Sample_new <- paste0("\u0394","0 DDA-A")

####
DDA_d4 <- DDA[which(DDA$Intensity.M > 0),]
DDA_d4$seqcharge <- paste0(DDA_d4$Modified.sequence, DDA_d4$Charge)
DDA_d4$seqcharge_run <- paste0(DDA_d4$seqcharge, "_", DDA_d4$Raw.file)
DDA_d4$Protein.Group <- gsub(";.*","",DDA_d4$Leading.razor.protein)
DDA_d4$prot_run <- paste0(DDA_d4$Leading.razor.protein, "_", DDA_d4$Raw.file)

DDA_d4_prot <- DDA_d4 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d4_prot$Sample <- paste0("B")
DDA_d4_prot$Sample_new <- paste0("\u0394","4 DDA-B")

DDA_d4_pep <- DDA_d4 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d4_pep$Sample <- paste0("B")
DDA_d4_pep$Sample_new <- paste0("\u0394","4 DDA-B")

####
DDA_d8 <- DDA[which(DDA$Intensity.H > 0),]
DDA_d8$seqcharge <- paste0(DDA_d8$Modified.sequence, DDA_d8$Charge)
DDA_d8$seqcharge_run <- paste0(DDA_d8$seqcharge, "_", DDA_d8$Raw.file)
DDA_d8$Protein.Group <- gsub(";.*","",DDA_d8$Leading.razor.protein)
DDA_d8$prot_run <- paste0(DDA_d8$Leading.razor.protein, "_", DDA_d8$Raw.file)

DDA_d8_prot <- DDA_d8 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d8_prot$Sample <- paste0("C")
DDA_d8_prot$Sample_new <- paste0("\u0394","8 DDA-C")

DDA_d8_pep <- DDA_d8 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d8_pep$Sample <- paste0("C")
DDA_d8_pep$Sample_new <- paste0("\u0394","8 DDA-C")


##########################
##########################  data-wrangling for plotting
all_DDA_pep <- rbind(DDA_d0_pep, DDA_d4_pep, DDA_d8_pep)
all_DDA_prot <- rbind(DDA_d0_prot, DDA_d4_prot, DDA_d8_prot)
all_DDA_prot <- all_DDA_prot[which(all_DDA_prot$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

p_DDA <- all_DDA_pep %>% dplyr::count(Raw.file, Sample_new)
colnames(p_DDA) <- c("Run", "Sample_new", "n")
p_DDA$Condition <- "mTRAQ\nDDA"

prot_DDA <- all_DDA_prot %>% dplyr::count(Raw.file, Sample_new)
colnames(prot_DDA) <- c("Run", "Sample_new", "n")
prot_DDA$Condition <- "mTRAQ\nDDA"

DDA_int <-all_DDA_prot %>% dplyr::group_by(Raw.file, Leading.razor.protein) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  dplyr::group_by(Raw.file) %>% dplyr::add_count() %>% dplyr::distinct(Raw.file, .keep_all=T)#require protein to be present across samples
DDA_int$prots_per_run <- DDA_int$nn/3
DDA_int$type <- "mTRAQ\nDDA"

########## intersected coverage plexDIA, LF-DIA
######## rep1
my_fin_colors2 <- c("turquoise3","palevioletred3","#ceca7c","turquoise3","palevioletred3","#ceca7c")

LF_rep1 <- all_unlab_prot[grepl("wJD756|759|762", all_unlab_prot$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
LF_rep1 <- dcast(LF_rep1, Protein.Group ~ Sample_new)
rownames(LF_rep1) <- LF_rep1$Protein.Group
LF_rep1 <- LF_rep1[,-1]
LF_rep1[!is.na(LF_rep1)] <- TRUE
LF_rep1[is.na(LF_rep1)] <- FALSE
LF_rep1$A <- as.logical(LF_rep1$A)
LF_rep1$B <- as.logical(LF_rep1$B)
LF_rep1$C <- as.logical(LF_rep1$C)
LF_rep1$PGs <- rownames(LF_rep1)

pD_rep1 <- mTRAQ_prots[grepl("803", mTRAQ_prots$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
pD_rep1$Protein.Group <- paste0("pD",pD_rep1$Protein.Group)
pD_rep1 <- dcast(pD_rep1, Protein.Group ~ Sample_new)
rownames(pD_rep1) <- pD_rep1$Protein.Group
pD_rep1 <- pD_rep1[,-1]
pD_rep1[!is.na(pD_rep1)] <- TRUE
pD_rep1[is.na(pD_rep1)] <- FALSE
pD_rep1[,1] <- as.logical(pD_rep1[,1])
pD_rep1[,2] <- as.logical(pD_rep1[,2])
pD_rep1[,3] <- as.logical(pD_rep1[,3])
pD_rep1$PGs <- rownames(pD_rep1)

pD_LF_rep1 <- pD_rep1 %>% full_join(LF_rep1, by =c("PGs" = "PGs"))
pD_LF_rep1[is.na(pD_LF_rep1)] <- FALSE
pD_LF_rep1 <- pD_LF_rep1[,-4]
pd <- euler(pD_LF_rep1, shape = "ellipse",quantities = TRUE)

pdf(file = "Fig2_venn_rep1.pdf",   # The directory you want to save the file in
    width = 7, # The width of the plot in inches
    height = 7)
plot(pd,
     quantities = TRUE,
     labels = list(font = 4), fill=my_fin_colors2, alpha = 1)
dev.off()

######## rep2
LF_rep2 <- all_unlab_prot[grepl("wJD757|760|763", all_unlab_prot$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
LF_rep2 <- dcast(LF_rep2, Protein.Group ~ Sample_new)
rownames(LF_rep2) <- LF_rep2$Protein.Group
LF_rep2 <- LF_rep2[,-1]
LF_rep2[!is.na(LF_rep2)] <- TRUE
LF_rep2[is.na(LF_rep2)] <- FALSE
LF_rep2$A <- as.logical(LF_rep2$A)
LF_rep2$B <- as.logical(LF_rep2$B)
LF_rep2$C <- as.logical(LF_rep2$C)
LF_rep2$PGs <- rownames(LF_rep2)

pD_rep2 <- mTRAQ_prots[grepl("804", mTRAQ_prots$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
pD_rep2$Protein.Group <- paste0("pD",pD_rep2$Protein.Group)
pD_rep2 <- dcast(pD_rep2, Protein.Group ~ Sample_new)
rownames(pD_rep2) <- pD_rep2$Protein.Group
pD_rep2 <- pD_rep2[,-1]
pD_rep2[!is.na(pD_rep2)] <- TRUE
pD_rep2[is.na(pD_rep2)] <- FALSE
pD_rep2[,1] <- as.logical(pD_rep2[,1])
pD_rep2[,2] <- as.logical(pD_rep2[,2])
pD_rep2[,3] <- as.logical(pD_rep2[,3])
pD_rep2$PGs <- rownames(pD_rep2)

pD_LF_rep2 <- pD_rep2 %>% full_join(LF_rep2, by =c("PGs" = "PGs"))
pD_LF_rep2[is.na(pD_LF_rep2)] <- FALSE
pD_LF_rep2 <- pD_LF_rep2[,-4]
pd <- euler(pD_LF_rep2, shape = "ellipse",quantities = TRUE)

pdf(file = "Fig2_venn_rep2.pdf", width = 7, height = 7)
plot(pd,
     quantities = TRUE,
     labels = list(font = 4), fill=my_fin_colors2, alpha = 1)
dev.off()


######## rep3
LF_rep3 <- all_unlab_prot[grepl("wJD758|761|764", all_unlab_prot$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
LF_rep3 <- dcast(LF_rep3, Protein.Group ~ Sample_new)
rownames(LF_rep3) <- LF_rep3$Protein.Group
LF_rep3 <- LF_rep3[,-1]
LF_rep3[!is.na(LF_rep3)] <- TRUE
LF_rep3[is.na(LF_rep3)] <- FALSE
LF_rep3$A <- as.logical(LF_rep3$A)
LF_rep3$B <- as.logical(LF_rep3$B)
LF_rep3$C <- as.logical(LF_rep3$C)
LF_rep3$PGs <- rownames(LF_rep3)

pD_rep3 <- mTRAQ_prots[grepl("815", mTRAQ_prots$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
pD_rep3$Protein.Group <- paste0("pD",pD_rep3$Protein.Group)
pD_rep3 <- dcast(pD_rep3, Protein.Group ~ Sample_new)
rownames(pD_rep3) <- pD_rep3$Protein.Group
pD_rep3 <- pD_rep3[,-1]
pD_rep3[!is.na(pD_rep3)] <- TRUE
pD_rep3[is.na(pD_rep3)] <- FALSE
pD_rep3[,1] <- as.logical(pD_rep3[,1])
pD_rep3[,2] <- as.logical(pD_rep3[,2])
pD_rep3[,3] <- as.logical(pD_rep3[,3])
pD_rep3$PGs <- rownames(pD_rep3)

pD_LF_rep3 <- pD_rep3 %>% full_join(LF_rep3, by =c("PGs" = "PGs"))
pD_LF_rep3[is.na(pD_LF_rep3)] <- FALSE
pD_LF_rep3 <- pD_LF_rep3[,-4]
pd <- euler(pD_LF_rep3, shape = "ellipse",quantities = TRUE)

pdf(file = "Fig2_venn_rep3.pdf", width = 7, height = 7)
plot(pd,
     quantities = TRUE,
     labels = list(font = 4), fill=my_fin_colors2, alpha = 1)
dev.off()

###################
all_pep <- rbind(p_mTRAQ, p_unlab, p_DDA)
all_pep$Sample <- gsub(".*-","",all_pep$Sample_new)

p_all_tab <- all_pep %>% dplyr::group_by(Sample_new) %>% dplyr::mutate(mean = mean(n)) %>%
  dplyr::mutate(se = sd(n)/sqrt(3)) %>% dplyr::distinct(Sample_new, .keep_all=T)

a <- p_all_tab[3,6]
b <- p_all_tab[2,6] + p_all_tab[3,6]
c <- p_all_tab[2,6] + p_all_tab[1,6] + p_all_tab[3,6]

p_all_tab$er_max <- p_all_tab$mean + p_all_tab$se
p_all_tab[1,8] <- c+as.numeric(p_all_tab[1,7])
p_all_tab[2,8] <- b+as.numeric(p_all_tab[2,7])
p_all_tab[3,8] <- a+p_all_tab[3,7]

p_all_tab$er_min <- p_all_tab$mean - p_all_tab$se
p_all_tab[1,9] <- c-as.numeric(p_all_tab[1,7])
p_all_tab[2,9] <- b-as.numeric(p_all_tab[2,7])
p_all_tab[3,9] <- a-p_all_tab[3,7]


## for DDA
a <- p_all_tab[9,6]
b <- p_all_tab[8,6] + p_all_tab[9,6]
c <- p_all_tab[8,6] + p_all_tab[7,6] + p_all_tab[9,6]

p_all_tab[7,8] <- c+as.numeric(p_all_tab[7,7])
p_all_tab[8,8] <- b+as.numeric(p_all_tab[8,7])
p_all_tab[9,8] <- a+p_all_tab[9,7]

p_all_tab[7,9] <- c-as.numeric(p_all_tab[7,7])
p_all_tab[8,9] <- b-as.numeric(p_all_tab[8,7])
p_all_tab[9,9] <- a-p_all_tab[9,7]


#### text 
df_n <- p_all_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+7000
my_fin_colors <- c("turquoise3","palevioletred3","#ceca7c")

p1_cov <- ggplot(p_all_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = p_all_tab %>% filter(Condition == "plexDIA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), color="black", width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "mTRAQ\nDDA"),#colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), color="black", width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "LF-DIA"),# colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), color="black", width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Precursors / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12),
        legend.position = c(0.7,0.8),
        legend.direction = "horizontal",
        legend.spacing.x = unit(1.0, 'line')) +
  guides(fill = guide_legend(override.aes= list(alpha = 0.7),
                             title = "Samples",
                             label.position = "bottom",
                             title.position = "top", title.vjust = 1, title.hjust=0.5)) +
  geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3
  ) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3, stat="identity", position=position_dodge(1.17))

p1_cov
ggsave("Figure2a.pdf",p1_cov,width=5.1,height=4.6, dpi=500)


############ prots

###################
all_prot <- rbind(prot_mTRAQ, prot_unlab, prot_DDA)
all_prot$Sample <- gsub(".*-","",all_prot$Sample_new)

all_prot_tab <- all_prot %>% dplyr::group_by(Sample_new) %>% dplyr::mutate(mean = mean(n)) %>%
  dplyr::mutate(se = sd(n)/sqrt(3)) %>% dplyr::distinct(Sample_new, .keep_all=T)

a <- all_prot_tab[3,6]
b <- all_prot_tab[2,6] + all_prot_tab[3,6]
c <- all_prot_tab[2,6] + all_prot_tab[1,6] + all_prot_tab[3,6]

all_prot_tab$er_max <- all_prot_tab$mean + all_prot_tab$se
all_prot_tab[1,8] <- c+as.numeric(all_prot_tab[1,7])
all_prot_tab[2,8] <- b+as.numeric(all_prot_tab[2,7])
all_prot_tab[3,8] <- a+all_prot_tab[3,7]

all_prot_tab$er_min <- all_prot_tab$mean - all_prot_tab$se
all_prot_tab[1,9] <- c-as.numeric(all_prot_tab[1,7])
all_prot_tab[2,9] <- b-as.numeric(all_prot_tab[2,7])
all_prot_tab[3,9] <- a-all_prot_tab[3,7]


## for DDA
a <- all_prot_tab[9,6]
b <- all_prot_tab[8,6] + all_prot_tab[9,6]
c <- all_prot_tab[8,6] + all_prot_tab[7,6] + all_prot_tab[9,6]

all_prot_tab[7,8] <- c+as.numeric(all_prot_tab[7,7])
all_prot_tab[8,8] <- b+as.numeric(all_prot_tab[8,7])
all_prot_tab[9,8] <- a+all_prot_tab[9,7]

all_prot_tab[7,9] <- c-as.numeric(all_prot_tab[7,7])
all_prot_tab[8,9] <- b-as.numeric(all_prot_tab[8,7])
all_prot_tab[9,9] <- a-all_prot_tab[9,7]


#### text 
df_n <- all_prot_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+700

p2_cov <- ggplot(all_prot_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = all_prot_tab %>% filter(Condition == "plexDIA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "mTRAQ\nDDA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Protein Data Points / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.position = "none") +
   geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2, stat="identity", position=position_dodge(1.17))

p2_cov
ggsave("Figure2b.pdf",p2_cov,width=5.1,height=4.6, dpi=500)


################## JACCARD 

mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]

mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0|mTRAQ$Precursor.Translated>0),]  #only keep precursors with Ms1 quant

ev2_0 <- mTRAQ[grepl("-0",mTRAQ$Precursor.Id),] %>% dplyr::distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("-4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Ms1.Area","Protein.Group") %>% dplyr::distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Ms1.Area_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("-8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Ms1.Area","Protein.Group") %>% dplyr::distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Ms1.Area_d8","Protein.Group")

d0_1 <- ev2_0[grepl("wJD803", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("wJD803", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_1 <- ev2_8[grepl("wJD803", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("wJD804", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("wJD804", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_2 <- ev2_8[grepl("wJD804", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("wJD815", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("wJD815", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_3 <- ev2_8[grepl("wJD815", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value

################################### unlabeled
unlab <- data.frame(fread(LF_fpath))
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]

df_t <- data.frame(c("wJD756_re", "wJD757_re", "wJD758_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0|unlab_1$Precursor.Translated>0),]  #only keep precursors with MS1 quant

##
df_t <- data.frame(c("wJD759_re", "wJD760_re", "wJD761_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0|unlab_2$Precursor.Translated>0),]  #only keep precursors with MS1 quant

##

df_t <- data.frame(c("wJD762_re", "wJD763_re", "wJD764_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_3 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_",unlab_3$rep)
unlab_3 <- unlab_3[which(unlab_3$Ms1.Area>0|unlab_3$Precursor.Translated>0),]  #only keep precursors with MS1 quant

#####
s1_1 <- unlab_1[grepl("wJD756_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("wJD757_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("wJD758_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)

s2_1 <- unlab_2[grepl("wJD759_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("wJD760_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("wJD761_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)

s3_1 <- unlab_3[grepl("wJD762_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_2 <- unlab_3[grepl("wJD763_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_3 <- unlab_3[grepl("wJD764_re", unlab_3$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3,s3_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3,s3_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s3_1","s1_2","s2_2","s3_2","s1_3","s2_3","s3_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################

############### Jaccard index  DDA

DDA <- data.frame(fread(DDA_fpath))

DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_1 <- ev2_8[grepl("wJD750", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_2 <- ev2_8[grepl("wJD751", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_3 <- ev2_8[grepl("wJD752", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")
jac_dist_DDA <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_DDA_m <- get_lower_tri(jac_dist_DDA)
jac_dist_DDA_m <- melt(jac_dist_DDA_m, na.rm=T)
jac_dist_DDA_m$value <- 1-jac_dist_DDA_m$value

########### plot:

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

pDDA <- ggplot(jac_dist_DDA_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black", size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="mTRAQ DDA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) +
  geom_text(aes(label = round(value, 2)*100))

pDDA
###########

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black", size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="plexDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) +
  geom_text(aes(label = round(value, 2)*100))
p2
############## LF

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black",size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="Label-Free DIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  #scale_fill_gradient2(high = "aquamarine", low = "maroon",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) 
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) +
   geom_text(aes(label = round(value, 2)*100))

p1

## combine
library(gridExtra)
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                            p1 + theme(legend.position="none"), NULL,
                            pDDA +theme(legend.position="none"),
                            nrow=1, widths=c(1,0.1,1,0.1,1)))

ggsave("Figure2d.pdf", p3, width=12.5, height=5.2, dpi=500)


############################ other plots:
jac_dist_mTRAQ_m$Type <- "plexDIA"
m_jac_LF$Type <- "LF-DIA"

jac_dist_mTRAQ_m$run_a <- stri_sub(jac_dist_mTRAQ_m$Var1,-1,-1)
jac_dist_mTRAQ_m$run_b <- stri_sub(jac_dist_mTRAQ_m$Var2,-1,-1)
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% dplyr::mutate("set" = ifelse(run_a == run_b, "Within a run", "Across runs"))

m_jac_LF$run_a <- stri_sub(m_jac_LF$Var1,-1,-1)
m_jac_LF$run_b <- stri_sub(m_jac_LF$Var2,-1,-1)
m_jac_LF <- m_jac_LF %>% mutate("set" = "Across runs")

both <- bind_rows(jac_dist_mTRAQ_m, m_jac_LF)
both <- both[which(both$value<1),]  #remove overlaps to self

both$Var1a <- substr(both$Var1,1,nchar(as.character(both$Var1))-1)
both$Var2a <- substr(both$Var2,1,nchar(as.character(both$Var2))-1)

both <- both %>% dplyr::mutate("Groups" = ifelse(Var1a == Var2a, "Same samples", "Different samples"))
both$set <- factor(both$set, levels=c("Within a run", "Across runs"))
both$Type <- factor(both$Type, levels=c("plexDIA", "LF-DIA"))
both$missingness <- (1-both$value)*100
both$Groups <- factor(both$Groups, levels=c("Same samples", "Different samples"))

ggplot(both, aes(x=as.factor(Type),y=(missingness))) + 
  geom_point(size=2.5, alpha =0.7, position=position_jitter(width=0.25, height=0),
             aes(fill=set), 
       colour="black",pch=21) +
  #geom_text(data=df_n, aes(x=as.factor(Label), y=log2(d0_d4_rat), label=n), col='black', size=4) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA")) + 
  #scale_fill_manual(values = c("seagreen2", "mediumpurple2")) + #"tan2"
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + #"tan2"
  theme_classic() +
  facet_grid(~Groups, scales = "free") +
  #ylim(y_d0d4_val[1],y_d0d4_val[2]) + 
  theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=16),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12),
        legend.text = element_text(size=14),
        legend.title = element_blank())+ 
  guides(fill = guide_legend(override.aes = list(size=4)))+
  #scale_fill_manual(values = c("lightslategrey", "darkorange", "mediumpurple1")) +
  labs(x = "", y = "Missing data between pairs, %", fill = "") + ylim(0, 35) 
ggsave("Figure2e.pdf", width=3.85, height=4, dpi=500)

```

Figure 3: Quantitative accuracy
```{r}
pD <- data.frame(fread(fpath))
LF <- data.frame(fread(LF_fpath))

## plexDIA
pD <- pD_isotopicCO(pD)  #correct for isotopic carryover across channels.. may take 2 minutes
pD_1 <- pD[grepl("wJD804", pD$Run),]
pD_1 <- pD_channel(pD_1) #Note which label is tagged on each precursor
pD_1 <- pD_1[which(pD_1$Lib.PG.Q.Value<0.01),]
pD_1 <- pD_seqcharge(pD_1)
pD_1 <- pD_rmMixSpec(pD_1) #Remove protein groups which map to >1 species
pD_1$File.Name <- paste0(pD_1$Run,"_", pD_1$channel_name)
pD_2 <- pD_diann_quant(pD_1, id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
pD_2 <- pD_2 %>% dplyr::rename("Protein.Group" = "PGs")
pD_list <- pD_prepPlot(pD_2, Run = "wJD804") #data wrangle and prepare for plotting


### LF-DIA
LF_1 <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("JD757", Run), "mTRAQ0",
                                             ifelse(grepl("JD760", Run), "mTRAQ4",
                                                    ifelse(grepl("JD763", Run), "mTRAQ8", "remove")))) %>%
  dplyr::filter(!grepl("remove", channel_name))
LF_1$Run <- "LF_runs"
LF_1 <- LF_1[which(LF_1$Lib.PG.Q.Value<0.01),]
LF_1 <- pD_rmMixSpec(LF_1) #Remove protein groups which map to >1 species
LF_1 <- pD_seqcharge(LF_1)
LF_1$File.Name <- paste0(LF_1$Run,"_", LF_1$channel_name)
LF_2 <- pD_diann_quant(LF_1, id.header = "seqcharge", quantity.header = "Ms1.Area")
LF_2 <- LF_2 %>% dplyr::rename("Protein.Group" = "PGs")
LF_list <- pD_prepPlot(LF_2, Run = "LF_runs") #data wrangle and prepare for plotting


## join the plexDIA and LF-DIA sets and plot
pD_LF <- pD_plotLFQBench(pD_list, LF_list, pD_list_name = "plexDIA", LF_list_name = "LF-DIA")
ggsave("Figure3abc.pdf", pD_LF, width=11,height=9, dpi=500)


################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########       Figure D: In-set vs Out-of-Set      ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 


df_names <- pD %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)
df <- pD
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$File.Name <- paste0(df$Run, df$channel_name)
df1 <- df[grepl("wJD803", df$Run),]
protein.groups1 <- diann_maxlfq(df1[df1$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
df2 <- df[grepl("wJD804", df$Run),]
protein.groups2 <- diann_maxlfq(df2[df2$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")

protein.groups1 <- data.frame(protein.groups1)
protein.groups2 <- data.frame(protein.groups2)

protein.groups1 <- protein.groups1 %>% dplyr::rename("d0" = "wJD803mTRAQ0",
                                                     "d4" = "wJD803mTRAQ4",
                                                     "d8" = "wJD803mTRAQ8")
protein.groups2 <- protein.groups2 %>% dplyr::rename("d0" = "wJD804mTRAQ0",
                                                     "d4" = "wJD804mTRAQ4",
                                                     "d8" = "wJD804mTRAQ8")

rep1_d0d4_pt <- protein.groups2
PGs <- rownames(rep1_d0d4_pt)
rep1_d0d4_pt <- cbind(rep1_d0d4_pt, PGs)
rep1_d0d4_pt$Prot_rat <- as.numeric(rep1_d0d4_pt$d0)/as.numeric(rep1_d0d4_pt$d4)
rep1_d0d4_pt <- rep1_d0d4_pt[!is.na(rep1_d0d4_pt$Prot_rat), ]
rep1_d0d4_pt <- rep1_d0d4_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_pt <- pD_rmMixSpec(rep1_d0d4_pt)
rep1_d0d4_pt <- pD_species(rep1_d0d4_pt)
med_human <- rep1_d0d4_pt[grepl("H. sapiens", rep1_d0d4_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_pt$Prot_rat <- rep1_d0d4_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_pt$Type <- "Within a run"

######## out of set:
rep1_d0d4_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- cbind(rep1_d0d4_out_pt, PGs)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% dplyr::select("d0", "PGs")
d4_set2 <- rep1_d0d4_pt %>% dplyr::select("d4", "PGs")

rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% inner_join(d4_set2, by =c("PGs" = "PGs"))
rep1_d0d4_out_pt$Prot_rat <- as.numeric(rep1_d0d4_out_pt$d0)/as.numeric(rep1_d0d4_out_pt$d4)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!is.na(rep1_d0d4_out_pt$Prot_rat), ]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_out_pt <- pD_rmMixSpec(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- pD_species(rep1_d0d4_out_pt)
med_human <- rep1_d0d4_out_pt[grepl("H. sapiens", rep1_d0d4_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_out_pt$Prot_rat <- rep1_d0d4_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_out_pt$Type <- "Across runs"


######################### d0:d8 ##################################

rep1_d0d8_pt <- protein.groups2
PGs <- rownames(rep1_d0d8_pt)
rep1_d0d8_pt <- cbind(rep1_d0d8_pt, PGs)
rep1_d0d8_pt <- data.frame(rep1_d0d8_pt)
rep1_d0d8_pt$Prot_rat <- as.numeric(rep1_d0d8_pt$d0)/as.numeric(rep1_d0d8_pt$d8)
rep1_d0d8_pt <- rep1_d0d8_pt[!is.na(rep1_d0d8_pt$Prot_rat), ]
rep1_d0d8_pt <- rep1_d0d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_pt <- pD_rmMixSpec(rep1_d0d8_pt)
rep1_d0d8_pt <- pD_species(rep1_d0d8_pt)
med_human <- rep1_d0d8_pt[grepl("H. sapiens", rep1_d0d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_pt$Prot_rat <- rep1_d0d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_pt_h <- rep1_d0d8_pt
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("H.", rep1_d0d8_pt$species),]
rep1_d0d8_pt$Type <- "Within a run"

######## out of set:
rep1_d0d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- cbind(rep1_d0d8_out_pt, PGs)
rep1_d0d8_out_pt <- data.frame(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% dplyr::select("d0", "PGs")
d8_set2 <- rep1_d0d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d0d8_out_pt$Prot_rat <- as.numeric(rep1_d0d8_out_pt$d0)/as.numeric(rep1_d0d8_out_pt$d8)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!is.na(rep1_d0d8_out_pt$Prot_rat), ]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_out_pt <- pD_rmMixSpec(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- pD_species(rep1_d0d8_out_pt)
med_human <- rep1_d0d8_out_pt[grepl("H. sapiens", rep1_d0d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_out_pt$Prot_rat <- rep1_d0d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("H.", rep1_d0d8_out_pt$species),]
rep1_d0d8_out_pt$Type <- "Across runs"


######################################## d4:d8 ###########################

rep1_d4d8_pt <- protein.groups2
PGs <- rownames(rep1_d4d8_pt)
rep1_d4d8_pt <- cbind(rep1_d4d8_pt, PGs)
rep1_d4d8_pt <- data.frame(rep1_d4d8_pt)
rep1_d4d8_pt$Prot_rat <- as.numeric(rep1_d4d8_pt$d4)/as.numeric(rep1_d4d8_pt$d8)
rep1_d4d8_pt <- rep1_d4d8_pt[!is.na(rep1_d4d8_pt$Prot_rat), ]
rep1_d4d8_pt <- rep1_d4d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_pt <- pD_rmMixSpec(rep1_d4d8_pt)
rep1_d4d8_pt <- pD_species(rep1_d4d8_pt)
med_human <- rep1_d4d8_pt[grepl("H. sapiens", rep1_d4d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_pt$Prot_rat <- rep1_d4d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_pt_h <- rep1_d4d8_pt
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("H.", rep1_d4d8_pt$species),]
rep1_d4d8_pt$Type <- "Within a run"

######## out of set:
rep1_d4d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- cbind(rep1_d4d8_out_pt, PGs)
rep1_d4d8_out_pt <- data.frame(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% dplyr::select("d4", "PGs")
d8_set2 <- rep1_d4d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d4d8_out_pt$Prot_rat <- as.numeric(rep1_d4d8_out_pt$d4)/as.numeric(rep1_d4d8_out_pt$d8)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!is.na(rep1_d4d8_out_pt$Prot_rat), ]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_out_pt <- pD_rmMixSpec(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- pD_species(rep1_d4d8_out_pt)
med_human <- rep1_d4d8_out_pt[grepl("H. sapiens", rep1_d4d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_out_pt$Prot_rat <- rep1_d4d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("H.", rep1_d4d8_out_pt$species),]
rep1_d4d8_out_pt$Type <- "Across runs"



###############
rep1_d0d4_pt <- rep1_d0d4_pt[rep1_d0d4_pt$PGs%in%rep1_d0d4_out_pt$PGs,]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[rep1_d0d4_out_pt$PGs%in%rep1_d0d4_pt$PGs,]

rep1_d0d8_pt <- rep1_d0d8_pt[rep1_d0d8_pt$PGs%in%rep1_d0d8_out_pt$PGs,]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[rep1_d0d8_out_pt$PGs%in%rep1_d0d8_pt$PGs,]

rep1_d4d8_pt <- rep1_d4d8_pt[rep1_d4d8_pt$PGs%in%rep1_d4d8_out_pt$PGs,]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[rep1_d4d8_out_pt$PGs%in%rep1_d4d8_pt$PGs,]

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))
df2<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(1/6), log2(6)))

d0d4 <- bind_rows(rep1_d0d4_pt, rep1_d0d4_out_pt)
d0d8 <- bind_rows(rep1_d0d8_pt, rep1_d0d8_out_pt)
d4d8 <- bind_rows(rep1_d4d8_pt, rep1_d4d8_out_pt)

d0d4 <- d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8 <- d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8 <- d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4$error <- abs(log2(d0d4$Prot_rat)-d0d4$Z)
d0d8$error <- abs(log2(d0d8$Prot_rat)-d0d8$Z)
d4d8$error <- abs(log2(d4d8$Prot_rat)-d4d8$Z)

all <- rbind(d0d4, d4d8, d0d8)
####

####


y_d0d4 <- all$error %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,0.99)))

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))

all$Type <- factor(all$Type, levels=c("Within a run", "Across runs"))

D_fig <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + 
  geom_boxplot(alpha =0.9, outlier.shape=NA) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  #scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "") +
  coord_cartesian(ylim = c(0,4))
D_fig

D_fig_noL <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + 
  geom_boxplot(alpha =0.9, outlier.shape=NA) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  #scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "",subtitle="",title="") +
  coord_cartesian(ylim = c(0,4))
D_fig_noL
ggsave("Figure3d_nolegend.pdf", width=3.8,height=3.7)

################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########        Figure E: R vs K quant MS2         ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 

ev <- pD[grepl("wJD804", pD$Run),]
ev_MS1 <- pD_PrecRatios(ev, quant.header = "Ms1.Area_iso")
ev_MS1$Quant <- "MS1"
ev_MS2 <- pD_PrecRatios(ev, quant.header = "Precursor.Translated")
ev_MS2$Quant <- "MS2"
ev <- rbind(ev_MS1, ev_MS2)
ev <- pD_species(ev)
ev$lab_species <- paste0(ev$variable, "_", ev$species)
df1<- data.frame(species = c("d0_d4_E. coli", "d0_d4_H. sapiens", "d0_d4_S. cerevisiae", "d0_d8_E. coli", "d0_d8_S. cerevisiae", "d4_d8_E. coli", "d4_d8_S. cerevisiae"), Z = c(2, 0, -1, log2(2/3), log2(3), log2(1/6), log2(6)))
ev <- ev %>% left_join(df1, by =c("lab_species" = "species"))
ev$seq_run_var <- paste0(ev$seqcharge_file, ev$variable)
ev <- ev %>% dplyr::group_by(seq_run_var) %>% dplyr::add_count() %>% dplyr::filter(n==2) #require quant in MS1 and MS2
ev <- ev[!is.na(ev$Z),] #removes human in d0_d8 and d4_d8 comparisons because they compare U937 and Jurkat cell lines
ev <- ev %>% dplyr::select(-"n")
#####
ev <- pD_Cterm(ev)
all_MS2_K <- ev[grepl("K", ev$Cterm),] %>% dplyr::filter(grepl("MS2", Quant))
all_MS2_R <- ev[grepl("R", ev$Cterm),] %>% dplyr::filter(grepl("MS2", Quant))

all_MS2_K$Type <- "Lys precursors"
all_MS2_K <- all_MS2_K %>% dplyr::add_count(species)
all_MS2_K$cond <- paste0("n=", all_MS2_K$n)

all_MS2_R$Type <- "Arg precursors"
all_MS2_R <- all_MS2_R %>% dplyr::add_count(species)
all_MS2_R$cond <- paste0("n=", all_MS2_R$n)

####
all_MS1 <- ev[grepl("MS1", ev$Quant),] %>% dplyr::add_count(species)
all_MS1$Type <- "All precursors"
all_MS1$cond <- paste0("n=", all_MS1$n)

K_vs_R_MS1_mDIA_d0d4 <- bind_rows(all_MS2_K, all_MS2_R, all_MS1)
K_vs_R_MS1_mDIA_d0d4$error <- abs(log2(K_vs_R_MS1_mDIA_d0d4$value)-K_vs_R_MS1_mDIA_d0d4$Z)

############ add in the MS1 quant for multiDIA precursors:
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% dplyr::distinct(cond,.keep_all=T)
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% dplyr::distinct(cond,.keep_all=T)
K_vs_R_MS1_mDIA_d0d4$Type <- factor(K_vs_R_MS1_mDIA_d0d4$Type, levels=c("Lys precursors", "Arg precursors", "All precursors"))

E_fig <- ggplot(K_vs_R_MS1_mDIA_d0d4, aes(x=as.factor(Type),y=error, fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  scale_x_discrete(limits=c("Lys precursors","Arg precursors", "All precursors")) + 
  scale_fill_manual(values = c("dodgerblue3", "deepskyblue", "tan2")) +
  theme_classic() +
  facet_grid(~species, scales = "free")  +
  coord_cartesian(ylim = c(0,5)) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 16, hjust = 0.5, face="bold"),
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y=expression(paste("|",Log["2"],", Precursor Ratio Error|")), fill = "",subtitle="",title="")
E_fig
ggsave("Figure3e.pdf", width=5.7,height=3.7)

```

Figure 4: Quantitative reproducibility (CVs, correlation to LF, Dif. ab. testing)
```{r}
pD <- data.frame(fread(fpath))
LF <- data.frame(fread(LF_fpath))

pD <- pD_channel(pD)
pD <- pD_seqcharge(pD)
pD$File.Name <- paste0(pD$Run, pD$channel_name)
pD <- pD_isotopicCO(pD)
pD_keep <-pD
pD <- pD[grepl("wJD803|804|815", pD$Run),]
protein.groups <- diann_maxlfq(pD[pD$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- robustbase::colMedians(protein.groups, na.rm=T)
protein.groups_n_plex <- sweep(protein.groups, 2, cmeds, FUN = '/')
PGs <- rownames(protein.groups)
protein.groups_n_plex <- cbind(protein.groups_n_plex,PGs)

df_names <- pD %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)

protein.groups_n_plex <- data.frame(protein.groups_n_plex)
protein.groups_n_plex <- protein.groups_n_plex %>% inner_join(df_names, by =c("PGs"="Protein.Group"))
protein.groups_n_plex <- pD_rmMixSpec(protein.groups_n_plex)
protein.groups_n_plex <- pD_species(protein.groups_n_plex)


######################## LF ########################################

LF <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("wJD757_re|wJD756_re|wJD758_re", Run), "mTRAQ0",
                                            ifelse(grepl("wJD759_re|wJD760_re|wJD761_re", Run), "mTRAQ4", "mTRAQ8")))
LF <- pD_seqcharge(LF)
LF$File.Name <- paste0(LF$Run, LF$channel_name)
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(LF[LF$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- colMedians(protein.groups, na.rm=T)
protein.groups_n <- sweep(protein.groups, 2, cmeds, FUN = '/')
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)
df_names <- LF %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)
protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% left_join(df_names, by =c("PGs"="Protein.Group"))
protein.groups_n <- pD_rmMixSpec(protein.groups_n)
protein.groups_n <- pD_species(protein.groups_n)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n_plex <- data.frame(protein.groups_n_plex)
## intersect plexDIA and LF-DIA ###
protein.groups_n <- protein.groups_n[protein.groups_n$PGs%in%protein.groups_n_plex$PGs,]
protein.groups_n_plex <- protein.groups_n_plex[protein.groups_n_plex$PGs%in%protein.groups_n$PGs,]

rownames(protein.groups_n) <- protein.groups_n$PGs
rownames(protein.groups_n_plex) <- protein.groups_n_plex$PGs
####### plex cont.

protein.groups_n_plex <- protein.groups_n_plex %>% dplyr::select(-matches(c("PGs", "species", "HY", "Protein.Names")))
m_PGs <- melt(as.matrix(protein.groups_n_plex))
m_PGs <- m_PGs %>% dplyr::mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                                                       ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs <- m_PGs[!is.na(m_PGs$value),]
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()
  
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_mean" = mean(as.numeric(value))) %>% dplyr::ungroup()
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("Prot_CV" = prot_sd/prot_mean) %>% dplyr::ungroup()
m_PGs$uni <- paste0(m_PGs$Var1,m_PGs$label)
m_PGs <- m_PGs %>% dplyr::distinct(uni, .keep_all=T)
m_PGs$Cond <- "plexDIA"

###### LF cont.
protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),] %>% dplyr::select(-matches(c("PGs","species", "HY", "Protein.Names")))

m_PGs_LF <- reshape2::melt(as.matrix(protein.groups_n))
m_PGs_LF <- m_PGs_LF %>% dplyr::mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                           ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs_LF <- m_PGs_LF[!is.na(m_PGs_LF$value),]
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::add_count() %>% filter(n==3) %>% ungroup()

m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("Prot_CV" = prot_sd/prot_mean) %>% dplyr::ungroup()
m_PGs_LF$uni <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs_LF <- m_PGs_LF %>% dplyr::distinct(uni, .keep_all=T)
m_PGs_LF$Cond <- "LF-DIA"

m_PGs_LF$Var1_lab <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs$Var1_lab <- paste0(m_PGs$Var1,m_PGs$label)

## last intersection:
m_PGs_LF <- m_PGs_LF[m_PGs_LF$Var1_lab%in%m_PGs$Var1_lab,]
m_PGs <- m_PGs[m_PGs$Var1_lab%in%m_PGs_LF$Var1_lab,]


##
################ select and bind:

mDIA_LF <- bind_rows(m_PGs,m_PGs_LF)
mDIA_LF$Cond <- factor(mDIA_LF$Cond, levels=c("plexDIA", "LF-DIA"))
df1 <- mDIA_LF %>% dplyr::group_by(Cond) %>% dplyr::mutate("med_CV" = round(median(Prot_CV,na.rm=T),3)) %>% dplyr::distinct(Cond,.keep_all=T)

ymax <- mDIA_LF$Prot_CV %>% sort()
ymax <- as.numeric(quantile(ymax,probs=c(.01,.99)))


cvs <- c(round(median(m_PGs$Prot_CV),3), round(median(m_PGs_LF$Prot_CV),3)) 

CV1_fig <- ggplot(mDIA_LF, aes(x=Cond,y=Prot_CV)) + 
  geom_quasirandom(alpha=0.1, groupOnX = T)+
  geom_boxplot(width=0.25, outlier.shape=NA, color="grey50") +
  theme_classic() +
  geom_text(data=df1, aes(x=Cond, y=cvs+0.035, label=paste0(cvs)), color="black",parse = FALSE, size=3.4) +
  theme(axis.text.x = element_text(size=16, face = "bold", color = "black"),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        axis.ticks.x = element_blank(),
        legend.position = "none",
        legend.text = element_text(size=12))+
  labs(x = "", y = "Sample-specific protein CV", fill = "")

CV1_fig
ggsave("Figure4a_beeswarm.png", width=4,height=4.5, dpi=300)

########################################################################################
########################################################################################
############################ Protein differential abundance ############################
########################################################################################
########################################################################################
cnames <- c("Protein.Group", "Protein.Ids", "Protein.Names", "Genes", "First.Protein.Description", "Proteotypic", "Stripped.Sequence", "Modified.Sequence", "Precursor.Charge", "Precursor.Id")
######################### LF:

LF <- fread(LF_fpath, select = c(cnames,"Run","Lib.PG.Q.Value","Ms1.Area"))
keep_LF <- LF
plex <- keep_LF
plex <- plex[which(plex$Lib.PG.Q.Value<0.01),]
plex <- pD_rmMixSpec(plex) # remove any proteins groups with >1 species

#######
plex <- plex %>% dplyr::mutate("channel_name" = ifelse(grepl("wJD759|wJD760|wJD761", Run), "mTRAQ0",
                                            ifelse(grepl("wJD762|wJD763|wJD764", Run), "mTRAQ4", "mTRAQ8")))
plex <- pD_seqcharge(plex)
plex$precRun <- paste0(plex$seqcharge, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] #%>% add_count(precRun) %>% filter(n==2)
plex <- plex %>% dplyr::filter(grepl("HUMAN", Protein.Names))
h <- plex %>% dplyr::filter(grepl("HUMAN", Protein.Names)) %>% 
  dplyr::group_by(chanRun) %>% dplyr::mutate(med_MS1 = median(Ms1.Area)) %>% 
  dplyr::distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area>0),]
plex <- plex %>% dplyr::mutate(Ms1.Area_cnorm = Ms1.Area/med_MS1)
plex <- plex %>% dplyr::group_by(seqcharge) %>% 
  dplyr::mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% dplyr::ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("seqcharge", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% dplyr::add_count(Protein.Group) %>%
  dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% dplyr::add_count(Protein.Group) %>% 
  dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group)
plex1 <- plex1 %>% dplyr::group_by(channel_name, Protein.Group) %>% dplyr::mutate("med_pep_ab" = median(Ms1.Area))%>% dplyr::ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),] 

prots_LF <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots_LF <- prots_LF[prots_LF$Protein.Group%in%int$Protein.Group,]
prots_LF$pval <- 10000
unique_prots <- as.vector(prots_LF$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots_LF[i,7] <- as.numeric(pval_temp)
  
}

prots_LF$FC_qval <-  p.adjust(prots_LF$pval, method = 'BH') # qvalue
prots_LF <- pD_species(prots_LF)
write.table(prots_LF, "U_J_DA_LF_02252022.txt", sep = "\t", row.names = FALSE)


###################
###################
################### alternative PGs quant
meta_ben <- read.delim(meta_bench)
meta_ben <- meta_ben %>% dplyr::mutate("channel_name" = ifelse(grepl("0", Label), "mTRAQ0",
                                                               ifelse(grepl("4", Label), "mTRAQ4", "mTRAQ8")))
meta_ben$runchan <- paste0(meta_ben$Raw, meta_ben$channel_name)
pD <- pD_keep
pD <- pD[grepl("804|806|808", pD$Run),]
df <- pD[!grepl("YEAST|ECOLI", pD$Protein.Names),]
df_pD <- df
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)
protein.groups_melt <- reshape2::melt(protein.groups_n)
protein.groups_melt <- protein.groups_melt %>% left_join(meta_ben, by =c("Var2" = "runchan"))
protein.groups_melt <- protein.groups_melt[!grepl("A", protein.groups_melt$Celltype),]
protein.groups_melt <- protein.groups_melt[,!grepl("X", colnames(protein.groups_melt))]
protein.groups_melt <- na.omit(protein.groups_melt)
protein.groups_melt$value <- as.numeric(protein.groups_melt$value)
protein.groups_melt <- protein.groups_melt %>% 
  dplyr::group_by(Var2) %>% 
  dplyr::mutate("med_norm" = value/median(value, na.rm=T)) %>%
  dplyr::ungroup() %>% 
  dplyr::add_count(Celltype, Var1) %>% 
  dplyr::filter(n==3) %>%
  dplyr::ungroup() %>%
  dplyr::group_by(Celltype, Var1) %>% 
  dplyr::mutate("mean_PG" = mean(med_norm)) %>% 
  dplyr::distinct(Celltype, Var1, .keep_all = T) %>%
  dplyr::select("Var1", "Celltype", "mean_PG")
PG_cast <- reshape2::dcast(protein.groups_melt, Var1 ~ Celltype, value.var = "mean_PG", fill=NA)
PG_cast$U_J <- log2(PG_cast$B/PG_cast$C)
PG_cast <- na.omit(PG_cast)

### LF

meta_ben <- read.delim(meta_bench)
LF <- data.frame(fread(LF_fpath))
LF <- LF[grepl("759|760|761|762|763|764", LF$Run),]
LF <- pD_seqcharge(LF)
LF <- LF[!grepl("YEAST|ECOLI", LF$Protein.Names),]
df <- LF
df$File.Name <- df$Run
#df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area")
protein.groups_n <- protein.groups                                      #protein.groups/protein.groups_med
PGs <- rownames(protein.groups)
protein.groups_n <- cbind(protein.groups_n,PGs)
protein.groups_melt <- reshape2::melt(protein.groups_n)
protein.groups_melt <- protein.groups_melt %>% left_join(meta_ben, by =c("Var2" = "Raw"))
protein.groups_melt <- protein.groups_melt[!grepl("A", protein.groups_melt$Celltype),]
protein.groups_melt <- protein.groups_melt[,!grepl("X", colnames(protein.groups_melt))]
protein.groups_melt <- na.omit(protein.groups_melt)
protein.groups_melt$value <- as.numeric(protein.groups_melt$value)
protein.groups_melt <- protein.groups_melt %>% 
  dplyr::group_by(Var2) %>% 
  dplyr::mutate("med_norm" = value/median(value, na.rm=T)) %>%
  dplyr::ungroup() %>% 
  dplyr::add_count(Celltype, Var1) %>% 
  dplyr::filter(n==3) %>%
  dplyr::ungroup() %>%
  dplyr::group_by(Celltype, Var1) %>% 
  dplyr::mutate("mean_PG" = mean(med_norm)) %>% 
  dplyr::distinct(Celltype, Var1, .keep_all = T) %>%
  dplyr::select("Var1", "Celltype", "mean_PG")
LF_PG_cast <- reshape2::dcast(protein.groups_melt, Var1 ~ Celltype, value.var = "mean_PG", fill=NA)
LF_PG_cast$U_J <- log2(LF_PG_cast$B/LF_PG_cast$C)
LF_PG_cast <- na.omit(LF_PG_cast)

### intersect
LF_PG_cast <- LF_PG_cast %>% dplyr::select("Var1", "U_J")
colnames(LF_PG_cast) <- c("PG", "U_J_LF")

PG_cast <- PG_cast %>% dplyr::select("Var1", "U_J")
colnames(PG_cast) <- c("PG", "U_J_pD")
both_cast <- LF_PG_cast %>% inner_join(PG_cast, by =c("PG" = "PG"))
DA <- read.delim("U_J_DA_LF_02252022.txt")
both_cast <- both_cast %>% left_join(DA, by =c("PG" ="Protein.Group"))

#calculate cors
plexDIA_LFDIA_cor <- cor(both_cast$U_J_LF, both_cast$U_J_pD, method="spearman")
DE_m_LF <- both_cast[which(both_cast$FC_qval < 0.01),]
plexDIA_LFDIA_cor_DE <- cor(DE_m_LF$U_J_LF, DE_m_LF$U_J_pD, method="spearman")

nrow(DE_m_LF)
nrow(both_cast)

both_cast<- both_cast %>% dplyr::mutate("FDR" = ifelse(FC_qval < 0.01, "yes", "no"))

ggplot(both_cast, aes(x=U_J_LF, y=U_J_pD)) +
  geom_point(data = both_cast %>% filter(FC_qval>=0.01), alpha=0.2)+
#  geom_point(data = j_multiDIA_LF_human %>% filter(FC_qval<0.01), alpha=0.4, color="red")+
  geom_pointdensity(data = both_cast %>% filter(FC_qval<0.01), aes(x=U_J_LF, y=U_J_pD), alpha=1)+
scale_colour_gradientn(
  colors =c("purple2","steelblue2", "springgreen3", "yellow", "mediumvioletred"),
  values = NULL,
  space = "Lab",
  na.value = "grey50",
  guide = "colourbar",
  aesthetics = "colour",
  colors
) +

  theme_classic() + theme(legend.position = "none") + 
  theme(axis.text.x = element_text(size=16),
        axis.text.y = element_text(size=16),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18))+
  annotate("text", x=-4, y=4.5, label=paste0("\U03C1 = ", round(plexDIA_LFDIA_cor, 2)), size=3) +
    annotate("text", x=-4, y=4, label=paste0("\U03C1 = ", round(plexDIA_LFDIA_cor_DE, 2)), size=3) +

  #annotate("text", x=-4, y=4, label=expression(paste(rho,"=", plexDIA_LFDIA_cor_DE)), size=3) +

  labs(x = expression(paste(Log["2"],", U-937 / Jurkat")), y=expression(paste(Log["2"],", U-937 / Jurkat")),
       colors = "dfd")#+ ylim(-5,5) + xlim(-5,5)# +

ggsave("Figure4b.pdf", width=4.8,height=4.8, dpi=500, device = "pdf")

bothcast2 <- both_cast %>% dplyr::select("PG", "Protein.Names", "FC_qval", "U_J_LF", "U_J_pD")
write.table(bothcast2, "U937_Jurkat_DA_pD_LFDIA.txt", sep = "\t", row.names = FALSE)

###################################################
################################################### 
################# Empirical FDR #################
################################################### 
################################################### 
library(PRROC)

plex <- pD_keep[which(pD_keep$Lib.PG.Q.Value<0.01),]

# remove any proteins groups with >1 species
plex <- pD_rmMixSpec(plex)
#######
#plex1 <- plex[grepl("803|804", plex$Run),]
#plex2 <- plex[grepl("805|806", plex$Run),]
#plex3 <- plex[grepl("807|808", plex$Run),]

plex1 <- plex[grepl("804", plex$Run),]
plex2 <- plex[grepl("806", plex$Run),]
plex3 <- plex[grepl("808", plex$Run),]

plex1 <- plex1 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("-4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

plex2 <- plex2 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ8",
                                            ifelse(grepl("-4", Modified.Sequence), "mTRAQ0", "mTRAQ4")))

plex3 <- plex3 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ4",
                                            ifelse(grepl("-4", Modified.Sequence), "mTRAQ8", "mTRAQ0")))

plex <- rbind(plex1,plex2,plex3)

plex <- pD_seqcharge(plex)
plex$precRun <- paste0(plex$seqcharge, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] %>% 
  dplyr::add_count(precRun) %>% dplyr::filter(n==2)
h <- plex %>% dplyr::filter(grepl("HUMAN", Protein.Names)) %>% 
  dplyr::group_by(chanRun) %>% dplyr::mutate(med_MS1 = median(Ms1.Area_iso)) %>% 
  distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area_iso>0),]
plex <- plex %>% dplyr::mutate(Ms1.Area_cnorm = Ms1.Area_iso/med_MS1)
plex <- plex %>% dplyr::group_by(seqcharge) %>% 
  dplyr::mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% dplyr::ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("seqcharge", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")

d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% 
  dplyr::add_count(Protein.Group) %>% dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% 
  dplyr::add_count(Protein.Group) %>% dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group)
plex1 <- plex1 %>% dplyr::group_by(channel_name, Protein.Group) %>% 
  dplyr::mutate("med_pep_ab" = median(Ms1.Area)) %>% dplyr::ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),]

prots <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots <- prots[prots$Protein.Group%in%int$Protein.Group,]
prots$pval <- 10000
unique_prots <- as.vector(prots$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots[i,7] <- as.numeric(pval_temp)
  
}
prots$FC_qval <-  prots$pval #p.adjust(prots$pval, method = 'BH') # qvalue
prots$FC_qval <-  p.adjust(prots$pval, method = 'BH') # qvalue

prots <- prots%>% dplyr::mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli",
                                                                  ifelse(grepl("HUMAN", Protein.Names), "H. sapiens",
                                                                         ifelse(grepl("YEAST", Protein.Names), "S. cerevisiae", "remove"))))

######################### LF:
LF <- fread(LF_fpath, select = c(cnames,"Run","Lib.PG.Q.Value","Ms1.Area"))
keep_LF <- LF
plex <- keep_LF
#plex <- plex[grepl("wJD804", plex$Run),]
plex <- plex[which(plex$Lib.PG.Q.Value<0.01),]

# remove any proteins groups with >1 species
plex <- pD_rmMixSpec(plex)
#######

plex <- plex %>% dplyr::mutate("channel_name" = ifelse(grepl("wJD756|wJD757|wJD758", Run), "mTRAQ0",
                                            ifelse(grepl("wJD759|wJD760|wJD761", Run), "mTRAQ4", "mTRAQ8")))
plex <- pD_seqcharge(plex)
plex$precRun <- paste0(plex$seqcharge, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)
plex <- plex[!grepl("mTRAQ8", plex$channel_name),] #%>% add_count(precRun) %>% filter(n==2)
h <- plex %>% dplyr::filter(grepl("HUMAN", Protein.Names)) %>% dplyr::group_by(chanRun) %>% dplyr::mutate(med_MS1 = median(Ms1.Area)) %>% 
  dplyr::distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_MS1")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Ms1.Area>0),]
plex <- plex %>% dplyr::mutate(Ms1.Area_cnorm = Ms1.Area/med_MS1)
plex <- plex %>% dplyr::group_by(seqcharge) %>% 
  dplyr::mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% dplyr::ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("seqcharge", "Ms1.Area", "Protein.Group", "Protein.Names", "channel_name")
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% dplyr::add_count(Protein.Group) %>% dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% dplyr::add_count(Protein.Group) %>% dplyr::filter(n>1) %>% dplyr::distinct(Protein.Group)
plex1 <- plex1 %>% dplyr::group_by(channel_name, Protein.Group) %>% 
  dplyr::mutate("med_pep_ab" = median(Ms1.Area))%>% dplyr::ungroup()
int <- d0[which(d0$Protein.Group%in%d4$Protein.Group),] 

prots_LF <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots_LF <- prots_LF[prots_LF$Protein.Group%in%int$Protein.Group,]
prots_LF$pval <- 10000
unique_prots <- as.vector(prots_LF$Protein.Group)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Protein.Group),]
  aav_test <- aov(Ms1.Area~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots_LF[i,7] <- as.numeric(pval_temp)
  
}

prots_LF$FC_qval <-  prots_LF$pval #p.adjust(prots_LF$pval, method = 'BH') # qvalue
prots_LF <- pD_species(prots_LF)


##### plot PR curves:
DE_LF_lim <- na.omit(prots_LF)#[which(DE_LF$n>2),]
DE_lim <- na.omit(prots)#[which(DE$n>2),]


### LF DIA
neg <- DE_LF_lim[grepl("HUMAN", DE_LF_lim$Protein.Names),]
neg <- 1/(as.vector(neg$FC_qval))

pos <- DE_LF_lim[!grepl("HUMAN", DE_LF_lim$Protein.Names),]
pos <- 1/(as.vector(pos$FC_qval))

pr<-pr.curve(scores.class0 = pos, scores.class1 = neg,  curve=T)
#pr
#plot(pr)

##### plexDIA
neg_plex <- DE_lim[grepl("HUMAN", DE_lim$Protein.Names),]
neg_plex <- 1/as.vector(neg_plex$FC_qval)

pos_plex <- DE_lim[!grepl("HUMAN", DE_lim$Protein.Names),]
pos_plex <- 1/as.vector(pos_plex$FC_qval)

pr_plex<-pr.curve(scores.class0 = pos_plex, scores.class1 = neg_plex,  curve=T)
pr_plex

#plot(pr_plex)
####
#plot(pr_plex, max.plot = TRUE, min.plot = TRUE, rand.plot = T, fill.area = F, color=3, auc.main = T)
#plot(pr, add = TRUE, color = 1)


####### remove human
prots_LF_YE <- prots_LF[!grepl("H. sapiens",prots_LF$species),]
prots_YE <- prots[!grepl("H. sapiens",prots$species),]

#####
plexdia <- data.frame(pr_plex$curve)
plexdia$numprots <- nrow(prots_YE)*(plexdia$X1)# - (1-plexdia$X2)*nrow(prots)*(plexdia$X1) #total proteins - human proteins = YE proteins

lfdia <- data.frame(pr$curve)
lfdia$numprots <- nrow(prots_LF_YE)*(lfdia$X1)# - (1-lfdia$X2)*nrow(prots_LF)*(lfdia$X1) #total proteins - human proteins = YE proteins

plexdia$type <- "plexDIA"
lfdia$type <- "LF-DIA"

both <- rbind(plexdia, lfdia)

ggplot(both, aes(x=100*(1-X2), y=as.numeric(numprots), colour=type)) + geom_point() + 
  labs(x = "Empirical FDR(%)", y = "Protein groups (true positives)", color="") +
  theme_bw()+
    theme(legend.position = "bottom",
        legend.text = element_text(size=16),
        axis.title.x = element_text(size=16),
        axis.title.y = element_text(size=16),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14)) + 
      scale_color_manual(values = c("black", "red"))

#make it so the lines only increases (so it's not jagged)
b <- seq(0, 1, 0.001)
d <- data.frame(b)
d$max_prot <- 0

plexdia1 <- plexdia
plexdia1$FDR <- 1-plexdia1$X2
for(i in 1:nrow(d)){
  a <- d[i,1]
  temp <- plexdia1[which(plexdia1$FDR<=a),]
  m_temp <- max(temp$numprots)
  d[i,2] <- as.numeric(m_temp)
  
}

d$type <- "plexDIA"

b <- seq(0, 1, 0.001)
lf <- data.frame(b)
lf$max_prot <- 0

lfdia1 <- lfdia
lfdia1$FDR <- 1-lfdia1$X2
for(i in 1:nrow(lf)){
  a <- lf[i,1]
  temp <- lfdia1[which(lfdia1$FDR<=a),]
  m_temp <- max(temp$numprots)
  lf[i,2] <- as.numeric(m_temp)
  
}

lf$type <- "LF-DIA"
both1 <- rbind(d,lf)

ggplot(both1, aes(x=100*b, y=as.numeric(max_prot), colour=type)) + geom_line(size=3.9, alpha=0.6)+#alpha=0.025) + 
  labs(x = "Empirical FDR (%)", y = ("Differentially abundant protein groups\n(TPs: S. cerevisiae & E. coli)"), color="") +
  theme_classic() + xlim(0,5) +
      theme(legend.position = c(0.2,0.88),
        legend.text = element_text(size=18),
        axis.title.x = element_text(size=18),
        axis.title.y = element_text(size=18),
        axis.text.x = element_text(size=16),
        axis.text.y = element_text(size=16)) + ylim(0,1400)+
  scale_color_manual(values = c("gray8", "red")) #+ geom_smooth(method = "loess", method.args = list(family = "symmetric"), size=1.25)
ggsave("Figure4c.pdf",width=5,height=5)

```

Figure 5: MS1 and MS2 cell cycle
```{r}

df_CC <- data.frame(fread(CC_data))
fil <- df_CC[which(df_CC$Lib.PG.Q.Value<0.01),]
length(unique(fil$Protein.Group))
fil <- df_CC[which(df_CC$GG.Q.Value<0.01),]
length(unique(fil$Genes))
fil$run_gene <- paste0(fil$Run, fil$Genes)
fil <- fil %>% dplyr::distinct(run_gene,.keep_all=T) %>% add_count(Genes) %>%
  dplyr::filter(n==2)
length(unique(fil$Genes))
con <- c("P04745", "P13645", "P35527","P04264","P35908","Q15323","Q14532","O76011","Q92764","O76013","O76014","O76015","O76009","Q14525","Q14533","Q9NSB4","P78385","Q9NSB2","P78386","O43790")

df <- df_CC[which(df_CC$Protein.Group%!in%con),]
df <- df[grepl("eJD903_re", df$Run),]
df <- pD_isotopicCO(df)
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$GG.Q.Value <= 0.01,], group.header="Genes", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
GGs <- protein.groups
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
GGs <- GGs[complete.cases(GGs),]
GGs[,1] <- GGs[,1]/median(GGs[,1], na.rm=T)
GGs[,2] <- GGs[,2]/median(GGs[,2], na.rm=T)
GGs[,3] <- GGs[,3]/median(GGs[,3], na.rm=T)

rmeans <- rowMeans(GGs, na.rm=T)
GGsx <- GGs/rmeans

Genes <- rownames(GGsx)
GG_n <- as.data.frame(cbind(GGsx,Genes))

GGs_G1 <- GG_n %>% dplyr::select("eJD903_remTRAQ0", "Genes")
GGs_G1$Condition <- "G1"
colnames(GGs_G1) <- c("Intensity", "Gene", "Condition")
GGs_S <- GG_n %>% dplyr::select("eJD903_remTRAQ4", "Genes")
GGs_S$Condition <- "S"
colnames(GGs_S) <- c("Intensity", "Gene", "Condition")
GGs_G2 <- GG_n %>% dplyr::select("eJD903_remTRAQ8", "Genes")
GGs_G2$Condition <- "G2"
colnames(GGs_G2) <- c("Intensity", "Gene", "Condition")
#------------
# Data import
#------------
# Your data should be formatted in terms of a melted list with the following characteristics:
# 1. Condition column
# 2. Relative Protein Abundance Column
# 3. Column of Uniprot IDs
# 4. (optional) Column of Genes
#
# Make sure to title the relevant columns: [Protein, Intensity, Condition, Gene]
DataPath <- rbind(GGs_G1, GGs_S, GGs_G2)
DataPath$Intensity <- as.numeric(DataPath$Intensity)

df_prot <- df %>% dplyr::select("Protein.Ids", "Genes") %>% dplyr::distinct(Genes, .keep_all=T)
df_prot$Protein <- str_extract(df_prot$Protein.Ids,"[^-]+")
df_prot$Protein <- gsub("(.*);.*", "\\1", df_prot$Protein)
df_prot <-  df_prot %>% dplyr::select("Protein", "Genes")
DataPath <- DataPath %>% left_join(df_prot, by = c("Gene" = "Genes"))

data <- DataPath

#Import the GO db
GO_db <- read.delim(GO_DB_Path, sep = " ")

####### code for reactome analysis
#----------------
GO_db <- GO_db %>% dplyr::select(Gene,GO_term_name)
GO_db <- unique(GO_db)
GO_db$Gene <- toupper(GO_db$Gene)

#----------
# Kruskall Wallis GSEA
#----------

GO_db <- GO_db %>% dplyr::group_by(GO_term_name) %>% dplyr::add_count() %>% dplyr::filter(n>1)

unique_GO_terms <- unique(GO_db$GO_term_name)
KW_output <- data.frame(GO_term = character(), pVal = numeric(),numberOfMatches= numeric(),fractionOfDB_Observed = numeric(),stringsAsFactors = FALSE)
for (i in 1:length(unique_GO_terms)){
  GO_term <- unique_GO_terms[i]
  GO_db_lim <- GO_db %>% dplyr::filter(GO_term_name == GO_term)
  
  data_matches <- data %>% dplyr::filter(Gene %in% GO_db_lim$Gene)
  matches_number <- length(unique(data_matches$Protein))
  total_Proteins_in_GOterm <- length(unique(GO_db_lim$Gene))
  fractionObserved <- matches_number/total_Proteins_in_GOterm
  dataProt_matches_medInt <- data_matches %>% dplyr::group_by(Condition)  %>% dplyr::summarize(medianInt = median(Intensity, na.rm=T))
  dataProt_matches_medInt$GO_term <- GO_term
  
  # Kruskall Wallis
  if(matches_number > 1){
    aav_test <- aov(Intensity ~ Condition, data = data_matches)
    sum_aav1 <- summary(aav_test)
    sum_aav1 <- data.frame(sum_aav1[[1]])
    pVal <- paste0(sum_aav1[1,5]) 
  }
  
  if(matches_number <2){
    pVal = NA
  }
    KW_output[i,1] <- GO_term
    KW_output[i,2] <- pVal 
    KW_output[i,3] <- matches_number
    KW_output[i,4] <- fractionObserved
    
    if(i == 1){
      KW_Int_output <- dataProt_matches_medInt[0,]
    }
    if(i > 1){
      KW_Int_output <- rbind(KW_Int_output, dataProt_matches_medInt)
    }
    
}  

# Multiple hypothesis testing correction
KW_output_MS1_keep <- KW_output
#KW_output <- KW_output_MS1_keep

KW_output <- KW_output[which(KW_output$numberOfMatches>3),]

#my changes
KW_output$qVal = p.adjust(KW_output$pVal, method = 'BH') # qvalue
KW_output1 <- KW_output %>% mutate("passes_5perc_FDR" = ifelse(qVal < 0.05, "TRUE", "FALSE"))
KW_output1 <- KW_output1[grepl("TRUE",KW_output1$passes_5perc_FDR),]
qval5_MS1 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.05),1]))
qval1_MS1 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.01),1]))

# Join Effect size
KW_output_comb <- KW_Int_output %>% left_join(KW_output, by = "GO_term")
KW_output_comb <- na.omit(KW_output_comb)
KW_output_comb <- ungroup(KW_output_comb)
KW_output_comb_effectSize <- KW_output_comb %>% dplyr::group_by(GO_term) %>% dplyr::summarize(EffectSize = max(abs(medianInt)))
KW_output_comb <- KW_output_comb %>% left_join(KW_output_comb_effectSize, by = "GO_term")
# bounding effect size into quintiles
KW_Effect_quantile <- quantile(KW_output_comb_effectSize$EffectSize,c(.80,.60,.40,.20))

# Dot plot
KW_output_comb$Condition <- factor(KW_output_comb$Condition, levels=c("G1", "S", "G2"))
KW_output_comb_MS1 <- KW_output_comb
KW_output_comb_MS1$Quant <- "MS1"

########################################################################  
########################################################################  
########################      MS2 data         ######################### 
########################################################################  
########################################################################  
df <- df_CC[which(df_CC$Protein.Group%!in%con),]
df <- df[grepl("eJD904_re", df$Run),]
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$File.Name <- paste0(df$Run, df$channel_name)
df <- df[which(df$Proteotypic==1),]
protein.groups <- diann_maxlfq(df[df$GG.Q.Value <= 0.01,], group.header="Genes", id.header = "seqcharge", quantity.header = "Precursor.Translated")
GGs <- protein.groups
#protein.groups_med <- apply(protein.groups,2,median, na.rm=T)
GGs <- GGs[complete.cases(GGs),]
GGs[,1] <- GGs[,1]/median(GGs[,1], na.rm=T)
GGs[,2] <- GGs[,2]/median(GGs[,2], na.rm=T)
GGs[,3] <- GGs[,3]/median(GGs[,3], na.rm=T)

rmeans <- rowMeans(GGs, na.rm=T)
GGsx <- GGs/rmeans

Genes <- rownames(GGsx)
GG_n <- as.data.frame(cbind(GGsx,Genes))

GGs_G1 <- GG_n %>% dplyr::select("eJD904_remTRAQ0", "Genes")
GGs_G1$Condition <- "G1"
colnames(GGs_G1) <- c("Intensity", "Gene", "Condition")
GGs_S <- GG_n %>% dplyr::select("eJD904_remTRAQ4", "Genes")
GGs_S$Condition <- "S"
colnames(GGs_S) <- c("Intensity", "Gene", "Condition")
GGs_G2 <- GG_n %>% dplyr::select("eJD904_remTRAQ8", "Genes")
GGs_G2$Condition <- "G2"
colnames(GGs_G2) <- c("Intensity", "Gene", "Condition")
#------------
# Data import
#------------
# Your data should be formatted in terms of a melted list with the following characteristics:
# 1. Condition column
# 2. Relative Protein Abundance Column
# 3. Column of Uniprot IDs
# 4. (optional) Column of Genes
#
# Make sure to title the relevant columns: [Protein, Intensity, Condition, Gene]
DataPath <- rbind(GGs_G1, GGs_S, GGs_G2)
DataPath$Intensity <- as.numeric(DataPath$Intensity)

df_prot <- df %>% dplyr::select("Protein.Ids", "Genes") %>% dplyr::distinct(Genes, .keep_all=T)
df_prot$Protein <- str_extract(df_prot$Protein.Ids,"[^-]+")
df_prot$Protein <- gsub("(.*);.*", "\\1", df_prot$Protein)
df_prot <-  df_prot %>% dplyr::select("Protein", "Genes")
DataPath <- DataPath %>% left_join(df_prot, by = c("Gene" = "Genes"))

data <- DataPath
GO_DB_Path <- "/Volumes/GoogleDrive/My Drive/MS/DBs_Params/GO/alt/GOA_db.txt"

#Import the GO db
GO_db <- read.delim(GO_DB_Path, sep = " ")

####### code for reactome analysis
#----------------
GO_db <- GO_db %>% dplyr::select(Gene,GO_term_name)
GO_db <- unique(GO_db)
GO_db$Gene <- toupper(GO_db$Gene)

#----------
# Kruskall Wallis GSEA
#----------

GO_db <- GO_db %>% dplyr::group_by(GO_term_name) %>% dplyr::add_count() %>% dplyr::filter(n>1)

unique_GO_terms <- unique(GO_db$GO_term_name)
KW_output <- data.frame(GO_term = character(), pVal = numeric(),numberOfMatches= numeric(),fractionOfDB_Observed = numeric(),stringsAsFactors = FALSE)
for (i in 1:length(unique_GO_terms)){
  GO_term <- unique_GO_terms[i]
  GO_db_lim <- GO_db %>% dplyr::filter(GO_term_name == GO_term)
  
  data_matches <- data %>% dplyr::filter(Gene %in% GO_db_lim$Gene)
  matches_number <- length(unique(data_matches$Protein))
  total_Proteins_in_GOterm <- length(unique(GO_db_lim$Gene))
  fractionObserved <- matches_number/total_Proteins_in_GOterm
  dataProt_matches_medInt <- data_matches %>% dplyr::group_by(Condition)  %>% dplyr::summarize(medianInt = median(Intensity, na.rm=T))
  dataProt_matches_medInt$GO_term <- GO_term
  
  # Kruskall Wallis
  if(matches_number > 1){
    aav_test <- aov(Intensity ~ Condition, data = data_matches)
    sum_aav1 <- summary(aav_test)
    sum_aav1 <- data.frame(sum_aav1[[1]])
    pVal <- paste0(sum_aav1[1,5]) 
  }
  
  if(matches_number <2){
    pVal = NA
  }
    KW_output[i,1] <- GO_term
    KW_output[i,2] <- pVal 
    KW_output[i,3] <- matches_number
    KW_output[i,4] <- fractionObserved
    
    if(i == 1){
      KW_Int_output <- dataProt_matches_medInt[0,]
    }
    if(i > 1){
      KW_Int_output <- rbind(KW_Int_output, dataProt_matches_medInt)
    }
    
}  

# Multiple hypothesis testing correction
KW_output_MS2_keep <- KW_output
KW_output <- KW_output[which(KW_output$numberOfMatches>3),]

#my changes
KW_output$qVal = p.adjust(KW_output$pVal, method = 'BH') # qvalue
KW_output1 <- KW_output %>% dplyr::mutate("passes_5perc_FDR" = ifelse(qVal < 0.05, "TRUE", "FALSE"))
KW_output1 <- KW_output1[grepl("TRUE",KW_output1$passes_5perc_FDR),]
qval5_MS2 <- as.vector((KW_output1[which(KW_output1$qVal < 0.05),1]))
qval1_MS2 <- as.vector(as.matrix(KW_output1[which(KW_output1$qVal < 0.01),1]))

# Join Effect size
KW_output_comb <- KW_Int_output %>% left_join(KW_output, by = "GO_term")
KW_output_comb <- na.omit(KW_output_comb)
KW_output_comb <- ungroup(KW_output_comb)
KW_output_comb_effectSize <- KW_output_comb %>% dplyr::group_by(GO_term) %>% dplyr::summarize(EffectSize = max(abs(medianInt)))
KW_output_comb <- KW_output_comb %>% left_join(KW_output_comb_effectSize, by = "GO_term")
# bounding effect size into quintiles
KW_Effect_quantile <- quantile(KW_output_comb_effectSize$EffectSize,c(.80,.60,.40,.20))

# Dot plot
KW_output_comb$Condition <- factor(KW_output_comb$Condition, levels=c("G1", "S", "G2"))
#KW_output_comb <- KW_output_comb[which(KW_output_comb$qVal < 0.05),]

KW_output_comb_MS2 <- KW_output_comb
KW_output_comb_MS2$Quant <- "MS2"

uni_5qval <- unique(append(qval5_MS1,qval5_MS2))
uni_1qval <- unique(append(qval1_MS1,qval1_MS2))

KW_output_comb_both <- rbind(KW_output_comb_MS1, KW_output_comb_MS2)
KW_output_comb_both$C_Q <- paste0(KW_output_comb_both$Condition, "_",KW_output_comb_both$Quant)

KW_output_comb_both5 <- KW_output_comb_both[KW_output_comb_both$GO_term%in%uni_5qval,] #keep GO-terms which have atleast 1 condition at 5% FDR
KW_output_comb_both1 <- KW_output_comb_both[KW_output_comb_both$GO_term%in%uni_1qval,] #keep GO-terms which have atleast 1 condition at 1% FDR

GO_db <- read.delim(GO_DB_Path, sep = " ")
GO_db <- GO_db %>% dplyr::select("GoID", "GO_term_name") %>% dplyr::distinct(GoID,.keep_all=T)
#cluster:
#1) unmelt
umelt <- KW_output_comb_both5 %>% dplyr::select("GO_term", "medianInt", "C_Q")
umelt <- umelt %>% left_join(GO_db, by =c("GO_term" = "GO_term_name"))
write.table(umelt, "PSEA_CC.txt", sep = "\t", row.names = FALSE)

keep <- c("GO:0005882", "GO:0004601", "GO:0042744", "GO:0016272", "GO:0000281", "GO:0000186", "GO:0000922", "GO:0015986", "GO:0005747", "GO:0051287", "GO:0006928", "GO:0042555", "GO:0008137", "GO:0000070", "GO:0006099", "GO:0006102", "GO:0006120", "GO:0015758", "GO:0051017", "GO:0061025", "GO:0022904", "GO:0050918", "GO:0042645", "GO:0070628","GO:0051084", "GO:0005876", "GO:0050786", "GO:0007077", "GO:0042776", "GO:0051084", "GO:0003857", "GO:0055086","GO:0006635","GO:0005753","GO:0008026")
umelt <- umelt[umelt$GoID%in%keep,] %>% dplyr::select(-"GoID")

umelt <- tidyr::spread(umelt, GO_term, medianInt)
rownames(umelt) <- umelt$C_Q

#2) calculate euclidian dist
t_umelt <- t(as.matrix(umelt))#[-c(1:3),]
t_umelt <- t_umelt[-1,]
ord <- hclust( dist(t_umelt, method = "euclidean"), method = "ward.D" )$order
rev_order <- rev(ord)
umelt <- melt(t_umelt)


umelt$Var1 <- factor(umelt$Var1, levels= rownames(t_umelt)[rev_order])

umelt$value[log2(as.numeric(umelt$value))>0.5] <- 2^0.4999  #set ceiling 
umelt$value[log2(as.numeric(umelt$value))< -0.5] <- 2^-0.4999 #set floor

cond_order <- factor(umelt$Var2, level = c('G1_MS1', 'G1_MS2', 'S_MS1', 'S_MS2', 'G2_MS1', 'G2_MS2'))

Cellcycles <- umelt %>%
  ggplot(aes(x = cond_order, y = Var1, fill = log2(as.numeric(value)))) + geom_tile(color = "gray5",size=0.15) +   
  scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0, limits=c(-0.5,0.5), breaks = c(-0.5,0,0.5)) + 
  theme_classic()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.x = element_text(size = 12, face = "bold"),
            legend.position = "top") 
Cellcycles

ggsave("Figure5b_MS1_MS2.pdf", Cellcycles, width = 6.65, height = 6.8)


#########################################################################
#########################################################################
############################ D.A. testing  ##############################
#########################################################################
#########################################################################
# 

#############
con <- c("P04745", "P13645", "P35527","P04264","P35908","Q15323","Q14532","O76011","Q92764","O76013","O76014","O76015","O76009","Q14525","Q14533","Q9NSB4","P78385","Q9NSB2","P78386","O43790") #contaminants

cnames <- c("Protein.Group", "Protein.Ids", "Protein.Names", "Genes", "First.Protein.Description", "Proteotypic", "Stripped.Sequence", "Modified.Sequence", "Precursor.Charge", "Precursor.Id")

plex <- fread(CC_data, select = c(cnames, "Run","GG.Q.Value","Ms1.Area"))
plex <- plex[which(plex$Protein.Group%!in%con),]
plex_MS1 <- plex[grepl("903",plex$Run),]
plex_MS1 <- plex_MS1[which(plex_MS1$GG.Q.Value<0.01),]
plex_MS1 <- pD_isotopicCO(plex_MS1)
plex_MS1 <- plex_MS1 %>% dplyr::rename("Intensity" = "Ms1.Area_iso")
plex_MS1 <- plex_MS1 %>% dplyr::select(-c("Ms1.Area"))

plex <- fread(CC_data, select = c(cnames,"Run","GG.Q.Value","Precursor.Translated"))
plex <- plex[which(plex$Protein.Group%!in%con),]
plex_MS2 <- plex[grepl("904",plex$Run),]
plex_MS2 <- plex_MS2[which(plex_MS2$GG.Q.Value<0.01),]
plex_MS2 <- plex_MS2 %>% dplyr::rename("Intensity" = "Precursor.Translated")

plex <- rbind(plex_MS1,plex_MS2)
plex <- pD_channel(plex)
plex <- pD_seqcharge(plex)
plex$precRun <- paste0(plex$seqcharge, plex$Run)
plex$chanRun <- paste0(plex$channel_name, plex$Run)

h <- plex %>% dplyr::group_by(chanRun) %>% dplyr::mutate(med_int = median(Intensity)) %>% 
  dplyr::distinct(chanRun, .keep_all=T) %>% dplyr::select("chanRun", "med_int")
plex <- plex %>% left_join(h, by = c("chanRun" = "chanRun"))
plex <- plex[which(plex$Intensity>0),]
plex <- plex %>% mutate(Intensity_cnorm = Intensity/med_int)
plex <- plex %>% dplyr::group_by(seqcharge, Run) %>% dplyr::mutate(Intensity = Intensity_cnorm/mean(Intensity_cnorm)) %>% dplyr::ungroup()
plex <- plex[which(plex$Proteotypic==T),]
plex1 <- plex %>% dplyr::select("seqcharge", "Intensity", "Protein.Group", "Genes", "Protein.Names","First.Protein.Description", "channel_name","Run")
d0_A <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% dplyr::group_by(Genes) %>% dplyr::mutate(med_ab_d0 = median(Intensity)) %>% dplyr::ungroup() %>%
  dplyr::select("Genes", "med_ab_d0") %>% dplyr::distinct(Genes, .keep_all=T) 
d4_A <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% dplyr::group_by(Genes) %>% dplyr::mutate(med_ab_d4 = median(Intensity)) %>% dplyr::ungroup() %>%
  dplyr::select("Genes", "med_ab_d4") %>% dplyr::distinct(Genes, .keep_all=T) 
d8_A <- plex1[grepl("mTRAQ8", plex1$channel_name),] %>% dplyr::group_by(Genes) %>% dplyr::mutate(med_ab_d8 = median(Intensity)) %>% dplyr::ungroup() %>%
  dplyr::select("Genes", "med_ab_d8") %>% dplyr::distinct(Genes, .keep_all=T) 
d0_d4 <- d0_A %>% inner_join(d4_A, by =c("Genes" = "Genes"))
d0_d4_d8 <- d0_d4 %>% inner_join(d8_A, by =c("Genes" = "Genes"))

#require >1 precursor per protein group per sample
d0 <- plex1[grepl("mTRAQ0", plex1$channel_name),] %>% add_count(Genes) %>% filter(n>1) %>% distinct(Genes) 
d4 <- plex1[grepl("mTRAQ4", plex1$channel_name),] %>% add_count(Genes) %>% filter(n>1) %>% distinct(Genes)
d8 <- plex1[grepl("mTRAQ8", plex1$channel_name),] %>% add_count(Genes) %>% filter(n>1) %>% distinct(Genes)

plex1 <- plex1 %>% dplyr::group_by(channel_name, Protein.Group) %>% dplyr::mutate("med_pep_ab" = median(Intensity)) %>% dplyr::ungroup()
int <- d0[which(d0$Genes%in%d4$Genes),]
int <- int[which(int$Genes%in%d8$Genes),]

prots <- plex1 %>% dplyr::distinct(Genes, .keep_all=T)
prots <- prots[prots$Genes%in%int$Genes,]
prots$pval <- 10000
unique_prots <- as.vector(prots$Genes)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$Genes),]
  aav_test <- aov(Intensity~channel_name, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots[i,10] <- as.numeric(pval_temp)
  
}

prots$FC_qval <-  p.adjust(prots$pval, method = 'BH') # qvalue
prots_sig <- prots[which(prots$FC_qval<0.01),]
prots_sig <- prots%>% dplyr::select("Genes", "FC_qval")
plex2 <- plex1 %>% inner_join(prots_sig, by =c("Genes"= "Genes"))
plex2 <- plex2 %>% dplyr::group_by(Run, channel_name, Genes) %>% dplyr::mutate("median_prot" = median(Intensity, na.rm=T)) %>% dplyr::ungroup()
plex_prot <- plex2 %>% dplyr::mutate("prot_chanrun" = paste0(Run,channel_name,Genes)) %>% dplyr::distinct(prot_chanrun, .keep_all=T)
plex_prot <- plex_prot %>% dplyr::group_by(Run, Genes) %>% dplyr::mutate("mean_prot_run" = mean(median_prot, na.rm=T)) %>% #take mean across channels
  dplyr::mutate("Prot_norm_int" = median_prot/mean_prot_run) %>% dplyr::ungroup()
plex_prot <- plex_prot %>% dplyr::mutate("Condition" = ifelse(grepl("mTRAQ0", channel_name), "G1",
                                                       ifelse(grepl("mTRAQ4", channel_name), "S", "G2"))) %>%
  dplyr::mutate("Quant" = ifelse(grepl("903", plex_prot$Run), "MS1", "MS2"))
plex_prot$C_Q <- paste0(plex_prot$Condition, plex_prot$Quant)
plex_prot_omit <- plex_prot[!grepl("Fez|C5orf", plex_prot$First.Protein.Description),]  #these proteins have missingness at MS1 and/or MS2
plex_prot_omit$prot_info <- paste0(plex_prot_omit$First.Protein.Description,", [",plex_prot_omit$Genes,"]")
#plex_prot_omit$prot_info <- paste0(plex_prot_omit$First.Protein.Description,", [",plex_prot_omit$Protein.Names,"]")
prots_sig <- prots[which(prots$FC_qval<0.01),]#%>% dplyr::select("Protein.Group", "FC_qval")
umelt <- plex_prot_omit %>% dplyr::select("prot_info","Prot_norm_int", "Condition","Quant","FC_qval")
write.table(umelt, "all_CC_prots.txt", sep = "\t", row.names = FALSE)

################# plot ##########################
umelt <- plex_prot_omit %>% dplyr::select("Condition", "prot_info","Prot_norm_int", "Quant", "C_Q")
umelt <- tidyr::spread(umelt, prot_info, Prot_norm_int)
rownames(umelt) <- umelt$C_Q
umelt1 <- umelt
rownames(umelt1) <- umelt1$C_Q

#2) calculate euclidian dist
t_umelt <- t(as.matrix(umelt1))[-c(1:3),]
t_umelt <- na.omit(t_umelt)
ord <- hclust( dist(t_umelt, method = "euclidean"), method = "ward.D" )$order
umelt <- melt(t_umelt)
umelt$Var1 <- factor(umelt$Var1, levels= rownames(t_umelt)[ord])
umelt$value[log2(as.numeric(umelt$value))>1] <- 2^0.999
umelt$value[log2(as.numeric(umelt$value))< -1] <- 2^-0.999
cond_order <- factor(umelt$Var2, level = c('G1MS1', 'G1MS2', 'SMS1', 'SMS2', 'G2MS1', 'G2MS2'))

length(unique(prots_sig$Protein.Group))
################ 
prots_sig <- prots_sig %>% dplyr::arrange(FC_qval)
denovo <- prots_sig[grepl("CMPK1|PFAS|ATIC", prots_sig$Genes),]
glyc <- prots_sig[grepl("PGAM1|TPI1|GLO1|ALDOA|ENO1|PGK1|TKT|GAPDH", prots_sig$Genes),]
dna_synth_catab <- prots_sig[grepl("HINT1|^RRM2$", prots_sig$Genes),]
cyto_org <- prots_sig[grepl("DSP|JUP|FLG2", prots_sig$Genes),]
noinfo <-  prots_sig[grepl("CDV3|JPT2", prots_sig$Genes),]
chemotaxis <- prots_sig[grepl("S100A7|^ANXA1$|PPIA|ACTN4|S100A9", prots_sig$Genes),]
spindle <- prots_sig[grepl("TPX2|KIF11|TACC3|TOP2A|MKI67|CLTC|CKAP5|NUSAP1", prots_sig$Genes),]
DNA_repair_rep <- prots_sig[grepl("^FH$|DHX9|PRKDC|PCLAF|BAZ1B|MSH2|MCM2|MCM6", prots_sig$Genes),]
protect_ROS <- prots_sig[grepl("PARK7|PRDX1", prots_sig$Genes),]
misfold <- prots_sig[grepl("HSPA8|PDIA3|HSPA5|PFDN4|PFDN1|PFDN2|VBP1", prots_sig$Genes),]
APC <- prots_sig[grepl("PLK1|UBE2S|CCNA2", prots_sig$Genes),]
  
denovo$type <- "'de novo' nucleotide synthesis"
glyc$type <- "Glycolysis, gluconeogenesis, PPP"
protect_ROS$type <- "Protection against ROS"
cyto_org$type <- "Intermediate filament organization"
noinfo$type <- "poorly characterized"
chemotaxis$type <- "Chemotaxis, migration, & motility"
spindle$type <- "Spindle interacting"
DNA_repair_rep$type <- "DNA repair"
misfold$type <- "Protein misfolding"
APC$type <- "Mitotic entry exit"

all_types <- rbind(denovo, glyc, protect_ROS,cyto_org, noinfo, chemotaxis, spindle, DNA_repair_rep,misfold, APC)
q <- plex_prot_omit %>% inner_join(all_types, by =c("Genes" = "Genes"))
q_info <- q %>% dplyr::select("prot_info", "type") %>% dplyr::distinct(prot_info, .keep_all = T)

#######
q_all <- q %>% dplyr::select("prot_info", "C_Q", "Prot_norm_int", "type")
colnames(q_all) <- c("Var1", "Var2", "value","type")

######### combine:
#########  plot:
cond_order <- factor(q_all$Var2, level = c('G1MS1', 'G1MS2', 'SMS1', 'SMS2', 'G2MS1', 'G2MS2'))

q_all$value[log2(as.numeric(q_all$value))>1] <- 2^0.9999  #set ceiling
q_all$value[log2(as.numeric(q_all$value))< -1] <- 2^-0.9999  #set floor

q_all_2 <- q_all
q_all_2$type<- factor(q_all_2$type, level = c("poorly characterized", "Chemotaxis, migration, & motility", "Glycolysis, gluconeogenesis, PPP", "Protection against ROS", "Protein misfolding","'de novo' nucleotide synthesis", "Intermediate filament organization", "DNA repair", "Spindle interacting", "Mitotic entry exit"))

DA_prot_HM <- q_all_2 %>%
  ggplot(aes(x = cond_order, y = reorder_within(Var1, value, type), fill = log2(as.numeric(value)))) + 
  geom_tile(color = "gray5",size=0.15) + 
  facet_grid(rows = vars(type), scales="free_y", space = "free_y", labeller = label_wrap_gen(width = 1, multi_line = TRUE))+
  scale_fill_gradient2(low = "#077DEC", high = "#EC7607", mid = "white",midpoint =0, limits=c(-1,1), breaks = c(-1,0,1)) +
  theme_classic()  + 
  labs(x = "Cell cycle phase", y = "", fill = "Log2, Median Abundance") + 
      theme(axis.text.x = element_text(size = 16),
            strip.text.y = element_text(angle = 0, size = 8),
            legend.position = "top",
            strip.background = element_rect(fill="white"))+
  scale_y_reordered()
DA_prot_HM

ggsave("Figure5c.pdf", DA_prot_HM, width = 7.5, height = 9.8)

##########
##########
############## XIC plots:

report <- data.frame(fread(CC_data))
XIC <-  data.frame(fread(CC_data_XIC))


JPT2_plot <- plot_XIC(report, XIC, sample1 = "G0", 
                     sample2 = "S", Run = "eJD903", Protein_x = "JPT2",
                     Modified.Sequence_1 = "\\(mTRAQ-n-0\\)TSDIFGSPVTATSR2",
                     Modified.Sequence_2 = "\\(mTRAQ-n-4\\)TSDIFGSPVTATSR2",
                     Stripped.Sequence = "TSDIFGSPVTATSR",
                     RT_start = 73.8,
                     RT_end = 74.2,
                     Remove_yions=T)
JPT2_plot

CDV3_plot <- plot_XIC(report, XIC, sample1 = "G0", 
                     sample2 = "S", Run = "eJD903", Protein_x = "CDV3",
                     Modified.Sequence_1 = "\\(mTRAQ-n-0\\)SLDNFFAK\\(mTRAQ-K-0\\)2",
                     Modified.Sequence_2 = "\\(mTRAQ-n-4\\)SLDNFFAK\\(mTRAQ-K-4\\)2",
                     Stripped.Sequence = "SLDNFFAK",
                     RT_start = 84.9,
                     RT_end = 85.3,
                     Remove_yions=F)
CDV3_plot

CDV3_JPT2 <- ggarrange(CDV3_plot, JPT2_plot, ncol=2)
ggsave("Figure5d.pdf", CDV3_JPT2, width=6,height=8,dpi=500)
```

Figure 6: Single cell data (QE and timsTOF), remove cells with >60% missing data
```{r}

Ms1Ext_df <- data.frame(fread(tims_SC_fpath_ms1Ext))
df <- data.frame(fread(tims_SC_fpath))
df$Run <- str_replace_all(df$Run, "-", ".")
df <- df[!grepl("_214", df$Run),] #remove the diluted bulk runs
PGs_0.01FDR <- df[which(df$Lib.PG.Q.Value<0.01),] %>% dplyr::distinct(Protein.Group)
SC1 <- pD_isoCO_Ms1Ext_tims(df, Ms1Ext_df) # correct the isotopic carryover across channel for Ms1.Extracted
SC <- SC1
meta <- read.delim(meta_PCA_fpath)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
meta <- meta[!grepl("0", meta$Celltype),] %>% dplyr::select(-c("Raw","Label") )
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$seqrun_chan <- paste0(df$seqcharge,df$Run,df$channel_name)
SC1$seqrun_chan <- paste0(SC1$seqcharge,SC1$Run,SC1$channel_name)
df_RTs <- df %>% dplyr::select("RT", "seqrun_chan")
SC <- SC1 
SC$Label <- "mTRAQ"
SC$seqcharge_run <- paste0(SC$seqcharge, "_", SC$Run)
SC$prot_run_chan <- paste0(SC$Protein.Group, "_", SC$Run,"_",SC$channel_name)
SC$run_chan <- paste0(SC$Run, SC$channel_name)
SC <-SC %>% left_join(meta, by =c("run_chan"="run_chan"))
SC <- SC[!grepl("_DB", SC$Celltype),]
ev2 <- SC

dat_missing <- SC %>% dplyr::group_by(run_chan) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>%
  dplyr::distinct(run_chan, .keep_all=T) %>% dplyr::ungroup() %>% dplyr::select("missing", "run_chan") %>%
  dplyr::filter(missing<0.6) #require <60% missing data

ev2 <-ev2[ev2$run_chan%in%dat_missing$run_chan,]
ev2 <- ev2[which(ev2$Ms1.Extracted_iso>0),]

#filter protein group 1% FDR
ev2_prot <- ev2[which(ev2$Protein.Group%in%PGs_0.01FDR$Protein.Group),] %>% distinct(prot_run_chan, .keep_all=T)
ev2_pep <- ev2 %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run", "Celltype")
mTRAQ_prots <- ev2_prot
mTRAQ_peps <- ev2_pep

p_mTRAQ_Ms1Ext <- mTRAQ_peps %>% dplyr::count(Run, Celltype)
p_mTRAQ_Ms1Ext$Condition <- "plexDIA"

prot_mTRAQ_Ms1Ext <- mTRAQ_prots %>% dplyr::count(Run, Celltype)
prot_mTRAQ_Ms1Ext$Condition <- "plexDIA"

my_fin_colors <- c("#52B788","#E07A5F","#457B9D")
prot_mTRAQ_Ms1Ext$type <- "Protein Groups"
p_mTRAQ_Ms1Ext$type <- "Precursors"

PGs_precs <- rbind(prot_mTRAQ_Ms1Ext, p_mTRAQ_Ms1Ext)
PGs_precs <- PGs_precs[!grepl("Neg", PGs_precs$Celltype),] #remove neg. controls (if any)
PGs <- PGs_precs[grepl("Prot", PGs_precs$type),]
PGs$PGs_perminAct <- PGs$n/15
PGs_tab <- PGs %>% dplyr::group_by(Celltype) %>% dplyr::mutate(mean = mean(PGs_perminAct)) %>% 
  dplyr::add_count() %>%
  dplyr::mutate(se = sd(PGs_perminAct)/sqrt(nn)) %>% dplyr::distinct(Celltype, .keep_all=T)
PGs_tab$Condition <- "plexDIA"
PGs_tab$y_se <- PGs_tab$mean
PGs_tab$y_se[2] <- PGs_tab$mean[3]+PGs_tab$mean[2]
PGs_tab$y_se[1] <- PGs_tab$mean[3]+PGs_tab$mean[2] + PGs_tab$mean[1] 



n <- length(unique(SC$Run))
scaleaxis <- function(x){ 
    x/n
}
SC <- SC1 %>% left_join(df_RTs, by =c("seqrun_chan" = "seqrun_chan"))
SC$Label <- "mTRAQ"
SC$seqcharge_run <- paste0(SC$seqcharge, "_", SC$Run)
SC$prot_run_chan <- paste0(SC$Protein.Group, "_", SC$Run,"_",SC$channel_name)
SC <- SC[which(SC$Ms1.Extracted_iso>0),]
SC$run_chan <- paste0(SC$Run, SC$channel_name)
SC <-SC %>% left_join(meta, by =c("run_chan"="run_chan"))
SC <- SC[!grepl("_DB", SC$Celltype),]
Precs_permin_plot <- ggplot(SC, aes(y=RT, fill=Celltype)) + geom_histogram(alpha=0.8, bins=30) +
  scale_x_continuous(labels=scaleaxis, n.breaks = 3) +
  theme_classic() + 
  theme(axis.text.x = element_text(size=16),
        axis.title.x = element_text(size=20),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) + ylim(0,30)+
  labs(x = "Precursors / minute", y="Retention time (min)") +
  scale_fill_manual(values = my_fin_colors) 
Precs_permin_plot
ggsave("Rev_Fig6d_coverage_Prec_v2.pdf", width=3.65, height=6)

PGs_plot <- ggplot(PGs, aes(x=Celltype, y =n, fill=Celltype)) +
  #facet_grid(cols = vars(type)) + 
  geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    geom_beeswarm(show.legend = F, alpha=0.7)+
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) +
  ylim(0,max(PGs$n+50)) +
  labs(y = "Protein groups per cell") +
  scale_fill_manual(values = my_fin_colors) +
  scale_y_continuous(n.breaks = 5, limits =c(0,max(PGs$n+50)))
PGs_plot
ggsave("Rev_Fig6c_coverage_PGs_v2.pdf", width=2.3, height=4)

Precs <- PGs_precs[!grepl("Prot", PGs_precs$type),]
Precs_plot <- ggplot(Precs, aes(x=Celltype, y =n, fill=Celltype)) +
  geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    geom_beeswarm(show.legend = F, alpha=0.7)+
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) +
  ylim(0,max(Precs$n+50)) +
  labs(y = "Precursors per cell") +
  scale_fill_manual(values = my_fin_colors)+
  scale_y_continuous(n.breaks = 5, limits =c(0,max(Precs$n+50)))

Precs_plot
ggsave("Rev_Fig6b_coverage_Precs_v2.pdf", width=2.3, height=4)

################### ################### ################### 
################### Figure 6d: Jaccard  ################### 
################### ################### ################### 

mTRAQ <- ev2[which(ev2$Protein.Group%in%PGs_0.01FDR$Protein.Group),] %>% filter(Ms1.Extracted_iso>0)
mTRAQ <- mTRAQ[!grepl("Neg", mTRAQ$Celltype),]
mTRAQ_keep_runs <- mTRAQ %>% distinct(Run, Celltype,.keep_all=T) %>% 
  dplyr::group_by(Run) %>%
  dplyr::add_count() %>% filter(n==3) %>% ungroup() %>%
  dplyr::distinct(Run)# only keep runs that did not include a neg control
mTRAQ <- mTRAQ[mTRAQ$Run%in%mTRAQ_keep_runs$Run,]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
ev2_u <- mTRAQ[grepl("U-937",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)
ev2_m <- mTRAQ[grepl("Mel",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)
ev2_p <- mTRAQ[grepl("PDAC",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)

uni_run <- unique(mTRAQ$Run)
jac_l <- vector(mode = "list", length = length(uni_run)*3)

for(i in 1:length(uni_run)){
  u_temp <- ev2_u[grepl(paste0(uni_run[i]), ev2_u$Run),] %>% dplyr::select(Protein.Group)
  m_temp <- ev2_m[grepl(paste0(uni_run[i]), ev2_m$Run),] %>% dplyr::select(Protein.Group)
  p_temp <- ev2_p[grepl(paste0(uni_run[i]), ev2_p$Run),] %>% dplyr::select(Protein.Group)
  jac_l[[3*i-2]] <- u_temp
  jac_l[[3*i-1]] <- m_temp
  jac_l[[3*i]] <- p_temp
}


uni_pep_mTRAq <- unique(mTRAQ$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- jac_l
for(i in 1:length(jac_l)){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- seq.int(1,nrow(jac), by =1)

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$Var1 != jac_dist_mTRAQ_m$Var2),]

#show jaccard distributions in-set and out-of-set 
nsets <- length(jac_l)/3
nsets <- length(jac_l)/3
jac_dist_mTRAQ_m$var1_set <- ceiling(jac_dist_mTRAQ_m$Var1/length(jac_l)*nsets)
jac_dist_mTRAQ_m$var2_set <- ceiling(jac_dist_mTRAQ_m$Var2/length(jac_l)*nsets)

in_set <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$var1_set==jac_dist_mTRAQ_m$var2_set),] %>% mutate("type" = "Within a run")
out_set <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$var1_set!=jac_dist_mTRAQ_m$var2_set),] %>% mutate("type" = "Across runs")

both_set <- rbind(in_set, out_set)

median(in_set$value)
median(out_set$value)

both_set$type <- factor(both_set$type, levels=c("Within a run", "Across runs"))

ggplot(both_set, aes(x=type, y =value, fill=type)) + 
    geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    #geom_jitter(show.legend = F, alpha=0.1)+ 
  theme_classic()+
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=18),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=14),
        legend.title = element_blank()
                     ) +
  ylim(0,1) +
  labs(y = "Protein Jaccard Index") +
    scale_fill_manual(values = c("#CC6677", "#40B0A6")) #"tan2"
ggsave("Fig6e_jaccard_timsTOF_v2.pdf", width=3.6,height=5.3)

################### ################### ################### 
################### Figure 6c: Quantitative accuracy relative to 100-cell bulk material
################### ################### ################### 

bulk_df <- data.frame(fread(b_fpath))
bulk_df_ms1ex <- data.frame(fread(b_ms1ext_fpath))
B1 <- pD_isoCO_Ms1Ext(bulk_df, bulk_df_ms1ex) # correct the isotopic carryover across channel for Ms1.Extracted
B <- B1
bulk_df_prot <- bulk_df %>% dplyr::select("Protein.Group", "Lib.PG.Q.Value") %>% distinct()
B$File.Name <- paste0(B$Run, B$channel_name)
B <- B %>% left_join(bulk_df_prot, by =c("Protein.Group" = "Protein.Group"))
B <- pD_diann_quant(B, id.header = "seqcharge", quantity.header = "Ms1.Extracted_iso")
B <- B %>% dplyr::select(-c("Protein.Names", "HY", "species"))
B_m <- reshape2::melt(B, id.vars = "PGs")
B_m$value <- as.numeric(B_m$value)
metaB <- read.delim(meta_PCA_fpath)
metaB$run_chan <- paste0(metaB$Raw, "mTRAQ",metaB$Label)
B_m <- B_m %>% inner_join(metaB, by =c("variable" = "run_chan"))
 
##### SC: 
Ms1Ext_df <- data.frame(fread(tims_SC_fpath_ms1Ext))
df <- data.frame(fread(tims_SC_fpath))
unique(df$Run)
df$Run <- str_replace_all(df$Run, "-", ".")
df <- df[!grepl("_214", df$Run),] #remove the diluted bulk runs
Ms1Ext_df <- Ms1Ext_df[,!grepl("_214", colnames(Ms1Ext_df))]  #remove the diluted bulk runs
SC1 <- pD_isoCO_Ms1Ext_tims(df, Ms1Ext_df) # correct the isotopic carryover across channel for Ms1.Extracted
SC <- SC1
SC_df_prot <- df %>% dplyr::select("Protein.Group", "Lib.PG.Q.Value") %>% distinct()
SC$File.Name <- paste0(SC$Run, SC$channel_name)
SC <- SC %>% left_join(SC_df_prot, by =c("Protein.Group" = "Protein.Group"))

dat_missing <- SC %>% dplyr::group_by(File.Name) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>%
  dplyr::distinct(File.Name, .keep_all=T) %>% dplyr::ungroup() %>% dplyr::select("missing", "File.Name") %>%
  dplyr::filter(missing<0.6) #require <60% missing data

SC <-SC[SC$File.Name%in%dat_missing$File.Name,]
SC <- SC[which(SC$Ms1.Extracted_iso>0),]

SC <- pD_diann_quant(SC, id.header = "seqcharge", quantity.header = "Ms1.Extracted_iso")
SC <- SC %>% dplyr::select(-c("Protein.Names", "HY", "species"))
SC_m <- reshape2::melt(SC, id.vars = "PGs")
SC_m$value <- as.numeric(SC_m$value)
meta <- read.delim(meta_PCA_fpath)
meta$run_chan <- paste0("X",meta$Raw, "mTRAQ",meta$Label)
SC_m <- SC_m %>% inner_join(meta, by =c("variable" = "run_chan"))
SC_m <- SC_m[!grepl("Neg", SC_m$Celltype),]

prot_SC <- SC_m %>% dplyr::filter(value>0) %>% dplyr::add_count(PGs, Celltype) %>% 
  dplyr::distinct(PGs, Celltype, .keep_all=T) %>%
  dplyr::filter(n>=5) %>%
  dplyr::add_count(PGs) %>% dplyr::filter(nn==3) #keep PGs present in >= 5 of single cells 

prot_B <- B_m %>% dplyr::filter(value>0) %>% dplyr::add_count(PGs, Celltype) %>% 
  dplyr::distinct(PGs, Celltype, .keep_all=T) %>%
  dplyr::filter(n>=2) %>%
  dplyr::add_count(PGs) %>% dplyr::filter(nn==3)#keep PGs present in >= 2/3 runs

B_m1 <- B_m[B_m$PGs%in%prot_B$PGs,]
SC_m1 <- SC_m[SC_m$PGs%in%prot_SC$PGs,]

B_m1 <- B_m1[B_m1$PGs%in%SC_m1$PGs,]
B_m1$value <- na_if(B_m1$value, 0) #replace 0 values with NA
B_m1[is.na(B_m1)] <- min(B_m1$value, na.rm=T) #replace NA with lowest non-zero value
B_m1 <- B_m1 %>% dplyr::group_by(Celltype, PGs) %>%
  dplyr::mutate("mean_PG" = mean(value)) %>%
  distinct(Celltype, PGs, .keep_all=T) %>% ungroup()

SC_m1 <- SC_m1[SC_m1$PGs%in%B_m$PGs,]
SC_m1$value <- na_if(SC_m1$value, 0) #replace 0 values with NA
SC_m1[is.na(SC_m1)] <- min(SC_m1$value, na.rm=T) #replace NA with lowest non-zero value
SC_m1 <- SC_m1 %>% dplyr::group_by(Celltype, PGs) %>%
  dplyr::mutate("mean_PG" = mean(value)) %>%
  distinct(Celltype, PGs, .keep_all=T) %>% ungroup()

B_m1$Celltype <- paste0("Bulk_",B_m1$Celltype)

SC_B <- rbind(SC_m1,B_m1) %>% dplyr::select("PGs", "Celltype", "mean_PG")
SC_B$mean_PG <- log10(SC_B$mean_PG)
SC_B <- dcast(SC_B, PGs~Celltype) %>% na.omit()

SC <- c("Melanoma", "PDAC", "U-937")
Bulk <- c("Bulk_Melanoma", "Bulk_PDAC", "Bulk_U-937")



############### fold-change comparisons U-937/PDAC
SC_B <- rbind(SC_m1,B_m1) %>% dplyr::select("PGs", "Celltype", "mean_PG")
SC_B <- dcast(SC_B, PGs~Celltype) %>% na.omit()
SC_B$Bulk_PDAC <- SC_B$Bulk_PDAC/median(SC_B$Bulk_PDAC)
SC_B$`Bulk_U-937` <- SC_B$`Bulk_U-937` /median(SC_B$`Bulk_U-937`)
SC_B$Bulk_Melanoma <- SC_B$Bulk_Melanoma /median(SC_B$Bulk_Melanoma)
SC_B$PDAC <- SC_B$PDAC /median(SC_B$PDAC )
SC_B$`U-937` <- SC_B$`U-937` /median(SC_B$`U-937`)
SC_B$P_U_b <- SC_B$Bulk_PDAC/SC_B$`Bulk_U-937`
SC_B$P_U_sc <- SC_B$PDAC/SC_B$`U-937`

SC_B$P_U_b <- log2(SC_B$P_U_b)
SC_B$P_U_sc <- log2(SC_B$P_U_sc)
b_sc_cor <- cor(SC_B$P_U_b, SC_B$P_U_sc, method="spearman")


coeff.tls <- COEF::tls(SC_B$P_U_sc ~ SC_B$P_U_b +0, SC_B)
coeff.tls <- coeff.tls[[1]]

PGs <- c("P17096;P17096-3", "P07437", "P08729")
PG_names <- c("HMGA1", "TUBB", "KRT7")

df_names <- data.frame(PGs, PG_names)
SC_B <- SC_B %>% left_join(df_names, by =c("PGs" = "PGs"))
SC_B <- SC_B %>% left_join(prots, by =c("PGs" = "Protein.Group"))

library(ggrepel)
hist(SC_B$P_U_b, breaks=20)
hist(SC_B$P_U_sc, breaks=20)

ggplot(SC_B, aes(x=P_U_b, y=P_U_sc)) +
  geom_pointdensity()+
  scale_color_viridis() +
  theme_classic() + theme(legend.position = "none") + 
  theme(axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14))+
  labs(x = expression(paste(Log["2"],", PDAC/U-937 100 cells")), y=expression(paste(Log["2"],", PDAC/U-937 1 cell")),
       colors = "dfd")+ ylim(-6,9.1) + xlim(-6,9.1) +
  geom_label_repel(aes(label = PG_names),
                   min.segment.length = unit(0, 'lines'),
                   box.padding = 2,
                   size = 2.75) +
  annotate("text", x=-3, y=5, label=paste0("rho=",round(b_sc_cor,2)), size=3)

ggsave("Figure6f_timsTOF_v2.pdf", width=3.8,height=3.8, dpi=500)

####################################
##################################
################## QE  ##########
####################################
##################################

Ms1Ext_df <- data.frame(fread(SC_ms1ext_fpath))
df <- data.frame(fread(SC_fpath))
df$Run <- str_replace_all(df$Run, "-", ".")
df <- df[!grepl("wJD12", df$Run),] #remove the diluted bulk runs
PGs_0.01FDR <- df[which(df$Lib.PG.Q.Value<0.01),] %>% dplyr::distinct(Protein.Group)
SC1 <- pD_isoCO_Ms1Ext(df, Ms1Ext_df) # correct the isotopic carryover across channel for Ms1.Extracted
SC <- SC1
meta <- read.delim(meta_PCA_fpath)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
meta <- meta[!grepl("Neg", meta$Celltype),] %>% dplyr::select(-c("Raw","Label") )
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$seqrun_chan <- paste0(df$seqcharge,df$Run,df$channel_name)
SC1$seqrun_chan <- paste0(SC1$seqcharge,SC1$Run,SC1$channel_name)
df_RTs <- df %>% dplyr::select("RT", "seqrun_chan")
SC <- SC1 
SC$Label <- "mTRAQ"
SC$seqcharge_run <- paste0(SC$seqcharge, "_", SC$Run)
SC$prot_run_chan <- paste0(SC$Protein.Group, "_", SC$Run,"_",SC$channel_name)
SC$run_chan <- paste0(SC$Run, SC$channel_name)
SC <-SC %>% left_join(meta, by =c("run_chan"="run_chan"))
SC <- SC[!grepl("_DB", SC$Celltype),]
ev2 <- SC

dat_missing <- SC %>% dplyr::group_by(run_chan) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>%
  dplyr::distinct(run_chan, .keep_all=T) %>% dplyr::ungroup() %>% dplyr::select("missing", "run_chan") %>%
  dplyr::filter(missing<0.6) #require <60% missing data

ev2 <-ev2[ev2$run_chan%in%dat_missing$run_chan,]
ev2 <- ev2[which(ev2$Ms1.Extracted_iso>0),]


#filter protein group 1% FDR
ev2_prot <- ev2[which(ev2$Protein.Group%in%PGs_0.01FDR$Protein.Group),] %>% distinct(prot_run_chan, .keep_all=T)
ev2_pep <- ev2 %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run", "Celltype")
mTRAQ_prots <- ev2_prot
mTRAQ_peps <- ev2_pep

p_mTRAQ_Ms1Ext <- mTRAQ_peps %>% dplyr::count(Run, Celltype)
p_mTRAQ_Ms1Ext$Condition <- "plexDIA"

prot_mTRAQ_Ms1Ext <- mTRAQ_prots %>% dplyr::count(Run, Celltype)
prot_mTRAQ_Ms1Ext$Condition <- "plexDIA"

my_fin_colors <- c("#52B788","#E07A5F","#457B9D")
prot_mTRAQ_Ms1Ext$type <- "Protein Groups"
p_mTRAQ_Ms1Ext$type <- "Precursors"

PGs_precs <- rbind(prot_mTRAQ_Ms1Ext, p_mTRAQ_Ms1Ext)
PGs_precs <- PGs_precs[!grepl("Neg", PGs_precs$Celltype),]
PGs <- PGs_precs[grepl("Prot", PGs_precs$type),]
PGs$PGs_perminAct <- PGs$n/30
PGs_tab <- PGs %>% dplyr::group_by(Celltype) %>% dplyr::mutate(mean = mean(PGs_perminAct)) %>% 
  dplyr::add_count() %>%
  dplyr::mutate(se = sd(PGs_perminAct)/sqrt(nn)) %>% dplyr::distinct(Celltype, .keep_all=T)
PGs_tab$Condition <- "plexDIA"
PGs_tab$y_se <- PGs_tab$mean
PGs_tab$y_se[2] <- PGs_tab$mean[3]+PGs_tab$mean[2]
PGs_tab$y_se[1] <- PGs_tab$mean[3]+PGs_tab$mean[2] + PGs_tab$mean[1] 


#12 negative controls, so subtract total number of runs (12/3=4).. also 7 real cells had >60% missing data, and were removed... so that is 7/3=2.33 runs worth
n <- length(unique(SC$Run)) - 4-2.33333
scaleaxis <- function(x){ 
    x/n
}
SC <- SC1 %>% left_join(df_RTs, by =c("seqrun_chan" = "seqrun_chan"))
SC$Label <- "mTRAQ"
SC$seqcharge_run <- paste0(SC$seqcharge, "_", SC$Run)
SC$prot_run_chan <- paste0(SC$Protein.Group, "_", SC$Run,"_",SC$channel_name)
SC <- SC[which(SC$Ms1.Extracted_iso>0),]
SC$run_chan <- paste0(SC$Run, SC$channel_name)
SC <-SC %>% left_join(meta, by =c("run_chan"="run_chan"))
SC <- SC[!grepl("_DB", SC$Celltype),]
SC <- na.omit(SC) #remove neg controls
Precs_permin_plot <- ggplot(SC, aes(y=RT, fill=Celltype)) + geom_histogram(alpha=0.8, bins=53) +
  scale_x_continuous(labels=scaleaxis, breaks =c(0, 100*n, 200*n, 300*n)) +
  scale_y_continuous(breaks=c(0,15,30,45,53), limits=c(0,53))+
  theme_classic() + 
  theme(axis.text.x = element_text(size=16),
        axis.title.x = element_text(size=20),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) +
  labs(x = "Mean Precursors / minute", y="Retention time (min)") +
  scale_fill_manual(values = my_fin_colors) 
Precs_permin_plot
ggsave("Rev_Fig6d_coverage_Prec_QE_v2.pdf", width=3.65, height=6)

PGs <- na.omit(PGs)
PGs_plot <- ggplot(PGs, aes(x=Celltype, y =n, fill=Celltype)) +
  #facet_grid(cols = vars(type)) + 
  geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    geom_beeswarm(show.legend = F, alpha=0.7)+
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) +
  ylim(0,max(PGs$n+50)) +
  labs(y = "Protein groups per cell") +
  scale_fill_manual(values = my_fin_colors) +
  scale_y_continuous(n.breaks = 5, limits =c(0,max(PGs$n+50)))
PGs_plot
ggsave("Rev_Fig6c_coverage_PGs_QE_v2.pdf", width=2.3, height=4)

Precs <- PGs_precs[!grepl("Prot", PGs_precs$type),]
Precs <- na.omit(Precs)
Precs_plot <- ggplot(Precs, aes(x=Celltype, y =n, fill=Celltype)) +
  #facet_grid(cols = vars(type)) + 
  #geom_violin(alpha=0.4)+ 
  geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    geom_beeswarm(show.legend = F, alpha=0.7)+
  theme_classic() + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=20),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=18),
        legend.title = element_blank()
                     ) +
  ylim(0,max(Precs$n+50)) +
  labs(y = "Precursors per cell") +
  scale_fill_manual(values = my_fin_colors)+
  scale_y_continuous(n.breaks = 5, limits =c(0,max(Precs$n+50)))

Precs_plot
ggsave("Rev_Fig6b_coverage_Precs_QE_v2.pdf", width=2.3, height=4)

################### ################### ################### 
################### Figure 6d: Jaccard  ################### 
################### ################### ################### 

mTRAQ <- ev2[which(ev2$Protein.Group%in%PGs_0.01FDR$Protein.Group),] %>% dplyr::filter(Ms1.Extracted_iso>0)
mTRAQ <- na.omit(mTRAQ) #remove neg controls
#mTRAQ <- mTRAQ[!grepl("Neg", mTRAQ$Celltype),]
mTRAQ_keep_runs <- mTRAQ %>% dplyr::distinct(Run, Celltype,.keep_all=T) %>% 
  dplyr::group_by(Run) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup() %>%
  dplyr::distinct(Run)# only keep runs that did not include a neg control and no failed cells
mTRAQ <- mTRAQ[mTRAQ$Run%in%mTRAQ_keep_runs$Run,]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
ev2_u <- mTRAQ[grepl("U-937",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)
ev2_m <- mTRAQ[grepl("Mel",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)
ev2_p <- mTRAQ[grepl("PDAC",mTRAQ$Celltype),] %>% distinct(PG_run, .keep_all=T)

uni_run <- unique(mTRAQ$Run)
jac_l <- vector(mode = "list", length = length(uni_run)*3)

for(i in 1:length(uni_run)){
  u_temp <- ev2_u[grepl(paste0(uni_run[i]), ev2_u$Run),] %>% dplyr::select(Protein.Group)
  m_temp <- ev2_m[grepl(paste0(uni_run[i]), ev2_m$Run),] %>% dplyr::select(Protein.Group)
  p_temp <- ev2_p[grepl(paste0(uni_run[i]), ev2_p$Run),] %>% dplyr::select(Protein.Group)
  jac_l[[3*i-2]] <- u_temp
  jac_l[[3*i-1]] <- m_temp
  jac_l[[3*i]] <- p_temp
}


uni_pep_mTRAq <- unique(mTRAQ$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- jac_l
for(i in 1:length(jac_l)){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- seq.int(1,nrow(jac), by =1)

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$Var1 != jac_dist_mTRAQ_m$Var2),]

#show jaccard distributions in-set and out-of-set 
nsets <- length(jac_l)/3
nsets <- length(jac_l)/3
jac_dist_mTRAQ_m$var1_set <- ceiling(jac_dist_mTRAQ_m$Var1/length(jac_l)*nsets)
jac_dist_mTRAQ_m$var2_set <- ceiling(jac_dist_mTRAQ_m$Var2/length(jac_l)*nsets)

in_set <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$var1_set==jac_dist_mTRAQ_m$var2_set),] %>% mutate("type" = "Within a run")
out_set <- jac_dist_mTRAQ_m[which(jac_dist_mTRAQ_m$var1_set!=jac_dist_mTRAQ_m$var2_set),] %>% mutate("type" = "Across runs")

both_set <- rbind(in_set, out_set)


both_set$type <- factor(both_set$type, levels=c("Within a run", "Across runs"))

ggplot(both_set, aes(x=type, y =value, fill=type)) + 
    geom_boxplot(alpha=0.8)+ #geom_jitter(width=0.2,alpha=0.2)+
    #geom_jitter(show.legend = F, alpha=0.1)+ 
  theme_classic()+
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=18),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=14),
        legend.text = element_text(size=14),
        legend.title = element_blank()
                     ) +
  ylim(0,1) +
  labs(y = "Protein Jaccard Index") +
    scale_fill_manual(values = c("#CC6677", "#40B0A6")) #"tan2"
ggsave("Fig6e_jaccard_QE_v2.pdf", width=3.6,height=5.3)

################### ################### ################### 
################### Figure 6c: Quantitative accuracy relative to 100-cell bulk material
################### ################### ################### 
bulk_df <- data.frame(fread(b_fpath))
bulk_df_ms1ex <- data.frame(fread(b_ms1ext_fpath))
B1 <- pD_isoCO_Ms1Ext(bulk_df, bulk_df_ms1ex) # correct the isotopic carryover across channel for Ms1.Extracted
B <- B1
bulk_df_prot <- bulk_df %>% dplyr::select("Protein.Group", "Lib.PG.Q.Value") %>% distinct()
B$File.Name <- paste0(B$Run, B$channel_name)
B <- B %>% left_join(bulk_df_prot, by =c("Protein.Group" = "Protein.Group"))
B <- pD_diann_quant(B, id.header = "seqcharge", quantity.header = "Ms1.Extracted_iso")
B <- B %>% dplyr::select(-c("Protein.Names", "HY", "species"))
B_m <- reshape2::melt(B, id.vars = "PGs")
B_m$value <- as.numeric(B_m$value)
metaB <- read.delim(meta_PCA_fpath)
metaB$run_chan <- paste0(metaB$Raw, "mTRAQ",metaB$Label)
B_m <- B_m %>% inner_join(metaB, by =c("variable" = "run_chan"))
 
##### SC: 
Ms1Ext_df <- data.frame(fread(SC_ms1ext_fpath))
df <- data.frame(fread(SC_fpath))
unique(df$Run)
df$Run <- str_replace_all(df$Run, "-", ".")
df <- df[!grepl("wJD12", df$Run),] #remove the diluted bulk runs
Ms1Ext_df <- Ms1Ext_df[,!grepl("wJD12", colnames(Ms1Ext_df))]  #remove the diluted bulk runs
SC1 <- pD_isoCO_Ms1Ext(df, Ms1Ext_df) # correct the isotopic carryover across channel for Ms1.Extracted
SC <- SC1
SC_df_prot <- df %>% dplyr::select("Protein.Group", "Lib.PG.Q.Value") %>% distinct()
SC$File.Name <- paste0(SC$Run, SC$channel_name)
SC <- SC %>% left_join(SC_df_prot, by =c("Protein.Group" = "Protein.Group"))

dat_missing <- SC %>% dplyr::group_by(File.Name) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>%
  dplyr::distinct(File.Name, .keep_all=T) %>% dplyr::ungroup() %>% dplyr::select("missing", "File.Name") %>%
  dplyr::filter(missing<0.6) #require <60% missing data

SC <-SC[SC$File.Name%in%dat_missing$File.Name,]
SC <- SC[which(SC$Ms1.Extracted_iso>0),]

SC <- pD_diann_quant(SC, id.header = "seqcharge", quantity.header = "Ms1.Extracted_iso")
SC <- SC %>% dplyr::select(-c("Protein.Names", "HY", "species"))
SC_m <- reshape2::melt(SC, id.vars = "PGs")
SC_m$value <- as.numeric(SC_m$value)
meta <- read.delim(meta_PCA_fpath)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
SC_m <- SC_m %>% inner_join(meta, by =c("variable" = "run_chan"))
SC_m <- SC_m[!grepl("Neg", SC_m$Celltype),]

prot_SC <- SC_m %>% dplyr::filter(value>0) %>% dplyr::add_count(PGs, Celltype) %>% 
  dplyr::distinct(PGs, Celltype, .keep_all=T) %>%
  dplyr::filter(n>=5) %>%
  dplyr::add_count(PGs) %>% dplyr::filter(nn==3) #keep PGs present in >= 5 single cells 

prot_B <- B_m %>% dplyr::filter(value>0) %>% dplyr::add_count(PGs, Celltype) %>% 
  dplyr::distinct(PGs, Celltype, .keep_all=T) %>%
  dplyr::filter(n>=2) %>%
  dplyr::add_count(PGs) %>% dplyr::filter(nn==3)#keep PGs present in >= 2/3 runs

B_m1 <- B_m[B_m$PGs%in%prot_B$PGs,]
SC_m1 <- SC_m[SC_m$PGs%in%prot_SC$PGs,]

B_m1 <- B_m1[B_m1$PGs%in%SC_m1$PGs,]
B_m1$value <- na_if(B_m1$value, 0) #replace 0 values with NA
B_m1[is.na(B_m1)] <- min(B_m1$value, na.rm=T) #replace NA with lowest non-zero value
B_m1 <- B_m1 %>% dplyr::group_by(Celltype, PGs) %>%
  dplyr::mutate("mean_PG" = mean(value)) %>%
  distinct(Celltype, PGs, .keep_all=T) %>% ungroup()

SC_m1 <- SC_m1[SC_m1$PGs%in%B_m$PGs,]
SC_m1$value <- na_if(SC_m1$value, 0) #replace 0 values with NA
SC_m1[is.na(SC_m1)] <- min(SC_m1$value, na.rm=T) #replace NA with lowest non-zero value
SC_m1 <- SC_m1 %>% dplyr::group_by(Celltype, PGs) %>%
  dplyr::mutate("mean_PG" = mean(value)) %>%
  distinct(Celltype, PGs, .keep_all=T) %>% ungroup()

B_m1$Celltype <- paste0("Bulk_",B_m1$Celltype)

SC_B <- rbind(SC_m1,B_m1) %>% dplyr::select("PGs", "Celltype", "mean_PG")
SC_B$mean_PG <- log10(SC_B$mean_PG)
SC_B <- dcast(SC_B, PGs~Celltype) %>% na.omit()

SC <- c("Melanoma", "PDAC", "U-937")
Bulk <- c("Bulk_Melanoma", "Bulk_PDAC", "Bulk_U-937")

############### fold-change comparisons U-937/PDAC
SC_B <- rbind(SC_m1,B_m1) %>% dplyr::select("PGs", "Celltype", "mean_PG")
SC_B <- dcast(SC_B, PGs~Celltype) %>% na.omit()
SC_B$Bulk_PDAC <- SC_B$Bulk_PDAC/median(SC_B$Bulk_PDAC)
SC_B$`Bulk_U-937` <- SC_B$`Bulk_U-937` /median(SC_B$`Bulk_U-937`)
SC_B$Bulk_Melanoma <- SC_B$Bulk_Melanoma /median(SC_B$Bulk_Melanoma)
SC_B$PDAC <- SC_B$PDAC /median(SC_B$PDAC )
SC_B$`U-937` <- SC_B$`U-937` /median(SC_B$`U-937`)
SC_B$P_U_b <- SC_B$Bulk_PDAC/SC_B$`Bulk_U-937`
SC_B$P_U_sc <- SC_B$PDAC/SC_B$`U-937`


SC_B$P_U_b <- log2(SC_B$P_U_b)
SC_B$P_U_sc <- log2(SC_B$P_U_sc)
b_sc_cor <- cor(SC_B$P_U_b, SC_B$P_U_sc, method="spearman")


coeff.tls <- COEF::tls(SC_B$P_U_sc ~ SC_B$P_U_b +0, SC_B)
coeff.tls <- coeff.tls[[1]]

PGs <- c("P17096;P17096-3", "P07437", "P08729")
PG_names <- c("HMGA1", "TUBB", "KRT7")

df_names <- data.frame(PGs, PG_names)
SC_B <- SC_B %>% left_join(df_names, by =c("PGs" = "PGs"))

library(ggrepel)
hist(SC_B$P_U_b, breaks=20)
hist(SC_B$P_U_sc, breaks=20)

ggplot(SC_B, aes(x=P_U_b, y=P_U_sc)) +
  geom_pointdensity()+
  #geom_pointdensity(data = SC_B %>% filter(FC_qval<0.01), alpha=1)+
  scale_color_viridis() +
 # geom_abline(intercept = coeff.tls[1], slope = coeff.tls[2], col="black",linetype =2, size=0.6) +
  theme_classic() + theme(legend.position = "none") + 
  theme(axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=14),
        axis.title.y = element_text(size=14))+
  labs(x = expression(paste(Log["2"],", PDAC/U-937 100 cells")), y=expression(paste(Log["2"],", PDAC/U-937 1 cell")),
       colors = "dfd")+ ylim(-6,9.1) + xlim(-6,9.1) +
  geom_label_repel(aes(label = PG_names),
                   min.segment.length = unit(0, 'lines'),
                   box.padding = 2,
                   size = 2.75) +
  annotate("text", x=-3, y=5, label=paste0("rho=",round(b_sc_cor,2)), size=3)

ggsave("Figure6f_QE_v2.pdf", width=3.8,height=3.8, dpi=500)



################################################
################################################
################################################
##################  XIC  #######################
################################################
################################################
################################################

XIC <- read.delim(SC_vis_XIC_fpath)
report <- data.frame(fread(SC_fpath))

TUBB <- plot_XIC(report, XIC, sample1 = "PDAC", sample2 = "U-937",
              Run = "wJD1176", Protein_x="TUBB", 
              Modified.Sequence_1 = "\\(mTRAQ-n-8\\)ISVYYNEATGGK\\(mTRAQ-K-8\\)3",
              Modified.Sequence_2 = "\\(mTRAQ-n-0\\)ISVYYNEATGGK\\(mTRAQ-K-0\\)3",
              Stripped.Sequence = "ISVYYNEATGGK",
              RT_start = 33.84,
              RT_end = 34.05, 
              Remove_yions=F)
TUBB

HMGA1 <- plot_XIC(report, XIC, sample1 = "PDAC", sample2 = "U-937",
              Run = "wJD1166", Protein_x="HMGA1", 
              Modified.Sequence_1 = "\\(mTRAQ-n-4\\)SSQPLASK\\(mTRAQ-K-4\\)2",
              Modified.Sequence_2 = "\\(mTRAQ-n-8\\)SSQPLASK\\(mTRAQ-K-8\\)2",
              Stripped.Sequence = "SSQPLASK",
              RT_start = 20.14,
              RT_end = 20.43, 
              Remove_yions=F)
HMGA1

KRT7 <- plot_XIC(report, XIC, sample1 = "PDAC", sample2 = "U-937",
              Run = "wJD1153", Protein_x="KRT7", 
              Modified.Sequence_1 = "\\(mTRAQ-n-4\\)TAAENEFVVLK\\(mTRAQ-K-4\\)K\\(mTRAQ-K-4\\)4",
              Modified.Sequence_2 = "\\(mTRAQ-n-8\\)TAAENEFVVLK\\(mTRAQ-K-8\\)K\\(mTRAQ-K-8\\)4",
              Stripped.Sequence = "TAAENEFVVLKK",
              RT_start = 34.24,
              RT_end = 34.67,
              Remove_yions=F)
KRT7

#check Cscores of KRT7 precursors
report_1153 <- report[grepl("1153", report$Run),]
KRT7_4 <- report_1153[grepl("\\(mTRAQ-n-4\\)TAAENEFVVLK\\(mTRAQ-K-4\\)K\\(mTRAQ-K-4\\)4", report_1153$Precursor.Id),]
KRT7_8 <- report_1153[grepl("\\(mTRAQ-n-8\\)TAAENEFVVLK\\(mTRAQ-K-8\\)K\\(mTRAQ-K-8\\)4", report_1153$Precursor.Id),]
KRT7_4$CScore
KRT7_8$CScore

all <- ggarrange(HMGA1, TUBB, KRT7, ncol=3)
all

ggsave("Fig6_SC_XIC_03072022.pdf", width=9, height=9, dpi=500)

################################
################################
########### ion  counts
###############################
###############################


# Load single cell copy numbers from tsv
sn <- as.data.frame(read.delim(sn_fpath))
#sn <- top_n(sn, 1000)

# calculate new columns like Label, Sequence etc. from the Precursor Id
sn$Precursor.Id.Old <- sn$Precursor.Id
sn <- translate_diann_channel_format(sn, columns = c('Precursor.Id'))
sn$Precursor.Id.Labelfree <- sapply(sn$Precursor.Id, .lf_channel)
sn$Label <- sapply(sn$Precursor.Id, .get_channel)
sn$Sequence <- sapply(sn$Precursor.Id.Labelfree, .get_sequence)
sn$Id <- paste(sn$Run,sn$Label, sep='-')

# Load and merge metadata
meta <- as.data.frame(read.delim(meta_PCA_fpath))
meta <- meta[!grepl("_DB|_t", meta$Celltype),]
meta$Id <- paste(meta$Raw,meta$Label, sep='-')

df <- merge(x = sn, y = meta, by = "Id", all = TRUE)
df <- dplyr::filter(df, !(df$Set %in% c('X1','X2','X3','X4', 'X5','X6')))

# Create peptide copy number distribution
peptide_df <- df %>% dplyr::group_by(Celltype, Sequence) %>%
  dplyr::add_count() %>% 
  dplyr::filter(n>=5) %>% 
  dplyr::ungroup() # require precursor to be present in atleast 5 of the cells

peptide_df <- peptide_df %>% 
  dplyr::group_by(Id, Sequence, Celltype) %>%                            # multiple group columns
  dplyr::summarise(Counts = sum(Extracted.Counts)) %>% dplyr::ungroup()

peptide_df <- na.omit(peptide_df)
peptide_df <- peptide_df %>% 
  dplyr::group_by(Sequence, Celltype) %>%                            # multiple group columns
  dplyr::summarise(Counts = median(Counts, na.rm=T)) %>% dplyr::ungroup()

peptide_df <- dplyr::filter(peptide_df, !grepl('Neg', Celltype))

peptide_df_fin <- peptide_df %>%
  dplyr::group_by(Celltype) %>%
  dplyr::summarise(med_count = median(Counts, na.rm=T)) %>% dplyr::ungroup()

ggplot(peptide_df, aes(y=log10(Counts), x = Celltype, color = Celltype)) + 
  geom_quasirandom(alpha=0.8,width=0.3,groupOnX=T)+
  geom_boxplot(width=0.45, outlier.shape=NA, color="black",alpha=1) +
  theme_classic() +
  geom_text(data=peptide_df_fin, aes(x=Celltype, y=log10(med_count)+0.13, label=paste0(round(log10(med_count),1))), color="black",parse = F, size=4) +
  theme(#axis.text.x = element_text(size=16, face = "bold", color = "black"),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=18),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        #plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        #strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
      labs(y=expression(paste(Log["10"],", Copies per peptide per cell")))+
      ylim(0,4.5) +
    scale_color_manual(values = my_fin_colors)
ggsave("Rev_Fig6_Precs_counts.pdf", width=2.3, height=5)


pep_df <- peptide_df      
# Load and merge report to get Protein groups
report <- as.data.frame(read_delim(SC_fpath))

report$Precursor.Run.Id <- paste(report$Precursor.Id, report$Run, sep='-')

protein_df <-cbind(df)
protein_df$Precursor.Run.Id <- paste(protein_df$Precursor.Id.Old, protein_df$Run, sep='-')
protein_df <- merge(x = protein_df, y = report, by = "Precursor.Run.Id",  all.x = TRUE)


# Create Protein copy number plot
#protein_df <- dplyr::filter(protein_df, Proteotypic==1)
protein_df <- protein_df %>% dplyr::group_by(Celltype, Protein.Group) %>%
  dplyr::add_count() %>% 
  dplyr::filter(n>=5) %>% 
  dplyr::ungroup() # require precursor to be present in atleast 5 of the cells

protein_df <- protein_df %>% 
  dplyr::group_by(Id, Protein.Group, Celltype) %>%                            # multiple group columns
  dplyr::summarise(Counts = sum(Extracted.Counts)) %>% dplyr::ungroup()

protein_df <- na.omit(protein_df)

protein_df <- protein_df %>% 
  dplyr::group_by(Protein.Group, Celltype) %>%                            # multiple group columns
  dplyr::summarise(Counts = median(Counts, na.rm=T)) %>% dplyr::ungroup()

protein_df <- dplyr::filter(protein_df, !grepl('Neg', Celltype))

protein_df_fin <- protein_df %>%
  dplyr::group_by(Celltype) %>%
  dplyr::summarise(med_count = median(Counts, na.rm=T)) %>% dplyr::ungroup()

ggplot(protein_df, aes(y=log10(Counts), x = Celltype, color = Celltype)) + 
  geom_quasirandom(alpha=0.8,width=0.3,groupOnX=T)+
  geom_boxplot(width=0.45, outlier.shape=NA, color="black",alpha=1) +
  theme_classic() +
  geom_text(data=protein_df_fin, aes(x=Celltype, y=log10(med_count)+0.13, label=paste0(round(log10(med_count),1))), color="black",parse = F, size=4) +
  theme(#axis.text.x = element_text(size=16, face = "bold", color = "black"),
        axis.text.x = element_blank(),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=18),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        #plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        #strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
      labs(y=expression(paste(Log["10"],", Copies per protein per cell")))+
      ylim(0,4.5) +
    scale_color_manual(values = my_fin_colors)
ggsave("Rev_Fig6_PG_counts.pdf", width=2.3, height=5)

###########################################
###########################################
###########################################
#########  PCA w/ bulk
###########################################
###########################################

m <- read.delim(meta_PCA_fpath)
m <- m[!grepl("_214", m$Raw),]
SC_ms1ext <- data.frame(fread(tims_SC_fpath_ms1Ext))
SC <- data.frame(fread(tims_SC_fpath))
PGs_0.01FDR <- SC[which(SC$Lib.PG.Q.Value<0.01),] %>% dplyr::distinct(Protein.Group)
SC$Run <- str_replace_all(SC$Run,"-",".")
meta <- m

qev <- pD_isoCO_Ms1Ext_tims(SC, SC_ms1ext) #correct for isotopic carryover
qev_keep <- qev
qev <- qev[which(qev$Protein.Group%in%PGs_0.01FDR$Protein.Group),]
qev <- pD_channel(qev)
qev$runchan <- paste0(qev$Run, qev$channel_name)

colnames(qev)
meta <- m

par(mfrow=c(1,1))
qev$pep<-paste0(qev$Modified.Sequence, qev$Precursor.Charge)
qev$prot <- qev$Protein.Group
qev$seqcharge<-paste0(qev$Stripped.Sequence, qev$Precursor.Charge)
q0<-qev %>% dplyr::select("Run","channel_name","Ms1.Extracted_iso", "pep", "prot", "seqcharge")
q0$run_chan <- paste0(q0$Run, q0$channel_name)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
#dat<-as.matrix(q0)
no_neg <- meta[!grepl("Neg", meta$Celltype),] 
no_neg1 <- no_neg %>% dplyr::select(-c("Raw","Label") )
dat_tims<-q0 %>% inner_join(no_neg1, by =c("run_chan"="run_chan"))
dat <- dat_tims
dat$Ms1.Area<-log2(dat$Ms1.Extracted_iso)
dat$Ms1.Area[dat$Ms1.Area==Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==-Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==0]<-NA
dat_tims <- dat
no_neg_tims <- no_neg

#### QE data
SC <- data.frame(fread(SC_fpath))
SC_ms1ext <- data.frame(fread(SC_ms1ext_fpath))
SC <- SC[!grepl("wJD12", SC$Run),] #remove the diluted bulk runs
SC_ms1ext <- SC_ms1ext[,!grepl("wJD12", colnames(SC_ms1ext))]  #remove the diluted bulk runs
PGs_0.01FDR <- SC[which(SC$Lib.PG.Q.Value<0.01),] %>% distinct(Protein.Group)
qev <- pD_isoCO_Ms1Ext(SC, SC_ms1ext) #correct for isotopic carryover
qev_keep <- qev
qev <- qev[which(qev$Protein.Group%in%PGs_0.01FDR$Protein.Group),]
qev <- pD_channel(qev)
qev$runchan <- paste0(qev$Run, qev$channel_name)

colnames(qev)
meta <- read.delim(meta_PCA_fpath)

par(mfrow=c(1,1))
qev$pep<-paste0(qev$Modified.Sequence, qev$Precursor.Charge)
qev$prot <- qev$Protein.Group
qev$seqcharge<-paste0(qev$Stripped.Sequence, qev$Precursor.Charge)
q0<-qev %>% dplyr::select("Run","channel_name","Ms1.Extracted_iso", "pep", "prot", "seqcharge")
q0$run_chan <- paste0(q0$Run, q0$channel_name)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
#dat<-as.matrix(q0)
no_neg <- meta[!grepl("Neg", meta$Celltype),] 
no_neg1 <- no_neg %>% dplyr::select(-c("Raw","Label") )
dat<-q0 %>% inner_join(no_neg1, by =c("run_chan"="run_chan"))

dat$Ms1.Area<-log2(dat$Ms1.Extracted_iso)
dat$Ms1.Area[dat$Ms1.Area==Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==-Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==0]<-NA
no_neg_SC <- no_neg
dat_SC <- dat

###########  QE bulk
B <- data.frame(fread(b_fpath))
B_ms1ext <- data.frame(fread(b_ms1ext_fpath))
PGs_0.01FDR <- SC[which(SC$Lib.PG.Q.Value<0.01),] %>% distinct(Protein.Group)
qev <- pD_isoCO_Ms1Ext(B, B_ms1ext) #correct for isotopic carryover
qev_keep <- qev
qev <- qev_keep
qev <- qev[which(qev$Protein.Group%in%PGs_0.01FDR$Protein.Group),]
qev <- pD_channel(qev)
qev$runchan <- paste0(qev$Run, qev$channel_name)

colnames(qev)
meta <- read.delim(meta_PCA_fpath)

par(mfrow=c(1,1))
qev$pep<-paste0(qev$Modified.Sequence, qev$Precursor.Charge)
qev$prot <- qev$Protein.Group
qev$seqcharge<-paste0(qev$Stripped.Sequence, qev$Precursor.Charge)
q0<-qev %>% dplyr::select("Run","channel_name","Ms1.Extracted_iso", "pep", "prot", "seqcharge")
q0$run_chan <- paste0(q0$Run, q0$channel_name)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
no_neg <- meta[!grepl("Neg", meta$Celltype),] 
no_neg1 <- no_neg %>% dplyr::select(-c("Raw","Label") )
dat<-q0 %>% inner_join(no_neg1, by =c("run_chan"="run_chan"))

dat$Ms1.Area<-log2(dat$Ms1.Extracted_iso)
dat$Ms1.Area[dat$Ms1.Area==Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==-Inf]<-NA
dat$Ms1.Area[dat$Ms1.Area==0]<-NA
dim(dat)

no_neg <- rbind(no_neg_SC, no_neg_tims, no_neg)
dat <- rbind(dat_SC, dat_tims, dat)
dat1 <- dat
dat <- dat1
dat_keep <- dat %>% dplyr::group_by(run_chan) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>% filter(missing<0.6) %>%
  dplyr::distinct(run_chan) %>% ungroup()

dat <- dat[dat$run_chan%in%dat_keep$run_chan,] #require <60% missing data per cell
  
t3m<-data.frame(dat)
t3m$Ms1.Area <- 2^(t3m$Ms1.Area)
t3m <- t3m %>% dplyr::group_by(Run, seqcharge) %>% dplyr::mutate("Ms1.Area"=Ms1.Area/mean(Ms1.Area, na.rm=T)) %>% ungroup()
t3m <- t3m %>% dplyr::group_by(run_chan) %>% dplyr::mutate("Ms1.Area" = Ms1.Area/median(Ms1.Area, na.rm=T)) %>% ungroup()
t3m <- t3m %>% dplyr::group_by(seqcharge) %>% dplyr::mutate("Ms1.Area" = Ms1.Area/mean(Ms1.Area, na.rm=T)) %>% ungroup()
#t3m$Ms1.Area <- log2(t3m$Ms1.Area)
t3m <- t3m %>% dplyr::select("pep", "prot", "run_chan", "Ms1.Area")
t3m<-melt(t3m, variable.names = c("pep", "prot", "Celltype"))
t3m <- t3m %>% select(-"variable")
colnames(t3m) <-c("pep","prot","id","quantitation")
t3m2<- t3m %>% dplyr::group_by(prot, id) %>% dplyr::summarize(qp = median(quantitation, na.rm=T))
t3m2$Run <- substr(t3m2$id,1,nchar(t3m2$id)-6)
t3m2$Channel <- substr(t3m2$id,nchar(t3m2$id),nchar(t3m2$id))

#### nromalize each protein in each channel to mean of the set
t3m2 <- t3m2 %>% dplyr::group_by(Run, prot) %>% dplyr::mutate("qp"=qp/mean(qp, na.rm=T)) %>% ungroup()
t3m2 <- t3m2 %>% dplyr::group_by(id) %>% dplyr::mutate("qp"=qp/median(qp, na.rm=T))  %>% ungroup()#column norm to median
t3m2 <- t3m2 %>% dplyr::group_by(prot) %>% dplyr::mutate("qp"=qp/mean(qp, na.rm=T))  %>% ungroup()#row norm to mean prot
t3m2$qp[t3m2$qp==0]<-NA

t3m2$qp <- log2(t3m2$qp)
t3m2$qp[t3m2$qp==Inf]<-NA
t3m2$qp[t3m2$qp==-Inf]<-NA
t3m2$qp[t3m2$qp==0]<-NA

t4m<-dcast(t3m2, prot ~ id, value.var = "qp", fill=NA)

t4<-as.matrix(t4m[,-1]); row.names(t4)<-t4m[,1]

dat2<-t4
dim(dat2)
colnames(dat)


## Impute single celldata
dat[dat==0]<-NA
imp.input<-(filt.mat.rc(dat2,0.95,0.95))
length(which(is.na(imp.input)))
length(which((imp.input==0)))
#imp.input[dat==imp.input]<-NA

dim(imp.input)

sc.imp <- hknn(imp.input, 3)

length(which(is.na(sc.imp)))
length(which((sc.imp==0)))

dim(sc.imp)

sc.imp[is.na(sc.imp)]<-0

# # Define the batches and model:
# 
# 
batch.covs <-no_neg$Label[match(colnames(sc.imp), no_neg$run_chan)]

mod<-data.frame(no_neg$Celltype[match(colnames(sc.imp), no_neg$run_chan)]); colnames(mod)<-"celltype"
mod<-model.matrix(~as.factor(celltype), data=mod)

matrix.sc.batch <- ComBat(sc.imp, batch=batch.covs, mod=mod)

# 
# # Data to use:
mat.sc.imp<-cr_norm_log((matrix.sc.batch))
mat.sc.imp1 <- melt(mat.sc.imp); colnames(mat.sc.imp1)<-c("prot","id","value")
# Re map ...
mat.sc.imp1 <- dcast(mat.sc.imp1, id ~ prot, value.var = "value", fill=NA)
mat.sc.imp1$celltype<-no_neg$Celltype[match(mat.sc.imp1$id, no_neg$run_chan)]
mat.sc.imp1 <- mat.sc.imp1 %>% dplyr::select(-"id")
mat.sc.imp1$celltype <- make.names(mat.sc.imp1$celltype, unique=T)
mat.sc.imp1 <- melt(mat.sc.imp1); colnames(mat.sc.imp1)<-c("cell","prot","value")
mat.sc.imp1$cell <- as.factor(mat.sc.imp1$cell)
mat.sc.imp2 <- dcast(mat.sc.imp1, prot~cell, value.var = "value", fill=NA)

write.table(mat.sc.imp2, "Proteins_SC.txt", sep = "\t", row.names = FALSE)

# 
# 
# # Dot product of each protein correlation vector with itself
r1<-cor(t(mat.sc.imp))
rsum<-rowSums(r1^2)
# Calculate the weighted data matrix:
X.m <- mat.sc.imp
X.m <- diag(rsum) %*%  X.m
# pca.imp.cor <- cor(X.m, use="complete.obs")
pca.imp.cor <- cor(X.m, use="complete.obs")
# PCA
sc.pca<-eigen(pca.imp.cor)
scx<-as.data.frame(sc.pca$vectors)
colnames(scx)<-paste0("PC",1:ncol(scx))
scx$cells<-colnames(pca.imp.cor)
# Percent of variance explained by each principle component
pca_var <- sc.pca$values
percent_var<- pca_var/sum(pca_var)*100
plot(1:length(percent_var), percent_var, xlab="PC", ylab="% of variance explained")
# Map meta data
pca.melt <- melt(scx); colnames(pca.melt)<-c("id","pc","value")
# Re map ...
pca.display <- dcast(pca.melt, id ~ pc, value.var = "value", fill=NA)
pca.display$celltype<-no_neg$Celltype[match(pca.display$id, no_neg$run_chan)]
pca.display$label<-no_neg$Label[match(pca.display$id, no_neg$run_chan)]

# PC's to display:
PCx<-"PC1"
PCy<-"PC2"
PCz<-"PC3"
PCza<-"PC4"
PCzb<-"PC5"
# Display
pca.display$lab <- paste0("d",pca.display$label)
pca.display <- pca.display %>% dplyr::mutate("celltype2" = ifelse(grepl("Melanoma", celltype), "Melanoma",
                                                                  ifelse(grepl("U-937", celltype), "U-937", "PDAC")))
pca.display <- pca.display %>% dplyr::mutate("SC_DB" = ifelse(grepl("_DB", celltype), "Bulk", 
                                                              ifelse(grepl("_t", celltype), "timsTOF SCP", "QE")))

check <- pca.display %>% dplyr::select("id","PC1","PC2","celltype")
pxlab<-ggplot(pca.display,aes(x =PC1, y = PC2, color =lab, shape=SC_DB, alpha =SC_DB),  size = 3, alpha=0.5) +
  labs(color = "Label", shape="",x = paste0(PCx,"  (", round(percent_var[1],0),"%)"), y = paste0(PCy,"  (", round(percent_var[2],0),"%)")) +
  scale_shape_manual(values = c(23,16,15)) + 
  scale_alpha_manual(values = c(0.66,0.66,0.66)) +
  font("ylab",size=20) +
  font("xlab",size=20) +
  font("xy.text", size=15) +
   geom_point()+#aes(fill=celltype2),colour="black",pch=21,alpha=0.66) +
   geom_point(colour="black",alpha=0.15) +
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2),size = 1, color = "black", shape = 23,stroke = 1.2,alpha=0.7)+
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2),size = 2, color = "black", shape = 23,stroke = 1.2, alpha=0.7)+
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2), size = 1.5,shape = 23,stroke = 0.6,alpha=1) +
     theme_classic() +
      theme(
        legend.position = "top",
        legend.background = element_rect(color = "transparent", fill = "transparent")) +
    scale_color_manual(values = my_fin_colors)+
      scale_fill_manual(values = my_fin_colors)+
    annotate("text", x=-0.072, y=-0.1, label=paste0(ncol(mat.sc.imp)-9, " cells"), size=3) +
  annotate("text", x=-0.04, y=-0.12, label=paste0(dim(mat.sc.imp)[1], " protein groups"), size=3)+
  guides(fill = "none", size = "none",alpha = "none") #labs(shape = "",x="PC1",y = "PC2") +
pxlab
ggsave("Rev_Fig6_labBias_v2.pdf",
       width=3,height=3.2)

## PC1 and PC2 
px<-ggplot(pca.display,aes(x =PC1, y = PC2, color =celltype2, shape=SC_DB, alpha =SC_DB),  size = 3, alpha=0.5) +
  labs(shape="",x = paste0(PCx,"  (", round(percent_var[1],0),"%)"), y = paste0(PCy,"  (", round(percent_var[2],0),"%)")) +
  scale_shape_manual(values = c(23,16,15)) + 
  scale_alpha_manual(values = c(0.66,0.66,0.66)) +
  font("ylab",size=20) +
  font("xlab",size=20) +
  font("xy.text", size=15) +
   geom_point()+#aes(fill=celltype2),colour="black",pch=21,alpha=0.66) +
   geom_point(colour="black",alpha=0.15) +
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2),size = 1, color = "black", shape = 23,stroke = 1.2,alpha=0.7)+
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2),size = 2, color = "black", shape = 23,stroke = 1.2, alpha=0.7)+
  geom_point(data = pca.display %>% dplyr::filter(grepl("_DB", celltype)),aes(x = PC1, y = PC2), size = 1.5,shape = 23,stroke = 0.6,alpha=1) +
     theme_classic() +
      theme(
        legend.position = "top",
        legend.background = element_rect(color = "transparent", fill = "transparent")) +
    scale_color_manual(values = my_fin_colors)+
      scale_fill_manual(values = my_fin_colors)+
    annotate("text", x=-0.072, y=-0.1, label=paste0(ncol(mat.sc.imp)-9, " cells"), size=3) +
  annotate("text", x=-0.04, y=-0.12, label=paste0(dim(mat.sc.imp)[1], " protein groups"), size=3)+
  guides(color = "none",fill = "none", size = "none",alpha = "none") #labs(shape = "",x="PC1",y = "PC2") +
px
ggsave("Fig6_PCA_1_2_v3_bulk.pdf",
       width=3.5,height=4)


############### differential proteins


bulk_df <- data.frame(fread(b_fpath))
bulk_df_ms1ex <- data.frame(fread(b_ms1ext_fpath))
metaB <- read.delim(meta_PCA_fpath)

PGs_0.01FDR <- bulk_df[which(bulk_df$Lib.PG.Q.Value<0.01),] %>% distinct(Protein.Group)
B1 <- pD_isoCO_Ms1Ext(bulk_df, bulk_df_ms1ex) # correct the isotopic carryover across channel for Ms1.Extracted
#B1 <- B1[grepl("wJD1098|wJD1099|wJD1100", B1$Run),]
B <- B1
bulk_df_proteotypic <- pD_seqcharge(bulk_df) #%>% filter(Proteotypic==1)
B <- B[B$seqcharge%in%bulk_df_proteotypic$seqcharge,]
B <- pD_channel(B)
B$run_chan <- paste0(B$Run, B$channel_name)
metaB$run_chan <- paste0(metaB$Raw, "mTRAQ", metaB$Label)
metaB_lim <- metaB %>% dplyr::select("run_chan", "Celltype")
B <- B %>% left_join(metaB_lim, by =c("run_chan" = "run_chan"))
B$run_cell <- paste0(B$Run, B$Celltype)
B <- B[!grepl("Mel", B$Celltype),]
B <- B[which(B$Ms1.Extracted_iso>0),]
B <- B %>% dplyr::group_by(seqcharge, Run) %>% dplyr::add_count() %>% dplyr::filter(n>1) #all cell types and all runs
plex <- B
plex <- plex %>% dplyr::group_by(run_cell) %>% dplyr::mutate(Ms1.Area_cnorm = Ms1.Extracted_iso/median(Ms1.Extracted_iso, na.rm=T))
plex <- plex %>% dplyr::group_by(seqcharge) %>% dplyr::mutate(Ms1.Area = Ms1.Area_cnorm/mean(Ms1.Area_cnorm)) %>% dplyr::ungroup()
plex1 <- plex %>% dplyr::select("seqcharge", "Ms1.Area", "Protein.Group", "Celltype")
int <- plex1 %>% dplyr::group_by(Protein.Group, Celltype) %>% dplyr::add_count() %>% 
  dplyr::filter(n>1) %>% dplyr::ungroup() %>%
  dplyr::distinct(Protein.Group, Celltype) 
plex1 <- plex1 %>% dplyr::group_by(Celltype, Protein.Group) %>% dplyr::mutate("med_pep_ab" = median(Ms1.Area))%>% dplyr::ungroup()
prots <- plex1 %>% dplyr::distinct(Protein.Group, .keep_all=T)
prots <- prots[prots$Protein.Group%in%int$Protein.Group,]
prots$pval <- 10000
prots$matchprot <- paste0("_",prots$Protein.Group,"_")
plex1$matchprot <- paste0("_",plex1$Protein.Group,"_")
unique_prots <- as.vector(prots$matchprot)

for(i in 1:length(unique_prots)){
  temp_interest <- as.character(unique_prots[i])  #for  a given protein
  prot_int <- plex1[grepl(paste0(temp_interest), plex1$matchprot),]
  aav_test <- aov(Ms1.Area~Celltype, data = prot_int)
  sum_aav1 <- summary(aav_test)
  sum_aav1 <- data.frame(sum_aav1[[1]])
  pval_temp <- paste0(sum_aav1[1,5]) 
  temp_interest <- temp_interest
  prots[i,6] <- as.numeric(pval_temp)
  
}

prots$FC_qval <-  p.adjust(prots$pval, method = 'BH') # qvalue

write.table(prots, "DA_PDAC_U937.txt", sep = "\t", row.names = FALSE)



```


#############
Supplementary
#############

Supp. Figure 1.... translated quant and IDs schematic

Supp. Figure 2: Missing E. coli experiment to benchmark translated FDR
```{r}

ev <- data.frame(fread(YE_fpath))
ev1 <- ev[which(ev$Channel.Q.Value<0.01),]  #filter for channel-q-value < 0.01 
ev1 <- pD_seqcharge(ev1)
ev1 <- pD_isotopicCO(ev1)
ev1$seqcharge_run <- paste0(ev1$seqcharge,ev1$Run)

ev2_lim <- ev1
ev2_lim <- ev2_lim[grepl("wJD813", ev2_lim$Run),]
ev2_lim <- ev2_lim[!grepl("-8", ev2_lim$Precursor.Id),]
ev2_lim <- pD_rmMixSpec(ev2_lim)
ev2_lim <- ev2_lim %>% dplyr::mutate("species" = ifelse(grepl("ECOLI", Protein.Names), "E. coli", "S. cerevisiae"))
ev2_lim <- ev2_lim %>% dplyr::mutate("Label" = ifelse(grepl("-4", Precursor.Id), paste0("\u0394","4"),  paste0("\u0394","0")))

d0 <- ev2_lim[grepl("0", ev2_lim$Label),] %>% dplyr::select("seqcharge", "Ms1.Area_iso", "Precursor.Translated","Ms1.Profile.Corr", "Channel.Q.Value", "species", "Label")
d4 <- ev2_lim[grepl("4", ev2_lim$Label),] %>% dplyr::select("seqcharge", "Ms1.Area_iso", "Precursor.Translated","Ms1.Profile.Corr", "Channel.Q.Value", "species", "Label")

all <- rbind(d0, d4)

all$spec_lab <- paste0(all$Label," ",all$species)
all <- all %>% dplyr::mutate(type = ifelse(grepl(paste0("\u0394","4 E. coli"), spec_lab), "3",
                                    ifelse(grepl(paste0("\u0394","0 E. coli"), spec_lab), "1",
                                    ifelse(grepl(paste0("\u0394","0 S. cerevisiae"), spec_lab), "2",
                                           ifelse(grepl(paste0("\u0394","4 S. cerevisiae"), spec_lab), "4","wrong")))))

all_lim <- all %>% dplyr::select("Ms1.Area_iso", "type")
all_lim_non0 <- all_lim[which(all_lim$Ms1.Area_iso>0),]
ec4 <- nrow(all_lim_non0[grepl("3", all_lim_non0$type),])
ec0 <- nrow(all_lim_non0[grepl("1", all_lim_non0$type),])

write.table(all_lim, "d0_d4_missingecoli.txt", sep = "\t", row.names = FALSE)

all_lim_MS2 <- all %>% dplyr::select("Precursor.Translated", "type")
all_lim_non0 <- all_lim_MS2[which(all_lim_MS2$Precursor.Translated>0),]
ec4 <- nrow(all_lim_non0[grepl("3", all_lim_non0$type),])
ec0 <- nrow(all_lim_non0[grepl("1", all_lim_non0$type),])

write.table(all_lim_MS2, "d0_d4_missingecoli_MS2.txt", sep = "\t", row.names = FALSE)
# this data was then plotted in matlab

```

Supp Figure 3: MS2-optimized, Coverage and completeness removed s3 rep 3 for LFDIA
```{r}

plexDIA_run <- "eJD906" #plexDIA run name you are interested in plotting
LF_runs <- c("eJD909", "eJD912", "eJD915") # LF-DIA runs you are interested in plotting

# Figure 2, Proteomic coverage and overlap between samples and runs of plexDIA and LF-DIA:
mTRAQ <- fread(MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value","Precursor.Quantity", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated"))

mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0|mTRAQ$Precursor.Translated>0),]  # must have non-zero quant
mTRAQ$Label <- "mTRAQ"
mTRAQ$seqcharge <- paste0(mTRAQ$Modified.Sequence, mTRAQ$Precursor.Charge)
mTRAQ$seqcharge_run <- paste0(mTRAQ$seqcharge, "_", mTRAQ$Run)
mTRAQ$prot_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ0\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ4\\)")
mTRAQ$seqcharge <- str_remove_all(mTRAQ$seqcharge, "\\(mTRAQ8\\)")
mTRAQ_uni_prot <- unique(mTRAQ$Protein.Group)

ev2_0 <- mTRAQ[grepl("-0",mTRAQ$Precursor.Id),] %>% dplyr::distinct(seqcharge_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("-4",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% dplyr::distinct(seqcharge_run, .keep_all=T)
colnames(ev2_4) <- c("Run","Protein.Group_d4","seqcharge_d4","Precursor.Quantity_d4","Precursor.Translated_d4", "seqcharge_run_d4", "prot_run_d4","Lib.PG.Q.Value_d4")
ev2_8 <- mTRAQ[grepl("-8",mTRAQ$Precursor.Id),] %>% dplyr::select("Run","Protein.Group","seqcharge","Precursor.Quantity","Precursor.Translated", "seqcharge_run", "prot_run", "Lib.PG.Q.Value") %>% dplyr::distinct(seqcharge_run, .keep_all=T)
colnames(ev2_8) <- c("Run","Protein.Group_d8","seqcharge_d8","Precursor.Quantity_d8","Precursor.Translated_d8", "seqcharge_run_d8", "prot_run_d8","Lib.PG.Q.Value_d8")

ev2_0_prot <- ev2_0[which(ev2_0$Lib.PG.Q.Value < 0.01),] %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
ev2_0_prot$Sample <- paste0("\u0394","0-A")
ev2_4_prot <- ev2_4[which(ev2_4$Lib.PG.Q.Value_d4 < 0.01),] %>% dplyr::distinct(prot_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4","prot_run_d4")
colnames(ev2_4_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_4_prot$Sample <- paste0("\u0394","4-B")
ev2_8_prot <- ev2_8[which(ev2_8$Lib.PG.Q.Value_d8 < 0.01),] %>%dplyr::distinct(prot_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8","prot_run_d8")
colnames(ev2_8_prot) <- c("Run", "Protein.Group", "prot_run")
ev2_8_prot$Sample <- paste0("\u0394","8-C")

ev2_0_pep <- ev2_0 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_0_pep$Sample <- paste0("\u0394","0-A")
ev2_4_pep <- ev2_4 %>% dplyr::distinct(seqcharge_run_d4, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d4", "seqcharge_d4", "seqcharge_run_d4")
colnames(ev2_4_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_4_pep$Sample <- paste0("\u0394","4-B")
ev2_8_pep <- ev2_8 %>% dplyr::distinct(seqcharge_run_d8, .keep_all=T)%>% dplyr::select("Run", "Protein.Group_d8", "seqcharge_d8", "seqcharge_run_d8")
colnames(ev2_8_pep) <- c("Run", "Protein.Group", "seqcharge", "seqcharge_run")
ev2_8_pep$Sample <- paste0("\u0394","8-C")

mTRAQ_prots <- rbind(ev2_0_prot,ev2_4_prot,ev2_8_prot)
mTRAQ_prots$Sample_new <- mTRAQ_prots$Sample

mTRAQ_peps <- rbind(ev2_0_pep,ev2_4_pep,ev2_8_pep)
mTRAQ_peps$Sample_new <- mTRAQ_peps$Sample

p_mTRAQ <- mTRAQ_peps %>% dplyr::count(Run, Sample_new)
p_mTRAQ$Condition <- "plexDIA"

prot_mTRAQ <- mTRAQ_prots %>% dplyr::count(Run, Sample_new)
prot_mTRAQ$Condition <- "plexDIA"

mTRAQ_int <-mTRAQ_prots %>% dplyr::group_by(Run, Protein.Group) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  dplyr::group_by(Run) %>% dplyr::add_count() %>% dplyr::distinct(Run, .keep_all=T)#require protein to be present across samples
mTRAQ_int$prots_per_run <- mTRAQ_int$nn/3
mTRAQ_int$type <- "plexDIA"


##############
##############
#######################
########################
##########################  Label-free

unlab <- fread(LF_MS2_fpath, select = c("Run","Protein.Group","Protein.Names","Genes", "Modified.Sequence", "Stripped.Sequence","Precursor.Id",
                               "Precursor.Charge", "Q.Value","Lib.PG.Q.Value", "PG.Q.Value", "Translated.Q.Value", "Proteotypic","Precursor.Translated", "Ms1.Area", "Ms1.Translated"))

unlab <- unlab[which(unlab$Ms1.Area>0|unlab$Precursor.Translated>0),]  # must have non-zero Ms1 quant
unlab_1 <- unlab[grepl("eJD908|eJD909|eJD910", unlab$Run),]
unlab_1$seqcharge <- paste0(unlab_1$Modified.Sequence, unlab_1$Precursor.Charge)
unlab_1$seqcharge_run <- paste0(unlab_1$seqcharge, "_", unlab_1$Run)
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_", unlab_1$Run)

unlab_1_prot <- unlab_1[which(unlab_1$Lib.PG.Q.Value < 0.01),] %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_1_prot$Sample <- paste0("\u0394","0-A")
unlab_1_prot$Sample_new <- paste0("A")

unlab_1_pep <- unlab_1 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_1_pep$Sample <- paste0("\u0394","0-A")
unlab_1_pep$Sample_new <- paste0("A")

##############
unlab_2 <- unlab[grepl("eJD911|eJD912|eJD913", unlab$Run),]
unlab_2$seqcharge <- paste0(unlab_2$Modified.Sequence, unlab_2$Precursor.Charge)
unlab_2$seqcharge_run <- paste0(unlab_2$seqcharge, "_", unlab_2$Run)
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_", unlab_2$Run)

unlab_2_prot <- unlab_2[which(unlab_2$Lib.PG.Q.Value < 0.01),] %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_2_prot$Sample <- paste0("\u0394","4-B")
unlab_2_prot$Sample_new <- paste0("B")

unlab_2_pep <- unlab_2 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_2_pep$Sample <- paste0("\u0394","4-B")
unlab_2_pep$Sample_new <- paste0("B")

##############
unlab_3 <- unlab[grepl("eJD914|eJD915", unlab$Run),]
unlab_3$seqcharge <- paste0(unlab_3$Modified.Sequence, unlab_3$Precursor.Charge)
unlab_3$seqcharge_run <- paste0(unlab_3$seqcharge, "_", unlab_3$Run)
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_", unlab_3$Run)

unlab_3_prot <- unlab_3[which(unlab_3$Lib.PG.Q.Value < 0.01),] %>% distinct(prot_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group","prot_run")
unlab_3_prot$Sample <- paste0("\u0394","8-C")
unlab_3_prot$Sample_new <- paste0("C")


unlab_3_pep <- unlab_3 %>% distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Run", "Protein.Group", "seqcharge", "seqcharge_run")
unlab_3_pep$Sample <- paste0("\u0394","8-C")
unlab_3_pep$Sample_new <- paste0("C")

###################

all_unlab_pep <- rbind(unlab_1_pep, unlab_2_pep, unlab_3_pep)
all_unlab_prot <- rbind(unlab_1_prot, unlab_2_prot, unlab_3_prot)

p_unlab <- all_unlab_pep %>% dplyr::count(Run, Sample_new)
p_unlab$Condition <- "LF-DIA"

prot_unlab <- all_unlab_prot %>% dplyr::count(Run, Sample_new)
prot_unlab$Condition <- "LF-DIA"

all_unlab_prot <- all_unlab_prot %>% 
  dplyr::mutate("rep" = ifelse(grepl("908|911|914", Run),"rep1",
                               ifelse(grepl("909|912|915", Run),"rep2",
                                      ifelse(grepl("910|913", Run),"rep3", "remove"))))

LF_int <-all_unlab_prot %>% dplyr::group_by(rep, Protein.Group) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  dplyr::group_by(rep) %>% dplyr::add_count() %>% dplyr::distinct(rep, .keep_all=T)#require protein to be present across samples
LF_int$prots_per_run <- LF_int$nn/3
LF_int$type <- "LF-DIA"


##############
##############
#######################
########################
##########################  mTRAQ DDA

#get prots which are 1% FDR
prot_DDAFDR <- read.delim(DDA_prot_fpath)
prot_DDAFDR <- prot_DDAFDR[which(prot_DDAFDR$Q.value<0.01),] %>% dplyr::filter(!grepl("REV|CON", Protein.IDs))
prot_DDAFDR$Protein.Group <- gsub(";.*","",prot_DDAFDR$Majority.protein.IDs)

DDA <- read.delim(DDA_fpath)
DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA_d0 <- DDA[which(DDA$Intensity.L > 0),]
DDA_d0$seqcharge <- paste0(DDA_d0$Modified.sequence, DDA_d0$Charge)
DDA_d0$seqcharge_run <- paste0(DDA_d0$seqcharge, "_", DDA_d0$Raw.file)
DDA_d0$Protein.Group <- gsub(";.*","",DDA_d0$Leading.razor.protein)
DDA_d0$prot_run <- paste0(DDA_d0$Leading.razor.protein, "_", DDA_d0$Raw.file)

DDA_d0_prot <- DDA_d0 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d0_prot$Sample <- paste0("A")
DDA_d0_prot$Sample_new <- paste0("\u0394","0 DDA-A")

DDA_d0_pep <- DDA_d0 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d0_pep$Sample <- paste0("A")
DDA_d0_pep$Sample_new <- paste0("\u0394","0 DDA-A")

####
DDA_d4 <- DDA[which(DDA$Intensity.M > 0),]
DDA_d4$seqcharge <- paste0(DDA_d4$Modified.sequence, DDA_d4$Charge)
DDA_d4$seqcharge_run <- paste0(DDA_d4$seqcharge, "_", DDA_d4$Raw.file)
DDA_d4$Protein.Group <- gsub(";.*","",DDA_d4$Leading.razor.protein)
DDA_d4$prot_run <- paste0(DDA_d4$Leading.razor.protein, "_", DDA_d4$Raw.file)

DDA_d4_prot <- DDA_d4 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d4_prot$Sample <- paste0("B")
DDA_d4_prot$Sample_new <- paste0("\u0394","4 DDA-B")

DDA_d4_pep <- DDA_d4 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d4_pep$Sample <- paste0("B")
DDA_d4_pep$Sample_new <- paste0("\u0394","4 DDA-B")

####
DDA_d8 <- DDA[which(DDA$Intensity.H > 0),]
DDA_d8$seqcharge <- paste0(DDA_d8$Modified.sequence, DDA_d8$Charge)
DDA_d8$seqcharge_run <- paste0(DDA_d8$seqcharge, "_", DDA_d8$Raw.file)
DDA_d8$Protein.Group <- gsub(";.*","",DDA_d8$Leading.razor.protein)
DDA_d8$prot_run <- paste0(DDA_d8$Leading.razor.protein, "_", DDA_d8$Raw.file)

DDA_d8_prot <- DDA_d8 %>% dplyr::distinct(prot_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein","prot_run")
DDA_d8_prot$Sample <- paste0("C")
DDA_d8_prot$Sample_new <- paste0("\u0394","8 DDA-C")

DDA_d8_pep <- DDA_d8 %>% dplyr::distinct(seqcharge_run, .keep_all=T) %>% dplyr::select("Raw.file", "Leading.razor.protein", "seqcharge", "seqcharge_run")
DDA_d8_pep$Sample <- paste0("C")
DDA_d8_pep$Sample_new <- paste0("\u0394","8 DDA-C")


##########################
##########################  data-wrangling for plotting
all_DDA_pep <- rbind(DDA_d0_pep, DDA_d4_pep, DDA_d8_pep)
all_DDA_prot <- rbind(DDA_d0_prot, DDA_d4_prot, DDA_d8_prot)
all_DDA_prot <- all_DDA_prot[which(all_DDA_prot$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

p_DDA <- all_DDA_pep %>% dplyr::count(Raw.file, Sample_new)
colnames(p_DDA) <- c("Run", "Sample_new", "n")
p_DDA$Condition <- "mTRAQ\nDDA"

prot_DDA <- all_DDA_prot %>% dplyr::count(Raw.file, Sample_new)
colnames(prot_DDA) <- c("Run", "Sample_new", "n")
prot_DDA$Condition <- "mTRAQ\nDDA"

DDA_int <-all_DDA_prot %>% dplyr::group_by(Raw.file, Leading.razor.protein) %>%
  dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()  %>%
  group_by(Raw.file) %>% add_count() %>% dplyr::distinct(Raw.file, .keep_all=T)#require protein to be present across samples
DDA_int$prots_per_run <- DDA_int$nn/3
DDA_int$type <- "mTRAQ\nDDA"

########## intersected coverage plexDIA, LF-DIA, mTRAQ DDA

########## intersected coverage plexDIA, LF-DIA
######## rep1
my_fin_colors2 <- c("turquoise3","palevioletred3","#ceca7c","turquoise3","palevioletred3","#ceca7c")

LF_rep1 <- all_unlab_prot[grepl("JD908|911|914", all_unlab_prot$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
LF_rep1 <- dcast(LF_rep1, Protein.Group ~ Sample_new)
rownames(LF_rep1) <- LF_rep1$Protein.Group
LF_rep1 <- LF_rep1[,-1]
LF_rep1[!is.na(LF_rep1)] <- TRUE
LF_rep1[is.na(LF_rep1)] <- FALSE
LF_rep1$A <- as.logical(LF_rep1$A)
LF_rep1$B <- as.logical(LF_rep1$B)
LF_rep1$C <- as.logical(LF_rep1$C)
LF_rep1$PGs <- rownames(LF_rep1)

pD_rep1 <- mTRAQ_prots[grepl("905", mTRAQ_prots$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
pD_rep1 <- data.frame(pD_rep1)
pD_rep1$Protein.Group <- paste0("pD",pD_rep1$Protein.Group)
pD_rep1 <- dcast(pD_rep1, Protein.Group ~ Sample_new)
rownames(pD_rep1) <- pD_rep1$Protein.Group
pD_rep1 <- pD_rep1[,-1]
pD_rep1[!is.na(pD_rep1)] <- TRUE
pD_rep1[is.na(pD_rep1)] <- FALSE
pD_rep1[,1] <- as.logical(pD_rep1[,1])
pD_rep1[,2] <- as.logical(pD_rep1[,2])
pD_rep1[,3] <- as.logical(pD_rep1[,3])
pD_rep1$PGs <- rownames(pD_rep1)

pD_LF_rep1 <- pD_rep1 %>% full_join(LF_rep1, by =c("PGs" = "PGs"))
pD_LF_rep1[is.na(pD_LF_rep1)] <- FALSE
pD_LF_rep1 <- pD_LF_rep1[,-4]
pd <- euler(pD_LF_rep1, shape = "ellipse",quantities = TRUE)

pdf(file = "SFig3_venn_rep1.pdf",   # The directory you want to save the file in
    width = 7, # The width of the plot in inches
    height = 7)
plot(pd,
     quantities = TRUE,
     labels = list(font = 4), fill=my_fin_colors2, alpha = 1)
dev.off()

######## rep2
LF_rep2 <- all_unlab_prot[grepl("JD909|912|915", all_unlab_prot$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
LF_rep2 <- dcast(LF_rep2, Protein.Group ~ Sample_new)
rownames(LF_rep2) <- LF_rep2$Protein.Group
LF_rep2 <- LF_rep2[,-1]
LF_rep2[!is.na(LF_rep2)] <- TRUE
LF_rep2[is.na(LF_rep2)] <- FALSE
LF_rep2$A <- as.logical(LF_rep2$A)
LF_rep2$B <- as.logical(LF_rep2$B)
LF_rep2$C <- as.logical(LF_rep2$C)
LF_rep2$PGs <- rownames(LF_rep2)

pD_rep2 <- mTRAQ_prots[grepl("906", mTRAQ_prots$Run),] %>% dplyr::select("Protein.Group", "Sample_new")
pD_rep2 <- data.frame(pD_rep2)
pD_rep2$Protein.Group <- paste0("pD",pD_rep2$Protein.Group)
pD_rep2 <- dcast(pD_rep2, Protein.Group ~ Sample_new)
rownames(pD_rep2) <- pD_rep2$Protein.Group
pD_rep2 <- pD_rep2[,-1]
pD_rep2[!is.na(pD_rep2)] <- TRUE
pD_rep2[is.na(pD_rep2)] <- FALSE
pD_rep2[,1] <- as.logical(pD_rep2[,1])
pD_rep2[,2] <- as.logical(pD_rep2[,2])
pD_rep2[,3] <- as.logical(pD_rep2[,3])
pD_rep2$PGs <- rownames(pD_rep2)

pD_LF_rep2 <- pD_rep2 %>% full_join(LF_rep2, by =c("PGs" = "PGs"))
pD_LF_rep2[is.na(pD_LF_rep2)] <- FALSE
pD_LF_rep2 <- pD_LF_rep2[,-4]
pd <- euler(pD_LF_rep2, shape = "ellipse",quantities = TRUE)

pdf(file = "SFig3_venn_rep2.pdf", width = 7, height = 7)
plot(pd,
     quantities = TRUE,
     labels = list(font = 4), fill=my_fin_colors2, alpha = 1)
dev.off()

###################
all_pep <- rbind(p_mTRAQ, p_unlab, p_DDA)
all_pep$Sample <- gsub(".*-","",all_pep$Sample_new)

all_pep <- all_pep %>% dplyr::group_by(Condition, Sample) %>% 
  dplyr::add_count() %>% dplyr::ungroup()
p_all_tab <- all_pep %>% dplyr::group_by(Sample_new) %>% dplyr::mutate(mean = mean(n)) %>%
  dplyr::mutate(se = sd(n)/sqrt(nn)) %>% dplyr::distinct(Sample_new, .keep_all=T)
p_all_tab <- p_all_tab[,-6]
a <- p_all_tab[3,6]
b <- p_all_tab[2,6] + p_all_tab[3,6]
c <- p_all_tab[2,6] + p_all_tab[1,6] + p_all_tab[3,6]

p_all_tab$er_max <- p_all_tab$mean + p_all_tab$se
p_all_tab[1,8] <- c+as.numeric(p_all_tab[1,7])
p_all_tab[2,8] <- b+as.numeric(p_all_tab[2,7])
p_all_tab[3,8] <- a+p_all_tab[3,7]

p_all_tab$er_min <- p_all_tab$mean - p_all_tab$se
p_all_tab[1,9] <- c-as.numeric(p_all_tab[1,7])
p_all_tab[2,9] <- b-as.numeric(p_all_tab[2,7])
p_all_tab[3,9] <- a-p_all_tab[3,7]


## for DDA
a <- p_all_tab[9,6]
b <- p_all_tab[8,6] + p_all_tab[9,6]
c <- p_all_tab[8,6] + p_all_tab[7,6] + p_all_tab[9,6]

p_all_tab[7,8] <- c+as.numeric(p_all_tab[7,7])
p_all_tab[8,8] <- b+as.numeric(p_all_tab[8,7])
p_all_tab[9,8] <- a+p_all_tab[9,7]

p_all_tab[7,9] <- c-as.numeric(p_all_tab[7,7])
p_all_tab[8,9] <- b-as.numeric(p_all_tab[8,7])
p_all_tab[9,9] <- a-p_all_tab[9,7]


#### text 
df_n <- p_all_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+10000
my_fin_colors <- c("turquoise3","palevioletred3","#ceca7c")

p1_cov <- ggplot(p_all_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = p_all_tab %>% filter(Condition == "plexDIA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "mTRAQ\nDDA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = p_all_tab %>% filter(Condition == "LF-DIA"), colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = p_all_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Precursors / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.text = element_text(size=12),
        legend.title = element_text(size=12),
        legend.position = c(0.7,0.8),
        legend.direction = "horizontal",
        legend.spacing.x = unit(1.0, 'line')) +
  guides(fill = guide_legend(override.aes= list(alpha = 0.7),
                             title = "Samples",
                             label.position = "bottom",
                             title.position = "top", title.vjust = 1, title.hjust=0.5)) +
  geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3
  ) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3, stat="identity", position=position_dodge(1.17))

p1_cov
ggsave("SFigure4a.pdf",p1_cov,width=5.6,height=5, dpi=500)


############ prots

###################
all_prot <- rbind(prot_mTRAQ, prot_unlab, prot_DDA)
all_prot$Sample <- gsub(".*-","",all_prot$Sample_new)
all_prot <- all_prot %>% dplyr::group_by(Condition, Sample) %>% 
  dplyr::add_count() %>% dplyr::ungroup()
all_prot_tab <- all_prot %>% dplyr::group_by(Sample_new) %>% dplyr::mutate(mean = mean(n)) %>%
  dplyr::mutate(se = sd(n)) %>% dplyr::distinct(Sample_new, .keep_all=T)
all_prot_tab <- all_prot_tab[,-6]

a <- all_prot_tab[3,6]
b <- all_prot_tab[2,6] + all_prot_tab[3,6]
c <- all_prot_tab[2,6] + all_prot_tab[1,6] + all_prot_tab[3,6]

all_prot_tab$er_max <- all_prot_tab$mean + all_prot_tab$se
all_prot_tab[1,8] <- c+as.numeric(all_prot_tab[1,7])
all_prot_tab[2,8] <- b+as.numeric(all_prot_tab[2,7])
all_prot_tab[3,8] <- a+all_prot_tab[3,7]

all_prot_tab$er_min <- all_prot_tab$mean - all_prot_tab$se
all_prot_tab[1,9] <- c-as.numeric(all_prot_tab[1,7])
all_prot_tab[2,9] <- b-as.numeric(all_prot_tab[2,7])
all_prot_tab[3,9] <- a-all_prot_tab[3,7]


## for DDA
a <- all_prot_tab[9,6]
b <- all_prot_tab[8,6] + all_prot_tab[9,6]
c <- all_prot_tab[8,6] + all_prot_tab[7,6] + all_prot_tab[9,6]

all_prot_tab[7,8] <- c+as.numeric(all_prot_tab[7,7])
all_prot_tab[8,8] <- b+as.numeric(all_prot_tab[8,7])
all_prot_tab[9,8] <- a+all_prot_tab[9,7]

all_prot_tab[7,9] <- c-as.numeric(all_prot_tab[7,7])
all_prot_tab[8,9] <- b-as.numeric(all_prot_tab[8,7])
all_prot_tab[9,9] <- a-all_prot_tab[9,7]


#### text 
df_n <- all_prot_tab[c(1,4,5,6,7),]
df_n$actual_text <- round(df_n$er_max-df_n$se)
df_n$text_place <- df_n$er_max+700

p2_cov <- ggplot(all_prot_tab, aes(x=Condition, y =mean, fill=Sample)) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA","mTRAQ\nDDA"))+
  geom_bar(data = all_prot_tab %>% filter(Condition == "plexDIA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "mTRAQ\nDDA"),colour="black", 
           stat="identity", aes(x=Condition, y =mean, fill=Sample), width = 0.35, alpha =1) + 
  geom_bar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), colour="black",
           stat="identity", position = position_dodge(width = 1.17), aes(x=Condition, y=mean, fill=Sample), width = 1, alpha =1) +
  scale_y_continuous(labels = scales::comma) +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(ymin=er_min, ymax=er_max), width=0.18, size=0.4,stat="identity") +
  geom_errorbar(data = all_prot_tab %>% filter(Condition == "LF-DIA"), aes(ymin=er_min, ymax=er_max), width=0.5, size=0.4,stat="identity", position=position_dodge(1.17)) +
  scale_fill_manual(values = my_fin_colors)+
  theme_classic() +
  labs(y = "Protein Data Points / Run", x="", fill ="") +
  theme(axis.text.x = element_text(size=14, color="black", face="bold"),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=16),
        legend.position = "none") +
   geom_text(data=df_n %>% filter(Condition == "plexDIA" | Condition == "mTRAQ\nDDA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2) +
  geom_text(data=df_n %>% filter(Condition == "LF-DIA"), aes(x=as.factor(Condition), y=text_place, label=scales::comma(actual_text)), col='black', size=3.2, stat="identity", position=position_dodge(1.17))

p2_cov
ggsave("SFigure4b.pdf",p2_cov,width=5.6,height=5, dpi=500)


################## JACCARD 

mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0|mTRAQ$Precursor.Translated>0),]  #only keep precursors with quant info

ev2_0 <- mTRAQ[grepl("-0",mTRAQ$Precursor.Id),] %>% dplyr::distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("-4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Protein.Group") %>% dplyr::distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Translated_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("-8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Translated","Protein.Group") %>% dplyr::distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Translated_d8","Protein.Group")

d0_1 <- ev2_0[grepl("eJD905", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("eJD905", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_1 <- ev2_8[grepl("eJD905", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("eJD906", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("eJD906", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_2 <- ev2_8[grepl("eJD906", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("eJD907", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("eJD907", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)
d8_3 <- ev2_8[grepl("eJD907", ev2_8$PG_d8),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")

jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value

################################### unlabeled
#unlab <- read.delim(LF_fpath)
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]

df_t <- data.frame(c("eJD908", "eJD909", "eJD910"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0|unlab_1$Precursor.Translated>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c("eJD911", "eJD912", "eJD913"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0|unlab_2$Precursor.Translated>0),]  #only keep precursors with quant info

##

df_t <- data.frame(c("eJD914", "eJD915"), 
                   c("1", "2"))
colnames(df_t) <- c("file", "rep")
unlab_3 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_3$prot_run <- paste0(unlab_3$Protein.Group, "_",unlab_3$rep)
unlab_3 <- unlab_3[which(unlab_3$Ms1.Area>0|unlab_3$Precursor.Translated>0),]  #only keep precursors with quant info

#####
s1_1 <- unlab_1[grepl("eJD908", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("eJD909", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("eJD910", unlab_1$Run),] %>% dplyr::select(Protein.Group)

s2_1 <- unlab_2[grepl("eJD911", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("eJD912", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("eJD913", unlab_2$Run),] %>% dplyr::select(Protein.Group)

s3_1 <- unlab_3[grepl("eJD914", unlab_3$Run),] %>% dplyr::select(Protein.Group)
s3_2 <- unlab_3[grepl("eJD915", unlab_3$Run),] %>% dplyr::select(Protein.Group)
#s3_3 <- unlab_3[grepl("eJD930", unlab_3$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s3_1,s1_2,s2_2,s3_2,s1_3,s2_3)
for(i in 1:8){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s3_1","s1_2","s2_2","s3_2","s1_3","s2_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################



############### Jaccard index  DDA

DDA <- data.frame(fread(DDA_fpath))

DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_1 <- ev2_8[grepl("wJD750", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_2 <- ev2_8[grepl("wJD751", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d8_3 <- ev2_8[grepl("wJD752", ev2_8$Raw.file),] %>% dplyr::select(Leading.razor.protein)


all_peps <- rbind(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d8_1,d0_2,d4_2,d8_2,d0_3,d4_3,d8_3)
for(i in 1:9){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d8_1","d0_2","d4_2","d8_2","d0_3","d4_3","d8_3")
jac_dist_DDA <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_DDA_m <- get_lower_tri(jac_dist_DDA)
jac_dist_DDA_m <- melt(jac_dist_DDA_m, na.rm=T)
jac_dist_DDA_m$value <- 1-jac_dist_DDA_m$value

########### plot:

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

pDDA <- ggplot(jac_dist_DDA_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black", size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="mTRAQ DDA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) +
  geom_text(aes(label = round(value, 2)*100))

pDDA

############### mTRAQ

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% dplyr::mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% dplyr::mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% dplyr::mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% dplyr::mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B", "C")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black", size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  #scale_fill_gradient(breaks=c(0.5,0.75,1),labels=c("0.5",0.75,"1.0"), high = 'yellow', low = "maroon",limits=c(0.45, 1)) +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="plexDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  #scale_fill_gradient2(high = "aquamarine", low = "maroon",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1)) 
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))+
   geom_text(aes(label = round(value, 2)*100))



############## LF
m_jac_LF <- m_jac_LF %>% dplyr::mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% dplyr::mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% dplyr::mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF <- m_jac_LF %>% dplyr::mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B", "C","A", "B", "C","A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black", size=0.5) + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "\nRep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3\n",
       fill = "Protein Jaccard Index",
       title="LF-DIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=28),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC", 
                       midpoint=0.79,limits=c(0.58, 1), breaks = c(0.6, 0.8, 1)) + 
  geom_text(aes(label = round(value, 2)*100))



## combine
library(gridExtra)
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                            p1 + theme(legend.position="none"), NULL,
                            pDDA +theme(legend.position="none"),
                            nrow=1, widths=c(1,0.1,1,0.1,1)))

#install.packages('svglite')

ggsave("SFigure4d.pdf", p3, width=12.5, height=5.2, dpi=500)





############################ other plots:
jac_dist_mTRAQ_m$Type <- "plexDIA"
m_jac_LF$Type <- "LF-DIA"

jac_dist_mTRAQ_m$run_a <- stri_sub(jac_dist_mTRAQ_m$Var1,-1,-1)
jac_dist_mTRAQ_m$run_b <- stri_sub(jac_dist_mTRAQ_m$Var2,-1,-1)
jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("set" = ifelse(run_a == run_b, "Within a run", "Across runs"))

m_jac_LF$run_a <- stri_sub(m_jac_LF$Var1,-1,-1)
m_jac_LF$run_b <- stri_sub(m_jac_LF$Var2,-1,-1)
m_jac_LF <- m_jac_LF %>% dplyr::mutate("set" = "Across runs")

both <- bind_rows(jac_dist_mTRAQ_m, m_jac_LF)
both <- both[which(both$value<1),]  #remove overlaps to self

both$Var1a <- substr(both$Var1,1,nchar(as.character(both$Var1))-1)
both$Var2a <- substr(both$Var2,1,nchar(as.character(both$Var2))-1)

both <- both %>% dplyr::mutate("Groups" = ifelse(Var1a == Var2a, "Same samples", "Different samples"))
both$set <- factor(both$set, levels=c("Within a run", "Across runs"))
both$Type <- factor(both$Type, levels=c("plexDIA", "LF-DIA"))
both$missingness <- (1-both$value)*100
both$Groups <- factor(both$Groups, levels=c("Same samples", "Different samples"))

ggplot(both, aes(x=as.factor(Type),y=(missingness))) + 
  geom_point(size=2.5, alpha =0.7, position=position_jitter(width=0.25, height=0),
             aes(fill=set), 
       colour="black",pch=21) +
  scale_x_discrete(limits=c("plexDIA","LF-DIA")) + 
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + 
  theme_classic() +
  facet_grid(~Groups, scales = "free") +
  theme(axis.text.x = element_text(size=12, face="bold", color="black"),
        axis.text.y = element_text(size=14),
        axis.title.y = element_text(size=16),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12),
        legend.text = element_text(size=14),
        legend.title = element_blank())+ 
  guides(fill = guide_legend(override.aes = list(size=4)))+
  labs(x = "", y = "Missing data between pairs, %", fill = "") + ylim(0, 35) 
ggsave("SFigure4e.pdf", width=3.85, height=4, dpi=500)

```

Supp. Figure 4: Jaccard index (data completeness) of our data and Navarro et al.
```{r}
############### Jaccard index

mTRAQ <- data.frame(fread(fpath))
mTRAQ <- mTRAQ[which(mTRAQ$Lib.PG.Q.Value < 0.01),]
mTRAQ$PG_run <- paste0(mTRAQ$Protein.Group, "_", mTRAQ$Run)
mTRAQ <- mTRAQ[which(mTRAQ$Ms1.Area>0|mTRAQ$Precursor.Translated>0),]  #only keep precursors with quant info

ev2_0 <- mTRAQ[grepl("-0",mTRAQ$Precursor.Id),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- mTRAQ[grepl("-4",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Normalised","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_4) <- c("PG_d4","Precursor.Quantity_d4","Precursor.Normalised_d4","Precursor.Translated_d4","Ms1.Area_d4","Protein.Group")
ev2_8 <- mTRAQ[grepl("-8",mTRAQ$Precursor.Id),] %>% dplyr::select("PG_run","Precursor.Quantity","Precursor.Normalised","Precursor.Translated","Ms1.Area","Protein.Group") %>% distinct(PG_run, .keep_all=T)
colnames(ev2_8) <- c("PG_d8","Precursor.Quantity_d8","Precursor.Normalised_d8","Precursor.Translated_d8","Ms1.Area_d8","Protein.Group")


d0_1 <- ev2_0[grepl("wJD803", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_1 <- ev2_4[grepl("wJD803", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

d0_2 <- ev2_0[grepl("wJD804", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_2 <- ev2_4[grepl("wJD804", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

d0_3 <- ev2_0[grepl("wJD815", ev2_0$Run),] %>% dplyr::select(Protein.Group)
d4_3 <- ev2_4[grepl("wJD815", ev2_4$PG_d4),] %>% dplyr::select(Protein.Group)

all_peps <- rbind(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d0_2","d4_2","d0_3","d4_3")


jac_dist_mTRAQ <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
jac_dist_mTRAQ_m <- get_lower_tri(jac_dist_mTRAQ)
jac_dist_mTRAQ_m <- melt(jac_dist_mTRAQ_m, na.rm=T)
jac_dist_mTRAQ_m$value <- 1-jac_dist_mTRAQ_m$value


################################### unlabeled
unlab <- data.frame(fread(LF_fpath))
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]

df_t <- data.frame(c("wJD756_re", "wJD757_re", "wJD758_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0|unlab_1$Precursor.Translated>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c("wJD759_re", "wJD760_re", "wJD761_re"), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0|unlab_2$Precursor.Translated>0),]  #only keep precursors with quant info


#####
s1_1 <- unlab_1[grepl("wJD756_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl("wJD757_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl("wJD758_re", unlab_1$Run),] %>% dplyr::select(Protein.Group)


s2_1 <- unlab_2[grepl("wJD759_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl("wJD760_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl("wJD761_re", unlab_2$Run),] %>% dplyr::select(Protein.Group)

all_peps <- rbind(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s1_2","s2_2","s1_3","s2_3")

jac_dist_LF_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF <- get_lower_tri(jac_dist_LF_df)
m_jac_LF <- melt(m_jac_LF, na.rm = TRUE)
m_jac_LF$value <- 1-m_jac_LF$value
#################


################################### unlabeled LFQBench
unlab <- data.frame(fread(Bench_fpath))
unlab <- unlab[which(unlab$Lib.PG.Q.Value < 0.01),]
uni_run <- unique(unlab$Run)
uni_run
df <- unlab


df_t <- data.frame(c(paste0(uni_run[3]), paste0(uni_run[2]), paste0(uni_run[5])), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_1 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_1$prot_run <- paste0(unlab_1$Protein.Group, "_",unlab_1$rep)
unlab_1 <- unlab_1[which(unlab_1$Ms1.Area>0|unlab_1$Precursor.Translated>0),]  #only keep precursors with quant info

##
df_t <- data.frame(c(paste0(uni_run[4]), paste0(uni_run[1]), paste0(uni_run[6])), 
                   c("1", "2", "3"))
colnames(df_t) <- c("file", "rep")
unlab_2 <- unlab %>% left_join(df_t, by =c("Run" = "file"))
unlab_2$prot_run <- paste0(unlab_2$Protein.Group, "_",unlab_2$rep)
unlab_2 <- unlab_2[which(unlab_2$Ms1.Area>0|unlab_2$Precursor.Translated>0),]  #only keep precursors with quant info

#####
s1_1 <- unlab_1[grepl(paste0(uni_run[3]), unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_2 <- unlab_1[grepl(paste0(uni_run[2]), unlab_1$Run),] %>% dplyr::select(Protein.Group)
s1_3 <- unlab_1[grepl(paste0(uni_run[5]), unlab_1$Run),] %>% dplyr::select(Protein.Group)


s2_1 <- unlab_2[grepl(paste0(uni_run[4]), unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_2 <- unlab_2[grepl(paste0(uni_run[1]), unlab_2$Run),] %>% dplyr::select(Protein.Group)
s2_3 <- unlab_2[grepl(paste0(uni_run[6]), unlab_2$Run),] %>% dplyr::select(Protein.Group)


all_peps <- rbind(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
uni_pep_mTRAq <- unique(all_peps$Protein.Group)
jac <- data.frame(matrix(ncol = length(uni_pep_mTRAq)))
colnames(jac) <- uni_pep_mTRAq

all <- list(s1_1,s2_1,s1_2,s2_2,s1_3,s2_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_mTRAq%in%temp$Protein.Group
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("s1_1","s2_1","s1_2","s2_2","s1_3","s2_3")

jac_dist_LF2_df <- (as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard")))
library(reshape2)

m_jac_LF2 <- get_lower_tri(jac_dist_LF2_df)
m_jac_LF2 <- melt(m_jac_LF2, na.rm = TRUE)
m_jac_LF2$value <- 1-m_jac_LF2$value
#################


############### Jaccard index  DDA

#get prots which are 1% FDR
DDA <- data.frame(fread(DDA_fpath))

DDA <- DDA[which(DDA$Leading.razor.protein%in%prot_DDAFDR$Protein.Group),] # this filtered for proteins at 1% FDR

DDA <- DDA[!grepl("CON|REV", DDA$Leading.razor.protein),]
DDA$PG_run <- paste0(DDA$Raw.file,DDA$Leading.razor.protein)

ev2_0 <- DDA[which(DDA$Intensity.L > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_4 <- DDA[which(DDA$Intensity.M > 0),] %>% distinct(PG_run, .keep_all=T)
ev2_8 <- DDA[which(DDA$Intensity.H > 0),] %>% distinct(PG_run, .keep_all=T)

d0_1 <- ev2_0[grepl("wJD750", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_1 <- ev2_4[grepl("wJD750", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_2 <- ev2_0[grepl("wJD751", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_2 <- ev2_4[grepl("wJD751", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

d0_3 <- ev2_0[grepl("wJD752", ev2_0$Raw.file),] %>% dplyr::select(Leading.razor.protein)
d4_3 <- ev2_4[grepl("wJD752", ev2_4$Raw.file),] %>% dplyr::select(Leading.razor.protein)

all_peps <- rbind(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
uni_pep_DDA <- unique(all_peps$Leading.razor.protein)
jac <- data.frame(matrix(ncol = length(uni_pep_DDA)))
colnames(jac) <- uni_pep_DDA

all <- list(d0_1,d4_1,d0_2,d4_2,d0_3,d4_3)
for(i in 1:6){
  temp <- data.frame(all[i])
  jac_temp <- uni_pep_DDA%in%temp$Leading.razor.protein
  
  jac <- rbind(jac, jac_temp)
}
jac <- jac[-c(1),]

rownames(jac) <- c("d0_1","d4_1","d0_2","d4_2","d0_3","d4_3")


jac_dist_DDA <- as.matrix(proxy::dist(jac, by_rows = TRUE, method = "Jaccard"))
library(reshape2)
jac_dist_DDA <- get_lower_tri(jac_dist_DDA)
jac_dist_DDA_m <- melt(jac_dist_DDA, na.rm=T)
jac_dist_DDA_m$value <- 1-jac_dist_DDA_m$value



#################### call it A B C



######################### DDA
jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                ifelse(grepl("_2", Var1),"Rep 2",
                                                                       ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                ifelse(grepl("_2", Var2),"Rep 2",
                                                                       ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_DDA_m <- jac_dist_DDA_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                   ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                          ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))

labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")

pDDA <- ggplot(jac_dist_DDA_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="mTRAQ DDA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))+
    geom_text(aes(label = round(value, 2)*100),size=6)


pDDA

############### mTRAQ


jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var1),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var1), "\u0394","8\nSample 3", "BS"))))

jac_dist_mTRAQ_m <- jac_dist_mTRAQ_m %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"\u0394","0\nSample 1", 
                                                                       ifelse(grepl("s2", Var2),"\u0394","4\nSample 2",
                                                                              ifelse(grepl("s3", Var2), "\u0394","8\nSample 3", "BS"))))
labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")

p2 <- ggplot(jac_dist_mTRAQ_m, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="plexDIA") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))+
    geom_text(aes(label = round(value, 2)*100),size=6)


############## LF
m_jac_LF <- m_jac_LF %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF <- m_jac_LF %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")



p1 <- ggplot(m_jac_LF, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="LF-DIA this study") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))+
    geom_text(aes(label = round(value, 2)*100), size=6)


############## LFQ Bench
m_jac_LF2 <- m_jac_LF2 %>% mutate("Var1_Rep" = ifelse(grepl("_1", Var1),"Rep 1", 
                                                    ifelse(grepl("_2", Var1),"Rep 2",
                                                           ifelse(grepl("_3", Var1), "Rep 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var2_Rep" = ifelse(grepl("_1", Var2),"Rep 1", 
                                                    ifelse(grepl("_2", Var2),"Rep 2",
                                                           ifelse(grepl("_3", Var2), "Rep 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var1_sample" = ifelse(grepl("s1", Var1),"Sample 1", 
                                                       ifelse(grepl("s2", Var1),"Sample 2",
                                                              ifelse(grepl("s3", Var1), "Sample 3", "BS"))))

m_jac_LF2 <- m_jac_LF2 %>% mutate("Var2_sample" = ifelse(grepl("s1", Var2),"Sample 1", 
                                                       ifelse(grepl("s2", Var2),"Sample 2",
                                                              ifelse(grepl("s3", Var2), "Sample 3", "BS"))))

labz <- c("A", "B","A", "B", "A", "B")
repz <- c("Rep 1", "Rep 2", "Rep 3")


p4 <- ggplot(m_jac_LF2, aes(x=Var1, y=Var2, fill=value)) + geom_tile(color = "black") + 
  scale_x_discrete(labels= labz) +
  scale_y_discrete(labels= labz) + theme_classic() +
  labs(x = "Rep. 1   Rep. 2   Rep. 3",
       y = "Rep. 1   Rep. 2   Rep. 3",
       fill = "Protein Jaccard Index",
       title="LF-DIA from Navarro et al.") + 
  theme(axis.text.x = element_text(size=15),
        axis.text.y = element_text(size=15),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        legend.text = element_text(size=20),
        legend.title = element_text(size=20),
        plot.title = element_text(hjust = 0.5,size=20),
        legend.spacing.x = unit(1.0, 'cm'),
        legend.position = "top")+
  scale_fill_gradient2(high = "#EC7607", low = "#077DEC",midpoint=0.8,limits=c(0.60, 1), breaks = c(0.6, 0.8, 1))+
    geom_text(aes(label = round(value, 2)*100), size=6)
p4

## combine
library(gridExtra)
library(ggpubr)
library(gridGraphics)

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend<-g_legend(p1)

p3 <- ggarrange(mylegend, NULL, nrow=3,heights=c(1, 1, 10),
                ggarrange(p2 + theme(legend.position="none"), NULL,
                          p1 + theme(legend.position="none"), NULL,
                          p4 + theme(legend.position="none"), NULL,
                          pDDA +theme(legend.position="none"),
                          nrow=1, widths=c(1,0.1,1,0.1,1, 0.1, 1)))

ggsave("SFigure4_Navarro.pdf", p3, width=16.5, height=5.2, dpi=500)

```

Supp. Figure 5: MS2 accuracy (V2 method)
```{r}
pD <- data.frame(fread(MS2_fpath))
LF <- data.frame(fread(LF_MS2_fpath))

## plexDIA
pD_1 <- pD[grepl("eJD906", pD$Run),]
pD_1 <- pD_channel(pD_1) #Note which label is tagged on each precursor
pD_1 <- pD_1[which(pD_1$Lib.PG.Q.Value<0.01),]
pD_1 <- pD_seqcharge(pD_1)
pD_1 <- pD_rmMixSpec(pD_1) #Remove protein groups which map to >1 species
pD_1$File.Name <- paste0(pD_1$Run,"_", pD_1$channel_name)
pD_2 <- pD_diann_quant(pD_1, id.header = "seqcharge", quantity.header = "Precursor.Translated")
pD_2 <- pD_2 %>% dplyr::rename("Protein.Group" = "PGs")
pD_list <- pD_prepPlot(pD_2, Run = "eJD906") #data wrangle and prepare for plotting


### LF-DIA
LF_1 <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("eJD909", Run), "mTRAQ0",
                                             ifelse(grepl("eJD912", Run), "mTRAQ4",
                                                    ifelse(grepl("eJD915", Run), "mTRAQ8", "remove")))) %>%
  dplyr::filter(!grepl("remove", channel_name))
LF_1$Run <- "LF_runs"
LF_1 <- LF_1[which(LF_1$Lib.PG.Q.Value<0.01),]
LF_1 <- pD_rmMixSpec(LF_1) #Remove protein groups which map to >1 species
LF_1 <- pD_seqcharge(LF_1)
LF_1$File.Name <- paste0(LF_1$Run,"_", LF_1$channel_name)
LF_2 <- pD_diann_quant(LF_1, id.header = "seqcharge", quantity.header = "Precursor.Translated")
LF_2 <- LF_2 %>% dplyr::rename("Protein.Group" = "PGs")
LF_list <- pD_prepPlot(LF_2, Run = "LF_runs") #data wrangle and prepare for plotting


## join the plexDIA and LF-DIA sets and plot
pD_LF <- pD_plotLFQBench(pD_list, LF_list, pD_list_name = "plexDIA", LF_list_name = "LF-DIA")
ggsave("SFigure5abc_MS2quant.pdf", pD_LF, width=11,height=9, dpi=500)


################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########       Figure D: In-set vs Out-of-Set      ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 


df_names <- pD %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)
df <- pD
df <- pD_channel(df)
df <- pD_seqcharge(df)
df$File.Name <- paste0(df$Run, df$channel_name)
df1 <- df[grepl("eJD905", df$Run),]
protein.groups1 <- diann_maxlfq(df1[df1$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Precursor.Translated")
df2 <- df[grepl("eJD906", df$Run),]
protein.groups2 <- diann_maxlfq(df2[df2$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Precursor.Translated")

protein.groups1 <- data.frame(protein.groups1)
protein.groups2 <- data.frame(protein.groups2)

protein.groups1 <- protein.groups1 %>% dplyr::rename("d0" = "eJD905mTRAQ0",
                                                     "d4" = "eJD905mTRAQ4",
                                                     "d8" = "eJD905mTRAQ8")
protein.groups2 <- protein.groups2 %>% dplyr::rename("d0" = "eJD906mTRAQ0",
                                                     "d4" = "eJD906mTRAQ4",
                                                     "d8" = "eJD906mTRAQ8")

rep1_d0d4_pt <- protein.groups2
PGs <- rownames(rep1_d0d4_pt)
rep1_d0d4_pt <- cbind(rep1_d0d4_pt, PGs)
rep1_d0d4_pt$Prot_rat <- as.numeric(rep1_d0d4_pt$d0)/as.numeric(rep1_d0d4_pt$d4)
rep1_d0d4_pt <- rep1_d0d4_pt[!is.na(rep1_d0d4_pt$Prot_rat), ]
rep1_d0d4_pt <- rep1_d0d4_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_pt <- pD_rmMixSpec(rep1_d0d4_pt)
rep1_d0d4_pt <- pD_species(rep1_d0d4_pt)
med_human <- rep1_d0d4_pt[grepl("H. sapiens", rep1_d0d4_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_pt$Prot_rat <- rep1_d0d4_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_pt$Type <- "Within a run"

######## out of set:
rep1_d0d4_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- cbind(rep1_d0d4_out_pt, PGs)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% dplyr::select("d0", "PGs")
d4_set2 <- rep1_d0d4_pt %>% dplyr::select("d4", "PGs")

rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% inner_join(d4_set2, by =c("PGs" = "PGs"))
rep1_d0d4_out_pt$Prot_rat <- as.numeric(rep1_d0d4_out_pt$d0)/as.numeric(rep1_d0d4_out_pt$d4)
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[!is.na(rep1_d0d4_out_pt$Prot_rat), ]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d4_out_pt <- pD_rmMixSpec(rep1_d0d4_out_pt)
rep1_d0d4_out_pt <- pD_species(rep1_d0d4_out_pt)
med_human <- rep1_d0d4_out_pt[grepl("H. sapiens", rep1_d0d4_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d4_out_pt$Prot_rat <- rep1_d0d4_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d4_out_pt$Type <- "Across runs"


######################### d0:d8 ##################################

rep1_d0d8_pt <- protein.groups2
PGs <- rownames(rep1_d0d8_pt)
rep1_d0d8_pt <- cbind(rep1_d0d8_pt, PGs)
rep1_d0d8_pt <- data.frame(rep1_d0d8_pt)
rep1_d0d8_pt$Prot_rat <- as.numeric(rep1_d0d8_pt$d0)/as.numeric(rep1_d0d8_pt$d8)
rep1_d0d8_pt <- rep1_d0d8_pt[!is.na(rep1_d0d8_pt$Prot_rat), ]
rep1_d0d8_pt <- rep1_d0d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_pt <- pD_rmMixSpec(rep1_d0d8_pt)
rep1_d0d8_pt <- pD_species(rep1_d0d8_pt)
med_human <- rep1_d0d8_pt[grepl("H. sapiens", rep1_d0d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_pt$Prot_rat <- rep1_d0d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_pt_h <- rep1_d0d8_pt
rep1_d0d8_pt <- rep1_d0d8_pt[!grepl("H.", rep1_d0d8_pt$species),]
rep1_d0d8_pt$Type <- "Within a run"

######## out of set:
rep1_d0d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- cbind(rep1_d0d8_out_pt, PGs)
rep1_d0d8_out_pt <- data.frame(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% dplyr::select("d0", "PGs")
d8_set2 <- rep1_d0d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d0d8_out_pt$Prot_rat <- as.numeric(rep1_d0d8_out_pt$d0)/as.numeric(rep1_d0d8_out_pt$d8)
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!is.na(rep1_d0d8_out_pt$Prot_rat), ]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d0d8_out_pt <- pD_rmMixSpec(rep1_d0d8_out_pt)
rep1_d0d8_out_pt <- pD_species(rep1_d0d8_out_pt)
med_human <- rep1_d0d8_out_pt[grepl("H. sapiens", rep1_d0d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d0d8_out_pt$Prot_rat <- rep1_d0d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[!grepl("H.", rep1_d0d8_out_pt$species),]
rep1_d0d8_out_pt$Type <- "Across runs"


######################################## d4:d8 ###########################

rep1_d4d8_pt <- protein.groups2
PGs <- rownames(rep1_d4d8_pt)
rep1_d4d8_pt <- cbind(rep1_d4d8_pt, PGs)
rep1_d4d8_pt <- data.frame(rep1_d4d8_pt)
rep1_d4d8_pt$Prot_rat <- as.numeric(rep1_d4d8_pt$d4)/as.numeric(rep1_d4d8_pt$d8)
rep1_d4d8_pt <- rep1_d4d8_pt[!is.na(rep1_d4d8_pt$Prot_rat), ]
rep1_d4d8_pt <- rep1_d4d8_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_pt <- pD_rmMixSpec(rep1_d4d8_pt)
rep1_d4d8_pt <- pD_species(rep1_d4d8_pt)
med_human <- rep1_d4d8_pt[grepl("H. sapiens", rep1_d4d8_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_pt$Prot_rat <- rep1_d4d8_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_pt_h <- rep1_d4d8_pt
rep1_d4d8_pt <- rep1_d4d8_pt[!grepl("H.", rep1_d4d8_pt$species),]
rep1_d4d8_pt$Type <- "Within a run"

######## out of set:
rep1_d4d8_out_pt <- protein.groups1
PGs <- rownames(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- cbind(rep1_d4d8_out_pt, PGs)
rep1_d4d8_out_pt <- data.frame(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% dplyr::select("d4", "PGs")
d8_set2 <- rep1_d4d8_pt_h %>% dplyr::select("d8", "PGs")

rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% inner_join(d8_set2, by =c("PGs" = "PGs"))
rep1_d4d8_out_pt$Prot_rat <- as.numeric(rep1_d4d8_out_pt$d4)/as.numeric(rep1_d4d8_out_pt$d8)
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!is.na(rep1_d4d8_out_pt$Prot_rat), ]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt %>% left_join(df_names, by =c("PGs"="Protein.Group"))
rep1_d4d8_out_pt <- pD_rmMixSpec(rep1_d4d8_out_pt)
rep1_d4d8_out_pt <- pD_species(rep1_d4d8_out_pt)
med_human <- rep1_d4d8_out_pt[grepl("H. sapiens", rep1_d4d8_out_pt$species),]
med_human <- median(med_human$Prot_rat, na.rm=T)
scalar <- 1/med_human
rep1_d4d8_out_pt$Prot_rat <- rep1_d4d8_out_pt$Prot_rat*scalar   #normalize human median on 1:1 ratio
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[!grepl("H.", rep1_d4d8_out_pt$species),]
rep1_d4d8_out_pt$Type <- "Across runs"



###############
rep1_d0d4_pt <- rep1_d0d4_pt[rep1_d0d4_pt$PGs%in%rep1_d0d4_out_pt$PGs,]
rep1_d0d4_out_pt <- rep1_d0d4_out_pt[rep1_d0d4_out_pt$PGs%in%rep1_d0d4_pt$PGs,]

rep1_d0d8_pt <- rep1_d0d8_pt[rep1_d0d8_pt$PGs%in%rep1_d0d8_out_pt$PGs,]
rep1_d0d8_out_pt <- rep1_d0d8_out_pt[rep1_d0d8_out_pt$PGs%in%rep1_d0d8_pt$PGs,]

rep1_d4d8_pt <- rep1_d4d8_pt[rep1_d4d8_pt$PGs%in%rep1_d4d8_out_pt$PGs,]
rep1_d4d8_out_pt <- rep1_d4d8_out_pt[rep1_d4d8_out_pt$PGs%in%rep1_d4d8_pt$PGs,]

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))
df2<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(2/3), log2(3)))
df3<- data.frame(species = c("E. coli", "S. cerevisiae"), Z = c(log2(1/6), log2(6)))

d0d4 <- bind_rows(rep1_d0d4_pt, rep1_d0d4_out_pt)
d0d8 <- bind_rows(rep1_d0d8_pt, rep1_d0d8_out_pt)
d4d8 <- bind_rows(rep1_d4d8_pt, rep1_d4d8_out_pt)

d0d4 <- d0d4 %>% left_join(df1, by =c("species" = "species"))
d0d8 <- d0d8 %>% left_join(df2, by =c("species" = "species"))
d4d8 <- d4d8 %>% left_join(df3, by =c("species" = "species"))

d0d4$error <- abs(log2(d0d4$Prot_rat)-d0d4$Z)
d0d8$error <- abs(log2(d0d8$Prot_rat)-d0d8$Z)
d4d8$error <- abs(log2(d4d8$Prot_rat)-d4d8$Z)

all <- rbind(d0d4, d4d8, d0d8)
####

####


y_d0d4 <- all$error %>% sort()
y_d0d4_val <- as.numeric(quantile(y_d0d4,probs=c(0,0.99)))

df1<- data.frame(species = c("E. coli", "H. sapiens", "S. cerevisiae"), Z = c(2, 0, -1))

all$Type <- factor(all$Type, levels=c("Within a run", "Across runs"))

D_fig <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + 
  geom_boxplot(alpha =0.9, outlier.shape=NA) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  #scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "top",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "") +
  coord_cartesian(ylim = c(0,4))
D_fig

D_fig_noL <- ggplot(all, aes(x=as.factor(Type),y=(error), fill=Type)) + 
  geom_boxplot(alpha =0.9, outlier.shape=NA) +
  scale_x_discrete(limits=c("Within a run","Across runs")) + 
  #scale_fill_manual(values = c("steelblue", "tomato1")) + #"tan2"
  scale_fill_manual(values = c("#CC6677", "#40B0A6")) + #"tan2"
  theme_classic() +
  facet_grid(~species, scales = "free") +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        # plot.title = element_text(size=14, hjust = 0.5),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        legend.position = "none",
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y = expression(paste("|",Log["2"],", Protein Ratio Error|")), fill = "",subtitle="",title="") +
  coord_cartesian(ylim = c(0,4))
D_fig_noL
ggsave("SFigure5d_nolegend.pdf", width=3.8,height=3.7)

################################################################# 
################################################################# 
################################################################# 
################################################################# 
#########        Figure E: R vs K quant MS2         ############# 
################################################################# 
################################################################# 
################################################################# 
################################################################# 

ev <- pD[grepl("eJD906", pD$Run),]

ev_MS2 <- pD_PrecRatios(ev, quant.header = "Precursor.Translated")
ev_MS2$Quant <- "MS2"
ev <- ev_MS2
ev <- pD_species(ev)
ev$lab_species <- paste0(ev$variable, "_", ev$species)
df1<- data.frame(species = c("d0_d4_E. coli", "d0_d4_H. sapiens", "d0_d4_S. cerevisiae", "d0_d8_E. coli", "d0_d8_S. cerevisiae", "d4_d8_E. coli", "d4_d8_S. cerevisiae"), Z = c(2, 0, -1, log2(2/3), log2(3), log2(1/6), log2(6)))
ev <- ev %>% left_join(df1, by =c("lab_species" = "species"))
ev$seq_run_var <- paste0(ev$seqcharge_file, ev$variable)
ev <- ev[!is.na(ev$Z),] #removes human in d0_d8 and d4_d8 comparisons because they compare U937 and Jurkat cell lines
#####
ev <- pD_Cterm(ev)
all_MS2_K <- ev[grepl("K", ev$Cterm),] %>% dplyr::filter(grepl("MS2", Quant))
all_MS2_R <- ev[grepl("R", ev$Cterm),] %>% dplyr::filter(grepl("MS2", Quant))

all_MS2_K$Type <- "Lys precursors"
all_MS2_K <- all_MS2_K %>% dplyr::add_count(species)
all_MS2_K$cond <- paste0("n=", all_MS2_K$n)

all_MS2_R$Type <- "Arg precursors"
all_MS2_R <- all_MS2_R %>% dplyr::add_count(species)
all_MS2_R$cond <- paste0("n=", all_MS2_R$n)

K_vs_R_MS1_mDIA_d0d4 <- bind_rows(all_MS2_K, all_MS2_R)
K_vs_R_MS1_mDIA_d0d4$error <- abs(log2(K_vs_R_MS1_mDIA_d0d4$value)-K_vs_R_MS1_mDIA_d0d4$Z)

############ add in the MS1 quant for multiDIA precursors:
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% dplyr::distinct(cond,.keep_all=T)
df_n <- K_vs_R_MS1_mDIA_d0d4 %>% dplyr::distinct(cond,.keep_all=T)
K_vs_R_MS1_mDIA_d0d4$Type <- factor(K_vs_R_MS1_mDIA_d0d4$Type, levels=c("Lys precursors", "Arg precursors"))

E_fig <- ggplot(K_vs_R_MS1_mDIA_d0d4, aes(x=as.factor(Type),y=error, fill=Type)) + geom_boxplot( alpha =0.7, outlier.shape=NA) +
  scale_x_discrete(limits=c("Lys precursors","Arg precursors")) + 
  scale_fill_manual(values = c("dodgerblue3", "deepskyblue")) +
  theme_classic() +
  facet_grid(~species, scales = "free")  +
  coord_cartesian(ylim = c(0,5)) +
  theme(axis.text.x = element_blank(),
        axis.text.y = element_text(size=12),
        axis.title.y = element_text(size=14),
        axis.ticks.x = element_blank(),
        plot.subtitle = element_text(size = 12, hjust = 0.5),
        plot.title = element_text(size = 16, hjust = 0.5, face="bold"),
        strip.text.x = element_text(size=12, face = "italic"),
        legend.text = element_text(size=12))+
  labs(x = "", y=expression(paste("|",Log["2"],", Precursor Ratio Error|")), fill = "",subtitle="",title="")
E_fig
ggsave("SFigure5e.pdf", width=5.7,height=3.7)


```

Supp. Figure 6: MS1 other replicates
```{r}
pD <- data.frame(fread(fpath))
LF <- data.frame(fread(LF_fpath))

## plexDIA
pD <- pD_isotopicCO(pD)  #correct for isotopic carryover across channels.. may take 2 minutes
pD_1 <- pD[grepl("wJD803", pD$Run),]
pD_1 <- pD_channel(pD_1) #Note which label is tagged on each precursor
pD_1 <- pD_1[which(pD_1$Lib.PG.Q.Value<0.01),]
pD_1 <- pD_seqcharge(pD_1)
pD_1 <- pD_rmMixSpec(pD_1) #Remove protein groups which map to >1 species
pD_1$File.Name <- paste0(pD_1$Run,"_", pD_1$channel_name)
pD_2 <- pD_diann_quant(pD_1, id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
pD_2 <- pD_2 %>% dplyr::rename("Protein.Group" = "PGs")
pD_list <- pD_prepPlot(pD_2, Run = "wJD803") #data wrangle and prepare for plotting


### LF-DIA
LF_1 <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("JD756", Run), "mTRAQ0",
                                             ifelse(grepl("JD759", Run), "mTRAQ4",
                                                    ifelse(grepl("JD762", Run), "mTRAQ8", "remove")))) %>%
  dplyr::filter(!grepl("remove", channel_name))
LF_1$Run <- "LF_runs"
LF_1 <- LF_1[which(LF_1$Lib.PG.Q.Value<0.01),]
LF_1 <- pD_rmMixSpec(LF_1) #Remove protein groups which map to >1 species
LF_1 <- pD_seqcharge(LF_1)
LF_1$File.Name <- paste0(LF_1$Run,"_", LF_1$channel_name)
LF_2 <- pD_diann_quant(LF_1, id.header = "seqcharge", quantity.header = "Ms1.Area")
LF_2 <- LF_2 %>% dplyr::rename("Protein.Group" = "PGs")
LF_list <- pD_prepPlot(LF_2, Run = "LF_runs") #data wrangle and prepare for plotting

## join the plexDIA and LF-DIA sets and plot
pD_LF <- pD_plotLFQBench(pD_list, LF_list, pD_list_name = "plexDIA", LF_list_name = "LF-DIA")
ggsave("SFigure6a.pdf", pD_LF, width=11,height=9, dpi=500)
################

## plexDIA
pD <- pD_isotopicCO(pD)  #correct for isotopic carryover across channels.. may take 2 minutes
pD_1 <- pD[grepl("wJD815", pD$Run),]
pD_1 <- pD_channel(pD_1) #Note which label is tagged on each precursor
pD_1 <- pD_1[which(pD_1$Lib.PG.Q.Value<0.01),]
pD_1 <- pD_seqcharge(pD_1)
pD_1 <- pD_rmMixSpec(pD_1) #Remove protein groups which map to >1 species
pD_1$File.Name <- paste0(pD_1$Run,"_", pD_1$channel_name)
pD_2 <- pD_diann_quant(pD_1, id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
pD_2 <- pD_2 %>% dplyr::rename("Protein.Group" = "PGs")
pD_list <- pD_prepPlot(pD_2, Run = "wJD815") #data wrangle and prepare for plotting


### LF-DIA
LF_1 <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("JD758", Run), "mTRAQ0",
                                             ifelse(grepl("JD761", Run), "mTRAQ4",
                                                    ifelse(grepl("JD764", Run), "mTRAQ8", "remove")))) %>%
  dplyr::filter(!grepl("remove", channel_name))
LF_1$Run <- "LF_runs"
LF_1 <- LF_1[which(LF_1$Lib.PG.Q.Value<0.01),]
LF_1 <- pD_rmMixSpec(LF_1) #Remove protein groups which map to >1 species
LF_1 <- pD_seqcharge(LF_1)
LF_1$File.Name <- paste0(LF_1$Run,"_", LF_1$channel_name)
LF_2 <- pD_diann_quant(LF_1, id.header = "seqcharge", quantity.header = "Ms1.Area")
LF_2 <- LF_2 %>% dplyr::rename("Protein.Group" = "PGs")
LF_list <- pD_prepPlot(LF_2, Run = "LF_runs") #data wrangle and prepare for plotting

## join the plexDIA and LF-DIA sets and plot
pD_LF <- pD_plotLFQBench(pD_list, LF_list, pD_list_name = "plexDIA", LF_list_name = "LF-DIA")
ggsave("SFigure6_b.pdf", pD_LF, width=11,height=9, dpi=500)

```

Supp Figure 7: MS1 out-of-set quant and CVs
```{r}
pD <- data.frame(fread(fpath))
LF <- data.frame(fread(LF_fpath))

pD <- pD_isotopicCO(pD)  #correct for isotopic carryover across channels.. may take 2 minutes
pD_1 <- pD
pD_1 <- pD_channel(pD_1) #Note which label is tagged on each precursor
pD_1 <- pD_1[which(pD_1$Lib.PG.Q.Value<0.01),]
pD_1 <- pD_seqcharge(pD_1)
pD_1 <- pD_rmMixSpec(pD_1) #Remove protein groups which map to >1 species
pD_1$File.Name <- paste0(pD_1$Run,"_", pD_1$channel_name)
pD_2 <- pD_diann_quant(pD_1, id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
pD_2 <- pD_2 %>% dplyr::rename("Protein.Group" = "PGs")
pD_list <- pD_prepPlot(pD_2, Run = "wJD815") #data wrangle and prepare for plotting

prot_df <- data.frame(pD_2)
df <- pD_species(prot_df)
df1 <- df
df1$d0_d4 <- as.numeric(df1$wJD803_mTRAQ0)/as.numeric(df1$wJD804_mTRAQ4)
df1 <- df1[!is.na(df1$d0_d4), ]
df1 <- df1[!is.infinite(df1$d0_d4), ]
df1 <- df1[which(df1$d0_d4>0), ]
med_human <- df1[grepl("H. sapiens", df1$species),]
med_human <- median(med_human$d0_d4, na.rm=T)
scalar <- 1/med_human
df1$d0_d4 <- df1$d0_d4*scalar   #normalize human median on 1:1 ratio
df1 <- df1 %>% dplyr::select("Protein.Group", "species", "wJD804_mTRAQ4", "d0_d4")
colnames(df1)[3] <- "sample2_int"
df1$sample2_int <- as.numeric(df1$sample2_int)
  
df2 <- df
df2$d0_d8 <- as.numeric(df2$wJD803_mTRAQ0)/as.numeric(df2$wJD804_mTRAQ8)
df2 <- df2[!is.na(df2$d0_d8), ]
df2 <- df2[!is.infinite(df2$d0_d8), ]
df2 <- df2[which(df2$d0_d8>0), ]
med_human <- df2[grepl("H. sapiens", df2$species),]
med_human <- median(med_human$d0_d8, na.rm=T)
scalar <- 1/med_human
df2$d0_d8 <- df2$d0_d8*scalar   #normalize human median on 1:1 ratio
df2 <- df2 %>% dplyr::select("Protein.Group", "species", "wJD804_mTRAQ8", "d0_d8")
colnames(df2)[3] <- "sample2_int"
df2$sample2_int <- as.numeric(df2$sample2_int)
  
df3 <- df
df3$d4_d8 <- as.numeric(df3$wJD803_mTRAQ4)/as.numeric(df3$wJD804_mTRAQ8)
df3 <- df3[!is.na(df3$d4_d8), ]
df3 <- df3[!is.infinite(df3$d4_d8), ]
df3 <- df3[which(df3$d4_d8>0), ]
med_human <- df3[grepl("H. sapiens", df3$species),]
med_human <- median(med_human$d4_d8, na.rm=T)
scalar <- 1/med_human
df3$d4_d8 <- df3$d4_d8*scalar   #normalize human median on 1:1 ratio
df3 <- df3 %>% dplyr::select("Protein.Group", "species", "wJD804_mTRAQ8", "d4_d8")
colnames(df3)[3] <- "sample2_int"
df3$sample2_int <- as.numeric(df3$sample2_int)
pDlist <- list(df1,df2,df3)


### LF-DIA
LF_1 <- LF %>% dplyr::mutate("channel_name" = ifelse(grepl("JD757", Run), "mTRAQ0",
                                             ifelse(grepl("JD760", Run), "mTRAQ4",
                                                    ifelse(grepl("JD763", Run), "mTRAQ8", "remove")))) %>%
  dplyr::filter(!grepl("remove", channel_name))
LF_1$Run <- "LF_runs"
LF_1 <- LF_1[which(LF_1$Lib.PG.Q.Value<0.01),]
LF_1 <- pD_rmMixSpec(LF_1) #Remove protein groups which map to >1 species
LF_1 <- pD_seqcharge(LF_1)
LF_1$File.Name <- paste0(LF_1$Run,"_", LF_1$channel_name)
LF_2 <- pD_diann_quant(LF_1, id.header = "seqcharge", quantity.header = "Ms1.Area")
LF_2 <- LF_2 %>% dplyr::rename("Protein.Group" = "PGs")
LF_list <- pD_prepPlot(LF_2, Run = "LF_runs") #data wrangle and prepare for plotting

## join the plexDIA and LF-DIA sets and plot
pD_LF <- pD_plotLFQBench(pDlist, LF_list, pD_list_name = "plexDIA", LF_list_name = "LF-DIA")
ggsave("SFigure7_out_of_set_quant.pdf", pD_LF, width=11,height=9, dpi=500)

##########################################################################
##########################################################################
################### CVs (within a run and across runs) ##################
##########################################################################
##########################################################################

pD <- data.frame(fread(fpath))

pD <- pD_channel(pD)
pD <- pD_seqcharge(pD)
pD$File.Name <- paste0(pD$Run, pD$channel_name)
pD <- pD_isotopicCO(pD)
pD_keep <-pD
#pD <- pD_keep
pD <- pD[grepl("wJD803|804|815", pD$Run),]
protein.groups <- diann_maxlfq(pD[pD$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
protein.groups_n_plex <- protein.groups    
cmeds <- robustbase::colMedians(protein.groups_n_plex, na.rm=T)
protein.groups_n_plex <- sweep(protein.groups_n_plex, 2, cmeds, FUN = '/')
PGs <- rownames(protein.groups_n_plex)
protein.groups_n_plex <- cbind(protein.groups_n_plex,PGs)

df_names <- pD %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)

protein.groups_n_plex <- data.frame(protein.groups_n_plex)
protein.groups_n_plex <- protein.groups_n_plex %>% inner_join(df_names, by =c("PGs"="Protein.Group"))
protein.groups_n_plex <- pD_rmMixSpec(protein.groups_n_plex)
protein.groups_n_plex <- pD_species(protein.groups_n_plex)


######################## label-swap  ########################################

df <- pD_keep
df1 <- df[grepl("wJD804", df$Run),]
df1 <- df1 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ0",
                                            ifelse(grepl("-4", Modified.Sequence), "mTRAQ4", "mTRAQ8")))

df2 <- df[grepl("wJD806", df$Run),]
df2 <- df2 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ8",
                                              ifelse(grepl("-4", Modified.Sequence), "mTRAQ0", "mTRAQ4")))

df3 <- df[grepl("wJD808", df$Run),]
df3 <- df3 %>% dplyr::mutate("channel_name" = ifelse(grepl("-0", Modified.Sequence), "mTRAQ4",
                                              ifelse(grepl("-4", Modified.Sequence), "mTRAQ8", "mTRAQ0")))

df <- rbind(df1,df2,df3)
df$File.Name <- paste0(df$Run, df$channel_name)
protein.groups <- diann_maxlfq(df[df$Lib.PG.Q.Value <= 0.01,], group.header="Protein.Group", id.header = "seqcharge", quantity.header = "Ms1.Area_iso")
protein.groups_n <- protein.groups    #protein.groups/protein.groups_med
cmeds <- robustbase::colMedians(protein.groups_n, na.rm=T)
protein.groups_n <- sweep(protein.groups_n, 2, cmeds, FUN = '/')
PGs <- rownames(protein.groups_n)
protein.groups_n <- cbind(protein.groups_n,PGs)

df_names <- pD %>% dplyr::select("Protein.Group", "Protein.Names") %>% dplyr::distinct(Protein.Group, .keep_all=T)

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n <- protein.groups_n %>% inner_join(df_names, by =c("PGs"="Protein.Group"))
protein.groups_n <- pD_rmMixSpec(protein.groups_n)
protein.groups_n <- pD_species(protein.groups_n)

########### combine/intersect same labels and label-swap plexDIA sets

protein.groups_n <- data.frame(protein.groups_n)
protein.groups_n_plex <- data.frame(protein.groups_n_plex)
## intersect plexDIA and LF-DIA ###
protein.groups_n <- protein.groups_n[protein.groups_n$PGs%in%protein.groups_n_plex$PGs,]
protein.groups_n_plex <- protein.groups_n_plex[protein.groups_n_plex$PGs%in%protein.groups_n$PGs,]

rownames(protein.groups_n) <- protein.groups_n$PGs
rownames(protein.groups_n_plex) <- protein.groups_n_plex$PGs
####### plex cont.

protein.groups_n_plex <- protein.groups_n_plex %>% dplyr::select(-matches(c("PGs", "species", "HY", "Protein.Names")))
m_PGs <- melt(as.matrix(protein.groups_n_plex))
m_PGs <- m_PGs %>% dplyr::mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                                                       ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs <- m_PGs[!is.na(m_PGs$value),]
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::add_count() %>% dplyr::filter(n==3) %>% dplyr::ungroup()
  
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_mean" = mean(as.numeric(value))) %>% dplyr::ungroup()
m_PGs <- m_PGs %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("Prot_CV" = prot_sd/prot_mean) %>% dplyr::ungroup()
m_PGs$uni <- paste0(m_PGs$Var1,m_PGs$label)
m_PGs <- m_PGs %>% dplyr::distinct(uni, .keep_all=T)
m_PGs$Cond <- "Same labels"

###### LF cont.
protein.groups_n <- protein.groups_n[!grepl("remove", protein.groups_n$species),] %>% dplyr::select(-matches(c("PGs","species", "HY", "Protein.Names")))

m_PGs_LF <- reshape2::melt(as.matrix(protein.groups_n))
m_PGs_LF <- m_PGs_LF %>% dplyr::mutate("label" = ifelse(grepl("mTRAQ0", Var2), "mTRAQ0",
                                           ifelse(grepl("mTRAQ4", Var2), "mTRAQ4", "mTRAQ8")))
m_PGs_LF <- m_PGs_LF[!is.na(m_PGs_LF$value),]
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::add_count() %>% filter(n==3) %>% ungroup()

m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_sd" = sd(value, na.rm=T)) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("prot_mean" = mean(as.numeric(value))) %>% ungroup
m_PGs_LF <- m_PGs_LF %>% dplyr::group_by(label, Var1) %>% dplyr::mutate("Prot_CV" = prot_sd/prot_mean) %>% dplyr::ungroup()
m_PGs_LF$uni <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs_LF <- m_PGs_LF %>% dplyr::distinct(uni, .keep_all=T)
m_PGs_LF$Cond <- "Different labels"

m_PGs_LF$Var1_lab <- paste0(m_PGs_LF$Var1,m_PGs_LF$label)
m_PGs$Var1_lab <- paste0(m_PGs$Var1,m_PGs$label)

## last intersection:
m_PGs_LF <- m_PGs_LF[m_PGs_LF$Var1_lab%in%m_PGs$Var1_lab,]
m_PGs <- m_PGs[m_PGs$Var1_lab%in%m_PGs_LF$Var1_lab,]

##
################ select and bind:

mDIA_LF <- bind_rows(m_PGs,m_PGs_LF)
mDIA_LF$Cond <- factor(mDIA_LF$Cond, levels=c("Same labels", "Different labels"))
df1 <- mDIA_LF %>% dplyr::group_by(Cond) %>% dplyr::mutate("med_CV" = round(median(Prot_CV,na.rm=T),3)) %>% dplyr::distinct(Cond,.keep_all=T)

ymax <- mDIA_LF$Prot_CV %>% sort()
ymax <- as.numeric(quantile(ymax,probs=c(.01,.99)))


cvs <- c(round(median(m_PGs$Prot_CV),3), round(median(m_PGs_LF$Prot_CV),3)) 

CV1_fig <- ggplot(mDIA_LF, aes(x=Cond,y=Prot_CV)) + 
  geom_quasirandom(alpha=0.1, groupOnX = T)+
  geom_boxplot(width=0.25, outlier.shape=NA, color="grey50") +
  theme_classic() +
  geom_text(data=df1, aes(x=Cond, y=cvs+0.045, label=paste0(cvs)), color="black",parse = FALSE, size=3.4) +
  theme(axis.text.x = element_text(size=16, face = "bold", color = "black"),
        axis.text.y = element_text(size=16),
        axis.title.y = element_text(size=18),
        axis.title.x = element_text(size=18),
        axis.ticks.x = element_blank(),
        legend.position = "none",
        legend.text = element_text(size=12))+
  labs(x = "", y = "Sample-specific protein CV", fill = "")
CV1_fig
ggsave("/Volumes/GoogleDrive/My Drive/Code/My_Data/PlexDIA/Revisions_small_samples_01122022/SFigureCV_labswap_beeswarm.png", width=4.7,height=4.7, dpi=300)

```

Supp. Figure 8: Data missingness (negative control and single)
```{r}

SC <- data.frame(fread(SC_fpath))
SC_ms1ext <- data.frame(fread(SC_ms1ext_fpath))
SC <- SC[!grepl("wJD12", SC$Run),] #remove the diluted bulk runs
SC_ms1ext <- SC_ms1ext[,!grepl("wJD12", colnames(SC_ms1ext))]  #remove the diluted bulk runs
PGs_0.01FDR <- SC[which(SC$Lib.PG.Q.Value<0.01),] %>% distinct(Protein.Group)
qev <- pD_isoCO_Ms1Ext(SC, SC_ms1ext) #correct for isotopic carryover
qev_keep <- qev
qev <- qev_keep
qev <- pD_channel(qev)
qev <- pD_seqcharge(qev)
qev$runchan <- paste0(qev$Run, qev$channel_name)
SC$runchan <- paste0(SC$Run, SC$channel_name)
colnames(qev)
meta <- read.delim(meta_PCA_fpath)

par(mfrow=c(1,1))
qev$pep<-paste0(qev$Modified.Sequence, qev$Precursor.Charge)
qev$prot <- qev$Protein.Group
qev$seqcharge<-paste0(qev$Stripped.Sequence, qev$Precursor.Charge)
q0<-qev %>% dplyr::select("Run","channel_name","Ms1.Extracted_iso", "Channel.Q.Value", "pep", "prot", "seqcharge")
q0$run_chan <- paste0(q0$Run, q0$channel_name)
meta$run_chan <- paste0(meta$Raw, "mTRAQ",meta$Label)
meta <- meta %>% dplyr::select(-c("Raw","Label") )
dat<-q0 %>% inner_join(meta, by =c("run_chan"="run_chan"))


dat_missing <- dat %>% dplyr::group_by(run_chan) %>% dplyr::add_count() %>% 
  dplyr::filter(Ms1.Extracted_iso==0) %>% dplyr::add_count() %>%
  dplyr::mutate(missing=nn/n) %>%
  dplyr::distinct(run_chan, .keep_all=T) %>% dplyr::ungroup() %>% dplyr::select("run_chan","missing", "Celltype")


ggplot(dat_missing, aes(x=Celltype, y =missing*100)) + geom_jitter(alpha=0.7,width=0.1) +
    theme_bw() +
  scale_x_discrete(limits=c("Melanoma", "PDAC", "U-937", "Neg"))+
  theme(axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=14)) + labs(y="Missing data (%)")
ggsave("MissingData_singleCells_QE.pdf",width=4,height=4)


```

Supp. Figure 9: PCA to assess labeling bias (or lack thereof)... Please find this in the Figure 6 section!

